diff --git a/packages/v5.1/output-branded/lib-files/lib.es2017.d.ts b/packages/v5.2/output-branded/lib-files/lib.es2017.d.ts
index 550bc3d..dddae9a 100644
--- a/packages/v5.1/output-branded/lib-files/lib.es2017.d.ts
+++ b/packages/v5.2/output-branded/lib-files/lib.es2017.d.ts
@@ -7,3 +7,4 @@
 /// <reference path="./lib.es2017.string.d.ts" />
 /// <reference path="./lib.es2017.intl.d.ts" />
 /// <reference path="./lib.es2017.typedarrays.d.ts" />
+/// <reference path="./lib.es2017.date.d.ts" />
