diff --git experimental/packages/utils/react-utils/src/hooks/reactive/use-merged2.mts apps/react-utils/src/hooks/reactive/use-merged2.mts
index a9f91343b..bdddb556c 100644
--- experimental/packages/utils/react-utils/src/hooks/reactive/use-merged2.mts
+++ apps/react-utils/src/hooks/reactive/use-merged2.mts
@@ -1,20 +1,20 @@
 import { useState } from 'better-react-use-state';
-import { useEffect } from 'react';
+import * as React from 'react';
 
-export function useMerged2<T1, T2>(
+export const useMerged2 = <T1, T2>(
   init: T1 | T2,
   value1: T1,
   value2: T2,
-): T1 | T2 {
-  const [merged, set] = useState<T1 | T2>(init);
+): T1 | T2 => {
+  const [merged, set] = useState(init);
 
-  useEffect(() => {
+  React.useEffect(() => {
     set(value1);
   }, [value1]);
 
-  useEffect(() => {
+  React.useEffect(() => {
     set(value2);
   }, [value2]);
 
   return merged;
-}
+};
