diff --git experimental/packages/apps/event-schedule-app-shared/src/v6/types/record/base/user.mts apps/event-schedule-app-shared/src/v6/types/record/base/user.mts
index 5241cf4e0..f017afb34 100644
--- experimental/packages/apps/event-schedule-app-shared/src/v6/types/record/base/user.mts
+++ apps/event-schedule-app-shared/src/v6/types/record/base/user.mts
@@ -1,4 +1,4 @@
-import { isRecord } from '@noshiro/ts-utils';
+import { hasKey, isRecord } from 'ts-data-forge';
 import {
   isUserId,
   isUserName,
@@ -18,9 +18,9 @@ export const userDefaultValue = {
 
 export const isUser = (a: unknown): a is User =>
   isRecord(a) &&
-  Object.hasOwn(a, 'id') &&
+  hasKey(a, 'id') &&
   isUserId(a.id) &&
-  Object.hasOwn(a, 'name') &&
+  hasKey(a, 'name') &&
   isUserName(a.name);
 
 const d = userDefaultValue;
@@ -28,7 +28,7 @@ const d = userDefaultValue;
 export const fillUser = (a?: unknown): User =>
   a === undefined || !isRecord(a)
     ? d
-    : {
-        id: Object.hasOwn(a, 'id') && isUserId(a.id) ? a.id : d.id,
-        name: Object.hasOwn(a, 'name') && isUserName(a.name) ? a.name : d.name,
-      };
+    : ({
+        id: hasKey(a, 'id') && isUserId(a.id) ? a.id : d.id,
+        name: hasKey(a, 'name') && isUserName(a.name) ? a.name : d.name,
+      } as const);
