diff --git experimental/packages/apps/event-schedule-app-shared/src/v6/types/record/base/year-month-date.mts apps/event-schedule-app-shared/src/v6/types/record/base/year-month-date.mts
index 2be90e165..c5f865908 100644
--- experimental/packages/apps/event-schedule-app-shared/src/v6/types/record/base/year-month-date.mts
+++ apps/event-schedule-app-shared/src/v6/types/record/base/year-month-date.mts
@@ -1,63 +1,59 @@
-import {
-  DateUtils,
-  Num,
-  Obj,
-  isNumber,
-  isRecord,
-  pipe,
-  type DateType,
-} from '@noshiro/ts-utils';
+import { Num, isNumber, isRecord, pipe } from 'ts-data-forge';
+import { DateUtils } from 'ts-fortress-types';
+import { type DateEnum, type MonthEnum, type SafeUint } from 'ts-type-forge';
+import { hasKeyValue } from '../../../../utils/index.mjs';
 
 export type YearMonthDate = Readonly<{
-  year: YearEnum;
+  year: SafeUint;
   month: MonthEnum;
   date: DateEnum;
 }>;
 
 export const yearMonthDateDefaultValue = {
-  year: pipe(DateUtils.today()).chain(DateUtils.getLocaleYear).value,
-  month: pipe(DateUtils.today()).chain(DateUtils.getLocaleMonth).value,
-  date: pipe(DateUtils.today()).chain(DateUtils.getLocaleDate).value,
+  year: pipe(DateUtils.today()).map(DateUtils.getLocaleYear).value,
+  month: pipe(DateUtils.today()).map(DateUtils.getLocaleMonth).value,
+  date: pipe(DateUtils.today()).map(DateUtils.getLocaleDate).value,
 } as const satisfies YearMonthDate;
 
-export const isYearEnum = (a: unknown): a is YearEnum =>
-  isNumber(a) && Number.isInteger(a) && a > 0;
+export const isYearEnum = (a: unknown): a is SafeUint =>
+  isNumber(a) && Number.isSafeInteger(a) && a > 0;
 
 export const isMonthEnum = (a: unknown): a is MonthEnum =>
-  isNumber(a) && Number.isInteger(a) && Num.isInRangeInclusive(1, 12)(a);
+  isNumber(a) && Number.isSafeInteger(a) && Num.isInRangeInclusive(1, 12)(a);
 
 export const isDateEnum = (a: unknown): a is DateEnum =>
-  isNumber(a) && Number.isInteger(a) && Num.isInRangeInclusive(1, 31)(a);
+  isNumber(a) && Number.isSafeInteger(a) && Num.isInRangeInclusive(1, 31)(a);
 
 export const isYearMonthDate = (a: unknown): a is YearMonthDate =>
   isRecord(a) &&
-  Obj.hasKeyValue(a, 'year', isYearEnum) &&
-  Obj.hasKeyValue(a, 'month', isMonthEnum) &&
-  Obj.hasKeyValue(a, 'date', isDateEnum);
+  hasKeyValue(a, 'year', isYearEnum) &&
+  hasKeyValue(a, 'month', isMonthEnum) &&
+  hasKeyValue(a, 'date', isDateEnum);
 
 const d = yearMonthDateDefaultValue;
 
 export const fillYearMonthDate = (a?: unknown): YearMonthDate =>
   a === undefined || !isRecord(a)
     ? d
-    : {
-        year: Obj.hasKeyValue(a, 'year', isYearEnum) ? a.year : d.year,
-        month: Obj.hasKeyValue(a, 'month', isMonthEnum) ? a.month : d.month,
-        date: Obj.hasKeyValue(a, 'date', isDateEnum) ? a.date : d.date,
-      };
-
-export const ymdFromDate = (date: DateType): YearMonthDate => ({
-  year: DateUtils.getLocaleYear(date),
-  month: DateUtils.getLocaleMonth(date),
-  date: DateUtils.getLocaleDate(date),
-});
+    : ({
+        year: hasKeyValue(a, 'year', isYearEnum) ? a.year : d.year,
+        month: hasKeyValue(a, 'month', isMonthEnum) ? a.month : d.month,
+        date: hasKeyValue(a, 'date', isDateEnum) ? a.date : d.date,
+      } as const);
+
+export const ymdFromDate = (date: Date): YearMonthDate =>
+  ({
+    year: DateUtils.getLocaleYear(date),
+    month: DateUtils.getLocaleMonth(date),
+    date: DateUtils.getLocaleDate(date),
+  }) as const;
 
 export const compareYmd = (a: YearMonthDate, b: YearMonthDate): -1 | 0 | 1 => {
-  if (a.year !== b.year)
-    return Num.mapNaN2Undefined(Math.sign(a.year - b.year)) ?? 0;
-  if (a.month !== b.month)
-    return Num.mapNaN2Undefined(Math.sign(a.month - b.month)) ?? 0;
-  if (a.date !== b.date)
-    return Num.mapNaN2Undefined(Math.sign(a.date - b.date)) ?? 0;
+  if (a.year !== b.year) return a.year < b.year ? -1 : 1;
+
+  if (a.month !== b.month) return a.month < b.month ? -1 : 1;
+
+  if (a.date !== b.date) return a.date < b.date ? -1 : 1;
+
   return 0;
 };
