diff --git experimental/packages/apps/event-schedule-app/src/utils/split-string-by-whitespace.test.ts apps/event-schedule-app/src/utils/split-string-by-whitespace.test.mts
index 351e23b5c..6f6ec4d1d 100644
--- experimental/packages/apps/event-schedule-app/src/utils/split-string-by-whitespace.test.ts
+++ apps/event-schedule-app/src/utils/split-string-by-whitespace.test.mts
@@ -1,4 +1,4 @@
-import { splitStringByWhitespace } from './split-string-by-whitespace';
+import { splitStringByWhitespace } from './split-string-by-whitespace.mjs';
 
 describe('splitStringByWhitespace', () => {
   test('1 whitespace', () => {
@@ -22,18 +22,16 @@ describe('splitStringByWhitespace', () => {
   test('2 sequential whitespaces', () => {
     expect(splitStringByWhitespace('aaa     bbb    ccc')).toStrictEqual([
       'aaa',
-      '     ',
+      ' '.repeat(5),
       'bbb',
-      '    ',
+      ' '.repeat(4),
       'ccc',
     ]);
   });
 
   test('2 sequential whitespaces and newline', () => {
     expect(
-      splitStringByWhitespace(`aaa
-    bbb
-    ccc`),
+      splitStringByWhitespace(['aaa', 'bbb', 'ccc'].join('\n    ')),
     ).toStrictEqual(['aaa', '\n    ', 'bbb', '\n    ', 'ccc']);
   });
 });
