diff --git experimental/packages/apps/event-schedule-app/src/env.ts apps/event-schedule-app/src/env.mts
index ff3c2eab4..f08dce947 100644
--- experimental/packages/apps/event-schedule-app/src/env.ts
+++ apps/event-schedule-app/src/env.mts
@@ -1,9 +1,12 @@
+import { type ReadonlyRecord } from 'ts-type-forge';
+
 export const isProduction = import.meta.env.PROD;
+
 export const isDevelopment = import.meta.env.DEV;
 
 export const useEmulators: boolean = true;
 
-export const experimentalFeature = {} as const satisfies Record<
+export const experimentalFeature = {} as const satisfies ReadonlyRecord<
   never,
   'hidden' | 'shown'
 >;
