diff --git experimental/packages/apps/event-schedule-app/src/components/organisms/detailed-filter/detailed-filter-filled-date-only.tsx apps/event-schedule-app/src/components/organisms/detailed-filter/detailed-filter-filled-date-only.tsx
index 6c69eed1d..44ee66520 100644
--- experimental/packages/apps/event-schedule-app/src/components/organisms/detailed-filter/detailed-filter-filled-date-only.tsx
+++ apps/event-schedule-app/src/components/organisms/detailed-filter/detailed-filter-filled-date-only.tsx
@@ -1,12 +1,11 @@
-import { CheckboxView } from '@noshiro/react-blueprintjs-utils';
-import { AnswerFilterAndSortStore } from '../../../store';
-import { CheckboxWrapper, FilterItem } from './styled';
+import { CheckboxView } from 'react-blueprintjs-utils';
+import { memoNamed } from 'react-utils';
+import { AnswerFilterAndSortStore } from '../../../store/index.mjs';
+import { CheckboxWrapper, FilterItem } from './styled.js';
 
 const dc = dict.answerPage.detailedFilter;
 
-type Props = Readonly<{
-  filledDateOnly: boolean;
-}>;
+type Props = Readonly<{ filledDateOnly: boolean }>;
 
 /* 全員回答済みの候補日のみ表示 */
 export const DetailedFilterFilledDateOnly = memoNamed<Props>(
