diff --git experimental/packages/apps/event-schedule-app/src/components/organisms/detailed-filter/detailed-filter-num-good-plus-fair-icon.tsx apps/event-schedule-app/src/components/organisms/detailed-filter/detailed-filter-num-good-plus-fair-icon.tsx
index a897cff80..3a1841524 100644
--- experimental/packages/apps/event-schedule-app/src/components/organisms/detailed-filter/detailed-filter-num-good-plus-fair-icon.tsx
+++ apps/event-schedule-app/src/components/organisms/detailed-filter/detailed-filter-num-good-plus-fair-icon.tsx
@@ -1,5 +1,7 @@
-import { AnswerFilterAndSortStore } from '../../../store';
-import { DetailedFilterNumIcon } from './detailed-filter-num-icon';
+import { memoNamed } from 'react-utils';
+import { type SafeUint } from 'ts-type-forge';
+import { AnswerFilterAndSortStore } from '../../../store/index.mjs';
+import { DetailedFilterNumIcon } from './detailed-filter-num-icon.js';
 
 type Props = Readonly<{
   enabled: boolean;
