diff --git experimental/packages/apps/event-schedule-app/src/components/organisms/detailed-filter/styled.tsx apps/event-schedule-app/src/components/organisms/detailed-filter/styled.tsx
index e1c025e73..9b0859a71 100644
--- experimental/packages/apps/event-schedule-app/src/components/organisms/detailed-filter/styled.tsx
+++ apps/event-schedule-app/src/components/organisms/detailed-filter/styled.tsx
@@ -1,4 +1,6 @@
 import { Icon } from '@blueprintjs/core';
+import styled from '@emotion/styled';
+import { memoNamed } from 'react-utils';
 
 export const FilterItem = styled.div`
   margin: 5px;
