diff --git a/packages/v5.9/output/lib-files/lib.esnext.array.d.ts b/packages/v6.0/output/lib-files/lib.esnext.array.d.ts
index 9b08357d..27ef5bf3 100644
--- a/packages/v5.9/output/lib-files/lib.esnext.array.d.ts
+++ b/packages/v6.0/output/lib-files/lib.esnext.array.d.ts
@@ -1,5 +1,3 @@
-/// <reference no-default-lib="true"/>
-
 interface ArrayConstructor {
   /**
    * Creates an array from an async iterator or iterable object.
