diff --git a/packages/v5.9/output/lib-files/lib.es2024.collection.d.ts b/packages/v6.0/output/lib-files/lib.es2024.collection.d.ts
index 956f3bc2..df03a102 100644
--- a/packages/v5.9/output/lib-files/lib.es2024.collection.d.ts
+++ b/packages/v6.0/output/lib-files/lib.es2024.collection.d.ts
@@ -1,5 +1,3 @@
-/// <reference no-default-lib="true"/>
-
 interface MapConstructor {
   /**
    * Groups members of an iterable according to the return value of the passed callback.
