diff --git a/packages/v5.9/output/lib-files/lib.es2021.weakref.d.ts b/packages/v6.0/output/lib-files/lib.es2021.weakref.d.ts
index f02bb0a0..89450d9b 100644
--- a/packages/v5.9/output/lib-files/lib.es2021.weakref.d.ts
+++ b/packages/v6.0/output/lib-files/lib.es2021.weakref.d.ts
@@ -1,5 +1,3 @@
-/// <reference no-default-lib="true"/>
-
 /// <reference path="./lib.es2015.symbol.wellknown.d.ts" />
 
 interface WeakRef<T extends WeakKey> {
