diff --git a/packages/v5.9/output/lib-files/lib.es2020.promise.d.ts b/packages/v6.0/output/lib-files/lib.es2020.promise.d.ts
index a6df09f8..d98d906f 100644
--- a/packages/v5.9/output/lib-files/lib.es2020.promise.d.ts
+++ b/packages/v6.0/output/lib-files/lib.es2020.promise.d.ts
@@ -1,5 +1,3 @@
-/// <reference no-default-lib="true"/>
-
 interface PromiseFulfilledResult<T> {
   readonly status: 'fulfilled';
   readonly value: T;
