diff --git a/packages/v5.9/output/lib-files/lib.es2015.symbol.wellknown.d.ts b/packages/v6.0/output/lib-files/lib.es2015.symbol.wellknown.d.ts
index 33e0ec53..38beeea9 100644
--- a/packages/v5.9/output/lib-files/lib.es2015.symbol.wellknown.d.ts
+++ b/packages/v6.0/output/lib-files/lib.es2015.symbol.wellknown.d.ts
@@ -1,5 +1,3 @@
-/// <reference no-default-lib="true"/>
-
 /// <reference path="./lib.es2015.symbol.d.ts" />
 
 interface SymbolConstructor {
