diff --git a/packages/v5.9/output/lib-files/lib.es2017.arraybuffer.d.ts b/packages/v6.0/output/lib-files/lib.es2017.arraybuffer.d.ts
index 881c9974..beec3703 100644
--- a/packages/v5.9/output/lib-files/lib.es2017.arraybuffer.d.ts
+++ b/packages/v6.0/output/lib-files/lib.es2017.arraybuffer.d.ts
@@ -1,5 +1,3 @@
-/// <reference no-default-lib="true"/>
-
 interface ArrayBufferConstructor {
   new (): ArrayBuffer;
 }
