diff --git a/packages/v5.9/output/lib-files/lib.es2019.object.d.ts b/packages/v6.0/output/lib-files/lib.es2019.object.d.ts
index 6aa54d1c..933560e4 100644
--- a/packages/v5.9/output/lib-files/lib.es2019.object.d.ts
+++ b/packages/v6.0/output/lib-files/lib.es2019.object.d.ts
@@ -1,6 +1,3 @@
-/// <reference no-default-lib="true"/>
-/// <reference lib="es5" />
-
 /// <reference path="./lib.es2015.iterable.d.ts" />
 
 declare namespace StrictLibInternals {
