diff --git experimental/packages/apps/event-schedule-app/src/components/organisms/button-with-confirm/archive-event-button.tsx apps/event-schedule-app/src/components/organisms/button-with-confirm/archive-event-button.tsx
index bc3eefc99..fa7be1da5 100644
--- experimental/packages/apps/event-schedule-app/src/components/organisms/button-with-confirm/archive-event-button.tsx
+++ apps/event-schedule-app/src/components/organisms/button-with-confirm/archive-event-button.tsx
@@ -1,7 +1,10 @@
 import { Button } from '@blueprintjs/core';
-import { AlertWithMaxWidth } from '@noshiro/react-blueprintjs-utils';
-import { eventListItemTextColor } from '../../../constants';
-import { Description } from '../../atoms';
+import { useBoolState } from 'better-react-use-state';
+import * as React from 'react';
+import { AlertWithMaxWidth } from 'react-blueprintjs-utils';
+import { memoNamed } from 'react-utils';
+import { eventListItemTextColor } from '../../../constants/index.mjs';
+import { Description } from '../../atoms/index.mjs';
 
 const dc = dict.eventListPage;
 
@@ -16,33 +19,39 @@ export const ArchiveEventButton = memoNamed<Props>(
     const [isOpen, { setTrue: handleOpen, setFalse: handleClose }] =
       useBoolState(false);
 
-    const archiveIconClick = useCallback(
+    const archiveIconClick = React.useCallback(
       // eslint-disable-next-line @typescript-eslint/prefer-readonly-parameter-types
       (ev: React.MouseEvent<HTMLElement>) => {
         ev.preventDefault();
+
         ev.stopPropagation();
+
         handleOpen();
       },
       [handleOpen],
     );
 
-    const onCancel = useCallback(
+    const onCancel = React.useCallback(
       // eslint-disable-next-line @typescript-eslint/prefer-readonly-parameter-types
       (ev?: React.SyntheticEvent<HTMLElement, Event>) => {
         ev?.preventDefault();
+
         ev?.stopPropagation();
+
         handleClose();
       },
       [handleClose],
     );
 
-    const onConfirmHandler = useCallback(
+    const onConfirmHandler = React.useCallback(
       // eslint-disable-next-line @typescript-eslint/prefer-readonly-parameter-types
       (ev?: React.SyntheticEvent<HTMLElement, Event>) => {
         ev?.preventDefault();
+
         ev?.stopPropagation();
 
         handleClose();
+
         onConfirm();
       },
       [onConfirm, handleClose],
@@ -53,13 +62,13 @@ export const ArchiveEventButton = memoNamed<Props>(
         <Button
           color={eventListItemTextColor}
           icon={archiveOrUnArchive}
-          minimal={true}
           title={dc[archiveOrUnArchive]}
+          variant={'minimal'}
           onClick={archiveIconClick}
         />
         <AlertWithMaxWidth
-          canEscapeKeyCancel={true}
-          canOutsideClickCancel={true}
+          canEscapeKeyCancel
+          canOutsideClickCancel
           cancelButtonText={dict.common.buttonText.cancel}
           confirmButtonText={dc[archiveOrUnArchive]}
           icon={archiveOrUnArchive}
