diff --git a/packages/v5.7/output/lib-files/lib.es2017.arraybuffer.d.ts b/packages/v5.7/output/lib-files/lib.es2017.arraybuffer.d.ts
new file mode 100644
index 0000000..93e10b3
--- /dev/null
+++ b/packages/v5.7/output/lib-files/lib.es2017.arraybuffer.d.ts
@@ -0,0 +1,6 @@
+/// <reference no-default-lib="true"/>
+/// <reference types="ts-type-forge/global" />
+
+interface ArrayBufferConstructor {
+  new (): ArrayBuffer;
+}
