diff --git a/packages/v5.6/output/lib-files/lib.decorators.d.ts b/packages/v5.7/output/lib-files/lib.decorators.d.ts
index 79aa328..8654eb2 100644
--- a/packages/v5.6/output/lib-files/lib.decorators.d.ts
+++ b/packages/v5.7/output/lib-files/lib.decorators.d.ts
@@ -101,9 +101,9 @@ interface ClassMethodDecoratorContext<
   };
 
   /**
-   * Adds a callback to be invoked either before static initializers are run (when
-   * decorating a `static` element), or before instance initializers are run (when
-   * decorating a non-`static` element).
+   * Adds a callback to be invoked either after static methods are defined but before
+   * static initializers are run (when decorating a `static` element), or before instance
+   * initializers are run (when decorating a non-`static` element).
    *
    * @example
    * ```ts
@@ -164,9 +164,9 @@ interface ClassGetterDecoratorContext<This = unknown, Value = unknown> {
   };
 
   /**
-   * Adds a callback to be invoked either before static initializers are run (when
-   * decorating a `static` element), or before instance initializers are run (when
-   * decorating a non-`static` element).
+   * Adds a callback to be invoked either after static methods are defined but before
+   * static initializers are run (when decorating a `static` element), or before instance
+   * initializers are run (when decorating a non-`static` element).
    */
   addInitializer(initializer: (this: This) => void): void;
 
@@ -208,9 +208,9 @@ interface ClassSetterDecoratorContext<This = unknown, Value = unknown> {
   };
 
   /**
-   * Adds a callback to be invoked either before static initializers are run (when
-   * decorating a `static` element), or before instance initializers are run (when
-   * decorating a non-`static` element).
+   * Adds a callback to be invoked either after static methods are defined but before
+   * static initializers are run (when decorating a `static` element), or before instance
+   * initializers are run (when decorating a non-`static` element).
    */
   addInitializer(initializer: (this: This) => void): void;
 
@@ -261,9 +261,8 @@ interface ClassAccessorDecoratorContext<This = unknown, Value = unknown> {
   };
 
   /**
-   * Adds a callback to be invoked either before static initializers are run (when
-   * decorating a `static` element), or before instance initializers are run (when
-   * decorating a non-`static` element).
+   * Adds a callback to be invoked immediately after the auto `accessor` being
+   * decorated is initialized (regardless if the `accessor` is `static` or not).
    */
   addInitializer(initializer: (this: This) => void): void;
 
@@ -355,9 +354,8 @@ interface ClassFieldDecoratorContext<This = unknown, Value = unknown> {
   };
 
   /**
-   * Adds a callback to be invoked either before static initializers are run (when
-   * decorating a `static` element), or before instance initializers are run (when
-   * decorating a non-`static` element).
+   * Adds a callback to be invoked immediately after the field being decorated
+   * is initialized (regardless if the field is `static` or not).
    */
   addInitializer(initializer: (this: This) => void): void;
 
