diff --git a/packages/v5.6/output/lib-files/lib.es2016.array.include.d.ts b/packages/v5.7/output/lib-files/lib.es2016.array.include.d.ts
index 9b1677c..95efd9d 100644
--- a/packages/v5.6/output/lib-files/lib.es2016.array.include.d.ts
+++ b/packages/v5.7/output/lib-files/lib.es2016.array.include.d.ts
@@ -25,7 +25,7 @@ interface ReadonlyArray<T> {
   ): searchElement is T;
 }
 
-interface Int8Array {
+interface Int8Array<TArrayBuffer extends ArrayBufferLike> {
   /**
    * Determines whether an array includes a certain element, returning true or false as appropriate.
    * @param searchElement The element to search for.
@@ -34,7 +34,7 @@ interface Int8Array {
   includes(searchElement: Int8, fromIndex?: number): boolean;
 }
 
-interface Uint8Array {
+interface Uint8Array<TArrayBuffer extends ArrayBufferLike> {
   /**
    * Determines whether an array includes a certain element, returning true or false as appropriate.
    * @param searchElement The element to search for.
@@ -43,7 +43,7 @@ interface Uint8Array {
   includes(searchElement: Uint8, fromIndex?: number): boolean;
 }
 
-interface Uint8ClampedArray {
+interface Uint8ClampedArray<TArrayBuffer extends ArrayBufferLike> {
   /**
    * Determines whether an array includes a certain element, returning true or false as appropriate.
    * @param searchElement The element to search for.
@@ -52,7 +52,7 @@ interface Uint8ClampedArray {
   includes(searchElement: Uint8, fromIndex?: number): boolean;
 }
 
-interface Int16Array {
+interface Int16Array<TArrayBuffer extends ArrayBufferLike> {
   /**
    * Determines whether an array includes a certain element, returning true or false as appropriate.
    * @param searchElement The element to search for.
@@ -61,7 +61,7 @@ interface Int16Array {
   includes(searchElement: number, fromIndex?: number): boolean;
 }
 
-interface Uint16Array {
+interface Uint16Array<TArrayBuffer extends ArrayBufferLike> {
   /**
    * Determines whether an array includes a certain element, returning true or false as appropriate.
    * @param searchElement The element to search for.
@@ -70,7 +70,7 @@ interface Uint16Array {
   includes(searchElement: number, fromIndex?: number): boolean;
 }
 
-interface Int32Array {
+interface Int32Array<TArrayBuffer extends ArrayBufferLike> {
   /**
    * Determines whether an array includes a certain element, returning true or false as appropriate.
    * @param searchElement The element to search for.
@@ -79,7 +79,7 @@ interface Int32Array {
   includes(searchElement: number, fromIndex?: number): boolean;
 }
 
-interface Uint32Array {
+interface Uint32Array<TArrayBuffer extends ArrayBufferLike> {
   /**
    * Determines whether an array includes a certain element, returning true or false as appropriate.
    * @param searchElement The element to search for.
@@ -88,7 +88,7 @@ interface Uint32Array {
   includes(searchElement: number, fromIndex?: number): boolean;
 }
 
-interface Float32Array {
+interface Float32Array<TArrayBuffer extends ArrayBufferLike> {
   /**
    * Determines whether an array includes a certain element, returning true or false as appropriate.
    * @param searchElement The element to search for.
@@ -97,7 +97,7 @@ interface Float32Array {
   includes(searchElement: number, fromIndex?: number): boolean;
 }
 
-interface Float64Array {
+interface Float64Array<TArrayBuffer extends ArrayBufferLike> {
   /**
    * Determines whether an array includes a certain element, returning true or false as appropriate.
    * @param searchElement The element to search for.
