diff --git a/packages/v5.6/output/lib-files/lib.es2023.array.d.ts b/packages/v5.7/output/lib-files/lib.es2023.array.d.ts
index 342e8bb..b5002b3 100644
--- a/packages/v5.6/output/lib-files/lib.es2023.array.d.ts
+++ b/packages/v5.7/output/lib-files/lib.es2023.array.d.ts
@@ -157,7 +157,7 @@ interface ReadonlyArray<T> {
   with(index: number, value: T): T[];
 }
 
-interface Int8Array {
+interface Int8Array<TArrayBuffer extends ArrayBufferLike> {
   /**
    * Returns the value of the last element in the array where predicate is true, and undefined
    * otherwise.
@@ -168,11 +168,11 @@ interface Int8Array {
    * predicate. If it is not provided, undefined is used instead.
    */
   findLast<S extends Int8>(
-    predicate: (value: Int8, index: number, array: Int8Array) => value is S,
+    predicate: (value: Int8, index: number, array: this) => value is S,
     thisArg?: unknown,
   ): S | undefined;
   findLast(
-    predicate: (value: Int8, index: number, array: Int8Array) => boolean,
+    predicate: (value: Int8, index: number, array: this) => boolean,
     thisArg?: unknown,
   ): Int8 | undefined;
 
@@ -186,14 +186,14 @@ interface Int8Array {
    * predicate. If it is not provided, undefined is used instead.
    */
   findLastIndex(
-    predicate: (value: Int8, index: number, array: Int8Array) => boolean,
+    predicate: (value: Int8, index: number, array: this) => boolean,
     thisArg?: unknown,
   ): number;
 
   /**
    * Returns a copy of an array with its elements reversed.
    */
-  toReversed(): Int8Array;
+  toReversed(): Int8Array<ArrayBuffer>;
 
   /**
    * Returns a copy of an array with its elements sorted.
@@ -201,11 +201,11 @@ interface Int8Array {
    * a negative value if the first argument is less than the second argument, zero if they're equal, and a positive
    * value otherwise. If omitted, the elements are sorted in ascending order.
    * ```ts
-   * const myNums = Int8Array.from([11, 2, 22, 1]);
-   * myNums.toSorted((a, b) => a - b) // Int8Array(4) [1, 2, 11, 22]
+   * const myNums = Int8Array<Buffer>.from([11, 2, 22, 1]);
+   * myNums.toSorted((a, b) => a - b) // Int8Array<Buffer>(4) [1, 2, 11, 22]
    * ```
    */
-  toSorted(compareFn?: (a: Int8, b: Int8) => number): Int8Array;
+  toSorted(compareFn?: (a: Int8, b: Int8) => number): Int8Array<ArrayBuffer>;
 
   /**
    * Copies an array, then overwrites the value at the provided index with the
@@ -216,10 +216,10 @@ interface Int8Array {
    * @param value The value to write into the copied array.
    * @returns The copied array with the updated value.
    */
-  with(index: number, value: Int8): Int8Array;
+  with(index: number, value: Int8): Int8Array<ArrayBuffer>;
 }
 
-interface Uint8Array {
+interface Uint8Array<TArrayBuffer extends ArrayBufferLike> {
   /**
    * Returns the value of the last element in the array where predicate is true, and undefined
    * otherwise.
@@ -230,11 +230,11 @@ interface Uint8Array {
    * predicate. If it is not provided, undefined is used instead.
    */
   findLast<S extends Uint8>(
-    predicate: (value: Uint8, index: number, array: Uint8Array) => value is S,
+    predicate: (value: Uint8, index: number, array: this) => value is S,
     thisArg?: unknown,
   ): S | undefined;
   findLast(
-    predicate: (value: Uint8, index: number, array: Uint8Array) => boolean,
+    predicate: (value: Uint8, index: number, array: this) => boolean,
     thisArg?: unknown,
   ): Uint8 | undefined;
 
@@ -248,14 +248,14 @@ interface Uint8Array {
    * predicate. If it is not provided, undefined is used instead.
    */
   findLastIndex(
-    predicate: (value: Uint8, index: number, array: Uint8Array) => boolean,
+    predicate: (value: Uint8, index: number, array: this) => boolean,
     thisArg?: unknown,
   ): number;
 
   /**
    * Returns a copy of an array with its elements reversed.
    */
-  toReversed(): Uint8Array;
+  toReversed(): Uint8Array<ArrayBuffer>;
 
   /**
    * Returns a copy of an array with its elements sorted.
@@ -263,11 +263,11 @@ interface Uint8Array {
    * a negative value if the first argument is less than the second argument, zero if they're equal, and a positive
    * value otherwise. If omitted, the elements are sorted in ascending order.
    * ```ts
-   * const myNums = Uint8Array.from([11, 2, 22, 1]);
-   * myNums.toSorted((a, b) => a - b) // Uint8Array(4) [1, 2, 11, 22]
+   * const myNums = Uint8Array<Buffer>.from([11, 2, 22, 1]);
+   * myNums.toSorted((a, b) => a - b) // Uint8Array<Buffer>(4) [1, 2, 11, 22]
    * ```
    */
-  toSorted(compareFn?: (a: Uint8, b: Uint8) => number): Uint8Array;
+  toSorted(compareFn?: (a: Uint8, b: Uint8) => number): Uint8Array<ArrayBuffer>;
 
   /**
    * Copies an array, then overwrites the value at the provided index with the
@@ -278,10 +278,10 @@ interface Uint8Array {
    * @param value The value to write into the copied array.
    * @returns The copied array with the updated value.
    */
-  with(index: number, value: Uint8): Uint8Array;
+  with(index: number, value: Uint8): Uint8Array<ArrayBuffer>;
 }
 
-interface Uint8ClampedArray {
+interface Uint8ClampedArray<TArrayBuffer extends ArrayBufferLike> {
   /**
    * Returns the value of the last element in the array where predicate is true, and undefined
    * otherwise.
@@ -292,19 +292,11 @@ interface Uint8ClampedArray {
    * predicate. If it is not provided, undefined is used instead.
    */
   findLast<S extends Uint8>(
-    predicate: (
-      value: Uint8,
-      index: number,
-      array: Uint8ClampedArray,
-    ) => value is S,
+    predicate: (value: Uint8, index: number, array: this) => value is S,
     thisArg?: unknown,
   ): S | undefined;
   findLast(
-    predicate: (
-      value: Uint8,
-      index: number,
-      array: Uint8ClampedArray,
-    ) => boolean,
+    predicate: (value: Uint8, index: number, array: this) => boolean,
     thisArg?: unknown,
   ): Uint8 | undefined;
 
@@ -318,18 +310,14 @@ interface Uint8ClampedArray {
    * predicate. If it is not provided, undefined is used instead.
    */
   findLastIndex(
-    predicate: (
-      value: Uint8,
-      index: number,
-      array: Uint8ClampedArray,
-    ) => boolean,
+    predicate: (value: Uint8, index: number, array: this) => boolean,
     thisArg?: unknown,
   ): number;
 
   /**
    * Returns a copy of an array with its elements reversed.
    */
-  toReversed(): Uint8ClampedArray;
+  toReversed(): Uint8ClampedArray<ArrayBuffer>;
 
   /**
    * Returns a copy of an array with its elements sorted.
@@ -337,11 +325,13 @@ interface Uint8ClampedArray {
    * a negative value if the first argument is less than the second argument, zero if they're equal, and a positive
    * value otherwise. If omitted, the elements are sorted in ascending order.
    * ```ts
-   * const myNums = Uint8ClampedArray.from([11, 2, 22, 1]);
-   * myNums.toSorted((a, b) => a - b) // Uint8ClampedArray(4) [1, 2, 11, 22]
+   * const myNums = Uint8ClampedArray<Buffer>.from([11, 2, 22, 1]);
+   * myNums.toSorted((a, b) => a - b) // Uint8ClampedArray<Buffer>(4) [1, 2, 11, 22]
    * ```
    */
-  toSorted(compareFn?: (a: Uint8, b: Uint8) => number): Uint8ClampedArray;
+  toSorted(
+    compareFn?: (a: Uint8, b: Uint8) => number,
+  ): Uint8ClampedArray<ArrayBuffer>;
 
   /**
    * Copies an array, then overwrites the value at the provided index with the
@@ -352,10 +342,10 @@ interface Uint8ClampedArray {
    * @param value The value to write into the copied array.
    * @returns The copied array with the updated value.
    */
-  with(index: number, value: Uint8): Uint8ClampedArray;
+  with(index: number, value: Uint8): Uint8ClampedArray<ArrayBuffer>;
 }
 
-interface Int16Array {
+interface Int16Array<TArrayBuffer extends ArrayBufferLike> {
   /**
    * Returns the value of the last element in the array where predicate is true, and undefined
    * otherwise.
@@ -366,11 +356,11 @@ interface Int16Array {
    * predicate. If it is not provided, undefined is used instead.
    */
   findLast<S extends number>(
-    predicate: (value: number, index: number, array: Int16Array) => value is S,
+    predicate: (value: number, index: number, array: this) => value is S,
     thisArg?: unknown,
   ): S | undefined;
   findLast(
-    predicate: (value: number, index: number, array: Int16Array) => boolean,
+    predicate: (value: number, index: number, array: this) => boolean,
     thisArg?: unknown,
   ): number | undefined;
 
@@ -384,14 +374,14 @@ interface Int16Array {
    * predicate. If it is not provided, undefined is used instead.
    */
   findLastIndex(
-    predicate: (value: number, index: number, array: Int16Array) => boolean,
+    predicate: (value: number, index: number, array: this) => boolean,
     thisArg?: unknown,
   ): number;
 
   /**
    * Returns a copy of an array with its elements reversed.
    */
-  toReversed(): Int16Array;
+  toReversed(): Int16Array<ArrayBuffer>;
 
   /**
    * Returns a copy of an array with its elements sorted.
@@ -399,11 +389,13 @@ interface Int16Array {
    * a negative value if the first argument is less than the second argument, zero if they're equal, and a positive
    * value otherwise. If omitted, the elements are sorted in ascending order.
    * ```ts
-   * const myNums = Int16Array.from([11, 2, -22, 1]);
-   * myNums.toSorted((a, b) => a - b) // Int16Array(4) [-22, 1, 2, 11]
+   * const myNums = Int16Array<Buffer>.from([11, 2, -22, 1]);
+   * myNums.toSorted((a, b) => a - b) // Int16Array<Buffer>(4) [-22, 1, 2, 11]
    * ```
    */
-  toSorted(compareFn?: (a: number, b: number) => number): Int16Array;
+  toSorted(
+    compareFn?: (a: number, b: number) => number,
+  ): Int16Array<ArrayBuffer>;
 
   /**
    * Copies an array, then overwrites the value at the provided index with the
@@ -414,10 +406,10 @@ interface Int16Array {
    * @param value The value to write into the copied array.
    * @returns The copied array with the updated value.
    */
-  with(index: number, value: number): Int16Array;
+  with(index: number, value: number): Int16Array<ArrayBuffer>;
 }
 
-interface Uint16Array {
+interface Uint16Array<TArrayBuffer extends ArrayBufferLike> {
   /**
    * Returns the value of the last element in the array where predicate is true, and undefined
    * otherwise.
@@ -428,11 +420,11 @@ interface Uint16Array {
    * predicate. If it is not provided, undefined is used instead.
    */
   findLast<S extends number>(
-    predicate: (value: number, index: number, array: Uint16Array) => value is S,
+    predicate: (value: number, index: number, array: this) => value is S,
     thisArg?: unknown,
   ): S | undefined;
   findLast(
-    predicate: (value: number, index: number, array: Uint16Array) => boolean,
+    predicate: (value: number, index: number, array: this) => boolean,
     thisArg?: unknown,
   ): number | undefined;
 
@@ -446,14 +438,14 @@ interface Uint16Array {
    * predicate. If it is not provided, undefined is used instead.
    */
   findLastIndex(
-    predicate: (value: number, index: number, array: Uint16Array) => boolean,
+    predicate: (value: number, index: number, array: this) => boolean,
     thisArg?: unknown,
   ): number;
 
   /**
    * Returns a copy of an array with its elements reversed.
    */
-  toReversed(): Uint16Array;
+  toReversed(): Uint16Array<ArrayBuffer>;
 
   /**
    * Returns a copy of an array with its elements sorted.
@@ -461,11 +453,13 @@ interface Uint16Array {
    * a negative value if the first argument is less than the second argument, zero if they're equal, and a positive
    * value otherwise. If omitted, the elements are sorted in ascending order.
    * ```ts
-   * const myNums = Uint16Array.from([11, 2, 22, 1]);
-   * myNums.toSorted((a, b) => a - b) // Uint16Array(4) [1, 2, 11, 22]
+   * const myNums = Uint16Array<Buffer>.from([11, 2, 22, 1]);
+   * myNums.toSorted((a, b) => a - b) // Uint16Array<Buffer>(4) [1, 2, 11, 22]
    * ```
    */
-  toSorted(compareFn?: (a: number, b: number) => number): Uint16Array;
+  toSorted(
+    compareFn?: (a: number, b: number) => number,
+  ): Uint16Array<ArrayBuffer>;
 
   /**
    * Copies an array, then overwrites the value at the provided index with the
@@ -476,10 +470,10 @@ interface Uint16Array {
    * @param value The value to write into the copied array.
    * @returns The copied array with the updated value.
    */
-  with(index: number, value: number): Uint16Array;
+  with(index: number, value: number): Uint16Array<ArrayBuffer>;
 }
 
-interface Int32Array {
+interface Int32Array<TArrayBuffer extends ArrayBufferLike> {
   /**
    * Returns the value of the last element in the array where predicate is true, and undefined
    * otherwise.
@@ -490,11 +484,11 @@ interface Int32Array {
    * predicate. If it is not provided, undefined is used instead.
    */
   findLast<S extends number>(
-    predicate: (value: number, index: number, array: Int32Array) => value is S,
+    predicate: (value: number, index: number, array: this) => value is S,
     thisArg?: unknown,
   ): S | undefined;
   findLast(
-    predicate: (value: number, index: number, array: Int32Array) => boolean,
+    predicate: (value: number, index: number, array: this) => boolean,
     thisArg?: unknown,
   ): number | undefined;
 
@@ -508,14 +502,14 @@ interface Int32Array {
    * predicate. If it is not provided, undefined is used instead.
    */
   findLastIndex(
-    predicate: (value: number, index: number, array: Int32Array) => boolean,
+    predicate: (value: number, index: number, array: this) => boolean,
     thisArg?: unknown,
   ): number;
 
   /**
    * Returns a copy of an array with its elements reversed.
    */
-  toReversed(): Int32Array;
+  toReversed(): Int32Array<ArrayBuffer>;
 
   /**
    * Returns a copy of an array with its elements sorted.
@@ -523,11 +517,13 @@ interface Int32Array {
    * a negative value if the first argument is less than the second argument, zero if they're equal, and a positive
    * value otherwise. If omitted, the elements are sorted in ascending order.
    * ```ts
-   * const myNums = Int32Array.from([11, 2, -22, 1]);
-   * myNums.toSorted((a, b) => a - b) // Int32Array(4) [-22, 1, 2, 11]
+   * const myNums = Int32Array<Buffer>.from([11, 2, -22, 1]);
+   * myNums.toSorted((a, b) => a - b) // Int32Array<Buffer>(4) [-22, 1, 2, 11]
    * ```
    */
-  toSorted(compareFn?: (a: number, b: number) => number): Int32Array;
+  toSorted(
+    compareFn?: (a: number, b: number) => number,
+  ): Int32Array<ArrayBuffer>;
 
   /**
    * Copies an array, then overwrites the value at the provided index with the
@@ -538,10 +534,10 @@ interface Int32Array {
    * @param value The value to write into the copied array.
    * @returns The copied array with the updated value.
    */
-  with(index: number, value: number): Int32Array;
+  with(index: number, value: number): Int32Array<ArrayBuffer>;
 }
 
-interface Uint32Array {
+interface Uint32Array<TArrayBuffer extends ArrayBufferLike> {
   /**
    * Returns the value of the last element in the array where predicate is true, and undefined
    * otherwise.
@@ -552,11 +548,11 @@ interface Uint32Array {
    * predicate. If it is not provided, undefined is used instead.
    */
   findLast<S extends number>(
-    predicate: (value: number, index: number, array: Uint32Array) => value is S,
+    predicate: (value: number, index: number, array: this) => value is S,
     thisArg?: unknown,
   ): S | undefined;
   findLast(
-    predicate: (value: number, index: number, array: Uint32Array) => boolean,
+    predicate: (value: number, index: number, array: this) => boolean,
     thisArg?: unknown,
   ): number | undefined;
 
@@ -570,14 +566,14 @@ interface Uint32Array {
    * predicate. If it is not provided, undefined is used instead.
    */
   findLastIndex(
-    predicate: (value: number, index: number, array: Uint32Array) => boolean,
+    predicate: (value: number, index: number, array: this) => boolean,
     thisArg?: unknown,
   ): number;
 
   /**
    * Returns a copy of an array with its elements reversed.
    */
-  toReversed(): Uint32Array;
+  toReversed(): Uint32Array<ArrayBuffer>;
 
   /**
    * Returns a copy of an array with its elements sorted.
@@ -585,11 +581,13 @@ interface Uint32Array {
    * a negative value if the first argument is less than the second argument, zero if they're equal, and a positive
    * value otherwise. If omitted, the elements are sorted in ascending order.
    * ```ts
-   * const myNums = Uint32Array.from([11, 2, 22, 1]);
-   * myNums.toSorted((a, b) => a - b) // Uint32Array(4) [1, 2, 11, 22]
+   * const myNums = Uint32Array<Buffer>.from([11, 2, 22, 1]);
+   * myNums.toSorted((a, b) => a - b) // Uint32Array<Buffer>(4) [1, 2, 11, 22]
    * ```
    */
-  toSorted(compareFn?: (a: number, b: number) => number): Uint32Array;
+  toSorted(
+    compareFn?: (a: number, b: number) => number,
+  ): Uint32Array<ArrayBuffer>;
 
   /**
    * Copies an array, then overwrites the value at the provided index with the
@@ -600,10 +598,10 @@ interface Uint32Array {
    * @param value The value to write into the copied array.
    * @returns The copied array with the updated value.
    */
-  with(index: number, value: number): Uint32Array;
+  with(index: number, value: number): Uint32Array<ArrayBuffer>;
 }
 
-interface Float32Array {
+interface Float32Array<TArrayBuffer extends ArrayBufferLike> {
   /**
    * Returns the value of the last element in the array where predicate is true, and undefined
    * otherwise.
@@ -614,15 +612,11 @@ interface Float32Array {
    * predicate. If it is not provided, undefined is used instead.
    */
   findLast<S extends number>(
-    predicate: (
-      value: number,
-      index: number,
-      array: Float32Array,
-    ) => value is S,
+    predicate: (value: number, index: number, array: this) => value is S,
     thisArg?: unknown,
   ): S | undefined;
   findLast(
-    predicate: (value: number, index: number, array: Float32Array) => boolean,
+    predicate: (value: number, index: number, array: this) => boolean,
     thisArg?: unknown,
   ): number | undefined;
 
@@ -636,14 +630,14 @@ interface Float32Array {
    * predicate. If it is not provided, undefined is used instead.
    */
   findLastIndex(
-    predicate: (value: number, index: number, array: Float32Array) => boolean,
+    predicate: (value: number, index: number, array: this) => boolean,
     thisArg?: unknown,
   ): number;
 
   /**
    * Returns a copy of an array with its elements reversed.
    */
-  toReversed(): Float32Array;
+  toReversed(): Float32Array<ArrayBuffer>;
 
   /**
    * Returns a copy of an array with its elements sorted.
@@ -651,11 +645,13 @@ interface Float32Array {
    * a negative value if the first argument is less than the second argument, zero if they're equal, and a positive
    * value otherwise. If omitted, the elements are sorted in ascending order.
    * ```ts
-   * const myNums = Float32Array.from([11.25, 2, -22.5, 1]);
-   * myNums.toSorted((a, b) => a - b) // Float32Array(4) [-22.5, 1, 2, 11.5]
+   * const myNums = Float32Array<Buffer>.from([11.25, 2, -22.5, 1]);
+   * myNums.toSorted((a, b) => a - b) // Float32Array<Buffer>(4) [-22.5, 1, 2, 11.5]
    * ```
    */
-  toSorted(compareFn?: (a: number, b: number) => number): Float32Array;
+  toSorted(
+    compareFn?: (a: number, b: number) => number,
+  ): Float32Array<ArrayBuffer>;
 
   /**
    * Copies an array, then overwrites the value at the provided index with the
@@ -666,10 +662,10 @@ interface Float32Array {
    * @param value The value to write into the copied array.
    * @returns The copied array with the updated value.
    */
-  with(index: number, value: number): Float32Array;
+  with(index: number, value: number): Float32Array<ArrayBuffer>;
 }
 
-interface Float64Array {
+interface Float64Array<TArrayBuffer extends ArrayBufferLike> {
   /**
    * Returns the value of the last element in the array where predicate is true, and undefined
    * otherwise.
@@ -680,15 +676,11 @@ interface Float64Array {
    * predicate. If it is not provided, undefined is used instead.
    */
   findLast<S extends number>(
-    predicate: (
-      value: number,
-      index: number,
-      array: Float64Array,
-    ) => value is S,
+    predicate: (value: number, index: number, array: this) => value is S,
     thisArg?: unknown,
   ): S | undefined;
   findLast(
-    predicate: (value: number, index: number, array: Float64Array) => boolean,
+    predicate: (value: number, index: number, array: this) => boolean,
     thisArg?: unknown,
   ): number | undefined;
 
@@ -702,14 +694,14 @@ interface Float64Array {
    * predicate. If it is not provided, undefined is used instead.
    */
   findLastIndex(
-    predicate: (value: number, index: number, array: Float64Array) => boolean,
+    predicate: (value: number, index: number, array: this) => boolean,
     thisArg?: unknown,
   ): number;
 
   /**
    * Returns a copy of an array with its elements reversed.
    */
-  toReversed(): Float64Array;
+  toReversed(): Float64Array<ArrayBuffer>;
 
   /**
    * Returns a copy of an array with its elements sorted.
@@ -717,11 +709,13 @@ interface Float64Array {
    * a negative value if the first argument is less than the second argument, zero if they're equal, and a positive
    * value otherwise. If omitted, the elements are sorted in ascending order.
    * ```ts
-   * const myNums = Float64Array.from([11.25, 2, -22.5, 1]);
-   * myNums.toSorted((a, b) => a - b) // Float64Array(4) [-22.5, 1, 2, 11.5]
+   * const myNums = Float64Array<Buffer>.from([11.25, 2, -22.5, 1]);
+   * myNums.toSorted((a, b) => a - b) // Float64Array<Buffer>(4) [-22.5, 1, 2, 11.5]
    * ```
    */
-  toSorted(compareFn?: (a: number, b: number) => number): Float64Array;
+  toSorted(
+    compareFn?: (a: number, b: number) => number,
+  ): Float64Array<ArrayBuffer>;
 
   /**
    * Copies an array, then overwrites the value at the provided index with the
@@ -732,10 +726,10 @@ interface Float64Array {
    * @param value The value to write into the copied array.
    * @returns The copied array with the updated value.
    */
-  with(index: number, value: number): Float64Array;
+  with(index: number, value: number): Float64Array<ArrayBuffer>;
 }
 
-interface BigInt64Array {
+interface BigInt64Array<TArrayBuffer extends ArrayBufferLike> {
   /**
    * Returns the value of the last element in the array where predicate is true, and undefined
    * otherwise.
@@ -746,15 +740,11 @@ interface BigInt64Array {
    * predicate. If it is not provided, undefined is used instead.
    */
   findLast<S extends bigint>(
-    predicate: (
-      value: bigint,
-      index: number,
-      array: BigInt64Array,
-    ) => value is S,
+    predicate: (value: bigint, index: number, array: this) => value is S,
     thisArg?: unknown,
   ): S | undefined;
   findLast(
-    predicate: (value: bigint, index: number, array: BigInt64Array) => boolean,
+    predicate: (value: bigint, index: number, array: this) => boolean,
     thisArg?: unknown,
   ): bigint | undefined;
 
@@ -768,14 +758,14 @@ interface BigInt64Array {
    * predicate. If it is not provided, undefined is used instead.
    */
   findLastIndex(
-    predicate: (value: bigint, index: number, array: BigInt64Array) => boolean,
+    predicate: (value: bigint, index: number, array: this) => boolean,
     thisArg?: unknown,
   ): number;
 
   /**
    * Returns a copy of an array with its elements reversed.
    */
-  toReversed(): BigInt64Array;
+  toReversed(): BigInt64Array<ArrayBuffer>;
 
   /**
    * Returns a copy of an array with its elements sorted.
@@ -783,11 +773,13 @@ interface BigInt64Array {
    * a negative value if the first argument is less than the second argument, zero if they're equal, and a positive
    * value otherwise. If omitted, the elements are sorted in ascending order.
    * ```ts
-   * const myNums = BigInt64Array.from([11n, 2n, -22n, 1n]);
-   * myNums.toSorted((a, b) => Number(a - b)) // BigInt64Array(4) [-22n, 1n, 2n, 11n]
+   * const myNums = BigInt64Array<Buffer>.from([11n, 2n, -22n, 1n]);
+   * myNums.toSorted((a, b) => Number(a - b)) // BigInt64Array<Buffer>(4) [-22n, 1n, 2n, 11n]
    * ```
    */
-  toSorted(compareFn?: (a: bigint, b: bigint) => number): BigInt64Array;
+  toSorted(
+    compareFn?: (a: bigint, b: bigint) => number,
+  ): BigInt64Array<ArrayBuffer>;
 
   /**
    * Copies an array, then overwrites the value at the provided index with the
@@ -798,10 +790,10 @@ interface BigInt64Array {
    * @param value The value to write into the copied array.
    * @returns The copied array with the updated value.
    */
-  with(index: number, value: bigint): BigInt64Array;
+  with(index: number, value: bigint): BigInt64Array<ArrayBuffer>;
 }
 
-interface BigUint64Array {
+interface BigUint64Array<TArrayBuffer extends ArrayBufferLike> {
   /**
    * Returns the value of the last element in the array where predicate is true, and undefined
    * otherwise.
@@ -812,15 +804,11 @@ interface BigUint64Array {
    * predicate. If it is not provided, undefined is used instead.
    */
   findLast<S extends bigint>(
-    predicate: (
-      value: bigint,
-      index: number,
-      array: BigUint64Array,
-    ) => value is S,
+    predicate: (value: bigint, index: number, array: this) => value is S,
     thisArg?: unknown,
   ): S | undefined;
   findLast(
-    predicate: (value: bigint, index: number, array: BigUint64Array) => boolean,
+    predicate: (value: bigint, index: number, array: this) => boolean,
     thisArg?: unknown,
   ): bigint | undefined;
 
@@ -834,14 +822,14 @@ interface BigUint64Array {
    * predicate. If it is not provided, undefined is used instead.
    */
   findLastIndex(
-    predicate: (value: bigint, index: number, array: BigUint64Array) => boolean,
+    predicate: (value: bigint, index: number, array: this) => boolean,
     thisArg?: unknown,
   ): number;
 
   /**
    * Returns a copy of an array with its elements reversed.
    */
-  toReversed(): BigUint64Array;
+  toReversed(): BigUint64Array<ArrayBuffer>;
 
   /**
    * Returns a copy of an array with its elements sorted.
@@ -849,11 +837,13 @@ interface BigUint64Array {
    * a negative value if the first argument is less than the second argument, zero if they're equal, and a positive
    * value otherwise. If omitted, the elements are sorted in ascending order.
    * ```ts
-   * const myNums = BigUint64Array.from([11n, 2n, 22n, 1n]);
-   * myNums.toSorted((a, b) => Number(a - b)) // BigUint64Array(4) [1n, 2n, 11n, 22n]
+   * const myNums = BigUint64Array<Buffer>.from([11n, 2n, 22n, 1n]);
+   * myNums.toSorted((a, b) => Number(a - b)) // BigUint64Array<Buffer>(4) [1n, 2n, 11n, 22n]
    * ```
    */
-  toSorted(compareFn?: (a: bigint, b: bigint) => number): BigUint64Array;
+  toSorted(
+    compareFn?: (a: bigint, b: bigint) => number,
+  ): BigUint64Array<ArrayBuffer>;
 
   /**
    * Copies an array, then overwrites the value at the provided index with the
@@ -864,5 +854,5 @@ interface BigUint64Array {
    * @param value The value to write into the copied array.
    * @returns The copied array with the updated value.
    */
-  with(index: number, value: bigint): BigUint64Array;
+  with(index: number, value: bigint): BigUint64Array<ArrayBuffer>;
 }
