diff --git a/packages/v5.6/output/lib-files/lib.es2015.core.d.ts b/packages/v5.7/output/lib-files/lib.es2015.core.d.ts
index 3fa95f0..8fa25b1 100644
--- a/packages/v5.6/output/lib-files/lib.es2015.core.d.ts
+++ b/packages/v5.7/output/lib-files/lib.es2015.core.d.ts
@@ -581,63 +581,63 @@ interface StringConstructor {
   ): string;
 }
 
-interface Int8Array {
+interface Int8Array<TArrayBuffer extends ArrayBufferLike> {
   toLocaleString(
     locales: string | readonly string[],
     options?: Intl.NumberFormatOptions,
   ): string;
 }
 
-interface Uint8Array {
+interface Uint8Array<TArrayBuffer extends ArrayBufferLike> {
   toLocaleString(
     locales: string | readonly string[],
     options?: Intl.NumberFormatOptions,
   ): string;
 }
 
-interface Uint8ClampedArray {
+interface Uint8ClampedArray<TArrayBuffer extends ArrayBufferLike> {
   toLocaleString(
     locales: string | readonly string[],
     options?: Intl.NumberFormatOptions,
   ): string;
 }
 
-interface Int16Array {
+interface Int16Array<TArrayBuffer extends ArrayBufferLike> {
   toLocaleString(
     locales: string | readonly string[],
     options?: Intl.NumberFormatOptions,
   ): string;
 }
 
-interface Uint16Array {
+interface Uint16Array<TArrayBuffer extends ArrayBufferLike> {
   toLocaleString(
     locales: string | readonly string[],
     options?: Intl.NumberFormatOptions,
   ): string;
 }
 
-interface Int32Array {
+interface Int32Array<TArrayBuffer extends ArrayBufferLike> {
   toLocaleString(
     locales: string | readonly string[],
     options?: Intl.NumberFormatOptions,
   ): string;
 }
 
-interface Uint32Array {
+interface Uint32Array<TArrayBuffer extends ArrayBufferLike> {
   toLocaleString(
     locales: string | readonly string[],
     options?: Intl.NumberFormatOptions,
   ): string;
 }
 
-interface Float32Array {
+interface Float32Array<TArrayBuffer extends ArrayBufferLike> {
   toLocaleString(
     locales: string | readonly string[],
     options?: Intl.NumberFormatOptions,
   ): string;
 }
 
-interface Float64Array {
+interface Float64Array<TArrayBuffer extends ArrayBufferLike> {
   toLocaleString(
     locales: string | readonly string[],
     options?: Intl.NumberFormatOptions,
