diff --git a/packages/v5.6/output/lib-files/lib.esnext.collection.d.ts b/packages/v5.7/output/lib-files/lib.esnext.collection.d.ts
index c057b57..d53cb82 100644
--- a/packages/v5.6/output/lib-files/lib.esnext.collection.d.ts
+++ b/packages/v5.7/output/lib-files/lib.esnext.collection.d.ts
@@ -1,17 +1,7 @@
 /// <reference no-default-lib="true"/>
 /// <reference types="ts-type-forge/global" />
 
-interface MapConstructor {
-  /**
-   * Groups members of an iterable according to the return value of the passed callback.
-   * @param items An iterable.
-   * @param keySelector A callback which will be invoked for each item in items.
-   */
-  groupBy<K, T>(
-    items: Iterable<T>,
-    keySelector: (item: T, index: number) => K,
-  ): Map<K, T[]>;
-}
+/// <reference path="./lib.es2024.collection.d.ts" />
 
 interface ReadonlySetLike<T> {
   /**
