diff --git experimental/packages/apps/event-schedule-app/src/functions/reducers/input-state/password-confirmation-input-state.ts apps/event-schedule-app/src/functions/reducers/input-state/password-confirmation-input-state.mts
index 1012f6d07..a97658116 100644
--- experimental/packages/apps/event-schedule-app/src/functions/reducers/input-state/password-confirmation-input-state.ts
+++ apps/event-schedule-app/src/functions/reducers/input-state/password-confirmation-input-state.mts
@@ -1,23 +1,15 @@
+import { pipe } from 'ts-data-forge';
+import { type DeepReadonly } from 'ts-type-forge';
+import { Obj, type Reducer } from '../../../utils-ported/index.mjs';
+
 export type PasswordWithConfirmationState = DeepReadonly<{
-  password: {
-    inputValue: string;
-    error: string | undefined;
-  };
-  passwordConfirmation: {
-    inputValue: string;
-    error: string | undefined;
-  };
+  password: { inputValue: string; error: string | undefined };
+  passwordConfirmation: { inputValue: string; error: string | undefined };
 }>;
 
 export const passwordWithConfirmationInitialState = {
-  password: {
-    inputValue: '',
-    error: undefined,
-  },
-  passwordConfirmation: {
-    inputValue: '',
-    error: undefined,
-  },
+  password: { inputValue: '', error: undefined },
+  passwordConfirmation: { inputValue: '', error: undefined },
 } as const satisfies PasswordWithConfirmationState;
 
 export const passwordWithConfirmationHasError = (
@@ -46,7 +38,7 @@ export const passwordWithConfirmationStateReducer: Reducer<
       const passwordInputValue = action.payload;
 
       return pipe(state)
-        .chain((draft) =>
+        .map((draft) =>
           Obj.setIn(
             draft,
             ['password', 'inputValue'],
@@ -60,8 +52,8 @@ export const passwordWithConfirmationStateReducer: Reducer<
               : passwordInputValue,
           ),
         )
-        .chain((draft) => Obj.setIn(draft, ['password', 'error'], undefined))
-        .chain((draft) =>
+        .map((draft) => Obj.setIn(draft, ['password', 'error'], undefined))
+        .map((draft) =>
           Obj.setIn(draft, ['passwordConfirmation', 'error'], undefined),
         ).value;
     }
@@ -70,7 +62,7 @@ export const passwordWithConfirmationStateReducer: Reducer<
       const passwordInputValue = action.payload;
 
       return pipe(state)
-        .chain((draft) =>
+        .map((draft) =>
           Obj.setIn(
             draft,
             ['passwordConfirmation', 'inputValue'],
@@ -84,8 +76,8 @@ export const passwordWithConfirmationStateReducer: Reducer<
               : passwordInputValue,
           ),
         )
-        .chain((draft) => Obj.setIn(draft, ['password', 'error'], undefined))
-        .chain((draft) =>
+        .map((draft) => Obj.setIn(draft, ['password', 'error'], undefined))
+        .map((draft) =>
           Obj.setIn(draft, ['passwordConfirmation', 'error'], undefined),
         ).value;
     }
