diff --git experimental/packages/utils/io-ts-types/src/date-time/ymdhm.mts apps/ts-fortress-types/src/date-time/ymdhm.mts
index bb19d6aff..f4c6e5808 100644
--- experimental/packages/utils/io-ts-types/src/date-time/ymdhm.mts
+++ apps/ts-fortress-types/src/date-time/ymdhm.mts
@@ -1,5 +1,5 @@
-import * as t from '@noshiro/io-ts';
-import { DateUtils, Num, type DateType } from '@noshiro/ts-utils';
+import * as t from 'ts-fortress';
+import { DateUtils } from '../utils/index.mjs';
 import { Dates, Hours, Minutes, Months, Years } from './time-enum.mjs';
 
 export const Ymdhm = t.record({
@@ -12,15 +12,16 @@ export const Ymdhm = t.record({
 
 export type Ymdhm = t.TypeOf<typeof Ymdhm>;
 
-export const YmdhmFromDate = (date: RawDateType): Ymdhm => ({
-  year: DateUtils.getLocaleYear(date),
-  month: DateUtils.getLocaleMonth(date),
-  date: DateUtils.getLocaleDate(date),
-  hours: DateUtils.getLocaleHours(date),
-  minutes: DateUtils.getLocaleMinutes(date),
-});
+export const YmdhmFromDate = (date: Date): Ymdhm =>
+  ({
+    year: DateUtils.getLocaleYear(date),
+    month: DateUtils.getLocaleMonth(date),
+    date: DateUtils.getLocaleDate(date),
+    hours: DateUtils.getLocaleHours(date),
+    minutes: DateUtils.getLocaleMinutes(date),
+  }) as const;
 
-export const Ymdhm2Date = (ymdhm: Ymdhm): DateType =>
+export const Ymdhm2Date = (ymdhm: Ymdhm): Date =>
   DateUtils.create(
     ymdhm.year,
     ymdhm.month,
@@ -30,15 +31,15 @@ export const Ymdhm2Date = (ymdhm: Ymdhm): DateType =>
   );
 
 export const compareYmdhm = (a: Ymdhm, b: Ymdhm): -1 | 0 | 1 => {
-  if (a.year !== b.year)
-    return Num.mapNaN2Undefined(Math.sign(a.year - b.year)) ?? 0;
-  if (a.month !== b.month)
-    return Num.mapNaN2Undefined(Math.sign(a.month - b.month)) ?? 0;
-  if (a.date !== b.date)
-    return Num.mapNaN2Undefined(Math.sign(a.date - b.date)) ?? 0;
-  if (a.hours !== b.hours)
-    return Num.mapNaN2Undefined(Math.sign(a.hours - b.hours)) ?? 0;
-  if (a.minutes !== b.minutes)
-    return Num.mapNaN2Undefined(Math.sign(a.minutes - b.minutes)) ?? 0;
+  if (a.year !== b.year) return a.year < b.year ? -1 : 1;
+
+  if (a.month !== b.month) return a.month < b.month ? -1 : 1;
+
+  if (a.date !== b.date) return a.date < b.date ? -1 : 1;
+
+  if (a.hours !== b.hours) return a.hours < b.hours ? -1 : 1;
+
+  if (a.minutes !== b.minutes) return a.minutes < b.minutes ? -1 : 1;
+
   return 0;
 };
