diff --git a/packages/v5.0/output/lib-files/lib.es5.d.ts b/packages/v5.1/output/lib-files/lib.es5.d.ts
index f93ba42..bef43a2 100644
--- a/packages/v5.0/output/lib-files/lib.es5.d.ts
+++ b/packages/v5.1/output/lib-files/lib.es5.d.ts
@@ -369,9 +369,14 @@ interface CallableFunction extends Function {
   /**
    * Calls the function with the specified object as the this value and the elements of specified array as the arguments.
    * @param thisArg The object to be used as the this object.
-   * @param args An array of argument values to be passed to the function.
    */
   apply<T, R>(this: (this: T) => R, thisArg: T): R;
+
+  /**
+   * Calls the function with the specified object as the this value and the elements of specified array as the arguments.
+   * @param thisArg The object to be used as the this object.
+   * @param args An array of argument values to be passed to the function.
+   */
   apply<T, A extends readonly unknown[], R>(
     this: (this: T, ...args: A) => R,
     thisArg: T,
@@ -393,49 +398,33 @@ interface CallableFunction extends Function {
    * For a given function, creates a bound function that has the same body as the original function.
    * The this object of the bound function is associated with the specified object, and has the specified initial parameters.
    * @param thisArg The object to be used as the this object.
-   * @param args Arguments to bind to the parameters of the function.
    */
   bind<T>(this: T, thisArg: ThisParameterType<T>): OmitThisParameter<T>;
-  bind<T, A0, A extends readonly unknown[], R>(
-    this: (this: T, arg0: A0, ...args: A) => R,
-    thisArg: T,
-    arg0: A0,
-  ): (...args: A) => R;
-  bind<T, A0, A1, A extends readonly unknown[], R>(
-    this: (this: T, arg0: A0, arg1: A1, ...args: A) => R,
-    thisArg: T,
-    arg0: A0,
-    arg1: A1,
-  ): (...args: A) => R;
-  bind<T, A0, A1, A2, A extends readonly unknown[], R>(
-    this: (this: T, arg0: A0, arg1: A1, arg2: A2, ...args: A) => R,
-    thisArg: T,
-    arg0: A0,
-    arg1: A1,
-    arg2: A2,
-  ): (...args: A) => R;
-  bind<T, A0, A1, A2, A3, A extends readonly unknown[], R>(
-    this: (this: T, arg0: A0, arg1: A1, arg2: A2, arg3: A3, ...args: A) => R,
-    thisArg: T,
-    arg0: A0,
-    arg1: A1,
-    arg2: A2,
-    arg3: A3,
-  ): (...args: A) => R;
-  bind<T, AX, R>(
-    this: (this: T, ...args: readonly AX[]) => R,
+
+  /**
+   * For a given function, creates a bound function that has the same body as the original function.
+   * The this object of the bound function is associated with the specified object, and has the specified initial parameters.
+   * @param thisArg The object to be used as the this object.
+   * @param args Arguments to bind to the parameters of the function.
+   */
+  bind<T, A extends readonly unknown[], B extends readonly unknown[], R>(
+    this: (this: T, ...args: readonly [...A, ...B]) => R,
     thisArg: T,
-    ...args: readonly AX[]
-  ): (...args: readonly AX[]) => R;
+    ...args: A
+  ): (...args: B) => R;
 }
 
 interface NewableFunction extends Function {
   /**
    * Calls the function with the specified object as the this value and the elements of specified array as the arguments.
    * @param thisArg The object to be used as the this object.
-   * @param args An array of argument values to be passed to the function.
    */
   apply<T>(this: new () => T, thisArg: T): void;
+  /**
+   * Calls the function with the specified object as the this value and the elements of specified array as the arguments.
+   * @param thisArg The object to be used as the this object.
+   * @param args An array of argument values to be passed to the function.
+   */
   apply<T, A extends readonly unknown[]>(
     this: new (...args: A) => T,
     thisArg: T,
@@ -457,40 +446,20 @@ interface NewableFunction extends Function {
    * For a given function, creates a bound function that has the same body as the original function.
    * The this object of the bound function is associated with the specified object, and has the specified initial parameters.
    * @param thisArg The object to be used as the this object.
-   * @param args Arguments to bind to the parameters of the function.
    */
   bind<T>(this: T, thisArg: unknown): T;
-  bind<A0, A extends readonly unknown[], R>(
-    this: new (arg0: A0, ...args: A) => R,
-    thisArg: unknown,
-    arg0: A0,
-  ): new (...args: A) => R;
-  bind<A0, A1, A extends readonly unknown[], R>(
-    this: new (arg0: A0, arg1: A1, ...args: A) => R,
-    thisArg: unknown,
-    arg0: A0,
-    arg1: A1,
-  ): new (...args: A) => R;
-  bind<A0, A1, A2, A extends readonly unknown[], R>(
-    this: new (arg0: A0, arg1: A1, arg2: A2, ...args: A) => R,
-    thisArg: unknown,
-    arg0: A0,
-    arg1: A1,
-    arg2: A2,
-  ): new (...args: A) => R;
-  bind<A0, A1, A2, A3, A extends readonly unknown[], R>(
-    this: new (arg0: A0, arg1: A1, arg2: A2, arg3: A3, ...args: A) => R,
-    thisArg: unknown,
-    arg0: A0,
-    arg1: A1,
-    arg2: A2,
-    arg3: A3,
-  ): new (...args: A) => R;
-  bind<AX, R>(
-    this: new (...args: readonly AX[]) => R,
+
+  /**
+   * For a given function, creates a bound function that has the same body as the original function.
+   * The this object of the bound function is associated with the specified object, and has the specified initial parameters.
+   * @param thisArg The object to be used as the this object.
+   * @param args Arguments to bind to the parameters of the function.
+   */
+  bind<A extends readonly unknown[], B extends readonly unknown[], R>(
+    this: new (...args: readonly [...A, ...B]) => R,
     thisArg: unknown,
-    ...args: readonly AX[]
-  ): new (...args: readonly AX[]) => R;
+    ...args: A
+  ): new (...args: B) => R;
 }
 
 interface IArguments {
@@ -2146,7 +2115,7 @@ interface DataView {
 interface DataViewConstructor {
   readonly prototype: DataView;
   new (
-    buffer: ArrayBufferLike,
+    buffer: ArrayBufferLike & { readonly BYTES_PER_ELEMENT?: never },
     byteOffset?: number,
     byteLength?: number,
   ): DataView;
@@ -2184,7 +2153,7 @@ interface Int8Array {
    * @param target If target is negative, it is treated as length+target where length is the
    * length of the array.
    * @param start If start is negative, it is treated as length+start. If end is negative, it
-   * is treated as length+end.
+   * is treated as length+end. If start is omitted, `0` is used.
    * @param end If not specified, length of the this object is used as its default value.
    */
   copyWithin(target: number, start: number, end?: number): this;
@@ -2539,7 +2508,7 @@ interface Uint8Array {
    * @param target If target is negative, it is treated as length+target where length is the
    * length of the array.
    * @param start If start is negative, it is treated as length+start. If end is negative, it
-   * is treated as length+end.
+   * is treated as length+end. If start is omitted, `0` is used.
    * @param end If not specified, length of the this object is used as its default value.
    */
   copyWithin(target: number, start: number, end?: number): this;
@@ -2895,7 +2864,7 @@ interface Uint8ClampedArray {
    * @param target If target is negative, it is treated as length+target where length is the
    * length of the array.
    * @param start If start is negative, it is treated as length+start. If end is negative, it
-   * is treated as length+end.
+   * is treated as length+end. If start is omitted, `0` is used.
    * @param end If not specified, length of the this object is used as its default value.
    */
   copyWithin(target: number, start: number, end?: number): this;
@@ -3267,7 +3236,7 @@ interface Int16Array {
    * @param target If target is negative, it is treated as length+target where length is the
    * length of the array.
    * @param start If start is negative, it is treated as length+start. If end is negative, it
-   * is treated as length+end.
+   * is treated as length+end. If start is omitted, `0` is used.
    * @param end If not specified, length of the this object is used as its default value.
    */
   copyWithin(target: number, start: number, end?: number): this;
@@ -3623,7 +3592,7 @@ interface Uint16Array {
    * @param target If target is negative, it is treated as length+target where length is the
    * length of the array.
    * @param start If start is negative, it is treated as length+start. If end is negative, it
-   * is treated as length+end.
+   * is treated as length+end. If start is omitted, `0` is used.
    * @param end If not specified, length of the this object is used as its default value.
    */
   copyWithin(target: number, start: number, end?: number): this;
@@ -3978,7 +3947,7 @@ interface Int32Array {
    * @param target If target is negative, it is treated as length+target where length is the
    * length of the array.
    * @param start If start is negative, it is treated as length+start. If end is negative, it
-   * is treated as length+end.
+   * is treated as length+end. If start is omitted, `0` is used.
    * @param end If not specified, length of the this object is used as its default value.
    */
   copyWithin(target: number, start: number, end?: number): this;
@@ -4334,7 +4303,7 @@ interface Uint32Array {
    * @param target If target is negative, it is treated as length+target where length is the
    * length of the array.
    * @param start If start is negative, it is treated as length+start. If end is negative, it
-   * is treated as length+end.
+   * is treated as length+end. If start is omitted, `0` is used.
    * @param end If not specified, length of the this object is used as its default value.
    */
   copyWithin(target: number, start: number, end?: number): this;
@@ -4690,7 +4659,7 @@ interface Float32Array {
    * @param target If target is negative, it is treated as length+target where length is the
    * length of the array.
    * @param start If start is negative, it is treated as length+start. If end is negative, it
-   * is treated as length+end.
+   * is treated as length+end. If start is omitted, `0` is used.
    * @param end If not specified, length of the this object is used as its default value.
    */
   copyWithin(target: number, start: number, end?: number): this;
@@ -5046,7 +5015,7 @@ interface Float64Array {
    * @param target If target is negative, it is treated as length+target where length is the
    * length of the array.
    * @param start If start is negative, it is treated as length+start. If end is negative, it
-   * is treated as length+end.
+   * is treated as length+end. If start is omitted, `0` is used.
    * @param end If not specified, length of the this object is used as its default value.
    */
   copyWithin(target: number, start: number, end?: number): this;
@@ -5307,12 +5276,21 @@ interface Float64Array {
   sort(compareFn?: (a: number, b: number) => number): this;
 
   /**
+   * Gets a new Float64Array view of the ArrayBuffer store for this array, referencing the elements
    * at begin, inclusive, up to end, exclusive.
    * @param begin The index of the beginning of the array.
    * @param end The index of the end of the array.
    */
   subarray(begin?: number, end?: number): Float64Array;
 
+  /**
+   * Converts a number to a string by using the current locale.
+   */
+  toLocaleString(): string;
+
+  /**
+   * Returns a string representation of an array.
+   */
   toString(): string;
 
   /** Returns the primitive value of the specified object. */
@@ -5368,11 +5346,30 @@ declare const Float64Array: Float64ArrayConstructor;
 
 declare namespace Intl {
   interface CollatorOptions {
-    readonly usage?: string | undefined;
-    readonly localeMatcher?: string | undefined;
+    readonly usage?: 'sort' | 'search' | undefined;
+    readonly localeMatcher?: 'lookup' | 'best fit' | undefined;
     readonly numeric?: boolean | undefined;
-    readonly caseFirst?: string | undefined;
-    readonly sensitivity?: string | undefined;
+    readonly caseFirst?: 'upper' | 'lower' | 'false' | undefined;
+    readonly sensitivity?: 'base' | 'accent' | 'case' | 'variant' | undefined;
+    readonly collation?:
+      | 'big5han'
+      | 'compat'
+      | 'dict'
+      | 'direct'
+      | 'ducet'
+      | 'emoji'
+      | 'eor'
+      | 'gb2312'
+      | 'phonebk'
+      | 'phonetic'
+      | 'pinyin'
+      | 'reformed'
+      | 'searchjl'
+      | 'stroke'
+      | 'trad'
+      | 'unihan'
+      | 'zhuyin'
+      | undefined;
     readonly ignorePunctuation?: boolean | undefined;
   }
 
