diff --git experimental/packages/apps/event-schedule-app/src/components/pages/create-event-schedule/index.ts apps/event-schedule-app/src/components/pages/create-event-schedule/index.mts
index 87b5c73e5..a6474e608 100644
--- experimental/packages/apps/event-schedule-app/src/components/pages/create-event-schedule/index.ts
+++ apps/event-schedule-app/src/components/pages/create-event-schedule/index.mts
@@ -1,6 +1,6 @@
-export * from './create-event-schedule';
-export * from './edit-event-schedule';
-export * from './edit-event-schedule-email-verified';
-export * from './edit-event-schedule-ok';
-export * from './error';
-export * from './event-schedule-setting-common';
+export * from './create-event-schedule.js';
+export * from './edit-event-schedule-email-verified.js';
+export * from './edit-event-schedule-ok.js';
+export * from './edit-event-schedule.js';
+export * from './error.js';
+export * from './event-schedule-setting-common.js';
