diff --git a/packages/v5.4/output-branded/lib-files/lib.es2022.regexp.d.ts b/packages/v5.5/output-branded/lib-files/lib.es2022.regexp.d.ts
index 5052de0..3b885c8 100644
--- a/packages/v5.4/output-branded/lib-files/lib.es2022.regexp.d.ts
+++ b/packages/v5.5/output-branded/lib-files/lib.es2022.regexp.d.ts
@@ -22,7 +22,7 @@ interface RegExpIndicesArray extends Array<
 
 interface RegExp {
   /**
-   * Returns a Boolean value indicating the state of the hasIndices flag (d) used with with a regular expression.
+   * Returns a Boolean value indicating the state of the hasIndices flag (d) used with a regular expression.
    * Default is false. Read-only.
    */
   readonly hasIndices: boolean;
