diff --git a/packages/v5.4/output-branded/lib-files/lib.es2015.core.d.ts b/packages/v5.5/output-branded/lib-files/lib.es2015.core.d.ts
index 0639656..078568f 100644
--- a/packages/v5.4/output-branded/lib-files/lib.es2015.core.d.ts
+++ b/packages/v5.5/output-branded/lib-files/lib.es2015.core.d.ts
@@ -74,6 +74,11 @@ interface Array<T> {
     start: NumberType.ArraySizeArg,
     end?: NumberType.ArraySizeArg,
   ): this;
+
+  toLocaleString(
+    locales: string | readonly string[],
+    options?: Intl.NumberFormatOptions & Intl.DateTimeFormatOptions,
+  ): string;
 }
 
 interface ArrayConstructor {
@@ -404,6 +409,11 @@ interface ReadonlyArray<T> {
     ) => boolean,
     thisArg?: unknown,
   ): NumberType.ArraySearchResult;
+
+  toLocaleString(
+    locales: string | readonly string[],
+    options?: Intl.NumberFormatOptions & Intl.DateTimeFormatOptions,
+  ): string;
 }
 
 interface RegExp {
@@ -611,3 +621,66 @@ interface StringConstructor {
     ...substitutions: readonly unknown[]
   ): string;
 }
+
+interface Int8Array {
+  toLocaleString(
+    locales: string | readonly string[],
+    options?: Intl.NumberFormatOptions,
+  ): string;
+}
+
+interface Uint8Array {
+  toLocaleString(
+    locales: string | readonly string[],
+    options?: Intl.NumberFormatOptions,
+  ): string;
+}
+
+interface Uint8ClampedArray {
+  toLocaleString(
+    locales: string | readonly string[],
+    options?: Intl.NumberFormatOptions,
+  ): string;
+}
+
+interface Int16Array {
+  toLocaleString(
+    locales: string | readonly string[],
+    options?: Intl.NumberFormatOptions,
+  ): string;
+}
+
+interface Uint16Array {
+  toLocaleString(
+    locales: string | readonly string[],
+    options?: Intl.NumberFormatOptions,
+  ): string;
+}
+
+interface Int32Array {
+  toLocaleString(
+    locales: string | readonly string[],
+    options?: Intl.NumberFormatOptions,
+  ): string;
+}
+
+interface Uint32Array {
+  toLocaleString(
+    locales: string | readonly string[],
+    options?: Intl.NumberFormatOptions,
+  ): string;
+}
+
+interface Float32Array {
+  toLocaleString(
+    locales: string | readonly string[],
+    options?: Intl.NumberFormatOptions,
+  ): string;
+}
+
+interface Float64Array {
+  toLocaleString(
+    locales: string | readonly string[],
+    options?: Intl.NumberFormatOptions,
+  ): string;
+}
