diff --git a/packages/v6.0/output/lib-files/lib.es2015.collection.d.ts b/packages/v7.0/output/lib-files/lib.es2015.collection.d.ts
index 9d46fa8c..b7963ffd 100644
--- a/packages/v6.0/output/lib-files/lib.es2015.collection.d.ts
+++ b/packages/v7.0/output/lib-files/lib.es2015.collection.d.ts
@@ -105,7 +105,7 @@ interface Set<T> {
    */
   has(value: T | (import('ts-type-forge').WidenLiteral<T> & {})): boolean;
   /**
-   * @returns the number of (unique) elements in Set.
+   * @returns the number of (unique) elements in the Set.
    */
   readonly size: number;
 }
@@ -117,11 +117,20 @@ interface SetConstructor {
 declare const Set: SetConstructor;
 
 interface ReadonlySet<T> {
+  /**
+   * Executes a provided function once per each value in the ReadonlySet object, in insertion order.
+   */
   forEach(
     callbackfn: (value: T, value2: T, set: ReadonlySet<T>) => void,
     thisArg?: unknown,
   ): void;
+  /**
+   * @returns a boolean indicating whether an element with the specified value exists in the Set or not.
+   */
   has(value: T | (import('ts-type-forge').WidenLiteral<T> & {})): boolean;
+  /**
+   * @returns the number of (unique) elements in the Set.
+   */
   readonly size: number;
 }
 
