diff --git a/packages/v6.0/output/lib-files/lib.es2015.symbol.d.ts b/packages/v7.0/output/lib-files/lib.es2015.symbol.d.ts
index 663725be..3df2235d 100644
--- a/packages/v6.0/output/lib-files/lib.es2015.symbol.d.ts
+++ b/packages/v7.0/output/lib-files/lib.es2015.symbol.d.ts
@@ -19,7 +19,7 @@ interface SymbolConstructor {
 
   /**
    * Returns a key from the global symbol registry matching the given Symbol if found.
-   * Otherwise, returns a undefined.
+   * Otherwise, returns undefined.
    * @param sym Symbol to find the key for.
    */
   keyFor(sym: symbol): string | undefined;
