diff --git experimental/packages/apps/event-schedule-app/src/functions/event-settings-page-diff.ts apps/event-schedule-app/src/functions/event-settings-page-diff.mts
index 31fb5d339..077956ecb 100644
--- experimental/packages/apps/event-schedule-app/src/functions/event-settings-page-diff.ts
+++ apps/event-schedule-app/src/functions/event-settings-page-diff.mts
@@ -1,35 +1,39 @@
-import { isKeyofNotificationSettings } from '@noshiro/event-schedule-app-shared';
-import { deepEqual } from '@noshiro/fast-deep-equal';
-import { hm2str, ymd2str, ymdhm2str } from '../constants';
-import { datetimeRangeFromMapKey, datetimeRangeToMapKey } from './map-key';
+import { isKeyofNotificationSettings } from 'event-schedule-app-shared';
+import { Arr, ISetMapped, expectType, fastDeepEqual } from 'ts-data-forge';
+import {
+  type DeepReadonly,
+  type ReadonlyRecord,
+  type RelaxedExclude,
+  type StrictExclude,
+  type StrictPick,
+} from 'ts-type-forge';
+import { hm2str, ymd2str, ymdhm2str } from '../constants/index.mjs';
+import {
+  Obj,
+  match,
+  type Paths,
+  type RecordValueAtPath,
+} from '../utils-ported/index.mjs';
+import {
+  datetimeRangeFromMapKey,
+  datetimeRangeToMapKey,
+} from './map-key/index.mjs';
 
 export type EventSettingsPageDiffResult = DeepReadonly<{
   title: string | undefined;
   notes: string | undefined;
   datetimeSpecification: string | undefined;
 
-  datetimeRangeList:
-    | {
-        added: string[];
-        deleted: string[];
-      }
-    | undefined;
+  datetimeRangeList: { added: string[]; deleted: string[] } | undefined;
 
   answerDeadline: string | undefined;
 
   notificationSettings: string[] | undefined;
 
   answerIcons: {
-    good: {
-      description: string | undefined;
-    };
-    fair: {
-      description: string | undefined;
-      point: string | undefined;
-    };
-    poor: {
-      description: string | undefined;
-    };
+    good: { description: string | undefined };
+    fair: { description: string | undefined; point: string | undefined };
+    poor: { description: string | undefined };
   };
 
   author: string | undefined;
@@ -44,48 +48,46 @@ const datetimeSpecificationOptions =
 const dc = dict.eventSettingsPage.diff;
 
 const map = (a: string, b: string): string =>
-  `"${a}" ${dict.common.mapTo} "${b}"`;
+  `"${a}" ${dict.common.mapTo} "${b}"` as const;
 
-type EventSchedulePaths = Exclude<Paths<EventSchedule>, readonly []>;
+type EventSchedulePaths = StrictExclude<Paths<EventSchedule>, readonly []>;
 
 {
   expectType<
-    Readonly<
-      | ['answerDeadline', 'date']
-      | ['answerDeadline', 'hours']
-      | ['answerDeadline', 'minutes']
-      | ['answerDeadline', 'month']
-      | ['answerDeadline', 'year']
-      | ['answerDeadline']
-      | ['answerIcons', 'fair', 'description']
-      | ['answerIcons', 'fair', 'point']
-      | ['answerIcons', 'fair']
-      | ['answerIcons', 'good', 'description']
-      | ['answerIcons', 'good', 'point']
-      | ['answerIcons', 'good']
-      | ['answerIcons', 'poor', 'description']
-      | ['answerIcons', 'poor', 'point']
-      | ['answerIcons', 'poor']
-      | ['answerIcons']
-      | ['archivedBy']
-      | ['author', 'id']
-      | ['author', 'name']
-      | ['author']
-      | ['datetimeRangeList']
-      | ['datetimeSpecification']
-      | ['notes']
-      | ['notificationSettings', 'notify00daysBeforeAnswerDeadline']
-      | ['notificationSettings', 'notify01daysBeforeAnswerDeadline']
-      | ['notificationSettings', 'notify03daysBeforeAnswerDeadline']
-      | ['notificationSettings', 'notify07daysBeforeAnswerDeadline']
-      | ['notificationSettings', 'notify14daysBeforeAnswerDeadline']
-      | ['notificationSettings', 'notify28daysBeforeAnswerDeadline']
-      | ['notificationSettings', 'notifyAfterAnswerDeadline']
-      | ['notificationSettings', 'notifyOnAnswerChange']
-      | ['notificationSettings']
-      | ['timezoneOffsetMinutes']
-      | ['title']
-    >,
+    | readonly ['answerDeadline', 'date']
+    | readonly ['answerDeadline', 'hours']
+    | readonly ['answerDeadline', 'minutes']
+    | readonly ['answerDeadline', 'month']
+    | readonly ['answerDeadline', 'year']
+    | readonly ['answerDeadline']
+    | readonly ['answerIcons', 'fair', 'description']
+    | readonly ['answerIcons', 'fair', 'point']
+    | readonly ['answerIcons', 'fair']
+    | readonly ['answerIcons', 'good', 'description']
+    | readonly ['answerIcons', 'good', 'point']
+    | readonly ['answerIcons', 'good']
+    | readonly ['answerIcons', 'poor', 'description']
+    | readonly ['answerIcons', 'poor', 'point']
+    | readonly ['answerIcons', 'poor']
+    | readonly ['answerIcons']
+    | readonly ['archivedBy']
+    | readonly ['author', 'id']
+    | readonly ['author', 'name']
+    | readonly ['author']
+    | readonly ['datetimeRangeList']
+    | readonly ['datetimeSpecification']
+    | readonly ['notes']
+    | readonly ['notificationSettings', 'notify00daysBeforeAnswerDeadline']
+    | readonly ['notificationSettings', 'notify01daysBeforeAnswerDeadline']
+    | readonly ['notificationSettings', 'notify03daysBeforeAnswerDeadline']
+    | readonly ['notificationSettings', 'notify07daysBeforeAnswerDeadline']
+    | readonly ['notificationSettings', 'notify14daysBeforeAnswerDeadline']
+    | readonly ['notificationSettings', 'notify28daysBeforeAnswerDeadline']
+    | readonly ['notificationSettings', 'notifyAfterAnswerDeadline']
+    | readonly ['notificationSettings', 'notifyOnAnswerChange']
+    | readonly ['notificationSettings']
+    | readonly ['timezoneOffsetMinutes']
+    | readonly ['title'],
     EventSchedulePaths
   >('=');
 }
@@ -100,11 +102,14 @@ const createDiffResult = <P extends EventSchedulePaths, R>(
   ) => RelaxedExclude<R, undefined>,
 ): R | undefined => {
   const a = Obj.getIn(prev, path);
+
   const b = Obj.getIn(curr, path);
-  return deepEqual(a, b) ? undefined : makeResult(a, b);
+
+  return fastDeepEqual(a, b) ? undefined : makeResult(a, b);
 };
 
 const ndc = dc.items.notificationSettings;
+
 const notificationSettings = dict.eventSettingsPage.section3.notification;
 
 const notificationSettingsDiff = (
@@ -113,7 +118,7 @@ const notificationSettingsDiff = (
   emailPrev: string | undefined,
   emailCurr: string | undefined,
 ): readonly string[] | undefined => {
-  if (deepEqual(a, b) && emailPrev === emailCurr) return undefined;
+  if (emailPrev === emailCurr && fastDeepEqual(a, b)) return undefined;
 
   if (a === 'none' || b === 'none') {
     return [
@@ -129,7 +134,7 @@ const notificationSettingsDiff = (
         (acc, key) =>
           a[key] === b[key]
             ? acc
-            : Arr.pushed(
+            : Arr.toPushed(
                 acc,
                 `${notificationSettings[key]}${dict.common.colon} ${map(
                   a[key] ? ndc.on : ndc.off,
@@ -166,16 +171,16 @@ const datetimeRange2str = (
   });
 
 const datetimeRangeListDiff = (
-  a: Pick<EventSchedule, 'datetimeRangeList' | 'datetimeSpecification'>,
-  b: Pick<EventSchedule, 'datetimeRangeList' | 'datetimeSpecification'>,
-): Record<'added' | 'deleted', readonly string[]> | undefined => {
-  const setA = ISetMapped.new(
+  a: StrictPick<EventSchedule, 'datetimeRangeList' | 'datetimeSpecification'>,
+  b: StrictPick<EventSchedule, 'datetimeRangeList' | 'datetimeSpecification'>,
+): ReadonlyRecord<'added' | 'deleted', readonly string[]> | undefined => {
+  const setA = ISetMapped.create(
     a.datetimeRangeList,
     datetimeRangeToMapKey,
     datetimeRangeFromMapKey,
   );
 
-  const setB = ISetMapped.new(
+  const setB = ISetMapped.create(
     b.datetimeRangeList,
     datetimeRangeToMapKey,
     datetimeRangeFromMapKey,
@@ -185,13 +190,13 @@ const datetimeRangeListDiff = (
 
   if (diff.added.isEmpty && diff.deleted.isEmpty) return undefined;
 
+  const added = diff.added.toArray();
+
+  const deleted = diff.deleted.toArray();
+
   return {
-    added: diff.added
-      .toArray()
-      .map((e) => datetimeRange2str(a.datetimeSpecification, e)),
-    deleted: diff.deleted
-      .toArray()
-      .map((e) => datetimeRange2str(b.datetimeSpecification, e)),
+    added: added.map((e) => datetimeRange2str(a.datetimeSpecification, e)),
+    deleted: deleted.map((e) => datetimeRange2str(b.datetimeSpecification, e)),
   };
 };
 
@@ -200,77 +205,78 @@ export const collectEventSettingsPageDiff = (
   curr: EventSchedule,
   emailPrev: string | undefined,
   emailCurr: string | undefined,
-): EventSettingsPageDiffResult => ({
-  title: createDiffResult(prev, curr, ['title'], map),
-  notes: createDiffResult(prev, curr, ['notes'], map),
-
-  datetimeSpecification: createDiffResult(
-    prev,
-    curr,
-    ['datetimeSpecification'],
-    (a, b) =>
-      map(datetimeSpecificationOptions[a], datetimeSpecificationOptions[b]),
-  ),
-
-  datetimeRangeList: datetimeRangeListDiff(prev, curr),
-
-  answerDeadline: createDiffResult(prev, curr, ['answerDeadline'], (a, b) =>
-    map(
-      a === 'none' ? dc.values.none : ymdhm2str(a),
-      b === 'none' ? dc.values.none : ymdhm2str(b),
+): EventSettingsPageDiffResult =>
+  ({
+    title: createDiffResult(prev, curr, ['title'], map),
+    notes: createDiffResult(prev, curr, ['notes'], map),
+
+    datetimeSpecification: createDiffResult(
+      prev,
+      curr,
+      ['datetimeSpecification'],
+      (a, b) =>
+        map(datetimeSpecificationOptions[a], datetimeSpecificationOptions[b]),
     ),
-  ),
 
-  notificationSettings: notificationSettingsDiff(
-    prev.notificationSettings,
-    curr.notificationSettings,
-    emailPrev,
-    emailCurr,
-  ),
+    datetimeRangeList: datetimeRangeListDiff(prev, curr),
 
-  answerIcons: {
-    good: {
-      description: createDiffResult(
-        prev,
-        curr,
-        ['answerIcons', 'good', 'description'],
-        map,
+    answerDeadline: createDiffResult(prev, curr, ['answerDeadline'], (a, b) =>
+      map(
+        a === 'none' ? dc.values.none : ymdhm2str(a),
+        b === 'none' ? dc.values.none : ymdhm2str(b),
       ),
-    },
+    ),
 
-    fair: {
-      description: createDiffResult(
-        prev,
-        curr,
-        ['answerIcons', 'fair', 'description'],
-        map,
-      ),
-      point: createDiffResult(
-        prev,
-        curr,
-        ['answerIcons', 'fair', 'point'],
-        (a, b) => map(a.toString(), b.toString()),
-      ),
-    },
+    notificationSettings: notificationSettingsDiff(
+      prev.notificationSettings,
+      curr.notificationSettings,
+      emailPrev,
+      emailCurr,
+    ),
 
-    poor: {
-      description: createDiffResult(
-        prev,
-        curr,
-        ['answerIcons', 'poor', 'description'],
-        map,
-      ),
+    answerIcons: {
+      good: {
+        description: createDiffResult(
+          prev,
+          curr,
+          ['answerIcons', 'good', 'description'],
+          map,
+        ),
+      },
+
+      fair: {
+        description: createDiffResult(
+          prev,
+          curr,
+          ['answerIcons', 'fair', 'description'],
+          map,
+        ),
+        point: createDiffResult(
+          prev,
+          curr,
+          ['answerIcons', 'fair', 'point'],
+          (a, b) => map(a.toString(), b.toString()),
+        ),
+      },
+
+      poor: {
+        description: createDiffResult(
+          prev,
+          curr,
+          ['answerIcons', 'poor', 'description'],
+          map,
+        ),
+      },
     },
-  },
-
-  author: createDiffResult(prev, curr, ['author'], (a, b) =>
-    map(a.name, b.name),
-  ),
-
-  timezoneOffsetMinutes: createDiffResult(
-    prev,
-    curr,
-    ['timezoneOffsetMinutes'],
-    (a, b) => map(a.toString(), b.toString()),
-  ),
-});
+
+    author: createDiffResult(prev, curr, ['author'], (a, b) =>
+      map(a.name, b.name),
+    ),
+
+    timezoneOffsetMinutes: createDiffResult(
+      prev,
+      curr,
+      ['timezoneOffsetMinutes'],
+      (a, b) => map(a.toString(), b.toString()),
+    ),
+  }) as const;
