diff --git experimental/packages/apps/event-schedule-app/src/functions/local-storage.ts apps/event-schedule-app/src/functions/local-storage.mts
index fc6453808..2a4137913 100644
--- experimental/packages/apps/event-schedule-app/src/functions/local-storage.ts
+++ apps/event-schedule-app/src/functions/local-storage.mts
@@ -1,6 +1,12 @@
-import { EventSchedule } from '@noshiro/event-schedule-app-shared';
+import { EventSchedule } from 'event-schedule-app-shared';
+import { Json, Result } from 'ts-data-forge';
+import {
+  type DeepReadonly,
+  type ReadonlyRecord,
+  type StrictPick,
+} from 'ts-type-forge';
 
-type EventSchedulePicked = Pick<
+type EventSchedulePicked = StrictPick<
   EventSchedule,
   | 'answerDeadline'
   | 'answerIcons'
@@ -9,23 +15,19 @@ type EventSchedulePicked = Pick<
   | 'notificationSettings'
   | 'title'
 > &
-  Readonly<{
-    datetimeRangeList: readonly DatetimeRange[];
-  }>;
+  Readonly<{ datetimeRangeList: readonly DatetimeRange[] }>;
 
 /** LocalStorage のキー重複回避のために suffix を付けている */
 const attachSuffix = (key: string): string =>
-  `${key}--event-schedule-app--2dc690b2-4b9b-49c3-921b-8c2a9451e415`;
+  `${key}--event-schedule-app--2dc690b2-4b9b-49c3-921b-8c2a9451e415` as const;
 
-type Store = DeepReadonly<{
-  createEventPage: EventSchedule;
-}>;
+type Store = DeepReadonly<{ createEventPage: EventSchedule }>;
 
-type StoreKeys = Record<keyof Store, string>;
+type StoreKeys = ReadonlyRecord<keyof Store, string>;
 
 const keys: StoreKeys = {
   createEventPage: attachSuffix('createEventPage'),
-};
+} as const;
 
 const restore = (at: string): string | undefined =>
   // eslint-disable-next-line unicorn/prefer-global-this
@@ -41,8 +43,11 @@ const restoreCreateEventPageTemp = (): Result<
   string
 > => {
   const fromDb = restore(keys.createEventPage);
+
   if (fromDb === undefined) return Result.ok(undefined);
+
   const obj = Json.parse(fromDb);
+
   return Result.isErr(obj)
     ? Result.err(obj.value)
     : Result.ok(EventSchedule.fill(obj.value));
@@ -52,12 +57,14 @@ const saveCreateEventPageTemp = (
   eventSchedule: EventSchedulePicked,
 ): Result<undefined, string> => {
   const str = Json.stringify(eventSchedule);
+
   if (Result.isErr(str)) {
     return Result.err(str.value);
-  } else {
-    save(keys.createEventPage, str.value);
-    return Result.ok(undefined);
   }
+
+  save(keys.createEventPage, str.value ?? '');
+
+  return Result.ok(undefined);
 };
 
 export const EventScheduleAppLocalStorage = {
