diff --git experimental/packages/apps/event-schedule-app/src/functions/normalize-event-schedule.ts apps/event-schedule-app/src/functions/normalize-event-schedule.mts
index 862b4d5ba..3539cb445 100644
--- experimental/packages/apps/event-schedule-app/src/functions/normalize-event-schedule.ts
+++ apps/event-schedule-app/src/functions/normalize-event-schedule.mts
@@ -1,19 +1,22 @@
-import { compareDatetimeRange } from '@noshiro/io-ts-types';
-import { datetimeRange2str } from '../constants';
+import { Arr, pipe } from 'ts-data-forge';
+import { compareDatetimeRange } from 'ts-fortress-types';
+import { datetimeRange2str } from '../constants/index.mjs';
+import { sortedTuple } from '../utils-ported/index.mjs';
 
 export const normalizeEventSchedule = (
   eventSchedule: EventSchedule,
-): EventSchedule => ({
-  title: eventSchedule.title.trim(),
-  notes: `${eventSchedule.notes.trim()}\n`,
-  datetimeSpecification: eventSchedule.datetimeSpecification,
-  datetimeRangeList: pipe(eventSchedule.datetimeRangeList)
-    .chain((list) => Arr.uniqBy(list, datetimeRange2str))
-    .chain((list) => Tpl.sorted(list, compareDatetimeRange)).value,
-  answerDeadline: eventSchedule.answerDeadline,
-  answerIcons: eventSchedule.answerIcons,
-  notificationSettings: eventSchedule.notificationSettings,
-  timezoneOffsetMinutes: eventSchedule.timezoneOffsetMinutes,
-  author: eventSchedule.author,
-  archivedBy: eventSchedule.archivedBy,
-});
+): EventSchedule =>
+  ({
+    title: eventSchedule.title.trim(),
+    notes: `${eventSchedule.notes.trim()}\n`,
+    datetimeSpecification: eventSchedule.datetimeSpecification,
+    datetimeRangeList: pipe(eventSchedule.datetimeRangeList)
+      .map((list) => Arr.uniqBy(list, datetimeRange2str))
+      .map((list) => sortedTuple(list, compareDatetimeRange)).value,
+    answerDeadline: eventSchedule.answerDeadline,
+    answerIcons: eventSchedule.answerIcons,
+    notificationSettings: eventSchedule.notificationSettings,
+    timezoneOffsetMinutes: eventSchedule.timezoneOffsetMinutes,
+    author: eventSchedule.author,
+    archivedBy: eventSchedule.archivedBy,
+  }) as const;
