diff --git experimental/packages/apps/event-schedule-app/src/functions/multiple-date-picker/generate-calendar.test.ts apps/event-schedule-app/src/functions/multiple-date-picker/generate-calendar.test.mts
index a953d3b98..9d610a307 100644
--- experimental/packages/apps/event-schedule-app/src/functions/multiple-date-picker/generate-calendar.test.ts
+++ apps/event-schedule-app/src/functions/multiple-date-picker/generate-calendar.test.mts
@@ -1,4 +1,6 @@
-import { generateCalendar } from './generate-calendar';
+import { asSafeUint } from 'ts-data-forge';
+import { type SafeUint } from 'ts-type-forge';
+import { generateCalendar } from './generate-calendar.mjs';
 
 test('generate-calendar', () => {
   /*
@@ -10,8 +12,10 @@ test('generate-calendar', () => {
    * |19|20|21|22|23|24|25|   |
    * |26|27|28|29|30|31| 1| --+
    */
-  const year: YearEnum = toSafeUint(2020);
+  const year: SafeUint = asSafeUint(2020);
+
   const month = 7;
+
   expect(generateCalendar(year, month)).toStrictEqual([
     [
       { year: 2020, month: 6, date: 28 },
