diff --git experimental/packages/apps/event-schedule-app/src/functions/answers-page/index.ts apps/event-schedule-app/src/functions/answers-page/index.mts
index 14874b35a..cdf44ac92 100644
--- experimental/packages/apps/event-schedule-app/src/functions/answers-page/index.ts
+++ apps/event-schedule-app/src/functions/answers-page/index.mts
@@ -1,3 +1,3 @@
-export * from './create-answer-selection-map-from-answers';
-export * from './create-answer-summary';
-export * from './create-answer-table';
+export * from './create-answer-selection-map-from-answers.mjs';
+export * from './create-answer-summary.mjs';
+export * from './create-answer-table.mjs';
