diff --git a/packages/v5.4/output/lib-files/lib.es2022.regexp.d.ts b/packages/v5.5/output/lib-files/lib.es2022.regexp.d.ts
index ec3157c..2058252 100644
--- a/packages/v5.4/output/lib-files/lib.es2022.regexp.d.ts
+++ b/packages/v5.5/output/lib-files/lib.es2022.regexp.d.ts
@@ -17,7 +17,7 @@ interface RegExpIndicesArray extends Array<readonly [number, number]> {
 
 interface RegExp {
   /**
-   * Returns a Boolean value indicating the state of the hasIndices flag (d) used with with a regular expression.
+   * Returns a Boolean value indicating the state of the hasIndices flag (d) used with a regular expression.
    * Default is false. Read-only.
    */
   readonly hasIndices: boolean;
