diff --git a/packages/v5.4/output/lib-files/lib.webworker.asynciterable.d.ts b/packages/v5.5/output/lib-files/lib.webworker.asynciterable.d.ts
index f4ee116..e0646f3 100644
--- a/packages/v5.4/output/lib-files/lib.webworker.asynciterable.d.ts
+++ b/packages/v5.5/output/lib-files/lib.webworker.asynciterable.d.ts
@@ -13,3 +13,10 @@ interface FileSystemDirectoryHandle {
   keys(): AsyncIterableIterator<string>;
   values(): AsyncIterableIterator<FileSystemHandle>;
 }
+
+interface ReadableStream<R = unknown> {
+  [Symbol.asyncIterator](
+    options?: ReadableStreamIteratorOptions,
+  ): AsyncIterableIterator<R>;
+  values(options?: ReadableStreamIteratorOptions): AsyncIterableIterator<R>;
+}
