diff --git experimental/packages/apps/event-schedule-app-shared/src/v9/types/record/answer.mts apps/event-schedule-app-shared/src/v9/types/record/answer.mts
index 6bc5965ef..58b662e75 100644
--- experimental/packages/apps/event-schedule-app-shared/src/v9/types/record/answer.mts
+++ apps/event-schedule-app-shared/src/v9/types/record/answer.mts
@@ -1,4 +1,4 @@
-import * as t from '@noshiro/io-ts';
+import * as t from 'ts-fortress';
 import { AnswerId, Weight } from '../named-primitive-types.mjs';
 import { AnswerSelection } from './answer-selection.mjs';
 import { User } from './base/index.mjs';
