diff --git experimental/packages/apps/event-schedule-app-shared/src/v9/types/record/event-schedule.test.mts apps/event-schedule-app-shared/src/v9/types/record/event-schedule.test.mts
index 4335b472a..a3c51bc0a 100644
--- experimental/packages/apps/event-schedule-app-shared/src/v9/types/record/event-schedule.test.mts
+++ apps/event-schedule-app-shared/src/v9/types/record/event-schedule.test.mts
@@ -1,11 +1,12 @@
-import { DatetimeRange, type Ymdhm } from '@noshiro/io-ts-types';
-import { DateUtils, expectType } from '@noshiro/ts-utils';
+import { Arr, expectType } from 'ts-data-forge';
+import { DateUtils, DatetimeRange, type Ymdhm } from 'ts-fortress-types';
+import { type NonEmptyArray } from 'ts-type-forge';
 import { type DatetimeSpecificationEnumType } from '../enum/index.mjs';
 import { type AnswerIconSettings } from './answer-icon-settings.mjs';
 import {
+  User,
   createAnswerIconSettingType,
   type NotificationSettings,
-  User,
 } from './base/index.mjs';
 import { EventSchedule } from './event-schedule.mjs';
 
@@ -31,7 +32,7 @@ describe('EventSchedule', () => {
       title: '',
       notes: '',
       datetimeSpecification: 'noStartEndSpecified',
-      datetimeRangeList: [DatetimeRange.defaultValue],
+      datetimeRangeList: Arr.asNonEmptyArray([DatetimeRange.defaultValue]),
       answerDeadline: 'none',
       answerIcons: {
         good: createAnswerIconSettingType(10).defaultValue,
@@ -42,20 +43,23 @@ describe('EventSchedule', () => {
       timezoneOffsetMinutes: DateUtils.today().getTimezoneOffset(),
       author: User.defaultValue,
       archivedBy: [],
-    };
+    } as const;
 
-    expect(EventSchedule.defaultValue).toStrictEqual(defaultValue);
+    assert.deepStrictEqual(EventSchedule.defaultValue, defaultValue);
   });
 
   describe('is', () => {
     test('defaultValue should be true', () => {
-      expect(EventSchedule.is(EventSchedule.defaultValue)).toBe(true);
+      assert.isTrue(EventSchedule.is(EventSchedule.defaultValue));
     });
   });
 
   describe('fill', () => {
     test('defaultValue should be true', () => {
-      expect(EventSchedule.fill({})).toStrictEqual(EventSchedule.defaultValue);
+      assert.deepStrictEqual(
+        EventSchedule.fill({}),
+        EventSchedule.defaultValue,
+      );
     });
   });
 });
