diff --git experimental/packages/apps/event-schedule-app-shared/src/v9/types/record/event-list-item.test.mts apps/event-schedule-app-shared/src/v9/types/record/event-list-item.test.mts
index cc83ef5b1..48933b477 100644
--- experimental/packages/apps/event-schedule-app-shared/src/v9/types/record/event-list-item.test.mts
+++ apps/event-schedule-app-shared/src/v9/types/record/event-list-item.test.mts
@@ -1,4 +1,4 @@
-import { expectType } from '@noshiro/ts-utils';
+import { expectType } from 'ts-data-forge';
 import { type Answer } from './answer.mjs';
 import { EventListItem } from './event-list-item.mjs';
 import { EventSchedule } from './event-schedule.mjs';
@@ -37,19 +37,23 @@ describe('EventListItem', () => {
       answersMetadata: {
         lastUpdate: '',
       },
-    };
-    expect(EventListItem.defaultValue).toStrictEqual(defaultValue);
+    } as const;
+
+    assert.deepStrictEqual(EventListItem.defaultValue, defaultValue);
   });
 
   describe('is', () => {
     test('defaultValue should be true', () => {
-      expect(EventListItem.is(EventListItem.defaultValue)).toBe(true);
+      assert.isTrue(EventListItem.is(EventListItem.defaultValue));
     });
   });
 
   describe('fill', () => {
     test('fill result should be the defaultValue', () => {
-      expect(EventListItem.fill({})).toStrictEqual(EventListItem.defaultValue);
+      assert.deepStrictEqual(
+        EventListItem.fill({}),
+        EventListItem.defaultValue,
+      );
     });
   });
 });
