diff --git experimental/packages/apps/event-schedule-app-shared/src/v9/types/record/event-list-item.mts apps/event-schedule-app-shared/src/v9/types/record/event-list-item.mts
index d6b322b72..de356361c 100644
--- experimental/packages/apps/event-schedule-app-shared/src/v9/types/record/event-list-item.mts
+++ apps/event-schedule-app-shared/src/v9/types/record/event-list-item.mts
@@ -1,4 +1,4 @@
-import * as t from '@noshiro/io-ts';
+import * as t from 'ts-fortress';
 import { Answer } from './answer.mjs';
 import { EventSchedule } from './event-schedule.mjs';
 
