diff --git experimental/packages/apps/event-schedule-app/src/components/pages/not-found-page.tsx apps/event-schedule-app/src/components/pages/not-found-page.tsx
index b26339a81..3cc5f0844 100644
--- experimental/packages/apps/event-schedule-app/src/components/pages/not-found-page.tsx
+++ apps/event-schedule-app/src/components/pages/not-found-page.tsx
@@ -1,7 +1,9 @@
 import { Icon, NonIdealState } from '@blueprintjs/core';
-import { Anchor } from '@noshiro/react-blueprintjs-utils';
-import { Routes } from '../../constants';
-import { Router } from '../../store';
+import { Anchor } from 'react-blueprintjs-utils';
+import { memoNamed } from 'react-utils';
+import { createRouterLinkClickHandler } from 'tiny-router-react-hooks';
+import { Routes } from '../../constants/index.mjs';
+import { Router } from '../../store/index.mjs';
 
 export const NotFoundPage = memoNamed('NotFoundPage', () => (
   <NonIdealState action={action} icon={'search'} title={dict.pageNotFound} />
