diff --git experimental/packages/apps/event-schedule-app/src/components/pages/reset-password-page.tsx apps/event-schedule-app/src/components/pages/reset-password-page.tsx
index 063e16810..377f71afd 100644
--- experimental/packages/apps/event-schedule-app/src/components/pages/reset-password-page.tsx
+++ apps/event-schedule-app/src/components/pages/reset-password-page.tsx
@@ -1,16 +1,17 @@
 import { Button, FormGroup } from '@blueprintjs/core';
-import { BpInput } from '@noshiro/react-blueprintjs-utils';
-import { ResetPasswordPageStore } from '../../store';
-import { Label } from '../atoms';
-import { SignInStyled } from '../styled';
+import styled from '@emotion/styled';
+import { BpInput } from 'react-blueprintjs-utils';
+import { memoNamed } from 'react-utils';
+import { useObservableValue } from 'synstate-react-hooks';
+import { ResetPasswordPageStore } from '../../store/index.mjs';
+import { Label } from '../atoms/index.mjs';
+import { SignInStyled } from '../styled/index.mjs';
 
 const dc = dict.register;
 
 const returnFalse = (): false => false;
 
-type Props = Readonly<{
-  hidePasswordResetForm: () => void;
-}>;
+type Props = Readonly<{ hidePasswordResetForm: () => void }>;
 
 export const ResetPasswordPage = memoNamed<Props>(
   'ResetPasswordPage',
@@ -25,7 +26,7 @@ export const ResetPasswordPage = memoNamed<Props>(
             <Button
               icon={'chevron-left'}
               intent={'none'}
-              minimal={true}
+              variant={'minimal'}
               onClick={hidePasswordResetForm}
             >
               {dc.resetPasswordMode.back}
@@ -44,7 +45,7 @@ export const ResetPasswordPage = memoNamed<Props>(
                 <BpInput
                   autoComplete={'email'}
                   // eslint-disable-next-line jsx-a11y/no-autofocus
-                  autoFocus={true}
+                  autoFocus
                   disabled={formState.isWaitingResponse}
                   intent={emailFormIntent}
                   type={'email'}
@@ -57,7 +58,7 @@ export const ResetPasswordPage = memoNamed<Props>(
             <SignInStyled.ButtonWrapper>
               <Button
                 disabled={enterButtonDisabled}
-                fill={true}
+                fill
                 intent={'primary'}
                 loading={formState.isWaitingResponse}
                 onClick={ResetPasswordPageStore.enterClickHandler}
