diff --git experimental/packages/apps/event-schedule-app-shared/src/v4/types/enum/answer-icon-id.mts apps/event-schedule-app-shared/src/v4/types/enum/answer-icon-id.mts
index 280770c65..d7177d552 100644
--- experimental/packages/apps/event-schedule-app-shared/src/v4/types/enum/answer-icon-id.mts
+++ apps/event-schedule-app-shared/src/v4/types/enum/answer-icon-id.mts
@@ -1,2 +1,3 @@
 export type AnswerIconId = 'fair' | 'good' | 'poor';
+
 export type AnswerIconIdWithNone = AnswerIconId | 'none';
