diff --git experimental/packages/apps/event-schedule-app/src/functions/reducers/sign-in-and-register-page/reset-password-page-state.ts apps/event-schedule-app/src/functions/reducers/sign-in-and-register-page/reset-password-page-state.mts
index 6cb83e823..7667ab239 100644
--- experimental/packages/apps/event-schedule-app/src/functions/reducers/sign-in-and-register-page/reset-password-page-state.ts
+++ apps/event-schedule-app/src/functions/reducers/sign-in-and-register-page/reset-password-page-state.mts
@@ -1,9 +1,10 @@
+import { type Reducer } from '../../../utils-ported/index.mjs';
 import {
   emailInputHasError,
   emailInputInitialState,
   emailInputStateReducer,
   type EmailInputState,
-} from '../input-state';
+} from '../input-state/index.mjs';
 
 export type ResetPasswordPageState = Readonly<{
   email: EmailInputState;
@@ -37,14 +38,13 @@ export const resetPasswordPageStateReducer: Reducer<
 > = (state, action) => {
   switch (action.type) {
     case 'inputEmail':
-      return Obj.set(
-        state,
-        'email',
-        emailInputStateReducer(state.email, {
+      return {
+        ...state,
+        email: emailInputStateReducer(state.email, {
           type: 'input',
           payload: action.payload,
         }),
-      );
+      };
 
     case 'setEmailError':
       return {
@@ -76,7 +76,7 @@ export const resetPasswordPageStateReducer: Reducer<
     }
 
     case 'done':
-      return Obj.set(state, 'isWaitingResponse', false);
+      return { ...state, isWaitingResponse: false };
 
     case 'reset':
       return resetPasswordPageInitialState;
