diff --git experimental/packages/apps/event-schedule-app-shared/src/v7/types/record/answer-icon-settings.mts apps/event-schedule-app-shared/src/v7/types/record/answer-icon-settings.mts
index 471d1c400..ce67823df 100644
--- experimental/packages/apps/event-schedule-app-shared/src/v7/types/record/answer-icon-settings.mts
+++ apps/event-schedule-app-shared/src/v7/types/record/answer-icon-settings.mts
@@ -1,4 +1,4 @@
-import * as t from '@noshiro/io-ts';
+import * as t from 'ts-fortress';
 import { answerIconSettingTypeDef } from './base/index.mjs';
 
 export const answerIconSettingsTypeDef = t.record({
