diff --git a/packages/v5.3/temp/copied/lib.es2021.intl.d.ts b/packages/v5.4/temp/copied/lib.es2021.intl.d.ts
index 4c3846e..637bec0 100644
--- a/packages/v5.3/temp/copied/lib.es2021.intl.d.ts
+++ b/packages/v5.4/temp/copied/lib.es2021.intl.d.ts
@@ -150,10 +150,7 @@ declare namespace Intl {
      *
      * [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/ListFormat).
      */
-    new (
-      locales?: BCP47LanguageTag | BCP47LanguageTag[],
-      options?: ListFormatOptions,
-    ): ListFormat;
+    new (locales?: LocalesArgument, options?: ListFormatOptions): ListFormat;
 
     /**
      * Returns an array containing those of the provided locales that are
@@ -172,8 +169,8 @@ declare namespace Intl {
      * [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/ListFormat/supportedLocalesOf).
      */
     supportedLocalesOf(
-      locales: BCP47LanguageTag | BCP47LanguageTag[],
+      locales: LocalesArgument,
       options?: Pick<ListFormatOptions, 'localeMatcher'>,
-    ): BCP47LanguageTag[];
+    ): UnicodeBCP47LocaleIdentifier[];
   };
 }
