diff --git a/packages/v5.3/temp/copied/lib.es2019.full.d.ts b/packages/v5.4/temp/copied/lib.es2019.full.d.ts
index 8a9e80c..6031b81 100644
--- a/packages/v5.3/temp/copied/lib.es2019.full.d.ts
+++ b/packages/v5.4/temp/copied/lib.es2019.full.d.ts
@@ -20,3 +20,4 @@ and limitations under the License.
 /// <reference lib="webworker.importscripts" />
 /// <reference lib="scripthost" />
 /// <reference lib="dom.iterable" />
+/// <reference lib="dom.asynciterable" />
