diff --git experimental/packages/apps/event-schedule-app/src/api/auth/send-password-reset-email.ts apps/event-schedule-app/src/api/auth/send-password-reset-email.mts
index cc5f964f6..5b5575878 100644
--- experimental/packages/apps/event-schedule-app/src/api/auth/send-password-reset-email.ts
+++ apps/event-schedule-app/src/api/auth/send-password-reset-email.mts
@@ -1,13 +1,15 @@
 import { sendPasswordResetEmail as sendPasswordResetEmail_ } from 'firebase/auth';
-import { fbAuth } from '../../initialize-firebase';
-import { assertIsCredentialError } from '../../types';
+import { Result } from 'ts-data-forge';
+import { fbAuth } from '../../initialize-firebase.mjs';
+import { assertIsCredentialError } from '../../types/index.mjs';
 
 export const sendPasswordResetEmail = (
   email: string,
 ): Promise<Result<void, Readonly<{ code: string; message: string }>>> =>
-  Result.fromPromise(sendPasswordResetEmail_(fbAuth, email)).then((a) =>
-    Result.mapErr(a, (error) => {
+  Result.fromPromise(sendPasswordResetEmail_(fbAuth, email)).then(
+    Result.mapErr((error) => {
       assertIsCredentialError(error);
+
       return error;
     }),
   );
