diff --git experimental/packages/utils/react-blueprintjs-utils/src/components/slider/range-slider-labels-hook.mts apps/react-blueprintjs-utils/src/components/slider/range-slider-labels-hook.mts
index 70373622f..a603b2cdd 100644
--- experimental/packages/utils/react-blueprintjs-utils/src/components/slider/range-slider-labels-hook.mts
+++ apps/react-blueprintjs-utils/src/components/slider/range-slider-labels-hook.mts
@@ -1,5 +1,6 @@
-import { Arr, Num, pipe } from '@noshiro/ts-utils';
-import { useMemo } from 'react';
+import * as React from 'react';
+import { Arr, Num, pipe } from 'ts-data-forge';
+import { type DeepReadonly, type UintRange } from 'ts-type-forge';
 import { formatPercentage } from './utils.mjs';
 
 export const useRangeSliderLabels = ({
@@ -13,7 +14,7 @@ export const useRangeSliderLabels = ({
   min: number;
   toRatio: (x: number) => number | undefined;
   labelFractionDigits?: UintRange<0, 21>;
-  customLabelValues?: readonly number[];
+  customLabelValues?: number[];
 }>): Readonly<{
   customLabelValuesWithRatio: DeepReadonly<
     {
@@ -25,19 +26,20 @@ export const useRangeSliderLabels = ({
   minLabel: string;
   maxLabel: string;
 }> => {
-  const customLabelValuesWithRatio = useMemo(() => {
+  const customLabelValuesWithRatio = React.useMemo(() => {
     if (customLabelValues === undefined) return [];
 
     const isInRangeInclusive = Num.isInRangeInclusive(min, max);
 
     if (!customLabelValues.every(isInRangeInclusive)) {
       console.error('all values should be in [min, max].');
+
       return [];
     }
 
     return pipe(customLabelValues)
-      .chain((list) => Arr.uniq(list))
-      .chain((list) =>
+      .map(Arr.uniq)
+      .map((list) =>
         list.map((x) => ({
           value: x,
           ratio: toRatio(x) ?? 0,
@@ -46,13 +48,14 @@ export const useRangeSliderLabels = ({
       ).value;
   }, [customLabelValues, min, max, toRatio]);
 
-  const toFixed = useMemo(
+  const toFixed = React.useMemo(
     () => (n: number) => n.toFixed(labelFractionDigits),
     [labelFractionDigits],
   );
 
-  const minLabel = useMemo(() => toFixed(min), [min, toFixed]);
-  const maxLabel = useMemo(() => toFixed(max), [max, toFixed]);
+  const minLabel = React.useMemo(() => toFixed(min), [min, toFixed]);
+
+  const maxLabel = React.useMemo(() => toFixed(max), [max, toFixed]);
 
   return {
     customLabelValuesWithRatio,
