diff --git experimental/packages/utils/react-blueprintjs-utils/src/components/wrapper/bp-password-input.tsx apps/react-blueprintjs-utils/src/components/wrapper/bp-password-input.tsx
index 207579d07..34ab4b363 100644
--- experimental/packages/utils/react-blueprintjs-utils/src/components/wrapper/bp-password-input.tsx
+++ apps/react-blueprintjs-utils/src/components/wrapper/bp-password-input.tsx
@@ -1,6 +1,6 @@
-import { memoNamed } from '@noshiro/react-utils';
 import { useBoolState } from 'better-react-use-state';
-import { useMemo } from 'react';
+import * as React from 'react';
+import { memoNamed } from 'react-utils';
 import { BpButton } from './bp-button.js';
 import { BpInput, type BpInputProps } from './bp-input.js';
 
@@ -17,11 +17,11 @@ export const BpPasswordInput = memoNamed<BpPasswordInputProps>(
     const [passwordIsOpen, { toggleState: onToggleVisibilityClick }] =
       useBoolState(false);
 
-    const rightElement = useMemo(
+    const rightElement = React.useMemo(
       () => (
         <BpButton
           icon={passwordIsOpen ? 'eye-open' : 'eye-off'}
-          minimal={true}
+          variant={'minimal'}
           onClick={onToggleVisibilityClick}
         />
       ),
