diff --git experimental/packages/apps/event-schedule-app/src/components/styled/buttons-wrapper.tsx apps/event-schedule-app/src/components/styled/buttons-wrapper.tsx
index 0c7940f18..2b0257223 100644
--- experimental/packages/apps/event-schedule-app/src/components/styled/buttons-wrapper.tsx
+++ apps/event-schedule-app/src/components/styled/buttons-wrapper.tsx
@@ -1,3 +1,5 @@
+import styled from '@emotion/styled';
+
 export const SingleButtonWrapper = styled.div`
   margin: 5px;
 `;
@@ -7,14 +9,10 @@ export const ButtonsWrapperNowrap = styled.div`
   display: flex;
   align-items: center;
 
-  & > * {
-    margin-right: 5px; // margin between this button and the next button
-    margin-bottom: 5px; // margin between this button and the buttons on next line
-  }
-  &::last-of-type {
-    margin-right: 0;
-    margin-bottom: 0;
-  }
+  & > * { margin-right: 5px; // margin between this button and the next button
+    margin-bottom: 5px; // margin between this button and the buttons on next line }
+  &::last-of-type { margin-right: 0;
+    margin-bottom: 0; }
 `;
 
 export const ButtonsWrapper = styled(ButtonsWrapperNowrap)`
