diff --git a/packages/v5.8/temp/copied/lib.esnext.d.ts b/packages/v5.9/temp/copied/lib.esnext.d.ts
index 4a8e433..a2af85f 100644
--- a/packages/v5.8/temp/copied/lib.esnext.d.ts
+++ b/packages/v5.9/temp/copied/lib.esnext.d.ts
@@ -24,3 +24,5 @@ and limitations under the License.
 /// <reference lib="esnext.iterator" />
 /// <reference lib="esnext.promise" />
 /// <reference lib="esnext.float16" />
+/// <reference lib="esnext.error" />
+/// <reference lib="esnext.sharedmemory" />
