diff --git experimental/packages/apps/lambda-calculus-interpreter-react/src/constants/dictionary/dictionary.ts experimental/packages/apps/lambda-calculus-interpreter-react/src/constants/dictionary/dictionary.ts
deleted file mode 100644
index 214d2afd4..000000000
--- experimental/packages/apps/lambda-calculus-interpreter-react/src/constants/dictionary/dictionary.ts
+++ /dev/null
@@ -1 +0,0 @@
-export const dict = {} as const;
