diff --git a/packages/v5.2/output-branded/lib-files/lib.es2023.array.d.ts b/packages/v5.3/output-branded/lib-files/lib.es2023.array.d.ts
index b76d631..032775f 100644
--- a/packages/v5.2/output-branded/lib-files/lib.es2023.array.d.ts
+++ b/packages/v5.3/output-branded/lib-files/lib.es2023.array.d.ts
@@ -249,7 +249,7 @@ interface Int8Array {
    * Returns a copy of an array with its elements sorted.
    * @param compareFn Function used to determine the order of the elements. It is expected to return
    * a negative value if the first argument is less than the second argument, zero if they're equal, and a positive
-   * value otherwise. If omitted, the elements are sorted in ascending, ASCII character order.
+   * value otherwise. If omitted, the elements are sorted in ascending order.
    * ```ts
    * const myNums = Uint8Array.from([11, 2, 22, 1]);
    * myNums.toSorted((a, b) => a - b) // Uint8Array(4) [1, 2, 11, 22]
@@ -323,7 +323,7 @@ interface Uint8Array {
    * Returns a copy of an array with its elements sorted.
    * @param compareFn Function used to determine the order of the elements. It is expected to return
    * a negative value if the first argument is less than the second argument, zero if they're equal, and a positive
-   * value otherwise. If omitted, the elements are sorted in ascending, ASCII character order.
+   * value otherwise. If omitted, the elements are sorted in ascending order.
    * ```ts
    * const myNums = Uint8Array.from([11, 2, 22, 1]);
    * myNums.toSorted((a, b) => a - b) // Uint8Array(4) [1, 2, 11, 22]
@@ -397,7 +397,7 @@ interface Uint8ClampedArray {
    * Returns a copy of an array with its elements sorted.
    * @param compareFn Function used to determine the order of the elements. It is expected to return
    * a negative value if the first argument is less than the second argument, zero if they're equal, and a positive
-   * value otherwise. If omitted, the elements are sorted in ascending, ASCII character order.
+   * value otherwise. If omitted, the elements are sorted in ascending order.
    * ```ts
    * const myNums = Uint8ClampedArray.from([11, 2, 22, 1]);
    * myNums.toSorted((a, b) => a - b) // Uint8ClampedArray(4) [1, 2, 11, 22]
@@ -471,7 +471,7 @@ interface Int16Array {
    * Returns a copy of an array with its elements sorted.
    * @param compareFn Function used to determine the order of the elements. It is expected to return
    * a negative value if the first argument is less than the second argument, zero if they're equal, and a positive
-   * value otherwise. If omitted, the elements are sorted in ascending, ASCII character order.
+   * value otherwise. If omitted, the elements are sorted in ascending order.
    * ```ts
    * const myNums = Int16Array.from([11, 2, -22, 1]);
    * myNums.toSorted((a, b) => a - b) // Int16Array(4) [-22, 1, 2, 11]
@@ -545,7 +545,7 @@ interface Uint16Array {
    * Returns a copy of an array with its elements sorted.
    * @param compareFn Function used to determine the order of the elements. It is expected to return
    * a negative value if the first argument is less than the second argument, zero if they're equal, and a positive
-   * value otherwise. If omitted, the elements are sorted in ascending, ASCII character order.
+   * value otherwise. If omitted, the elements are sorted in ascending order.
    * ```ts
    * const myNums = Uint16Array.from([11, 2, 22, 1]);
    * myNums.toSorted((a, b) => a - b) // Uint16Array(4) [1, 2, 11, 22]
@@ -619,7 +619,7 @@ interface Int32Array {
    * Returns a copy of an array with its elements sorted.
    * @param compareFn Function used to determine the order of the elements. It is expected to return
    * a negative value if the first argument is less than the second argument, zero if they're equal, and a positive
-   * value otherwise. If omitted, the elements are sorted in ascending, ASCII character order.
+   * value otherwise. If omitted, the elements are sorted in ascending order.
    * ```ts
    * const myNums = Int32Array.from([11, 2, -22, 1]);
    * myNums.toSorted((a, b) => a - b) // Int32Array(4) [-22, 1, 2, 11]
@@ -693,7 +693,7 @@ interface Uint32Array {
    * Returns a copy of an array with its elements sorted.
    * @param compareFn Function used to determine the order of the elements. It is expected to return
    * a negative value if the first argument is less than the second argument, zero if they're equal, and a positive
-   * value otherwise. If omitted, the elements are sorted in ascending, ASCII character order.
+   * value otherwise. If omitted, the elements are sorted in ascending order.
    * ```ts
    * const myNums = Uint32Array.from([11, 2, 22, 1]);
    * myNums.toSorted((a, b) => a - b) // Uint32Array(4) [1, 2, 11, 22]
@@ -767,7 +767,7 @@ interface Float32Array {
    * Returns a copy of an array with its elements sorted.
    * @param compareFn Function used to determine the order of the elements. It is expected to return
    * a negative value if the first argument is less than the second argument, zero if they're equal, and a positive
-   * value otherwise. If omitted, the elements are sorted in ascending, ASCII character order.
+   * value otherwise. If omitted, the elements are sorted in ascending order.
    * ```ts
    * const myNums = Float32Array.from([11.25, 2, -22.5, 1]);
    * myNums.toSorted((a, b) => a - b) // Float32Array(4) [-22.5, 1, 2, 11.5]
@@ -841,7 +841,7 @@ interface Float64Array {
    * Returns a copy of an array with its elements sorted.
    * @param compareFn Function used to determine the order of the elements. It is expected to return
    * a negative value if the first argument is less than the second argument, zero if they're equal, and a positive
-   * value otherwise. If omitted, the elements are sorted in ascending, ASCII character order.
+   * value otherwise. If omitted, the elements are sorted in ascending order.
    * ```ts
    * const myNums = Float64Array.from([11.25, 2, -22.5, 1]);
    * myNums.toSorted((a, b) => a - b) // Float64Array(4) [-22.5, 1, 2, 11.5]
@@ -915,7 +915,7 @@ interface BigInt64Array {
    * Returns a copy of an array with its elements sorted.
    * @param compareFn Function used to determine the order of the elements. It is expected to return
    * a negative value if the first argument is less than the second argument, zero if they're equal, and a positive
-   * value otherwise. If omitted, the elements are sorted in ascending, ASCII character order.
+   * value otherwise. If omitted, the elements are sorted in ascending order.
    * ```ts
    * const myNums = BigInt64Array.from([11n, 2n, -22n, 1n]);
    * myNums.toSorted((a, b) => Number(a - b)) // BigInt64Array(4) [-22n, 1n, 2n, 11n]
@@ -989,7 +989,7 @@ interface BigUint64Array {
    * Returns a copy of an array with its elements sorted.
    * @param compareFn Function used to determine the order of the elements. It is expected to return
    * a negative value if the first argument is less than the second argument, zero if they're equal, and a positive
-   * value otherwise. If omitted, the elements are sorted in ascending, ASCII character order.
+   * value otherwise. If omitted, the elements are sorted in ascending order.
    * ```ts
    * const myNums = BigUint64Array.from([11n, 2n, 22n, 1n]);
    * myNums.toSorted((a, b) => Number(a - b)) // BigUint64Array(4) [1n, 2n, 11n, 22n]
