diff --git experimental/packages/apps/blueprintjs-playground-styled/src/constants/dictionary/dictionary.test.ts experimental/packages/apps/blueprintjs-playground-styled/src/constants/dictionary/dictionary.test.ts
deleted file mode 100644
index cad882c33..000000000
--- experimental/packages/apps/blueprintjs-playground-styled/src/constants/dictionary/dictionary.test.ts
+++ /dev/null
@@ -1,7 +0,0 @@
-import { dict } from './dictionary';
-
-describe('dict', () => {
-  test('dict', () => {
-    expect(typeof dict).toBe('object');
-  });
-});
