diff --git experimental/packages/apps/event-schedule-app/src/components/organisms/update-user-info-dialog/update-password-dialog.tsx apps/event-schedule-app/src/components/organisms/update-user-info-dialog/update-password-dialog.tsx
index 149f1d744..0b9c32969 100644
--- experimental/packages/apps/event-schedule-app/src/components/organisms/update-user-info-dialog/update-password-dialog.tsx
+++ apps/event-schedule-app/src/components/organisms/update-user-info-dialog/update-password-dialog.tsx
@@ -1,22 +1,24 @@
 import { Button, FormGroup } from '@blueprintjs/core';
-import { BpInput } from '@noshiro/react-blueprintjs-utils';
+import { css } from '@emotion/react';
+import * as React from 'react';
+import { BpInput } from 'react-blueprintjs-utils';
+import { memoNamed } from 'react-utils';
+import { useObservableValue } from 'synstate-react-hooks';
 import {
   UpdatePasswordPageStore,
   UpdateUserInfoDialogStore,
-} from '../../../store';
-import { Label } from '../../atoms';
-import { LockButton } from '../../molecules';
-import { UpdateUserInfoDialogTemplate } from './update-user-info-dialog-template';
+} from '../../../store/index.mjs';
+import { noop } from '../../../utils-ported/index.mjs';
+import { Label } from '../../atoms/index.mjs';
+import { LockButton } from '../../molecules/index.mjs';
+import { UpdateUserInfoDialogTemplate } from './update-user-info-dialog-template.js';
 
 const dc = dict.accountSettings;
 
-type Props = Readonly<{
-  dialogIsOpen: boolean;
-  currentEmail: string | null;
-}>;
+type Props = Readonly<{ dialogIsOpen: boolean; currentEmail: string | null }>;
 
 // https://yuzu441.hateblo.jp/entry/2020/11/16/190229
-const hideStyle: React.CSSProperties = { display: 'none' };
+const hideStyle: React.CSSProperties = { display: 'none' } as const;
 
 export const UpdatePasswordDialog = memoNamed<Props>(
   'UpdatePasswordDialog',
@@ -30,7 +32,7 @@ export const UpdatePasswordDialog = memoNamed<Props>(
       newPasswordIsOpen,
     } = useObservableValue(UpdatePasswordPageStore.state);
 
-    const newPasswordLockButton = useMemo(
+    const newPasswordLockButton = React.useMemo(
       () => (
         <LockButton
           disabled={formState.isWaitingResponse}
@@ -41,7 +43,7 @@ export const UpdatePasswordDialog = memoNamed<Props>(
       [formState.isWaitingResponse, newPasswordIsOpen],
     );
 
-    const oldPasswordInputButton = useMemo(
+    const oldPasswordInputButton = React.useMemo(
       () => (
         <LockButton
           disabled={formState.isWaitingResponse}
@@ -52,7 +54,7 @@ export const UpdatePasswordDialog = memoNamed<Props>(
       [formState.isWaitingResponse, oldPasswordIsOpen],
     );
 
-    const body = useMemo(
+    const body = React.useMemo(
       () => (
         <div
           css={css`
@@ -83,7 +85,7 @@ export const UpdatePasswordDialog = memoNamed<Props>(
             <BpInput
               autoComplete={'current-password'}
               // eslint-disable-next-line jsx-a11y/no-autofocus
-              autoFocus={true}
+              autoFocus
               disabled={formState.isWaitingResponse}
               intent={oldPasswordFormIntent}
               rightElement={oldPasswordInputButton}
@@ -145,7 +147,7 @@ export const UpdatePasswordDialog = memoNamed<Props>(
       ],
     );
 
-    const submitButton = useMemo(
+    const submitButton = React.useMemo(
       () => (
         <Button
           disabled={enterButtonDisabled}
@@ -173,8 +175,11 @@ export const UpdatePasswordDialog = memoNamed<Props>(
 );
 
 const oldPasswordInputLabel = <Label>{dc.updatePassword.oldPassword}</Label>;
+
 const currentEmailInputLabel = <Label>{dc.updatePassword.currentEmail}</Label>;
+
 const newPasswordInputLabel = <Label>{dc.updatePassword.newPassword}</Label>;
+
 const verifyNewPasswordInputLabel = (
   <Label>{dc.updatePassword.verifyNewPassword}</Label>
 );
