diff --git experimental/packages/apps/lambda-calculus-interpreter-react/src/components/code-area.tsx apps/lambda-calculus-interpreter-react/src/components/code-area.tsx
index 595ab84ba..575b40b03 100644
--- experimental/packages/apps/lambda-calculus-interpreter-react/src/components/code-area.tsx
+++ apps/lambda-calculus-interpreter-react/src/components/code-area.tsx
@@ -1,3 +1,7 @@
+import styled from '@emotion/styled';
+import * as React from 'react';
+import { memoNamed } from 'react-utils';
+
 export const CodeArea = memoNamed(
   'CodeArea',
   ({
@@ -13,14 +17,16 @@ export const CodeArea = memoNamed(
     maxHeightPx: number;
     cyId: string;
   }>) => {
-    const onChange = useCallback<React.ChangeEventHandler<HTMLTextAreaElement>>(
+    const onChange = React.useCallback<
+      React.ChangeEventHandler<HTMLTextAreaElement>
+    >(
       (ev) => {
         valueChange(ev.target.value);
       },
       [valueChange],
     );
 
-    const style = useMemo<React.CSSProperties>(
+    const style = React.useMemo<React.CSSProperties>(
       () => ({
         minHeight: `${minHeightPx}px`,
         maxHeight: `${maxHeightPx}px`,
