diff --git a/packages/v5.7/output/lib-files/lib.es2020.bigint.d.ts b/packages/v5.8/output/lib-files/lib.es2020.bigint.d.ts
index 774532f..841128e 100644
--- a/packages/v5.7/output/lib-files/lib.es2020.bigint.d.ts
+++ b/packages/v5.8/output/lib-files/lib.es2020.bigint.d.ts
@@ -467,6 +467,11 @@ interface BigInt64ArrayConstructor {
     byteOffset?: number,
     length?: number,
   ): BigInt64Array<TArrayBuffer>;
+  new (
+    buffer: ArrayBuffer,
+    byteOffset?: number,
+    length?: number,
+  ): BigInt64Array<ArrayBuffer>;
   new (array: ArrayLike<bigint> | ArrayBuffer): BigInt64Array<ArrayBuffer>;
 
   /** The size in bytes of each element in the array. */
@@ -480,14 +485,13 @@ interface BigInt64ArrayConstructor {
 
   /**
    * Creates an array from an array-like or iterable object.
-   * @param arrayLike An array-like or iterable object to convert to an array.
-   * @param mapfn A mapping function to call on every element of the array.
-   * @param thisArg Value of 'this' used to invoke the mapfn.
+   * @param arrayLike An array-like object to convert to an array.
    */
   from(arrayLike: ArrayLike<bigint>): BigInt64Array<ArrayBuffer>;
+
   /**
    * Creates an array from an array-like or iterable object.
-   * @param arrayLike An array-like or iterable object to convert to an array.
+   * @param arrayLike An array-like object to convert to an array.
    * @param mapfn A mapping function to call on every element of the array.
    * @param thisArg Value of 'this' used to invoke the mapfn.
    */
@@ -496,6 +500,24 @@ interface BigInt64ArrayConstructor {
     mapfn: (v: U, k: number) => bigint,
     thisArg?: unknown,
   ): BigInt64Array<ArrayBuffer>;
+
+  /**
+   * Creates an array from an array-like or iterable object.
+   * @param elements An iterable object to convert to an array.
+   */
+  from(elements: Iterable<bigint>): BigInt64Array<ArrayBuffer>;
+
+  /**
+   * Creates an array from an array-like or iterable object.
+   * @param elements An iterable object to convert to an array.
+   * @param mapfn A mapping function to call on every element of the array.
+   * @param thisArg Value of 'this' used to invoke the mapfn.
+   */
+  from<T>(
+    elements: Iterable<T>,
+    mapfn?: (v: T, k: number) => bigint,
+    thisArg?: unknown,
+  ): BigInt64Array<ArrayBuffer>;
 }
 declare const BigInt64Array: BigInt64ArrayConstructor;
 
@@ -838,6 +860,11 @@ interface BigUint64ArrayConstructor {
     byteOffset?: number,
     length?: number,
   ): BigUint64Array<TArrayBuffer>;
+  new (
+    buffer: ArrayBuffer,
+    byteOffset?: number,
+    length?: number,
+  ): BigUint64Array<ArrayBuffer>;
   new (array: ArrayLike<bigint> | ArrayBuffer): BigUint64Array<ArrayBuffer>;
 
   /** The size in bytes of each element in the array. */
@@ -851,16 +878,39 @@ interface BigUint64ArrayConstructor {
 
   /**
    * Creates an array from an array-like or iterable object.
-   * @param arrayLike An array-like or iterable object to convert to an array.
+   * @param arrayLike An array-like object to convert to an array.
+   */
+  from(arrayLike: ArrayLike<bigint>): BigUint64Array<ArrayBuffer>;
+
+  /**
+   * Creates an array from an array-like or iterable object.
+   * @param arrayLike An array-like object to convert to an array.
    * @param mapfn A mapping function to call on every element of the array.
    * @param thisArg Value of 'this' used to invoke the mapfn.
    */
-  from(arrayLike: ArrayLike<bigint>): BigUint64Array;
   from<U>(
     arrayLike: ArrayLike<U>,
     mapfn: (v: U, k: number) => bigint,
     thisArg?: unknown,
-  ): BigUint64Array;
+  ): BigUint64Array<ArrayBuffer>;
+
+  /**
+   * Creates an array from an array-like or iterable object.
+   * @param elements An iterable object to convert to an array.
+   */
+  from(elements: Iterable<bigint>): BigUint64Array<ArrayBuffer>;
+
+  /**
+   * Creates an array from an array-like or iterable object.
+   * @param elements An iterable object to convert to an array.
+   * @param mapfn A mapping function to call on every element of the array.
+   * @param thisArg Value of 'this' used to invoke the mapfn.
+   */
+  from<T>(
+    elements: Iterable<T>,
+    mapfn?: (v: T, k: number) => bigint,
+    thisArg?: unknown,
+  ): BigUint64Array<ArrayBuffer>;
 }
 declare const BigUint64Array: BigUint64ArrayConstructor;
 
