diff --git experimental/packages/apps/event-schedule-app/src/store/update-user-info-page-state/update-password-page.ts apps/event-schedule-app/src/store/update-user-info-page-state/update-password-page.mts
index 3bdd5bf1a..76b47c7ca 100644
--- experimental/packages/apps/event-schedule-app/src/store/update-user-info-page-state/update-password-page.ts
+++ apps/event-schedule-app/src/store/update-user-info-page-state/update-password-page.mts
@@ -1,33 +1,36 @@
 import { EmailAuthProvider } from 'firebase/auth';
-import { api } from '../../api';
+import { type InitializedObservable, combine, map } from 'synstate';
+import { createBooleanState, createReducer } from 'synstate-react-hooks';
+import { Result } from 'ts-data-forge';
+import { api } from '../../api/index.mjs';
 import {
   createToaster,
   showToast,
   updatePasswordPageHasError,
   updatePasswordPageInitialState,
   updatePasswordPageStateReducer,
-} from '../../functions';
-import { Auth } from '../auth';
-import { UpdateUserInfoDialogStore } from './update-user-info-dialog-state';
+} from '../../functions/index.mjs';
+import { Auth } from '../auth.mjs';
+import { UpdateUserInfoDialogStore } from './update-user-info-dialog-state.mjs';
 
 const dc = dict.accountSettings;
 
 const toast = createToaster();
 
-const { state: formState$, dispatch } = createReducer(
+const [, dispatch, { state: formState$ }] = createReducer(
   updatePasswordPageStateReducer,
   updatePasswordPageInitialState,
 );
 
-const enterButtonDisabled$ = formState$.chain(
+const enterButtonDisabled$ = formState$.pipe(
   map((st) => st.isWaitingResponse || updatePasswordPageHasError(st)),
 );
 
-const oldPasswordFormIntent$: InitializedObservable<Intent> = formState$.chain(
+const oldPasswordFormIntent$: InitializedObservable<Intent> = formState$.pipe(
   map((st) => (st.oldPassword.error === undefined ? 'primary' : 'danger')),
 );
 
-const newPasswordFormIntent$: InitializedObservable<Intent> = formState$.chain(
+const newPasswordFormIntent$: InitializedObservable<Intent> = formState$.pipe(
   map((st) =>
     st.newPassword.password.error === undefined &&
     st.newPassword.passwordConfirmation.error === undefined
@@ -36,17 +39,23 @@ const newPasswordFormIntent$: InitializedObservable<Intent> = formState$.chain(
   ),
 );
 
-const {
-  state: oldPasswordIsOpen$,
-  setFalse: hideOldPassword,
-  toggle: toggleOldPasswordLock,
-} = createBooleanState(false);
-
-const {
-  state: newPasswordIsOpen$,
-  setFalse: hideNewPassword,
-  toggle: toggleNewPasswordLock,
-} = createBooleanState(false);
+const [
+  ,
+  {
+    state: oldPasswordIsOpen$,
+    setFalse: hideOldPassword,
+    toggle: toggleOldPasswordLock,
+  },
+] = createBooleanState(false);
+
+const [
+  ,
+  {
+    state: newPasswordIsOpen$,
+    setFalse: hideNewPassword,
+    toggle: toggleNewPasswordLock,
+  },
+] = createBooleanState(false);
 
 const state = combine([
   formState$,
@@ -55,7 +64,7 @@ const state = combine([
   newPasswordFormIntent$,
   oldPasswordIsOpen$,
   newPasswordIsOpen$,
-]).chain(
+]).pipe(
   map(
     ([
       formState,
@@ -96,6 +105,7 @@ const submit = async (user: FireAuthUser): Promise<void> => {
           type: 'setOldPasswordError',
           payload: dc.reauthenticate.message.wrongPassword,
         });
+
         break;
 
       default:
@@ -114,8 +124,10 @@ const submit = async (user: FireAuthUser): Promise<void> => {
           message: dc.reauthenticate.message.error,
           intent: 'danger',
         });
+
         break;
     }
+
     return;
   }
 
@@ -140,6 +152,7 @@ const submit = async (user: FireAuthUser): Promise<void> => {
       message: dc.updatePassword.message.error,
       intent: 'danger',
     });
+
     return;
   }
 
@@ -156,6 +169,7 @@ const submit = async (user: FireAuthUser): Promise<void> => {
 
 const enterClickHandler = (): void => {
   const enterButtonDisabled = enterButtonDisabled$.getSnapshot().value;
+
   const fireAuthUser = Auth.getFireAuthUserSnapshot();
 
   if (enterButtonDisabled || fireAuthUser === undefined) return;
@@ -165,29 +179,22 @@ const enterClickHandler = (): void => {
 };
 
 const inputOldPasswordHandler = (value: string): void => {
-  dispatch({
-    type: 'inputOldPassword',
-    payload: value,
-  });
+  dispatch({ type: 'inputOldPassword', payload: value });
 };
 
 const inputNewPasswordHandler = (value: string): void => {
-  dispatch({
-    type: 'inputNewPassword',
-    payload: value,
-  });
+  dispatch({ type: 'inputNewPassword', payload: value });
 };
 
 const inputNewPasswordConfirmationHandler = (value: string): void => {
-  dispatch({
-    type: 'inputNewPasswordConfirmation',
-    payload: value,
-  });
+  dispatch({ type: 'inputNewPasswordConfirmation', payload: value });
 };
 
 const resetAllDialogState = (): void => {
   dispatch({ type: 'reset' });
+
   hideOldPassword();
+
   hideNewPassword();
 };
 
