diff --git a/packages/v6.0/temp/copied/lib.es2015.collection.d.ts b/packages/v7.0/temp/copied/lib.es2015.collection.d.ts
index 6068004b..ac70757b 100644
--- a/packages/v6.0/temp/copied/lib.es2015.collection.d.ts
+++ b/packages/v7.0/temp/copied/lib.es2015.collection.d.ts
@@ -120,7 +120,7 @@ interface Set<T> {
    */
   has(value: T): boolean;
   /**
-   * @returns the number of (unique) elements in Set.
+   * @returns the number of (unique) elements in the Set.
    */
   readonly size: number;
 }
@@ -132,11 +132,20 @@ interface SetConstructor {
 declare var Set: SetConstructor;
 
 interface ReadonlySet<T> {
+  /**
+   * Executes a provided function once per each value in the ReadonlySet object, in insertion order.
+   */
   forEach(
     callbackfn: (value: T, value2: T, set: ReadonlySet<T>) => void,
     thisArg?: any,
   ): void;
+  /**
+   * @returns a boolean indicating whether an element with the specified value exists in the Set or not.
+   */
   has(value: T): boolean;
+  /**
+   * @returns the number of (unique) elements in the Set.
+   */
   readonly size: number;
 }
 
