diff --git a/packages/v5.3/output-branded/lib-files/lib.es2019.full.d.ts b/packages/v5.4/output-branded/lib-files/lib.es2019.full.d.ts
index 066f616..677204c 100644
--- a/packages/v5.3/output-branded/lib-files/lib.es2019.full.d.ts
+++ b/packages/v5.4/output-branded/lib-files/lib.es2019.full.d.ts
@@ -6,3 +6,4 @@
 /// <reference path="./lib.webworker.importscripts.d.ts" />
 /// <reference path="./lib.scripthost.d.ts" />
 /// <reference path="./lib.dom.iterable.d.ts" />
+/// <reference path="./lib.dom.asynciterable.d.ts" />
