diff --git experimental/packages/apps/housing-loan-calculator-app/src/components/numeric-input/yen-numeric-input.tsx apps/housing-loan-calculator-app/src/components/numeric-input/yen-numeric-input.tsx
index cea96cfbb..6c7f2162e 100644
--- experimental/packages/apps/housing-loan-calculator-app/src/components/numeric-input/yen-numeric-input.tsx
+++ apps/housing-loan-calculator-app/src/components/numeric-input/yen-numeric-input.tsx
@@ -1,6 +1,9 @@
-import { useNumericInputWithStepState } from '@noshiro/numeric-input-utils';
-import { NumericInputView } from '@noshiro/react-blueprintjs-utils';
-import { Yen } from '../../types';
+import { useNumericInputWithStepState } from 'numeric-input-utils';
+import * as React from 'react';
+import { NumericInputView } from 'react-blueprintjs-utils';
+import { memoNamed } from 'react-utils';
+import { Num, Result } from 'ts-data-forge';
+import { Yen } from '../../types/index.mjs';
 
 type Props = Readonly<{
   value: Yen;
@@ -32,14 +35,15 @@ export const YenNumericInput = memoNamed<Props>(
       onKeyDown,
     } = useNumericInputWithStepState<Yen>({
       onValueChange,
-      normalize: (n) => Yen.cast(Number(n)),
-      decode: (s) => Yen.cast(Number(s)),
+      normalize: (n) => Yen.cast(n),
+      decode: (s) =>
+        Yen.cast(Result.unwrapOkOr(Num.safeParseFloat(s), Number.NaN)),
       encode: (s) => s.toString(),
       valueFromProps,
       step,
     });
 
-    const inputProps = useMemo(
+    const inputProps = React.useMemo(
       () => ({ min, max, step, onKeyDown, 'data-e2e': cyId }),
       [max, min, step, onKeyDown, cyId],
     );
@@ -47,9 +51,9 @@ export const YenNumericInput = memoNamed<Props>(
     return (
       <NumericInputView
         disabled={disabled}
-        fillSpace={true}
+        fillSpace
         inputProps={inputProps}
-        selectOnFocus={true}
+        selectOnFocus
         valueAsStr={state}
         onDecrementMouseDown={onDecrementMouseDown}
         onIncrementMouseDown={onIncrementMouseDown}
