diff --git experimental/packages/apps/event-schedule-app/src/types/assert-is-credential-error.ts apps/event-schedule-app/src/types/assert-is-credential-error.mts
index edd0e409a..4abc82ee0 100644
--- experimental/packages/apps/event-schedule-app/src/types/assert-is-credential-error.ts
+++ apps/event-schedule-app/src/types/assert-is-credential-error.mts
@@ -1,13 +1,16 @@
+import { isRecord, isString, unknownToString } from 'ts-data-forge';
+import { hasKeyValue } from '../utils-ported/index.mjs';
+
 export const assertIsCredentialError: (
   e: unknown,
-) => asserts e is { code: string; message: string } = (e) => {
-  if (
-    !(
-      isRecord(e) &&
-      Obj.hasKeyValue(e, 'code', isString) &&
-      Obj.hasKeyValue(e, 'message', isString)
-    )
-  ) {
-    throw new Error(`object is not credential error type: ${Str.from(e)}`);
+) => asserts e is Readonly<{ code: string; message: string }> = (e) => {
+  if (!(
+    isRecord(e) &&
+    hasKeyValue(e, 'code', isString) &&
+    hasKeyValue(e, 'message', isString)
+  )) {
+    throw new Error(
+      `object is not credential error type: ${unknownToString(e)}`,
+    );
   }
 };
