diff --git experimental/packages/apps/lambda-calculus-interpreter-core/src/functions/is-lambda-term.mts apps/lambda-calculus-interpreter-core/src/functions/is-lambda-term.mts
index cc68a417f..f98dec006 100644
--- experimental/packages/apps/lambda-calculus-interpreter-core/src/functions/is-lambda-term.mts
+++ apps/lambda-calculus-interpreter-core/src/functions/is-lambda-term.mts
@@ -1,20 +1,23 @@
-import { Arr } from '@noshiro/ts-utils';
+import { Arr } from 'ts-data-forge';
 import {
   type LambdaAbstraction,
   type LambdaApplication,
   type LambdaTerm,
 } from '../types/index.mjs';
+import { hasLength } from '../utils/index.mjs';
 import { isVariable } from './is-variable.mjs';
 
 export const isLambdaTerm = (term: unknown): term is LambdaTerm =>
   isVariable(term) || isAbstraction(term) || isApplication(term);
 
 export const isAbstraction = (term: unknown): term is LambdaAbstraction => {
-  if (!Array.isArray(term) || !Arr.isArrayOfLength3(term)) return false;
+  if (!Arr.isArray(term) || !hasLength(term, 3)) return false;
+
   return term[0] === 'lambda' && isVariable(term[1]) && isLambdaTerm(term[2]);
 };
 
 export const isApplication = (term: unknown): term is LambdaApplication => {
-  if (!Array.isArray(term) || !Arr.isArrayOfLength2(term)) return false;
+  if (!Arr.isArray(term) || !hasLength(term, 2)) return false;
+
   return isLambdaTerm(term[0]) && isLambdaTerm(term[1]);
 };
