diff --git experimental/packages/apps/event-schedule-app/src/functions/reducers/icon-list-reducer.ts apps/event-schedule-app/src/functions/reducers/icon-list-reducer.mts
index 620336b62..5bb92f204 100644
--- experimental/packages/apps/event-schedule-app/src/functions/reducers/icon-list-reducer.ts
+++ apps/event-schedule-app/src/functions/reducers/icon-list-reducer.mts
@@ -1,14 +1,8 @@
+import { Obj, type Reducer } from '../../utils-ported/index.mjs';
+
 export type IconListReducerAction = Readonly<
-  | {
-      type: 'update-description';
-      iconId: AnswerIconId;
-      description: string;
-    }
-  | {
-      type: 'update-point';
-      iconId: AnswerIconId;
-      point: AnswerIconPoint;
-    }
+  | { type: 'update-description'; iconId: AnswerIconId; description: string }
+  | { type: 'update-point'; iconId: AnswerIconId; point: AnswerIconPoint }
 >;
 
 export type IconListReducerState = AnswerIconSettings;
@@ -19,12 +13,14 @@ export const iconListReducer: Reducer<
 > = (state, action) => {
   switch (action.type) {
     case 'update-description':
-      return Obj.update(state, action.iconId, (s) =>
-        Obj.set(s, 'description', action.description),
-      );
+      return Obj.update(state, action.iconId, (s) => ({
+        ...s,
+        description: action.description,
+      }));
     case 'update-point':
-      return Obj.update(state, action.iconId, (s) =>
-        Obj.set(s, 'point', action.point),
-      );
+      return Obj.update(state, action.iconId, (s) => ({
+        ...s,
+        point: action.point,
+      }));
   }
 };
