diff --git experimental/packages/apps/event-schedule-app-shared/src/v9/types/named-primitive-types.mts apps/event-schedule-app-shared/src/v9/types/named-primitive-types.mts
index 78c4fe235..d088af409 100644
--- experimental/packages/apps/event-schedule-app-shared/src/v9/types/named-primitive-types.mts
+++ apps/event-schedule-app-shared/src/v9/types/named-primitive-types.mts
@@ -1,10 +1,13 @@
-import * as t from '@noshiro/io-ts';
+import * as t from 'ts-fortress';
 
-export const AnswerId = t.simpleBrandedString('AnswerId', '');
+export const AnswerId = t.brandedString({
+  typeName: 'AnswerId',
+  defaultValue: '',
+});
 
 export type AnswerId = t.TypeOf<typeof AnswerId>;
 
-const NonNullUserId = t.simpleBrandedString('UserId', '');
+const NonNullUserId = t.brandedString({ typeName: 'UserId', defaultValue: '' });
 
 export const UserId = t.union([NonNullUserId, t.nullType], {
   defaultType: t.nullType,
@@ -12,10 +15,13 @@ export const UserId = t.union([NonNullUserId, t.nullType], {
 
 export type UserId = t.TypeOf<typeof UserId>;
 
-export const UserName = t.simpleBrandedString('UserName', '');
+export const UserName = t.brandedString({
+  typeName: 'UserName',
+  defaultValue: '',
+});
 
 export type UserName = t.TypeOf<typeof UserName>;
 
-export const Weight = t.simpleBrandedNumber('Weight', 1);
+export const Weight = t.brandedNumber({ typeName: 'Weight', defaultValue: 1 });
 
 export type Weight = t.TypeOf<typeof Weight>;
