diff --git a/packages/v5.7/temp/copied/lib.dom.iterable.d.ts b/packages/v5.8/temp/copied/lib.dom.iterable.d.ts
index 6e6acc8..ee75ddd 100644
--- a/packages/v5.7/temp/copied/lib.dom.iterable.d.ts
+++ b/packages/v5.8/temp/copied/lib.dom.iterable.d.ts
@@ -19,11 +19,6 @@ and limitations under the License.
 /// Window Iterable APIs
 /////////////////////////////
 
-interface AbortSignal {
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/AbortSignal/any_static) */
-  any(signals: Iterable<AbortSignal>): AbortSignal;
-}
-
 interface AudioParam {
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/AudioParam/setValueCurveAtTime) */
   setValueCurveAtTime(
@@ -220,6 +215,10 @@ interface IDBObjectStore {
   ): IDBIndex;
 }
 
+interface ImageTrackList {
+  [Symbol.iterator](): ArrayIterator<ImageTrack>;
+}
+
 interface MIDIInputMap extends ReadonlyMap<string, MIDIInput> {}
 
 interface MIDIOutput {
@@ -396,7 +395,7 @@ interface SubtleCrypto {
   ): Promise<CryptoKey>;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/SubtleCrypto/generateKey) */
   generateKey(
-    algorithm: 'Ed25519',
+    algorithm: 'Ed25519' | { name: 'Ed25519' },
     extractable: boolean,
     keyUsages: ReadonlyArray<'sign' | 'verify'>,
   ): Promise<CryptoKeyPair>;
@@ -492,6 +491,8 @@ interface URLSearchParams {
   values(): URLSearchParamsIterator<string>;
 }
 
+interface ViewTransitionTypeSet extends Set<string> {}
+
 interface WEBGL_draw_buffers {
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WEBGL_draw_buffers/drawBuffersWEBGL) */
   drawBuffersWEBGL(buffers: Iterable<GLenum>): void;
@@ -733,7 +734,7 @@ interface WebGL2RenderingContextOverloads {
     srcOffset?: number,
     srcLength?: GLuint,
   ): void;
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGLRenderingContext/uniformMatrix) */
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGL2RenderingContext/uniformMatrix) */
   uniformMatrix2fv(
     location: WebGLUniformLocation | null,
     transpose: GLboolean,
