diff --git a/packages/v5.5/output-branded/lib-files/lib.es2022.intl.d.ts b/packages/v5.6/output-branded/lib-files/lib.es2022.intl.d.ts
index 78f4485..b34f10a 100644
--- a/packages/v5.5/output-branded/lib-files/lib.es2022.intl.d.ts
+++ b/packages/v5.6/output-branded/lib-files/lib.es2022.intl.d.ts
@@ -31,6 +31,14 @@ declare namespace Intl {
     readonly granularity: 'grapheme' | 'word' | 'sentence';
   }
 
+  interface SegmentIterator<T> extends IteratorObject<
+    T,
+    BuiltinIteratorReturn,
+    unknown
+  > {
+    [Symbol.iterator](): SegmentIterator<T>;
+  }
+
   interface Segments {
     /**
      * Returns an object describing the segment in the original string that includes the code unit at a specified index.
@@ -40,7 +48,7 @@ declare namespace Intl {
     containing(codeUnitIndex?: number): SegmentData;
 
     /** Returns an iterator to iterate over the segments. */
-    [Symbol.iterator](): IterableIterator<SegmentData>;
+    [Symbol.iterator](): SegmentIterator<SegmentData>;
   }
 
   interface SegmentData {
