diff --git experimental/packages/apps/event-schedule-app/src/functions/reducers/update-user-info-dialog-state/update-password-page-state.ts apps/event-schedule-app/src/functions/reducers/update-user-info-dialog-state/update-password-page-state.mts
index 427190f8b..0f49bda58 100644
--- experimental/packages/apps/event-schedule-app/src/functions/reducers/update-user-info-dialog-state/update-password-page-state.ts
+++ apps/event-schedule-app/src/functions/reducers/update-user-info-dialog-state/update-password-page-state.mts
@@ -1,3 +1,4 @@
+import { type Reducer } from '../../../utils-ported/index.mjs';
 import {
   inputHasError,
   inputInitialState,
@@ -7,7 +8,7 @@ import {
   passwordWithConfirmationStateReducer,
   type InputState,
   type PasswordWithConfirmationState,
-} from '../input-state';
+} from '../input-state/index.mjs';
 
 export type UpdatePasswordPageState = Readonly<{
   oldPassword: InputState;
@@ -49,34 +50,31 @@ export const updatePasswordPageStateReducer: Reducer<
 > = (state, action) => {
   switch (action.type) {
     case 'inputOldPassword':
-      return Obj.set(
-        state,
-        'oldPassword',
-        inputStateReducer(state.oldPassword, {
+      return {
+        ...state,
+        oldPassword: inputStateReducer(state.oldPassword, {
           type: 'input',
           payload: action.payload,
         }),
-      );
+      };
 
     case 'inputNewPassword':
-      return Obj.set(
-        state,
-        'newPassword',
-        passwordWithConfirmationStateReducer(state.newPassword, {
+      return {
+        ...state,
+        newPassword: passwordWithConfirmationStateReducer(state.newPassword, {
           type: 'inputPassword',
           payload: action.payload,
         }),
-      );
+      };
 
     case 'inputNewPasswordConfirmation':
-      return Obj.set(
-        state,
-        'newPassword',
-        passwordWithConfirmationStateReducer(state.newPassword, {
+      return {
+        ...state,
+        newPassword: passwordWithConfirmationStateReducer(state.newPassword, {
           type: 'inputPasswordConfirmation',
           payload: action.payload,
         }),
-      );
+      };
 
     case 'setOldPasswordError':
       return {
@@ -122,9 +120,7 @@ export const updatePasswordPageStateReducer: Reducer<
     case 'submit': {
       const passwordNextState = passwordWithConfirmationStateReducer(
         state.newPassword,
-        {
-          type: 'submit',
-        },
+        { type: 'submit' },
       );
 
       return {
@@ -138,7 +134,7 @@ export const updatePasswordPageStateReducer: Reducer<
     }
 
     case 'done':
-      return Obj.set(state, 'isWaitingResponse', false);
+      return { ...state, isWaitingResponse: false };
 
     case 'reset':
       return updatePasswordPageInitialState;
