diff --git a/packages/v5.5/output/lib-files/lib.esnext.disposable.d.ts b/packages/v5.6/output/lib-files/lib.esnext.disposable.d.ts
index 0ba53c2..144b690 100644
--- a/packages/v5.5/output/lib-files/lib.esnext.disposable.d.ts
+++ b/packages/v5.6/output/lib-files/lib.esnext.disposable.d.ts
@@ -2,6 +2,8 @@
 /// <reference types="ts-type-forge/global" />
 
 /// <reference path="./lib.es2015.symbol.d.ts" />
+/// <reference path="./lib.es2015.iterable.d.ts" />
+/// <reference path="./lib.es2018.asynciterable.d.ts" />
 
 interface SymbolConstructor {
   /**
@@ -168,3 +170,7 @@ interface AsyncDisposableStackConstructor {
   readonly prototype: AsyncDisposableStack;
 }
 declare const AsyncDisposableStack: AsyncDisposableStackConstructor;
+
+interface IteratorObject<T, TReturn, TNext> extends Disposable {}
+
+interface AsyncIteratorObject<T, TReturn, TNext> extends AsyncDisposable {}
