diff --git a/packages/v5.5/output/lib-files/lib.es2018.asynciterable.d.ts b/packages/v5.6/output/lib-files/lib.es2018.asynciterable.d.ts
index a438b6f..c7813d9 100644
--- a/packages/v5.5/output/lib-files/lib.es2018.asynciterable.d.ts
+++ b/packages/v5.6/output/lib-files/lib.es2018.asynciterable.d.ts
@@ -12,10 +12,10 @@ interface SymbolConstructor {
   readonly asyncIterator: unique symbol;
 }
 
-interface AsyncIterator<T, TReturn = any, TNext = undefined> {
+interface AsyncIterator<T, TReturn = any, TNext = unknown> {
   // NOTE: 'next' is defined using a tuple to ensure we report the correct assignability errors in all places.
   next(
-    ...args: readonly [] | readonly [TNext]
+    ...[value]: readonly [] | readonly [TNext]
   ): Promise<IteratorResult<T, TReturn>>;
   return?(
     value?: TReturn | PromiseLike<TReturn>,
@@ -23,10 +23,28 @@ interface AsyncIterator<T, TReturn = any, TNext = undefined> {
   throw?(e?: unknown): Promise<IteratorResult<T, TReturn>>;
 }
 
-interface AsyncIterable<T> {
-  [Symbol.asyncIterator](): AsyncIterator<T>;
+interface AsyncIterable<T, TReturn = any, TNext = unknown> {
+  [Symbol.asyncIterator](): AsyncIterator<T, TReturn, TNext>;
 }
 
-interface AsyncIterableIterator<T> extends AsyncIterator<T> {
-  [Symbol.asyncIterator](): AsyncIterableIterator<T>;
+/**
+ * Describes a user-defined {@link AsyncIterator} that is also async iterable.
+ */
+interface AsyncIterableIterator<
+  T,
+  TReturn = any,
+  TNext = unknown,
+> extends AsyncIterator<T, TReturn, TNext> {
+  [Symbol.asyncIterator](): AsyncIterableIterator<T, TReturn, TNext>;
+}
+
+/**
+ * Describes an {@link AsyncIterator} produced by the runtime that inherits from the intrinsic `AsyncIterator.prototype`.
+ */
+interface AsyncIteratorObject<
+  T,
+  TReturn = any,
+  TNext = unknown,
+> extends AsyncIterator<T, TReturn, TNext> {
+  [Symbol.asyncIterator](): AsyncIteratorObject<T, TReturn, TNext>;
 }
