diff --git experimental/packages/apps/event-schedule-app/src/store/sign-in-and-register-page-state/reset-password-page.ts apps/event-schedule-app/src/store/sign-in-and-register-page-state/reset-password-page.mts
index 96967657f..48dc94604 100644
--- experimental/packages/apps/event-schedule-app/src/store/sign-in-and-register-page-state/reset-password-page.ts
+++ apps/event-schedule-app/src/store/sign-in-and-register-page-state/reset-password-page.mts
@@ -1,28 +1,31 @@
-import { api } from '../../api';
-import { Routes } from '../../constants';
+import { type InitializedObservable, combine, map } from 'synstate';
+import { createReducer } from 'synstate-react-hooks';
+import { Optional, Result } from 'ts-data-forge';
+import { api } from '../../api/index.mjs';
+import { Routes } from '../../constants/index.mjs';
 import {
   createToaster,
   resetPasswordPageHasError,
   resetPasswordPageInitialState,
   resetPasswordPageStateReducer,
   showToast,
-} from '../../functions';
-import { Router } from '../router';
+} from '../../functions/index.mjs';
+import { Router } from '../router.mjs';
 
 const dc = dict.register;
 
 const toast = createToaster();
 
-const { state: formState$, dispatch } = createReducer(
+const [, dispatch, { state: formState$ }] = createReducer(
   resetPasswordPageStateReducer,
   resetPasswordPageInitialState,
 );
 
-const enterButtonDisabled$ = formState$.chain(
+const enterButtonDisabled$ = formState$.pipe(
   map((st) => st.isWaitingResponse || resetPasswordPageHasError(st)),
 );
 
-const emailFormIntent$: InitializedObservable<Intent> = formState$.chain(
+const emailFormIntent$: InitializedObservable<Intent> = formState$.pipe(
   map((st) => (st.email.error === undefined ? 'primary' : 'danger')),
 );
 
@@ -30,7 +33,7 @@ const state = combine([
   formState$,
   enterButtonDisabled$,
   emailFormIntent$,
-]).chain(
+]).pipe(
   map(([formState, enterButtonDisabled, emailFormIntent]) => ({
     formState,
     enterButtonDisabled,
@@ -54,10 +57,12 @@ const submit = async (pageToBack: string | undefined): Promise<void> => {
           type: 'setEmailError',
           payload: dict.register.message.error.userNotFound,
         });
+
         return;
 
       default:
         console.error(sendPasswordResetEmailResult.value);
+
         dispatch({
           type: 'setOtherError',
           payload: sendPasswordResetEmailResult.value.message,
@@ -68,6 +73,7 @@ const submit = async (pageToBack: string | undefined): Promise<void> => {
           message: dc.message.error.unknownErrorOnSendingResetPasswordEmail,
           intent: 'danger',
         });
+
         return;
     }
   }
@@ -91,14 +97,13 @@ const enterClickHandler = (): void => {
   if (enterButtonDisabled$.getSnapshot().value) return;
 
   // TODO: use toast
-  submit(Maybe.unwrap(Router.pageToBack$.getSnapshot())).catch(console.error);
+  submit(Optional.unwrap(Router.pageToBack$.getSnapshot())).catch(
+    console.error,
+  );
 };
 
 const inputEmailHandler = (value: string): void => {
-  dispatch({
-    type: 'inputEmail',
-    payload: value,
-  });
+  dispatch({ type: 'inputEmail', payload: value });
 };
 
 const resetAllState = (): void => {
