diff --git a/packages/v5.5/temp/copied/lib.webworker.iterable.d.ts b/packages/v5.6/temp/copied/lib.webworker.iterable.d.ts
index 847d9ce..531b863 100644
--- a/packages/v5.5/temp/copied/lib.webworker.iterable.d.ts
+++ b/packages/v5.6/temp/copied/lib.webworker.iterable.d.ts
@@ -25,24 +25,24 @@ interface AbortSignal {
 }
 
 interface CSSNumericArray {
-  [Symbol.iterator](): IterableIterator<CSSNumericValue>;
-  entries(): IterableIterator<[number, CSSNumericValue]>;
-  keys(): IterableIterator<number>;
-  values(): IterableIterator<CSSNumericValue>;
+  [Symbol.iterator](): ArrayIterator<CSSNumericValue>;
+  entries(): ArrayIterator<[number, CSSNumericValue]>;
+  keys(): ArrayIterator<number>;
+  values(): ArrayIterator<CSSNumericValue>;
 }
 
 interface CSSTransformValue {
-  [Symbol.iterator](): IterableIterator<CSSTransformComponent>;
-  entries(): IterableIterator<[number, CSSTransformComponent]>;
-  keys(): IterableIterator<number>;
-  values(): IterableIterator<CSSTransformComponent>;
+  [Symbol.iterator](): ArrayIterator<CSSTransformComponent>;
+  entries(): ArrayIterator<[number, CSSTransformComponent]>;
+  keys(): ArrayIterator<number>;
+  values(): ArrayIterator<CSSTransformComponent>;
 }
 
 interface CSSUnparsedValue {
-  [Symbol.iterator](): IterableIterator<CSSUnparsedSegment>;
-  entries(): IterableIterator<[number, CSSUnparsedSegment]>;
-  keys(): IterableIterator<number>;
-  values(): IterableIterator<CSSUnparsedSegment>;
+  [Symbol.iterator](): ArrayIterator<CSSUnparsedSegment>;
+  entries(): ArrayIterator<[number, CSSUnparsedSegment]>;
+  keys(): ArrayIterator<number>;
+  values(): ArrayIterator<CSSUnparsedSegment>;
 }
 
 interface Cache {
@@ -67,33 +67,49 @@ interface CanvasPathDrawingStyles {
 }
 
 interface DOMStringList {
-  [Symbol.iterator](): IterableIterator<string>;
+  [Symbol.iterator](): ArrayIterator<string>;
 }
 
 interface FileList {
-  [Symbol.iterator](): IterableIterator<File>;
+  [Symbol.iterator](): ArrayIterator<File>;
 }
 
 interface FontFaceSet extends Set<FontFace> {}
 
+interface FormDataIterator<T> extends IteratorObject<
+  T,
+  BuiltinIteratorReturn,
+  unknown
+> {
+  [Symbol.iterator](): FormDataIterator<T>;
+}
+
 interface FormData {
-  [Symbol.iterator](): IterableIterator<[string, FormDataEntryValue]>;
+  [Symbol.iterator](): FormDataIterator<[string, FormDataEntryValue]>;
   /** Returns an array of key, value pairs for every entry in the list. */
-  entries(): IterableIterator<[string, FormDataEntryValue]>;
+  entries(): FormDataIterator<[string, FormDataEntryValue]>;
   /** Returns a list of keys in the list. */
-  keys(): IterableIterator<string>;
+  keys(): FormDataIterator<string>;
   /** Returns a list of values in the list. */
-  values(): IterableIterator<FormDataEntryValue>;
+  values(): FormDataIterator<FormDataEntryValue>;
+}
+
+interface HeadersIterator<T> extends IteratorObject<
+  T,
+  BuiltinIteratorReturn,
+  unknown
+> {
+  [Symbol.iterator](): HeadersIterator<T>;
 }
 
 interface Headers {
-  [Symbol.iterator](): IterableIterator<[string, string]>;
+  [Symbol.iterator](): HeadersIterator<[string, string]>;
   /** Returns an iterator allowing to go through all key/value pairs contained in this object. */
-  entries(): IterableIterator<[string, string]>;
+  entries(): HeadersIterator<[string, string]>;
   /** Returns an iterator allowing to go through all keys of the key/value pairs contained in this object. */
-  keys(): IterableIterator<string>;
+  keys(): HeadersIterator<string>;
   /** Returns an iterator allowing to go through all values of the key/value pairs contained in this object. */
-  values(): IterableIterator<string>;
+  values(): HeadersIterator<string>;
 }
 
 interface IDBDatabase {
@@ -125,11 +141,7 @@ interface IDBObjectStore {
 }
 
 interface MessageEvent<T = any> {
-  /**
-   * @deprecated
-   *
-   * [MDN Reference](https://developer.mozilla.org/docs/Web/API/MessageEvent/initMessageEvent)
-   */
+  /** @deprecated */
   initMessageEvent(
     type: string,
     bubbles?: boolean,
@@ -142,11 +154,23 @@ interface MessageEvent<T = any> {
   ): void;
 }
 
+interface StylePropertyMapReadOnlyIterator<T> extends IteratorObject<
+  T,
+  BuiltinIteratorReturn,
+  unknown
+> {
+  [Symbol.iterator](): StylePropertyMapReadOnlyIterator<T>;
+}
+
 interface StylePropertyMapReadOnly {
-  [Symbol.iterator](): IterableIterator<[string, Iterable<CSSStyleValue>]>;
-  entries(): IterableIterator<[string, Iterable<CSSStyleValue>]>;
-  keys(): IterableIterator<string>;
-  values(): IterableIterator<Iterable<CSSStyleValue>>;
+  [Symbol.iterator](): StylePropertyMapReadOnlyIterator<
+    [string, Iterable<CSSStyleValue>]
+  >;
+  entries(): StylePropertyMapReadOnlyIterator<
+    [string, Iterable<CSSStyleValue>]
+  >;
+  keys(): StylePropertyMapReadOnlyIterator<string>;
+  values(): StylePropertyMapReadOnlyIterator<Iterable<CSSStyleValue>>;
 }
 
 interface SubtleCrypto {
@@ -235,14 +259,22 @@ interface SubtleCrypto {
   ): Promise<CryptoKey>;
 }
 
+interface URLSearchParamsIterator<T> extends IteratorObject<
+  T,
+  BuiltinIteratorReturn,
+  unknown
+> {
+  [Symbol.iterator](): URLSearchParamsIterator<T>;
+}
+
 interface URLSearchParams {
-  [Symbol.iterator](): IterableIterator<[string, string]>;
+  [Symbol.iterator](): URLSearchParamsIterator<[string, string]>;
   /** Returns an array of key, value pairs for every entry in the search params. */
-  entries(): IterableIterator<[string, string]>;
+  entries(): URLSearchParamsIterator<[string, string]>;
   /** Returns a list of keys in the search params. */
-  keys(): IterableIterator<string>;
+  keys(): URLSearchParamsIterator<string>;
   /** Returns a list of values in the search params. */
-  values(): IterableIterator<string>;
+  values(): URLSearchParamsIterator<string>;
 }
 
 interface WEBGL_draw_buffers {
