diff --git a/packages/v5.5/temp/copied/lib.es2020.bigint.d.ts b/packages/v5.6/temp/copied/lib.es2020.bigint.d.ts
index 851f3fd..15dea27 100644
--- a/packages/v5.5/temp/copied/lib.es2020.bigint.d.ts
+++ b/packages/v5.6/temp/copied/lib.es2020.bigint.d.ts
@@ -278,7 +278,7 @@ interface BigInt64Array {
   copyWithin(target: number, start: number, end?: number): this;
 
   /** Yields index, value pairs for every entry in the array. */
-  entries(): IterableIterator<[number, bigint]>;
+  entries(): ArrayIterator<[number, bigint]>;
 
   /**
    * Determines whether all the members of an array satisfy the specified test.
@@ -378,7 +378,7 @@ interface BigInt64Array {
   join(separator?: string): string;
 
   /** Yields each index in the array. */
-  keys(): IterableIterator<number>;
+  keys(): ArrayIterator<number>;
 
   /**
    * Returns the index of the last occurrence of a value in an array.
@@ -539,9 +539,9 @@ interface BigInt64Array {
   valueOf(): BigInt64Array;
 
   /** Yields each value in the array. */
-  values(): IterableIterator<bigint>;
+  values(): ArrayIterator<bigint>;
 
-  [Symbol.iterator](): IterableIterator<bigint>;
+  [Symbol.iterator](): ArrayIterator<bigint>;
 
   readonly [Symbol.toStringTag]: 'BigInt64Array';
 
@@ -612,7 +612,7 @@ interface BigUint64Array {
   copyWithin(target: number, start: number, end?: number): this;
 
   /** Yields index, value pairs for every entry in the array. */
-  entries(): IterableIterator<[number, bigint]>;
+  entries(): ArrayIterator<[number, bigint]>;
 
   /**
    * Determines whether all the members of an array satisfy the specified test.
@@ -712,7 +712,7 @@ interface BigUint64Array {
   join(separator?: string): string;
 
   /** Yields each index in the array. */
-  keys(): IterableIterator<number>;
+  keys(): ArrayIterator<number>;
 
   /**
    * Returns the index of the last occurrence of a value in an array.
@@ -873,9 +873,9 @@ interface BigUint64Array {
   valueOf(): BigUint64Array;
 
   /** Yields each value in the array. */
-  values(): IterableIterator<bigint>;
+  values(): ArrayIterator<bigint>;
 
-  [Symbol.iterator](): IterableIterator<bigint>;
+  [Symbol.iterator](): ArrayIterator<bigint>;
 
   readonly [Symbol.toStringTag]: 'BigUint64Array';
 
