diff --git a/packages/v5.5/temp/copied/lib.webworker.d.ts b/packages/v5.6/temp/copied/lib.webworker.d.ts
index f213ac3..5d33ae5 100644
--- a/packages/v5.5/temp/copied/lib.webworker.d.ts
+++ b/packages/v5.6/temp/copied/lib.webworker.d.ts
@@ -338,10 +338,6 @@ interface ImageEncodeOptions {
   type?: string;
 }
 
-interface ImportMeta {
-  url: string;
-}
-
 interface JsonWebKey {
   alg?: string;
   crv?: string;
@@ -412,6 +408,10 @@ interface MediaEncodingConfiguration extends MediaConfiguration {
   type: MediaEncodingType;
 }
 
+interface MediaStreamTrackProcessorInit {
+  maxBufferSize?: number;
+}
+
 interface MessageEventInit<T = any> extends EventInit {
   data?: T;
   lastEventId?: string;
@@ -1863,6 +1863,8 @@ interface CanvasShadowStyles {
 }
 
 interface CanvasState {
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/CanvasRenderingContext2D/isContextLost) */
+  isContextLost(): boolean;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/CanvasRenderingContext2D/reset) */
   reset(): void;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/CanvasRenderingContext2D/restore) */
@@ -2221,14 +2223,12 @@ interface DOMMatrix extends DOMMatrixReadOnly {
   ): DOMMatrix;
   rotateFromVectorSelf(x?: number, y?: number): DOMMatrix;
   rotateSelf(rotX?: number, rotY?: number, rotZ?: number): DOMMatrix;
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMMatrix/scale3dSelf) */
   scale3dSelf(
     scale?: number,
     originX?: number,
     originY?: number,
     originZ?: number,
   ): DOMMatrix;
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMMatrix/scaleSelf) */
   scaleSelf(
     scaleX?: number,
     scaleY?: number,
@@ -2252,72 +2252,42 @@ declare var DOMMatrix: {
 
 /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMMatrixReadOnly) */
 interface DOMMatrixReadOnly {
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMMatrixReadOnly/a) */
   readonly a: number;
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMMatrixReadOnly/b) */
   readonly b: number;
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMMatrixReadOnly/c) */
   readonly c: number;
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMMatrixReadOnly/d) */
   readonly d: number;
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMMatrixReadOnly/e) */
   readonly e: number;
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMMatrixReadOnly/f) */
   readonly f: number;
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMMatrixReadOnly/is2D) */
   readonly is2D: boolean;
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMMatrixReadOnly/isIdentity) */
   readonly isIdentity: boolean;
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMMatrixReadOnly/m11) */
   readonly m11: number;
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMMatrixReadOnly/m12) */
   readonly m12: number;
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMMatrixReadOnly/m13) */
   readonly m13: number;
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMMatrixReadOnly/m14) */
   readonly m14: number;
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMMatrixReadOnly/m21) */
   readonly m21: number;
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMMatrixReadOnly/m22) */
   readonly m22: number;
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMMatrixReadOnly/m23) */
   readonly m23: number;
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMMatrixReadOnly/m24) */
   readonly m24: number;
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMMatrixReadOnly/m31) */
   readonly m31: number;
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMMatrixReadOnly/m32) */
   readonly m32: number;
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMMatrixReadOnly/m33) */
   readonly m33: number;
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMMatrixReadOnly/m34) */
   readonly m34: number;
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMMatrixReadOnly/m41) */
   readonly m41: number;
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMMatrixReadOnly/m42) */
   readonly m42: number;
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMMatrixReadOnly/m43) */
   readonly m43: number;
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMMatrixReadOnly/m44) */
   readonly m44: number;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMMatrixReadOnly/flipX) */
   flipX(): DOMMatrix;
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMMatrixReadOnly/flipY) */
   flipY(): DOMMatrix;
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMMatrixReadOnly/inverse) */
   inverse(): DOMMatrix;
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMMatrixReadOnly/multiply) */
   multiply(other?: DOMMatrixInit): DOMMatrix;
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMMatrixReadOnly/rotate) */
   rotate(rotX?: number, rotY?: number, rotZ?: number): DOMMatrix;
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMMatrixReadOnly/rotateAxisAngle) */
   rotateAxisAngle(
     x?: number,
     y?: number,
     z?: number,
     angle?: number,
   ): DOMMatrix;
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMMatrixReadOnly/rotateFromVector) */
   rotateFromVector(x?: number, y?: number): DOMMatrix;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMMatrixReadOnly/scale) */
   scale(
@@ -2328,29 +2298,19 @@ interface DOMMatrixReadOnly {
     originY?: number,
     originZ?: number,
   ): DOMMatrix;
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMMatrixReadOnly/scale3d) */
   scale3d(
     scale?: number,
     originX?: number,
     originY?: number,
     originZ?: number,
   ): DOMMatrix;
-  /**
-   * @deprecated
-   *
-   * [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMMatrixReadOnly/scaleNonUniform)
-   */
+  /** @deprecated */
   scaleNonUniform(scaleX?: number, scaleY?: number): DOMMatrix;
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMMatrixReadOnly/skewX) */
   skewX(sx?: number): DOMMatrix;
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMMatrixReadOnly/skewY) */
   skewY(sy?: number): DOMMatrix;
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMMatrixReadOnly/toFloat32Array) */
   toFloat32Array(): Float32Array;
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMMatrixReadOnly/toFloat64Array) */
   toFloat64Array(): Float64Array;
   toJSON(): any;
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMMatrixReadOnly/transformPoint) */
   transformPoint(point?: DOMPointInit): DOMPoint;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMMatrixReadOnly/translate) */
   translate(tx?: number, ty?: number, tz?: number): DOMMatrix;
@@ -2393,7 +2353,6 @@ interface DOMPointReadOnly {
   readonly y: number;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMPointReadOnly/z) */
   readonly z: number;
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMPointReadOnly/matrixTransform) */
   matrixTransform(matrix?: DOMMatrixInit): DOMPoint;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMPointReadOnly/toJSON) */
   toJSON(): any;
@@ -2408,15 +2367,10 @@ declare var DOMPointReadOnly: {
 
 /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMQuad) */
 interface DOMQuad {
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMQuad/p1) */
   readonly p1: DOMPoint;
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMQuad/p2) */
   readonly p2: DOMPoint;
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMQuad/p3) */
   readonly p3: DOMPoint;
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMQuad/p4) */
   readonly p4: DOMPoint;
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMQuad/getBounds) */
   getBounds(): DOMRect;
   toJSON(): any;
 }
@@ -2444,6 +2398,7 @@ interface DOMRect extends DOMRectReadOnly {
 declare var DOMRect: {
   prototype: DOMRect;
   new (x?: number, y?: number, width?: number, height?: number): DOMRect;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMRect/fromRect_static) */
   fromRect(other?: DOMRectInit): DOMRect;
 };
 
@@ -2696,15 +2651,10 @@ declare var EncodedVideoChunk: {
  * [MDN Reference](https://developer.mozilla.org/docs/Web/API/ErrorEvent)
  */
 interface ErrorEvent extends Event {
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/ErrorEvent/colno) */
   readonly colno: number;
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/ErrorEvent/error) */
   readonly error: any;
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/ErrorEvent/filename) */
   readonly filename: string;
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/ErrorEvent/lineno) */
   readonly lineno: number;
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/ErrorEvent/message) */
   readonly message: string;
 }
 
@@ -4221,7 +4171,7 @@ interface IDBTransaction extends EventTarget {
   /**
    * Returns a list of the names of object stores in the transaction's scope. For an upgrade transaction this is all object stores in the database.
    *
-   * [MDN Reference](https://developer.mozilla.org/docs/Web/API/IDBTransaction/objectStoreNames)
+   * [MDN Reference](https://developer.mozilla.org/docs/Web/API/IDBTransaction/ObjectStoreNames)
    */
   readonly objectStoreNames: DOMStringList;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/IDBTransaction/abort_event) */
@@ -4372,6 +4322,11 @@ declare var ImageData: {
   ): ImageData;
 };
 
+interface ImportMeta {
+  url: string;
+  resolve(specifier: string): string;
+}
+
 /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/KHR_parallel_shader_compile) */
 interface KHR_parallel_shader_compile {
   readonly COMPLETION_STATUS_KHR: 0x91b1;
@@ -4433,6 +4388,25 @@ declare var MediaCapabilities: {
   new (): MediaCapabilities;
 };
 
+/** [MDN Reference](https://developer.mozilla.org/docs/Web/API/MediaSourceHandle) */
+interface MediaSourceHandle {}
+
+declare var MediaSourceHandle: {
+  prototype: MediaSourceHandle;
+  new (): MediaSourceHandle;
+};
+
+/** [MDN Reference](https://developer.mozilla.org/docs/Web/API/MediaStreamTrackProcessor) */
+interface MediaStreamTrackProcessor {
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/MediaStreamTrackProcessor/readable) */
+  readonly readable: ReadableStream;
+}
+
+declare var MediaStreamTrackProcessor: {
+  prototype: MediaStreamTrackProcessor;
+  new (init: MediaStreamTrackProcessorInit): MediaStreamTrackProcessor;
+};
+
 /**
  * This Channel Messaging API interface allows us to create a new message channel and send data through it via its two MessagePort properties.
  *
@@ -4494,11 +4468,7 @@ interface MessageEvent<T = any> extends Event {
    * [MDN Reference](https://developer.mozilla.org/docs/Web/API/MessageEvent/source)
    */
   readonly source: MessageEventSource | null;
-  /**
-   * @deprecated
-   *
-   * [MDN Reference](https://developer.mozilla.org/docs/Web/API/MessageEvent/initMessageEvent)
-   */
+  /** @deprecated */
   initMessageEvent(
     type: string,
     bubbles?: boolean,
@@ -4889,7 +4859,9 @@ interface OffscreenCanvas extends EventTarget {
    * [MDN Reference](https://developer.mozilla.org/docs/Web/API/OffscreenCanvas/height)
    */
   height: number;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/OffscreenCanvas/contextlost_event) */
   oncontextlost: ((this: OffscreenCanvas, ev: Event) => any) | null;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/OffscreenCanvas/contextrestored_event) */
   oncontextrestored: ((this: OffscreenCanvas, ev: Event) => any) | null;
   /**
    * These attributes return the dimensions of the OffscreenCanvas object's bitmap.
@@ -4982,8 +4954,6 @@ interface OffscreenCanvasRenderingContext2D
     CanvasTextDrawingStyles,
     CanvasTransform {
   readonly canvas: OffscreenCanvas;
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/OffscreenCanvasRenderingContext2D/commit) */
-  commit(): void;
 }
 
 declare var OffscreenCanvasRenderingContext2D: {
@@ -5633,6 +5603,7 @@ declare var Report: {
 
 /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/ReportBody) */
 interface ReportBody {
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/ReportBody/toJSON) */
   toJSON(): any;
 }
 
@@ -5695,11 +5666,7 @@ interface Request extends Body {
    * [MDN Reference](https://developer.mozilla.org/docs/Web/API/Request/integrity)
    */
   readonly integrity: string;
-  /**
-   * Returns a boolean indicating whether or not request can outlive the global in which it was created.
-   *
-   * [MDN Reference](https://developer.mozilla.org/docs/Web/API/Request/keepalive)
-   */
+  /** Returns a boolean indicating whether or not request can outlive the global in which it was created. */
   readonly keepalive: boolean;
   /**
    * Returns request's HTTP method, which is "GET" by default.
@@ -6638,9 +6605,11 @@ declare var URL: {
   prototype: URL;
   new (url: string | URL, base?: string | URL): URL;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/URL/canParse_static) */
-  canParse(url: string | URL, base?: string): boolean;
+  canParse(url: string | URL, base?: string | URL): boolean;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/URL/createObjectURL_static) */
   createObjectURL(obj: Blob): string;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/URL/parse_static) */
+  parse(url: string | URL, base?: string | URL): URL | null;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/URL/revokeObjectURL_static) */
   revokeObjectURL(url: string): void;
 };
@@ -6733,6 +6702,7 @@ interface VideoDecoderEventMap {
 interface VideoDecoder extends EventTarget {
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/VideoDecoder/decodeQueueSize) */
   readonly decodeQueueSize: number;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/VideoDecoder/dequeue_event) */
   ondequeue: ((this: VideoDecoder, ev: Event) => any) | null;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/VideoDecoder/state) */
   readonly state: CodecState;
@@ -6771,6 +6741,7 @@ interface VideoDecoder extends EventTarget {
 declare var VideoDecoder: {
   prototype: VideoDecoder;
   new (init: VideoDecoderInit): VideoDecoder;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/VideoDecoder/isConfigSupported_static) */
   isConfigSupported(config: VideoDecoderConfig): Promise<VideoDecoderSupport>;
 };
 
@@ -6786,6 +6757,7 @@ interface VideoEncoderEventMap {
 interface VideoEncoder extends EventTarget {
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/VideoEncoder/encodeQueueSize) */
   readonly encodeQueueSize: number;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/VideoEncoder/dequeue_event) */
   ondequeue: ((this: VideoEncoder, ev: Event) => any) | null;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/VideoEncoder/state) */
   readonly state: CodecState;
@@ -6795,6 +6767,7 @@ interface VideoEncoder extends EventTarget {
   configure(config: VideoEncoderConfig): void;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/VideoEncoder/encode) */
   encode(frame: VideoFrame, options?: VideoEncoderEncodeOptions): void;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/VideoEncoder/flush) */
   flush(): Promise<void>;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/VideoEncoder/reset) */
   reset(): void;
@@ -6823,6 +6796,7 @@ interface VideoEncoder extends EventTarget {
 declare var VideoEncoder: {
   prototype: VideoEncoder;
   new (init: VideoEncoderInit): VideoEncoder;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/VideoEncoder/isConfigSupported_static) */
   isConfigSupported(config: VideoEncoderConfig): Promise<VideoEncoderSupport>;
 };
 
@@ -6854,6 +6828,7 @@ interface VideoFrame {
   clone(): VideoFrame;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/VideoFrame/close) */
   close(): void;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/VideoFrame/copyTo) */
   copyTo(
     destination: AllowSharedBufferSource,
     options?: VideoFrameCopyToOptions,
@@ -7715,7 +7690,7 @@ interface WebGL2RenderingContextBase {
   ): void;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGL2RenderingContext/clientWaitSync) */
   clientWaitSync(sync: WebGLSync, flags: GLbitfield, timeout: GLuint64): GLenum;
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGL2RenderingContext/compressedTexImage3D) */
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGLRenderingContext/compressedTexImage2D) */
   compressedTexImage3D(
     target: GLenum,
     level: GLint,
@@ -9143,6 +9118,7 @@ declare var WebGLRenderingContext: {
 };
 
 interface WebGLRenderingContextBase {
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGLRenderingContext/drawingBufferColorSpace) */
   drawingBufferColorSpace: PredefinedColorSpace;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGLRenderingContext/drawingBufferHeight) */
   readonly drawingBufferHeight: GLsizei;
@@ -10099,7 +10075,7 @@ declare var WebGLVertexArrayObject: {
   new (): WebGLVertexArrayObject;
 };
 
-/** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGLVertexArrayObjectOES) */
+/** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGLVertexArrayObject) */
 interface WebGLVertexArrayObjectOES {}
 
 interface WebSocketEventMap {
@@ -10329,24 +10305,24 @@ interface WindowOrWorkerGlobalScope {
   /**
    * Available only in secure contexts.
    *
-   * [MDN Reference](https://developer.mozilla.org/docs/Web/API/caches)
+   * [MDN Reference](https://developer.mozilla.org/docs/Web/API/Window/caches)
    */
   readonly caches: CacheStorage;
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/crossOriginIsolated) */
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/Window/crossOriginIsolated) */
   readonly crossOriginIsolated: boolean;
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/crypto_property) */
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/Window/crypto) */
   readonly crypto: Crypto;
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/indexedDB) */
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/Window/indexedDB) */
   readonly indexedDB: IDBFactory;
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/isSecureContext) */
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/Window/isSecureContext) */
   readonly isSecureContext: boolean;
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/origin) */
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/Window/origin) */
   readonly origin: string;
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/performance_property) */
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/Window/performance) */
   readonly performance: Performance;
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/atob) */
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/Window/atob) */
   atob(data: string): string;
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/btoa) */
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/Window/btoa) */
   btoa(data: string): string;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/clearInterval) */
   clearInterval(id: number | undefined): void;
@@ -10478,9 +10454,11 @@ interface WorkerGlobalScope
   onoffline: ((this: WorkerGlobalScope, ev: Event) => any) | null;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WorkerGlobalScope/online_event) */
   ononline: ((this: WorkerGlobalScope, ev: Event) => any) | null;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WorkerGlobalScope/rejectionhandled_event) */
   onrejectionhandled:
     | ((this: WorkerGlobalScope, ev: PromiseRejectionEvent) => any)
     | null;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WorkerGlobalScope/unhandledrejection_event) */
   onunhandledrejection:
     | ((this: WorkerGlobalScope, ev: PromiseRejectionEvent) => any)
     | null;
@@ -10915,7 +10893,7 @@ interface Console {
   clear(): void;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/console/count_static) */
   count(label?: string): void;
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/console/countReset_static) */
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/console/countreset_static) */
   countReset(label?: string): void;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/console/debug_static) */
   debug(...data: any[]): void;
@@ -10927,9 +10905,9 @@ interface Console {
   error(...data: any[]): void;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/console/group_static) */
   group(...data: any[]): void;
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/console/groupCollapsed_static) */
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/console/groupcollapsed_static) */
   groupCollapsed(...data: any[]): void;
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/console/groupEnd_static) */
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/console/groupend_static) */
   groupEnd(): void;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/console/info_static) */
   info(...data: any[]): void;
@@ -10939,9 +10917,9 @@ interface Console {
   table(tabularData?: any, properties?: string[]): void;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/console/time_static) */
   time(label?: string): void;
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/console/timeEnd_static) */
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/console/timeend_static) */
   timeEnd(label?: string): void;
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/console/timeLog_static) */
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/console/timelog_static) */
   timeLog(label?: string, ...data: any[]): void;
   timeStamp(label?: string): void;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/console/trace_static) */
@@ -10963,9 +10941,7 @@ declare namespace WebAssembly {
 
   /** [MDN Reference](https://developer.mozilla.org/docs/WebAssembly/JavaScript_interface/Global) */
   interface Global<T extends ValueType = ValueType> {
-    /** [MDN Reference](https://developer.mozilla.org/docs/WebAssembly/JavaScript_interface/Global/value) */
     value: ValueTypeMap[T];
-    /** [MDN Reference](https://developer.mozilla.org/docs/WebAssembly/JavaScript_interface/Global/valueOf) */
     valueOf(): ValueTypeMap[T];
   }
 
@@ -11283,9 +11259,11 @@ declare var onoffline:
 declare var ononline:
   | ((this: DedicatedWorkerGlobalScope, ev: Event) => any)
   | null;
+/** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WorkerGlobalScope/rejectionhandled_event) */
 declare var onrejectionhandled:
   | ((this: DedicatedWorkerGlobalScope, ev: PromiseRejectionEvent) => any)
   | null;
+/** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WorkerGlobalScope/unhandledrejection_event) */
 declare var onunhandledrejection:
   | ((this: DedicatedWorkerGlobalScope, ev: PromiseRejectionEvent) => any)
   | null;
@@ -11312,24 +11290,24 @@ declare var fonts: FontFaceSet;
 /**
  * Available only in secure contexts.
  *
- * [MDN Reference](https://developer.mozilla.org/docs/Web/API/caches)
+ * [MDN Reference](https://developer.mozilla.org/docs/Web/API/Window/caches)
  */
 declare var caches: CacheStorage;
-/** [MDN Reference](https://developer.mozilla.org/docs/Web/API/crossOriginIsolated) */
+/** [MDN Reference](https://developer.mozilla.org/docs/Web/API/Window/crossOriginIsolated) */
 declare var crossOriginIsolated: boolean;
-/** [MDN Reference](https://developer.mozilla.org/docs/Web/API/crypto_property) */
+/** [MDN Reference](https://developer.mozilla.org/docs/Web/API/Window/crypto) */
 declare var crypto: Crypto;
-/** [MDN Reference](https://developer.mozilla.org/docs/Web/API/indexedDB) */
+/** [MDN Reference](https://developer.mozilla.org/docs/Web/API/Window/indexedDB) */
 declare var indexedDB: IDBFactory;
-/** [MDN Reference](https://developer.mozilla.org/docs/Web/API/isSecureContext) */
+/** [MDN Reference](https://developer.mozilla.org/docs/Web/API/Window/isSecureContext) */
 declare var isSecureContext: boolean;
-/** [MDN Reference](https://developer.mozilla.org/docs/Web/API/origin) */
+/** [MDN Reference](https://developer.mozilla.org/docs/Web/API/Window/origin) */
 declare var origin: string;
-/** [MDN Reference](https://developer.mozilla.org/docs/Web/API/performance_property) */
+/** [MDN Reference](https://developer.mozilla.org/docs/Web/API/Window/performance) */
 declare var performance: Performance;
-/** [MDN Reference](https://developer.mozilla.org/docs/Web/API/atob) */
+/** [MDN Reference](https://developer.mozilla.org/docs/Web/API/Window/atob) */
 declare function atob(data: string): string;
-/** [MDN Reference](https://developer.mozilla.org/docs/Web/API/btoa) */
+/** [MDN Reference](https://developer.mozilla.org/docs/Web/API/Window/btoa) */
 declare function btoa(data: string): string;
 /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/clearInterval) */
 declare function clearInterval(id: number | undefined): void;
@@ -11470,6 +11448,7 @@ type Transferable =
   | OffscreenCanvas
   | ImageBitmap
   | MessagePort
+  | MediaSourceHandle
   | ReadableStream
   | WritableStream
   | TransformStream
