diff --git a/packages/v5.5/temp/copied/lib.es2017.object.d.ts b/packages/v5.6/temp/copied/lib.es2017.object.d.ts
index 54cd759..4960c2c 100644
--- a/packages/v5.5/temp/copied/lib.es2017.object.d.ts
+++ b/packages/v5.6/temp/copied/lib.es2017.object.d.ts
@@ -17,25 +17,25 @@ and limitations under the License.
 
 interface ObjectConstructor {
   /**
-   * Returns an array of values of the enumerable properties of an object
+   * Returns an array of values of the enumerable own properties of an object
    * @param o Object that contains the properties and methods. This can be an object that you created or an existing Document Object Model (DOM) object.
    */
   values<T>(o: { [s: string]: T } | ArrayLike<T>): T[];
 
   /**
-   * Returns an array of values of the enumerable properties of an object
+   * Returns an array of values of the enumerable own properties of an object
    * @param o Object that contains the properties and methods. This can be an object that you created or an existing Document Object Model (DOM) object.
    */
   values(o: {}): any[];
 
   /**
-   * Returns an array of key/values of the enumerable properties of an object
+   * Returns an array of key/values of the enumerable own properties of an object
    * @param o Object that contains the properties and methods. This can be an object that you created or an existing Document Object Model (DOM) object.
    */
   entries<T>(o: { [s: string]: T } | ArrayLike<T>): [string, T][];
 
   /**
-   * Returns an array of key/values of the enumerable properties of an object
+   * Returns an array of key/values of the enumerable own properties of an object
    * @param o Object that contains the properties and methods. This can be an object that you created or an existing Document Object Model (DOM) object.
    */
   entries(o: {}): [string, any][];
