diff --git a/packages/v5.5/temp/copied/lib.dom.asynciterable.d.ts b/packages/v5.6/temp/copied/lib.dom.asynciterable.d.ts
index db3dfb4..3273a50 100644
--- a/packages/v5.5/temp/copied/lib.dom.asynciterable.d.ts
+++ b/packages/v5.6/temp/copied/lib.dom.asynciterable.d.ts
@@ -19,16 +19,36 @@ and limitations under the License.
 /// Window Async Iterable APIs
 /////////////////////////////
 
+interface FileSystemDirectoryHandleAsyncIterator<T> extends AsyncIteratorObject<
+  T,
+  BuiltinIteratorReturn,
+  unknown
+> {
+  [Symbol.asyncIterator](): FileSystemDirectoryHandleAsyncIterator<T>;
+}
+
 interface FileSystemDirectoryHandle {
-  [Symbol.asyncIterator](): AsyncIterableIterator<[string, FileSystemHandle]>;
-  entries(): AsyncIterableIterator<[string, FileSystemHandle]>;
-  keys(): AsyncIterableIterator<string>;
-  values(): AsyncIterableIterator<FileSystemHandle>;
+  [Symbol.asyncIterator](): FileSystemDirectoryHandleAsyncIterator<
+    [string, FileSystemHandle]
+  >;
+  entries(): FileSystemDirectoryHandleAsyncIterator<[string, FileSystemHandle]>;
+  keys(): FileSystemDirectoryHandleAsyncIterator<string>;
+  values(): FileSystemDirectoryHandleAsyncIterator<FileSystemHandle>;
+}
+
+interface ReadableStreamAsyncIterator<T> extends AsyncIteratorObject<
+  T,
+  BuiltinIteratorReturn,
+  unknown
+> {
+  [Symbol.asyncIterator](): ReadableStreamAsyncIterator<T>;
 }
 
 interface ReadableStream<R = any> {
   [Symbol.asyncIterator](
     options?: ReadableStreamIteratorOptions,
-  ): AsyncIterableIterator<R>;
-  values(options?: ReadableStreamIteratorOptions): AsyncIterableIterator<R>;
+  ): ReadableStreamAsyncIterator<R>;
+  values(
+    options?: ReadableStreamIteratorOptions,
+  ): ReadableStreamAsyncIterator<R>;
 }
