diff --git experimental/packages/apps/event-schedule-app-shared/src/v1/types/record/answer.mts apps/event-schedule-app-shared/src/v1/types/record/answer.mts
index 6c514d665..da75468ee 100644
--- experimental/packages/apps/event-schedule-app-shared/src/v1/types/record/answer.mts
+++ apps/event-schedule-app-shared/src/v1/types/record/answer.mts
@@ -1,4 +1,5 @@
-import { DateUtils } from '@noshiro/ts-utils';
+import { DateUtils } from 'ts-fortress-types';
+import { type MergeIntersection, type StrictPick } from 'ts-type-forge';
 import {
   createAnswerId,
   createUserName,
@@ -28,7 +29,7 @@ export type Answer = Readonly<{
 
 export type PartialAnswer = Partial<
   MergeIntersection<
-    Pick<
+    StrictPick<
       Answer,
       | 'comment'
       | 'id'
@@ -56,15 +57,18 @@ export const defaultAnswer = {
 } as const satisfies Answer;
 
 const d = defaultAnswer;
-export const fillAnswer = (p?: PartialAnswer): Answer => ({
-  id: p?.id ?? d.id,
-  userName: p?.userName ?? d.userName,
-  comment: p?.comment ?? d.comment,
-  selection: (p?.selection ?? d.selection).map(fillAnswerSelection),
-  [ANSWER_KEY_CREATED_AT]:
-    (p === undefined ? undefined : p[ANSWER_KEY_CREATED_AT]) ??
-    d[ANSWER_KEY_CREATED_AT],
-  useWeight: p?.useWeight ?? d.useWeight,
-  weight: p?.weight ?? d.weight,
-  isRequiredParticipants: p?.isRequiredParticipants ?? d.isRequiredParticipants,
-});
+
+export const fillAnswer = (p?: PartialAnswer): Answer =>
+  ({
+    id: p?.id ?? d.id,
+    userName: p?.userName ?? d.userName,
+    comment: p?.comment ?? d.comment,
+    selection: (p?.selection ?? d.selection).map(fillAnswerSelection),
+    [ANSWER_KEY_CREATED_AT]:
+      (p === undefined ? undefined : p[ANSWER_KEY_CREATED_AT]) ??
+      d[ANSWER_KEY_CREATED_AT],
+    useWeight: p?.useWeight ?? d.useWeight,
+    weight: p?.weight ?? d.weight,
+    isRequiredParticipants:
+      p?.isRequiredParticipants ?? d.isRequiredParticipants,
+  }) as const;
