diff --git experimental/packages/apps/event-schedule-app/src/store/create-event-schedule-page/create-event-schedule.ts apps/event-schedule-app/src/store/create-event-schedule-page/create-event-schedule.mts
index ecc153276..8ea2f6d48 100644
--- experimental/packages/apps/event-schedule-app/src/store/create-event-schedule-page/create-event-schedule.ts
+++ apps/event-schedule-app/src/store/create-event-schedule-page/create-event-schedule.mts
@@ -2,39 +2,57 @@ import {
   NotificationSettings,
   UserId,
   UserName,
-} from '@noshiro/event-schedule-app-shared';
-import { deepEqual } from '@noshiro/fast-deep-equal';
-import { compareYearMonthDate } from '@noshiro/io-ts-types';
-import { toAbsolutePath } from '@noshiro/ts-utils-additional';
-import { api } from '../../api';
-import { Routes, eventScheduleInitialValue } from '../../constants';
+} from 'event-schedule-app-shared';
+import {
+  type InitializedObservable,
+  debounce,
+  filter,
+  map,
+  skip,
+} from 'synstate';
+import { createBooleanState, createState } from 'synstate-react-hooks';
+import { Result, fastDeepEqual, isNotUndefined, pipe } from 'ts-data-forge';
+import { compareYearMonthDate } from 'ts-fortress-types';
+import { api } from '../../api/index.mjs';
+import { Routes, eventScheduleInitialValue } from '../../constants/index.mjs';
 import {
   EventScheduleAppLocalStorage,
   createToaster,
   showToast,
-} from '../../functions';
-import { type EventScheduleSettingCommonState } from '../../types';
-import { now } from '../../utils';
-import { Auth } from '../auth';
-import { createEventScheduleSettingStore } from './event-schedule-setting-common';
+} from '../../functions/index.mjs';
+import { type EventScheduleSettingCommonState } from '../../types/index.mjs';
+import {
+  mapOptional,
+  noop,
+  toAbsolutePath,
+} from '../../utils-ported/index.mjs';
+import { now } from '../../utils/index.mjs';
+import { Auth } from '../auth.mjs';
+import { createEventScheduleSettingStore } from './event-schedule-setting-common.mjs';
 
 const toast = createToaster();
 
 const { commonState$, commonStateHandlers } = createEventScheduleSettingStore();
 
-const hasNoChanges$: InitializedObservable<boolean> = commonState$.chain(
+const hasNoChanges$: InitializedObservable<boolean> = commonState$.pipe(
   map(({ eventScheduleNormalized }) =>
-    deepEqual(eventScheduleInitialValue, eventScheduleNormalized),
+    fastDeepEqual(eventScheduleInitialValue, eventScheduleNormalized),
   ),
 );
 
 const resetAllState = (): void => {
   commonStateHandlers.resetTitle();
+
   commonStateHandlers.resetNotes();
+
   commonStateHandlers.resetDatetimeSpecification();
+
   commonStateHandlers.resetDatetimeRangeList();
+
   commonStateHandlers.resetAnswerDeadlineSection();
+
   commonStateHandlers.resetNotificationSettingsSection();
+
   commonStateHandlers.resetAnswerIcons();
 };
 
@@ -43,23 +61,29 @@ const restoreFromLocalStorage = (): void => {
 
   if (Result.isOk(fromStorage)) {
     const ev = fromStorage.value;
+
     if (ev === undefined) return;
 
     commonStateHandlers.setTitle(ev.title);
+
     commonStateHandlers.setNotes(ev.notes);
+
     commonStateHandlers.setDatetimeSpecification(ev.datetimeSpecification);
+
     // 過去日（今日含む）は復元しない
     commonStateHandlers.setDatetimeRangeList(
       ev.datetimeRangeList.filter(
         (d) => compareYearMonthDate(d.ymd, now()) > 0,
       ),
     );
+
     commonStateHandlers.setAnswerIcons(ev.answerIcons);
 
     if (ev.answerDeadline === 'none') {
       commonStateHandlers.turnOffAnswerDeadlineSection();
     } else {
       commonStateHandlers.turnOnAnswerDeadlineSection();
+
       commonStateHandlers.setAnswerDeadline(ev.answerDeadline);
     }
 
@@ -67,6 +91,7 @@ const restoreFromLocalStorage = (): void => {
       commonStateHandlers.turnOffNotificationSection();
     } else {
       commonStateHandlers.turnOnNotificationSection();
+
       commonStateHandlers.setNotificationSettingsWithEmail({
         ...ev.notificationSettings,
         email: Auth.getFireAuthUserSnapshot()?.email ?? '',
@@ -85,36 +110,33 @@ const saveToLocalStorage = (
     datetimeSpecification: commonState.datetimeSpecification,
     notes: commonState.notes,
     notificationSettings:
-      pipe(commonState.notificationSettingsWithEmail).chainOptional(
-        NotificationSettings.fill,
+      pipe(commonState.notificationSettingsWithEmail).map((__v) =>
+        mapOptional(__v, NotificationSettings.fill),
       ).value ?? 'none',
     title: commonState.title,
   });
+
   if (Result.isErr(saveResult)) {
     console.warn(saveResult.value);
   }
 };
 
-const {
-  useCurrentValue: useIsLoading,
-  setTrue: setIsLoadingTrue,
-  setFalse: setIsLoadingFalse,
-} = createBooleanState(false);
+const [
+  useIsLoading,
+  { setTrue: setIsLoadingTrue, setFalse: setIsLoadingFalse },
+] = createBooleanState(false);
 
-const {
-  useCurrentValue: useCreateResultDialogIsOpen,
-  setTrue: openCreateResultDialog,
-  setFalse: closeCreateResultDialog,
-} = createBooleanState(false);
+const [
+  useCreateResultDialogIsOpen,
+  { setTrue: openCreateResultDialog, setFalse: closeCreateResultDialog },
+] = createBooleanState(false);
 
-const {
-  useCurrentValue: useUrl,
-  getSnapshot: getUrlSnapshot,
-  setState: setUrl,
-} = createState<string>('');
+const [useUrl, setUrl, { getSnapshot: getUrlSnapshot }] =
+  createState<string>('');
 
 const createEvent = async (): Promise<Result<undefined, string>> => {
   const commonState = commonState$.getSnapshot().value;
+
   const fireAuthUser = Auth.getFireAuthUserSnapshot();
 
   const {
@@ -123,20 +145,21 @@ const createEvent = async (): Promise<Result<undefined, string>> => {
     notificationSettingsWithEmail,
   } = commonState;
 
-  const email = notificationSettingsWithEmail?.email ?? '';
-
   if (!eventScheduleValidationOk) return Result.ok(undefined);
 
+  const email = notificationSettingsWithEmail?.email ?? '';
+
   setIsLoadingTrue();
 
   openCreateResultDialog();
 
-  const res = await api.event.add(
-    Obj.set(eventScheduleNormalized, 'author', {
+  const res = await api.event.add({
+    ...eventScheduleNormalized,
+    author: {
       id: mapOptional(fireAuthUser?.uid, UserId.cast) ?? null,
       name: UserName.cast(fireAuthUser?.displayName ?? ''),
-    }),
-  );
+    },
+  });
 
   if (Result.isErr(res)) {
     return Result.err(res.value);
@@ -146,12 +169,14 @@ const createEvent = async (): Promise<Result<undefined, string>> => {
 
   if (email !== '') {
     const res2 = await api.event.setAuthorsEmail(eventId, email);
+
     if (Result.isErr(res2)) {
       return Result.err(res2.value);
     }
   }
 
   setIsLoadingFalse();
+
   setUrl(toAbsolutePath(`..${Routes.routes.answerPage(eventId)}`));
 
   // reset local storage
@@ -171,30 +196,30 @@ const createEvent = async (): Promise<Result<undefined, string>> => {
 const onCreateEventClick = (): void => {
   createEvent()
     .then((res) => {
-      if (Result.isErr(res)) {
-        console.error(res.value);
-        showToast({
-          toast,
-          message: dict.eventSettingsPage.createEventResultMessage.error,
-          intent: 'danger',
-        });
+      if (!Result.isErr(res)) {
+        return;
       }
+
+      console.error(res.value);
+
+      showToast({
+        toast,
+        message: dict.eventSettingsPage.createEventResultMessage.error,
+        intent: 'danger',
+      });
     })
     .catch(noop);
 };
 
 const onClipboardButtonClick = (): void => {
   const url = getUrlSnapshot();
+
   // eslint-disable-next-line unicorn/prefer-global-this
   const clipboard = window.navigator.clipboard;
 
   // https://stackoverflow.com/questions/51805395/navigator-clipboard-is-undefined
-  if (
-    isNotUndefined(clipboard) &&
-    // eslint-disable-next-line unicorn/prefer-global-this
-    window.isSecureContext &&
-    isNotUndefined(url)
-  ) {
+  // eslint-disable-next-line unicorn/prefer-global-this
+  if (window.isSecureContext) {
     // TODO: use toast
     clipboard.writeText(url).catch(console.error);
   }
@@ -203,9 +228,9 @@ const onClipboardButtonClick = (): void => {
 /* subscriptions */
 
 commonState$
-  .chain(debounceTime(500))
-  .chain(skip(1))
-  .chain(filter(isNotUndefined))
+  .pipe(debounce(500))
+  .pipe(skip(1))
+  .pipe(filter(isNotUndefined))
   .subscribe(saveToLocalStorage);
 
 export const CreateEventScheduleStore = {
