diff --git experimental/packages/apps/event-schedule-app/src/store/create-event-schedule-page/index.ts apps/event-schedule-app/src/store/create-event-schedule-page/index.mts
index e75fcf149..b0514bca0 100644
--- experimental/packages/apps/event-schedule-app/src/store/create-event-schedule-page/index.ts
+++ apps/event-schedule-app/src/store/create-event-schedule-page/index.mts
@@ -1,3 +1,3 @@
-export * from './create-event-schedule';
-export * from './edit-event-schedule';
-export * from './event-schedule-setting-common';
+export * from './create-event-schedule.mjs';
+export * from './edit-event-schedule.mjs';
+export * from './event-schedule-setting-common.mjs';
