diff --git experimental/packages/apps/housing-loan-calculator-app/src/components/summary-section/data-item.tsx apps/housing-loan-calculator-app/src/components/summary-section/data-item.tsx
index 06659cce4..c8ca6b8f8 100644
--- experimental/packages/apps/housing-loan-calculator-app/src/components/summary-section/data-item.tsx
+++ apps/housing-loan-calculator-app/src/components/summary-section/data-item.tsx
@@ -1,3 +1,7 @@
+import { css } from '@emotion/react';
+import { memoNamed } from 'react-utils';
+import { type DeepReadonly } from 'ts-type-forge';
+
 type Props = DeepReadonly<{
   title: string;
   description: string;
