diff --git a/packages/v5.1/temp/copied/lib.es2015.collection.d.ts b/packages/v5.2/temp/copied/lib.es2015.collection.d.ts
index e3851bf..8264382 100644
--- a/packages/v5.1/temp/copied/lib.es2015.collection.d.ts
+++ b/packages/v5.2/temp/copied/lib.es2015.collection.d.ts
@@ -64,7 +64,7 @@ interface ReadonlyMap<K, V> {
   readonly size: number;
 }
 
-interface WeakMap<K extends object, V> {
+interface WeakMap<K extends WeakKey, V> {
   /**
    * Removes the specified element from the WeakMap.
    * @returns true if the element was successfully removed, or false if it was not present.
@@ -80,16 +80,16 @@ interface WeakMap<K extends object, V> {
   has(key: K): boolean;
   /**
    * Adds a new element with a specified key and value.
-   * @param key Must be an object.
+   * @param key Must be an object or symbol.
    */
   set(key: K, value: V): this;
 }
 
 interface WeakMapConstructor {
-  new <K extends object = object, V = any>(
+  new <K extends WeakKey = WeakKey, V = any>(
     entries?: readonly (readonly [K, V])[] | null,
   ): WeakMap<K, V>;
-  readonly prototype: WeakMap<object, any>;
+  readonly prototype: WeakMap<WeakKey, any>;
 }
 declare var WeakMap: WeakMapConstructor;
 
@@ -137,9 +137,9 @@ interface ReadonlySet<T> {
   readonly size: number;
 }
 
-interface WeakSet<T extends object> {
+interface WeakSet<T extends WeakKey> {
   /**
-   * Appends a new object to the end of the WeakSet.
+   * Appends a new value to the end of the WeakSet.
    */
   add(value: T): this;
   /**
@@ -148,13 +148,13 @@ interface WeakSet<T extends object> {
    */
   delete(value: T): boolean;
   /**
-   * @returns a boolean indicating whether an object exists in the WeakSet or not.
+   * @returns a boolean indicating whether a value exists in the WeakSet or not.
    */
   has(value: T): boolean;
 }
 
 interface WeakSetConstructor {
-  new <T extends object = object>(values?: readonly T[] | null): WeakSet<T>;
-  readonly prototype: WeakSet<object>;
+  new <T extends WeakKey = WeakKey>(values?: readonly T[] | null): WeakSet<T>;
+  readonly prototype: WeakSet<WeakKey>;
 }
 declare var WeakSet: WeakSetConstructor;
