diff --git a/packages/v5.1/temp/copied/lib.es2021.weakref.d.ts b/packages/v5.2/temp/copied/lib.es2021.weakref.d.ts
index adf2b98..eb554fc 100644
--- a/packages/v5.1/temp/copied/lib.es2021.weakref.d.ts
+++ b/packages/v5.2/temp/copied/lib.es2021.weakref.d.ts
@@ -15,12 +15,13 @@ and limitations under the License.
 
 /// <reference no-default-lib="true"/>
 
-interface WeakRef<T extends object> {
+interface WeakRef<T extends WeakKey> {
   readonly [Symbol.toStringTag]: 'WeakRef';
 
   /**
-   * Returns the WeakRef instance's target object, or undefined if the target object has been
+   * Returns the WeakRef instance's target value, or undefined if the target value has been
    * reclaimed.
+   * In es2023 the value can be either a symbol or an object, in previous versions only object is permissible.
    */
   deref(): T | undefined;
 }
@@ -29,10 +30,11 @@ interface WeakRefConstructor {
   readonly prototype: WeakRef<any>;
 
   /**
-   * Creates a WeakRef instance for the given target object.
-   * @param target The target object for the WeakRef instance.
+   * Creates a WeakRef instance for the given target value.
+   * In es2023 the value can be either a symbol or an object, in previous versions only object is permissible.
+   * @param target The target value for the WeakRef instance.
    */
-  new <T extends object>(target: T): WeakRef<T>;
+  new <T extends WeakKey>(target: T): WeakRef<T>;
 }
 
 declare var WeakRef: WeakRefConstructor;
@@ -41,22 +43,23 @@ interface FinalizationRegistry<T> {
   readonly [Symbol.toStringTag]: 'FinalizationRegistry';
 
   /**
-   * Registers an object with the registry.
-   * @param target The target object to register.
-   * @param heldValue The value to pass to the finalizer for this object. This cannot be the
-   * target object.
+   * Registers a value with the registry.
+   * In es2023 the value can be either a symbol or an object, in previous versions only object is permissible.
+   * @param target The target value to register.
+   * @param heldValue The value to pass to the finalizer for this value. This cannot be the
+   * target value.
    * @param unregisterToken The token to pass to the unregister method to unregister the target
-   * object. If provided (and not undefined), this must be an object. If not provided, the target
-   * cannot be unregistered.
+   * value. If not provided, the target cannot be unregistered.
    */
-  register(target: object, heldValue: T, unregisterToken?: object): void;
+  register(target: WeakKey, heldValue: T, unregisterToken?: WeakKey): void;
 
   /**
-   * Unregisters an object from the registry.
+   * Unregisters a value from the registry.
+   * In es2023 the value can be either a symbol or an object, in previous versions only object is permissible.
    * @param unregisterToken The token that was used as the unregisterToken argument when calling
-   * register to register the target object.
+   * register to register the target value.
    */
-  unregister(unregisterToken: object): void;
+  unregister(unregisterToken: WeakKey): void;
 }
 
 interface FinalizationRegistryConstructor {
@@ -64,7 +67,7 @@ interface FinalizationRegistryConstructor {
 
   /**
    * Creates a finalization registry with an associated cleanup callback
-   * @param cleanupCallback The callback to call after an object in the registry has been reclaimed.
+   * @param cleanupCallback The callback to call after a value in the registry has been reclaimed.
    */
   new <T>(cleanupCallback: (heldValue: T) => void): FinalizationRegistry<T>;
 }
