diff --git a/packages/v5.1/temp/copied/lib.webworker.d.ts b/packages/v5.2/temp/copied/lib.webworker.d.ts
index f559733..a74b0d1 100644
--- a/packages/v5.1/temp/copied/lib.webworker.d.ts
+++ b/packages/v5.2/temp/copied/lib.webworker.d.ts
@@ -452,7 +452,7 @@ interface NotificationOptions {
   lang?: string;
   renotify?: boolean;
   requireInteraction?: boolean;
-  silent?: boolean;
+  silent?: boolean | null;
   tag?: string;
   timestamp?: EpochTimeStamp;
   vibrate?: VibratePattern;
@@ -537,7 +537,6 @@ interface RTCEncodedAudioFrameMetadata {
 }
 
 interface RTCEncodedVideoFrameMetadata {
-  contributingSources?: number[];
   dependencies?: number[];
   frameId?: number;
   height?: number;
@@ -707,8 +706,8 @@ interface TextDecoderOptions {
 }
 
 interface TextEncoderEncodeIntoResult {
-  read?: number;
-  written?: number;
+  read: number;
+  written: number;
 }
 
 interface Transformer<I = any, O = any> {
@@ -866,6 +865,32 @@ interface WebGLContextEventInit extends EventInit {
   statusMessage?: string;
 }
 
+interface WebTransportCloseInfo {
+  closeCode?: number;
+  reason?: string;
+}
+
+interface WebTransportErrorOptions {
+  source?: WebTransportErrorSource;
+  streamErrorCode?: number | null;
+}
+
+interface WebTransportHash {
+  algorithm?: string;
+  value?: BufferSource;
+}
+
+interface WebTransportOptions {
+  allowPooling?: boolean;
+  congestionControl?: WebTransportCongestionControl;
+  requireUnreliable?: boolean;
+  serverCertificateHashes?: WebTransportHash[];
+}
+
+interface WebTransportSendStreamOptions {
+  sendOrder?: number | null;
+}
+
 interface WorkerOptions {
   credentials?: RequestCredentials;
   name?: string;
@@ -977,9 +1002,9 @@ interface AbortSignal extends EventTarget {
 declare var AbortSignal: {
   prototype: AbortSignal;
   new (): AbortSignal;
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/AbortSignal/abort) */
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/AbortSignal/abort_static) */
   abort(reason?: any): AbortSignal;
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/AbortSignal/timeout) */
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/AbortSignal/timeout_static) */
   timeout(milliseconds: number): AbortSignal;
 };
 
@@ -1829,6 +1854,8 @@ interface CanvasShadowStyles {
 }
 
 interface CanvasState {
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/CanvasRenderingContext2D/reset) */
+  reset(): void;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/CanvasRenderingContext2D/restore) */
   restore(): void;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/CanvasRenderingContext2D/save) */
@@ -2333,7 +2360,7 @@ interface DOMPoint extends DOMPointReadOnly {
 declare var DOMPoint: {
   prototype: DOMPoint;
   new (x?: number, y?: number, z?: number, w?: number): DOMPoint;
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMPoint/fromPoint) */
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMPoint/fromPoint_static) */
   fromPoint(other?: DOMPointInit): DOMPoint;
 };
 
@@ -2356,7 +2383,7 @@ interface DOMPointReadOnly {
 declare var DOMPointReadOnly: {
   prototype: DOMPointReadOnly;
   new (x?: number, y?: number, z?: number, w?: number): DOMPointReadOnly;
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMPointReadOnly/fromPoint) */
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMPointReadOnly/fromPoint_static) */
   fromPoint(other?: DOMPointInit): DOMPointReadOnly;
 };
 
@@ -2430,7 +2457,7 @@ declare var DOMRectReadOnly: {
     width?: number,
     height?: number,
   ): DOMRectReadOnly;
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMRectReadOnly/fromRect) */
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMRectReadOnly/fromRect_static) */
   fromRect(other?: DOMRectInit): DOMRectReadOnly;
 };
 
@@ -3224,11 +3251,17 @@ interface FileSystemSyncAccessHandle {
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/FileSystemSyncAccessHandle/getSize) */
   getSize(): number;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/FileSystemSyncAccessHandle/read) */
-  read(buffer: BufferSource, options?: FileSystemReadWriteOptions): number;
+  read(
+    buffer: AllowSharedBufferSource,
+    options?: FileSystemReadWriteOptions,
+  ): number;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/FileSystemSyncAccessHandle/truncate) */
   truncate(newSize: number): void;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/FileSystemSyncAccessHandle/write) */
-  write(buffer: BufferSource, options?: FileSystemReadWriteOptions): number;
+  write(
+    buffer: AllowSharedBufferSource,
+    options?: FileSystemReadWriteOptions,
+  ): number;
 }
 
 declare var FileSystemSyncAccessHandle: {
@@ -3424,6 +3457,8 @@ interface Headers {
   delete(name: string): void;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/Headers/get) */
   get(name: string): string | null;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/Headers/getSetCookie) */
+  getSetCookie(): string[];
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/Headers/has) */
   has(name: string): boolean;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/Headers/set) */
@@ -3812,7 +3847,7 @@ declare var IDBKeyRange: {
   /**
    * Returns a new IDBKeyRange spanning from lower to upper. If lowerOpen is true, lower is not included in the range. If upperOpen is true, upper is not included in the range.
    *
-   * [MDN Reference](https://developer.mozilla.org/docs/Web/API/IDBKeyRange/bound)
+   * [MDN Reference](https://developer.mozilla.org/docs/Web/API/IDBKeyRange/bound_static)
    */
   bound(
     lower: any,
@@ -3823,19 +3858,19 @@ declare var IDBKeyRange: {
   /**
    * Returns a new IDBKeyRange starting at key with no upper bound. If open is true, key is not included in the range.
    *
-   * [MDN Reference](https://developer.mozilla.org/docs/Web/API/IDBKeyRange/lowerBound)
+   * [MDN Reference](https://developer.mozilla.org/docs/Web/API/IDBKeyRange/lowerBound_static)
    */
   lowerBound(lower: any, open?: boolean): IDBKeyRange;
   /**
    * Returns a new IDBKeyRange spanning only key.
    *
-   * [MDN Reference](https://developer.mozilla.org/docs/Web/API/IDBKeyRange/only)
+   * [MDN Reference](https://developer.mozilla.org/docs/Web/API/IDBKeyRange/only_static)
    */
   only(value: any): IDBKeyRange;
   /**
    * Returns a new IDBKeyRange with no lower bound and ending at key. If open is true, key is not included in the range.
    *
-   * [MDN Reference](https://developer.mozilla.org/docs/Web/API/IDBKeyRange/upperBound)
+   * [MDN Reference](https://developer.mozilla.org/docs/Web/API/IDBKeyRange/upperBound_static)
    */
   upperBound(upper: any, open?: boolean): IDBKeyRange;
 };
@@ -4644,6 +4679,8 @@ interface Notification extends EventTarget {
   onerror: ((this: Notification, ev: Event) => any) | null;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/Notification/show_event) */
   onshow: ((this: Notification, ev: Event) => any) | null;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/Notification/silent) */
+  readonly silent: boolean | null;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/Notification/tag) */
   readonly tag: string;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/Notification/title) */
@@ -4675,7 +4712,7 @@ interface Notification extends EventTarget {
 declare var Notification: {
   prototype: Notification;
   new (title: string, options?: NotificationOptions): Notification;
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/Notification/permission) */
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/Notification/permission_static) */
   readonly permission: NotificationPermission;
 };
 
@@ -5078,7 +5115,7 @@ interface PerformanceObserver {
 declare var PerformanceObserver: {
   prototype: PerformanceObserver;
   new (callback: PerformanceObserverCallback): PerformanceObserver;
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/PerformanceObserver/supportedEntryTypes) */
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/PerformanceObserver/supportedEntryTypes_static) */
   readonly supportedEntryTypes: ReadonlyArray<string>;
 };
 
@@ -5287,7 +5324,7 @@ interface PushManager {
 declare var PushManager: {
   prototype: PushManager;
   new (): PushManager;
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/PushManager/supportedContentEncodings) */
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/PushManager/supportedContentEncodings_static) */
   readonly supportedContentEncodings: ReadonlyArray<string>;
 };
 
@@ -5675,9 +5712,11 @@ interface Response extends Body {
 declare var Response: {
   prototype: Response;
   new (body?: BodyInit | null, init?: ResponseInit): Response;
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/Response/error) */
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/Response/error_static) */
   error(): Response;
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/Response/redirect) */
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/Response/json_static) */
+  json(data: any, init?: ResponseInit): Response;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/Response/redirect_static) */
   redirect(url: string | URL, status?: number): Response;
 };
 
@@ -5787,7 +5826,6 @@ interface ServiceWorkerContainer extends EventTarget {
   oncontrollerchange: ((this: ServiceWorkerContainer, ev: Event) => any) | null;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/ServiceWorkerContainer/message_event) */
   onmessage: ((this: ServiceWorkerContainer, ev: MessageEvent) => any) | null;
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/ServiceWorkerContainer/messageerror_event) */
   onmessageerror:
     | ((this: ServiceWorkerContainer, ev: MessageEvent) => any)
     | null;
@@ -6275,7 +6313,7 @@ interface TextDecoder extends TextDecoderCommon {
    *
    * [MDN Reference](https://developer.mozilla.org/docs/Web/API/TextDecoder/decode)
    */
-  decode(input?: BufferSource, options?: TextDecodeOptions): string;
+  decode(input?: AllowSharedBufferSource, options?: TextDecodeOptions): string;
 }
 
 declare var TextDecoder: {
@@ -6490,14 +6528,18 @@ interface URL {
 declare var URL: {
   prototype: URL;
   new (url: string | URL, base?: string | URL): URL;
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/URL/createObjectURL) */
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/URL/canParse_static) */
+  canParse(url: string | URL, base?: string): boolean;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/URL/createObjectURL_static) */
   createObjectURL(obj: Blob): string;
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/URL/revokeObjectURL) */
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/URL/revokeObjectURL_static) */
   revokeObjectURL(url: string): void;
 };
 
 /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/URLSearchParams) */
 interface URLSearchParams {
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/URLSearchParams/size) */
+  readonly size: number;
   /**
    * Appends a specified key/value pair as a new search parameter.
    *
@@ -6509,7 +6551,7 @@ interface URLSearchParams {
    *
    * [MDN Reference](https://developer.mozilla.org/docs/Web/API/URLSearchParams/delete)
    */
-  delete(name: string): void;
+  delete(name: string, value?: string): void;
   /**
    * Returns the first value associated to the given search parameter.
    *
@@ -6527,7 +6569,7 @@ interface URLSearchParams {
    *
    * [MDN Reference](https://developer.mozilla.org/docs/Web/API/URLSearchParams/has)
    */
-  has(name: string): boolean;
+  has(name: string, value?: string): boolean;
   /**
    * Sets the value associated to a given search parameter to the given value. If there were several values, delete the others.
    *
@@ -10046,6 +10088,100 @@ declare var WebSocket: {
   readonly CLOSED: 3;
 };
 
+/**
+ * Available only in secure contexts.
+ *
+ * [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebTransport)
+ */
+interface WebTransport {
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebTransport/closed) */
+  readonly closed: Promise<WebTransportCloseInfo>;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebTransport/datagrams) */
+  readonly datagrams: WebTransportDatagramDuplexStream;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebTransport/incomingBidirectionalStreams) */
+  readonly incomingBidirectionalStreams: ReadableStream;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebTransport/incomingUnidirectionalStreams) */
+  readonly incomingUnidirectionalStreams: ReadableStream;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebTransport/ready) */
+  readonly ready: Promise<undefined>;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebTransport/close) */
+  close(closeInfo?: WebTransportCloseInfo): void;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebTransport/createBidirectionalStream) */
+  createBidirectionalStream(
+    options?: WebTransportSendStreamOptions,
+  ): Promise<WebTransportBidirectionalStream>;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebTransport/createUnidirectionalStream) */
+  createUnidirectionalStream(
+    options?: WebTransportSendStreamOptions,
+  ): Promise<WritableStream>;
+}
+
+declare var WebTransport: {
+  prototype: WebTransport;
+  new (url: string | URL, options?: WebTransportOptions): WebTransport;
+};
+
+/**
+ * Available only in secure contexts.
+ *
+ * [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebTransportBidirectionalStream)
+ */
+interface WebTransportBidirectionalStream {
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebTransportBidirectionalStream/readable) */
+  readonly readable: ReadableStream;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebTransportBidirectionalStream/writable) */
+  readonly writable: WritableStream;
+}
+
+declare var WebTransportBidirectionalStream: {
+  prototype: WebTransportBidirectionalStream;
+  new (): WebTransportBidirectionalStream;
+};
+
+/**
+ * Available only in secure contexts.
+ *
+ * [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebTransportDatagramDuplexStream)
+ */
+interface WebTransportDatagramDuplexStream {
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebTransportDatagramDuplexStream/incomingHighWaterMark) */
+  incomingHighWaterMark: number;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebTransportDatagramDuplexStream/incomingMaxAge) */
+  incomingMaxAge: number;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebTransportDatagramDuplexStream/maxDatagramSize) */
+  readonly maxDatagramSize: number;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebTransportDatagramDuplexStream/outgoingHighWaterMark) */
+  outgoingHighWaterMark: number;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebTransportDatagramDuplexStream/outgoingMaxAge) */
+  outgoingMaxAge: number;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebTransportDatagramDuplexStream/readable) */
+  readonly readable: ReadableStream;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebTransportDatagramDuplexStream/writable) */
+  readonly writable: WritableStream;
+}
+
+declare var WebTransportDatagramDuplexStream: {
+  prototype: WebTransportDatagramDuplexStream;
+  new (): WebTransportDatagramDuplexStream;
+};
+
+/**
+ * Available only in secure contexts.
+ *
+ * [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebTransportError)
+ */
+interface WebTransportError extends DOMException {
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebTransportError/source) */
+  readonly source: WebTransportErrorSource;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebTransportError/streamErrorCode) */
+  readonly streamErrorCode: number | null;
+}
+
+declare var WebTransportError: {
+  prototype: WebTransportError;
+  new (message?: string, options?: WebTransportErrorOptions): WebTransportError;
+};
+
 /**
  * This ServiceWorker API interface represents the scope of a service worker client that is a document in a browser context, controlled by an active worker. The service worker client independently selects and uses a service worker for its own loading and sub-resources.
  *
@@ -10704,16 +10840,19 @@ declare namespace WebAssembly {
   };
 
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/WebAssembly/Global) */
-  interface Global {
+  interface Global<T extends ValueType = ValueType> {
     /** [MDN Reference](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/WebAssembly/Global/value) */
-    value: any;
+    value: ValueTypeMap[T];
     /** [MDN Reference](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/WebAssembly/Global/valueOf) */
-    valueOf(): any;
+    valueOf(): ValueTypeMap[T];
   }
 
   var Global: {
     prototype: Global;
-    new (descriptor: GlobalDescriptor, v?: any): Global;
+    new <T extends ValueType = ValueType>(
+      descriptor: GlobalDescriptor<T>,
+      v?: ValueTypeMap[T],
+    ): Global<T>;
   };
 
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/WebAssembly/Instance) */
@@ -10787,9 +10926,9 @@ declare namespace WebAssembly {
     new (descriptor: TableDescriptor, value?: any): Table;
   };
 
-  interface GlobalDescriptor {
+  interface GlobalDescriptor<T extends ValueType = ValueType> {
     mutable?: boolean;
-    value: ValueType;
+    value: T;
   }
 
   interface MemoryDescriptor {
@@ -10815,6 +10954,16 @@ declare namespace WebAssembly {
     maximum?: number;
   }
 
+  interface ValueTypeMap {
+    anyfunc: Function;
+    externref: any;
+    f32: number;
+    f64: number;
+    i32: number;
+    i64: bigint;
+    v128: never;
+  }
+
   interface WebAssemblyInstantiatedSource {
     instance: Instance;
     module: Module;
@@ -10822,19 +10971,12 @@ declare namespace WebAssembly {
 
   type ImportExportKind = 'function' | 'global' | 'memory' | 'table';
   type TableKind = 'anyfunc' | 'externref';
-  type ValueType =
-    | 'anyfunc'
-    | 'externref'
-    | 'f32'
-    | 'f64'
-    | 'i32'
-    | 'i64'
-    | 'v128';
   type ExportValue = Function | Global | Memory | Table;
   type Exports = Record<string, ExportValue>;
   type ImportValue = ExportValue | number;
   type Imports = Record<string, ModuleImports>;
   type ModuleImports = Record<string, ImportValue>;
+  type ValueType = keyof ValueTypeMap;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/WebAssembly/compile) */
   function compile(bytes: BufferSource): Promise<Module>;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/WebAssembly/compileStreaming) */
@@ -11139,6 +11281,7 @@ declare function removeEventListener(
   options?: boolean | EventListenerOptions,
 ): void;
 type AlgorithmIdentifier = Algorithm | string;
+type AllowSharedBufferSource = ArrayBuffer | ArrayBufferView;
 type BigInteger = Uint8Array;
 type BinaryData = ArrayBuffer | ArrayBufferView;
 type BlobPart = BufferSource | Blob | string;
@@ -11429,6 +11572,8 @@ type VideoPixelFormat =
   | 'RGBX';
 type VideoTransferCharacteristics = 'bt709' | 'iec61966-2-1' | 'smpte170m';
 type WebGLPowerPreference = 'default' | 'high-performance' | 'low-power';
+type WebTransportCongestionControl = 'default' | 'low-latency' | 'throughput';
+type WebTransportErrorSource = 'session' | 'stream';
 type WorkerType = 'classic' | 'module';
 type WriteCommandType = 'seek' | 'truncate' | 'write';
 type XMLHttpRequestResponseType =
