diff --git experimental/packages/apps/event-schedule-app/src/store/fetching-state/index.ts apps/event-schedule-app/src/store/fetching-state/index.mts
index 45626c0b4..c0f263b7f 100644
--- experimental/packages/apps/event-schedule-app/src/store/fetching-state/index.ts
+++ apps/event-schedule-app/src/store/fetching-state/index.mts
@@ -1,3 +1,3 @@
-export * from './answers';
-export * from './event-list';
-export * from './event-schedule';
+export * from './answers.mjs';
+export * from './event-list.mjs';
+export * from './event-schedule.mjs';
