diff --git experimental/packages/apps/event-schedule-app-shared/src/v6/types/enum/datetime-specification-type.mts apps/event-schedule-app-shared/src/v6/types/enum/datetime-specification-type.mts
index b25fc1b87..03763f16e 100644
--- experimental/packages/apps/event-schedule-app-shared/src/v6/types/enum/datetime-specification-type.mts
+++ apps/event-schedule-app-shared/src/v6/types/enum/datetime-specification-type.mts
@@ -1,3 +1,5 @@
+import { type ReadonlyRecord } from 'ts-type-forge';
+
 export type DatetimeSpecificationEnumType =
   | 'noStartEndSpecified'
   | 'startSpecified'
@@ -9,7 +11,7 @@ export const datetimeSpecificationOptions = {
   startSpecified: 'startSpecified',
   endSpecified: 'endSpecified',
   startAndEndSpecified: 'startAndEndSpecified',
-} as const satisfies Record<
+} as const satisfies ReadonlyRecord<
   DatetimeSpecificationEnumType,
   DatetimeSpecificationEnumType
 >;
