diff --git experimental/packages/apps/poll-discord-app/src/env.mts apps/poll-discord-app/src/env.mts
index 3d20a36f2..c45e9ae21 100644
--- experimental/packages/apps/poll-discord-app/src/env.mts
+++ apps/poll-discord-app/src/env.mts
@@ -1,6 +1,7 @@
 import * as dotenv from 'dotenv';
+import { mapOptional, match } from './utils/index.mjs';
 
-const NODE_ENV: 'development' | 'production' | (string & {}) | undefined =
+const NODE_ENV: 'development' | 'production' | undefined | (string & {}) =
   process.env['NODE_ENV'];
 
 console.log(`NODE_ENV = "${NODE_ENV ?? ''}"`);
