diff --git experimental/packages/apps/event-schedule-app/src/hooks/use-form-error-hook.ts apps/event-schedule-app/src/hooks/use-form-error-hook.mts
index e7d21574a..14a69b41d 100644
--- experimental/packages/apps/event-schedule-app/src/hooks/use-form-error-hook.ts
+++ apps/event-schedule-app/src/hooks/use-form-error-hook.mts
@@ -1,25 +1,29 @@
-export const useFormError = <T>(
+import { useBoolState } from 'better-react-use-state';
+import * as React from 'react';
+
+export const useFormError = <T,>(
   value: T,
   valueIsInvalid: (v: T) => boolean,
   onValueChange: (v: T) => void,
-): [boolean, (v: T) => void, () => void] => {
+): readonly [boolean, (v: T) => void, () => void] => {
   const [showError, { setTrue: show, setFalse: hide }] = useBoolState(false);
 
-  const onBlur = useCallback(() => {
+  const onBlur = React.useCallback(() => {
     if (valueIsInvalid(value)) {
       show();
     }
   }, [value, valueIsInvalid, show]);
 
-  useEffect(() => {
+  React.useEffect(() => {
     if (!valueIsInvalid(value)) {
       hide();
     }
   }, [value, valueIsInvalid, hide]);
 
-  const onChangeLocal = useCallback(
+  const onChangeLocal = React.useCallback(
     (v: T) => {
       onValueChange(v);
+
       if (valueIsInvalid(v)) {
         show();
       } else {
