diff --git a/packages/v5.1/output/lib-files/lib.es5.d.ts b/packages/v5.2/output/lib-files/lib.es5.d.ts
index bef43a2..cdcdcaa 100644
--- a/packages/v5.1/output/lib-files/lib.es5.d.ts
+++ b/packages/v5.2/output/lib-files/lib.es5.d.ts
@@ -1934,6 +1934,15 @@ type Uncapitalize<S extends string> = intrinsic;
  */
 interface ThisType<T> {}
 
+/**
+ * Stores types to be used with WeakSet, WeakMap, WeakRef, and FinalizationRegistry
+ */
+interface WeakKeyTypes {
+  readonly object: object;
+}
+
+type WeakKey = WeakKeyTypes[keyof WeakKeyTypes];
+
 /**
  * Represents a raw buffer of binary data, which is used to store data for the
  * different typed arrays. ArrayBuffers cannot be read from or written to directly,
@@ -2153,7 +2162,7 @@ interface Int8Array {
    * @param target If target is negative, it is treated as length+target where length is the
    * length of the array.
    * @param start If start is negative, it is treated as length+start. If end is negative, it
-   * is treated as length+end. If start is omitted, `0` is used.
+   * is treated as length+end.
    * @param end If not specified, length of the this object is used as its default value.
    */
   copyWithin(target: number, start: number, end?: number): this;
@@ -2508,7 +2517,7 @@ interface Uint8Array {
    * @param target If target is negative, it is treated as length+target where length is the
    * length of the array.
    * @param start If start is negative, it is treated as length+start. If end is negative, it
-   * is treated as length+end. If start is omitted, `0` is used.
+   * is treated as length+end.
    * @param end If not specified, length of the this object is used as its default value.
    */
   copyWithin(target: number, start: number, end?: number): this;
@@ -2864,7 +2873,7 @@ interface Uint8ClampedArray {
    * @param target If target is negative, it is treated as length+target where length is the
    * length of the array.
    * @param start If start is negative, it is treated as length+start. If end is negative, it
-   * is treated as length+end. If start is omitted, `0` is used.
+   * is treated as length+end.
    * @param end If not specified, length of the this object is used as its default value.
    */
   copyWithin(target: number, start: number, end?: number): this;
@@ -3236,7 +3245,7 @@ interface Int16Array {
    * @param target If target is negative, it is treated as length+target where length is the
    * length of the array.
    * @param start If start is negative, it is treated as length+start. If end is negative, it
-   * is treated as length+end. If start is omitted, `0` is used.
+   * is treated as length+end.
    * @param end If not specified, length of the this object is used as its default value.
    */
   copyWithin(target: number, start: number, end?: number): this;
@@ -3592,7 +3601,7 @@ interface Uint16Array {
    * @param target If target is negative, it is treated as length+target where length is the
    * length of the array.
    * @param start If start is negative, it is treated as length+start. If end is negative, it
-   * is treated as length+end. If start is omitted, `0` is used.
+   * is treated as length+end.
    * @param end If not specified, length of the this object is used as its default value.
    */
   copyWithin(target: number, start: number, end?: number): this;
@@ -3947,7 +3956,7 @@ interface Int32Array {
    * @param target If target is negative, it is treated as length+target where length is the
    * length of the array.
    * @param start If start is negative, it is treated as length+start. If end is negative, it
-   * is treated as length+end. If start is omitted, `0` is used.
+   * is treated as length+end.
    * @param end If not specified, length of the this object is used as its default value.
    */
   copyWithin(target: number, start: number, end?: number): this;
@@ -4303,7 +4312,7 @@ interface Uint32Array {
    * @param target If target is negative, it is treated as length+target where length is the
    * length of the array.
    * @param start If start is negative, it is treated as length+start. If end is negative, it
-   * is treated as length+end. If start is omitted, `0` is used.
+   * is treated as length+end.
    * @param end If not specified, length of the this object is used as its default value.
    */
   copyWithin(target: number, start: number, end?: number): this;
@@ -4659,7 +4668,7 @@ interface Float32Array {
    * @param target If target is negative, it is treated as length+target where length is the
    * length of the array.
    * @param start If start is negative, it is treated as length+start. If end is negative, it
-   * is treated as length+end. If start is omitted, `0` is used.
+   * is treated as length+end.
    * @param end If not specified, length of the this object is used as its default value.
    */
   copyWithin(target: number, start: number, end?: number): this;
@@ -5015,7 +5024,7 @@ interface Float64Array {
    * @param target If target is negative, it is treated as length+target where length is the
    * length of the array.
    * @param start If start is negative, it is treated as length+start. If end is negative, it
-   * is treated as length+end. If start is omitted, `0` is used.
+   * is treated as length+end.
    * @param end If not specified, length of the this object is used as its default value.
    */
   copyWithin(target: number, start: number, end?: number): this;
