diff --git a/packages/v5.4/temp/copied/lib.es2020.bigint.d.ts b/packages/v5.5/temp/copied/lib.es2020.bigint.d.ts
index eb6efba..851f3fd 100644
--- a/packages/v5.4/temp/copied/lib.es2020.bigint.d.ts
+++ b/packages/v5.5/temp/copied/lib.es2020.bigint.d.ts
@@ -527,7 +527,10 @@ interface BigInt64Array {
   subarray(begin?: number, end?: number): BigInt64Array;
 
   /** Converts the array to a string by using the current locale. */
-  toLocaleString(): string;
+  toLocaleString(
+    locales?: string | string[],
+    options?: Intl.NumberFormatOptions,
+  ): string;
 
   /** Returns a string representation of the array. */
   toString(): string;
@@ -858,7 +861,10 @@ interface BigUint64Array {
   subarray(begin?: number, end?: number): BigUint64Array;
 
   /** Converts the array to a string by using the current locale. */
-  toLocaleString(): string;
+  toLocaleString(
+    locales?: string | string[],
+    options?: Intl.NumberFormatOptions,
+  ): string;
 
   /** Returns a string representation of the array. */
   toString(): string;
@@ -948,6 +954,5 @@ interface DataView {
 declare namespace Intl {
   interface NumberFormat {
     format(value: number | bigint): string;
-    resolvedOptions(): ResolvedNumberFormatOptions;
   }
 }
