diff --git a/packages/v5.4/temp/copied/lib.dom.d.ts b/packages/v5.5/temp/copied/lib.dom.d.ts
index aa56a37..6042785 100644
--- a/packages/v5.4/temp/copied/lib.dom.d.ts
+++ b/packages/v5.5/temp/copied/lib.dom.d.ts
@@ -218,6 +218,9 @@ interface ChannelSplitterOptions extends AudioNodeOptions {
 interface CheckVisibilityOptions {
   checkOpacity?: boolean;
   checkVisibilityCSS?: boolean;
+  contentVisibilityAuto?: boolean;
+  opacityProperty?: boolean;
+  visibilityProperty?: boolean;
 }
 
 interface ClientQueryOptions {
@@ -284,6 +287,10 @@ interface ConstrainULongRange extends ULongRange {
   ideal?: number;
 }
 
+interface ContentVisibilityAutoStateChangeEventInit extends EventInit {
+  skipped?: boolean;
+}
+
 interface ConvolverOptions extends AudioNodeOptions {
   buffer?: AudioBuffer | null;
   disableNormalization?: boolean;
@@ -573,6 +580,8 @@ interface GainOptions extends AudioNodeOptions {
 
 interface GamepadEffectParameters {
   duration?: number;
+  leftTrigger?: number;
+  rightTrigger?: number;
   startDelay?: number;
   strongMagnitude?: number;
   weakMagnitude?: number;
@@ -1379,8 +1388,8 @@ interface RTCIceCandidateInit {
 }
 
 interface RTCIceCandidatePair {
-  local?: RTCIceCandidate;
-  remote?: RTCIceCandidate;
+  local: RTCIceCandidate;
+  remote: RTCIceCandidate;
 }
 
 interface RTCIceCandidatePairStats extends RTCStats {
@@ -1635,6 +1644,17 @@ interface ReadableStreamGetReaderOptions {
   mode?: ReadableStreamReaderMode;
 }
 
+interface ReadableStreamIteratorOptions {
+  /**
+   * Asynchronously iterates over the chunks in the stream's internal queue.
+   *
+   * Asynchronously iterating over the stream will lock it, preventing any other consumer from acquiring a reader. The lock will be released if the async iterator's return() method is called, e.g. by breaking out of the loop.
+   *
+   * By default, calling the async iterator's return() method will also cancel the stream. To prevent this, use the stream's values() method, passing true for the preventCancel option.
+   */
+  preventCancel?: boolean;
+}
+
 interface ReadableStreamReadDoneResult<T> {
   done: true;
   value?: T;
@@ -1766,21 +1786,22 @@ interface ScrollToOptions extends ScrollOptions {
 interface SecurityPolicyViolationEventInit extends EventInit {
   blockedURI?: string;
   columnNumber?: number;
-  disposition: SecurityPolicyViolationEventDisposition;
-  documentURI: string;
-  effectiveDirective: string;
+  disposition?: SecurityPolicyViolationEventDisposition;
+  documentURI?: string;
+  effectiveDirective?: string;
   lineNumber?: number;
-  originalPolicy: string;
+  originalPolicy?: string;
   referrer?: string;
   sample?: string;
   sourceFile?: string;
-  statusCode: number;
-  violatedDirective: string;
+  statusCode?: number;
+  violatedDirective?: string;
 }
 
 interface ShadowRootInit {
   delegatesFocus?: boolean;
   mode: ShadowRootMode;
+  serializable?: boolean;
   slotAssignment?: SlotAssignmentMode;
 }
 
@@ -2220,6 +2241,8 @@ interface ARIAMixin {
   ariaAtomic: string | null;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/Element/ariaAutoComplete) */
   ariaAutoComplete: string | null;
+  ariaBrailleLabel: string | null;
+  ariaBrailleRoleDescription: string | null;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/Element/ariaBusy) */
   ariaBusy: string | null;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/Element/ariaChecked) */
@@ -2368,6 +2391,8 @@ declare var AbortSignal: {
   new (): AbortSignal;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/AbortSignal/abort_static) */
   abort(reason?: any): AbortSignal;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/AbortSignal/any_static) */
+  any(signals: AbortSignal[]): AbortSignal;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/AbortSignal/timeout_static) */
   timeout(milliseconds: number): AbortSignal;
 };
@@ -4066,6 +4091,16 @@ declare var CSSScale: {
   new (x: CSSNumberish, y: CSSNumberish, z?: CSSNumberish): CSSScale;
 };
 
+interface CSSScopeRule extends CSSGroupingRule {
+  readonly end: string | null;
+  readonly start: string | null;
+}
+
+declare var CSSScopeRule: {
+  prototype: CSSScopeRule;
+  new (): CSSScopeRule;
+};
+
 /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/CSSSkew) */
 interface CSSSkew extends CSSTransformComponent {
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/CSSSkew/ax) */
@@ -4101,6 +4136,14 @@ declare var CSSSkewY: {
   new (ay: CSSNumericValue): CSSSkewY;
 };
 
+/** [MDN Reference](https://developer.mozilla.org/docs/Web/API/CSSStartingStyleRule) */
+interface CSSStartingStyleRule extends CSSGroupingRule {}
+
+declare var CSSStartingStyleRule: {
+  prototype: CSSStartingStyleRule;
+  new (): CSSStartingStyleRule;
+};
+
 /**
  * An object that is a CSS declaration block, and exposes style information and various style-related methods and properties.
  *
@@ -4326,6 +4369,7 @@ interface CSSStyleDeclaration {
   clipRule: string;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/CSS/color) */
   color: string;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/CSS/color-interpolation) */
   colorInterpolation: string;
   colorInterpolationFilters: string;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/CSS/color-scheme) */
@@ -4370,6 +4414,8 @@ interface CSSStyleDeclaration {
   containerType: string;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/CSS/content) */
   content: string;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/CSS/content-visibility) */
+  contentVisibility: string;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/CSS/counter-increment) */
   counterIncrement: string;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/CSS/counter-reset) */
@@ -4871,6 +4917,8 @@ interface CSSStyleDeclaration {
   textUnderlinePosition: string;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/CSS/text-wrap) */
   textWrap: string;
+  textWrapMode: string;
+  textWrapStyle: string;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/CSS/top) */
   top: string;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/CSS/touch-action) */
@@ -4885,6 +4933,8 @@ interface CSSStyleDeclaration {
   transformStyle: string;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/CSS/transition) */
   transition: string;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/CSS/transition-behavior) */
+  transitionBehavior: string;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/CSS/transition-delay) */
   transitionDelay: string;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/CSS/transition-duration) */
@@ -5306,6 +5356,8 @@ interface CSSStyleDeclaration {
   webkitUserSelect: string;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/CSS/white-space) */
   whiteSpace: string;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/CSS/white-space-collapse) */
+  whiteSpaceCollapse: string;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/CSS/widows) */
   widows: string;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/CSS/width) */
@@ -5324,6 +5376,8 @@ interface CSSStyleDeclaration {
   y: string;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/CSS/z-index) */
   zIndex: string;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/CSS/zoom) */
+  zoom: string;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/CSSStyleDeclaration/getPropertyPriority) */
   getPropertyPriority(property: string): string;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/CSSStyleDeclaration/getPropertyValue) */
@@ -6309,6 +6363,20 @@ declare var ConstantSourceNode: {
   ): ConstantSourceNode;
 };
 
+/** [MDN Reference](https://developer.mozilla.org/docs/Web/API/ContentVisibilityAutoStateChangeEvent) */
+interface ContentVisibilityAutoStateChangeEvent extends Event {
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/ContentVisibilityAutoStateChangeEvent/skipped) */
+  readonly skipped: boolean;
+}
+
+declare var ContentVisibilityAutoStateChangeEvent: {
+  prototype: ContentVisibilityAutoStateChangeEvent;
+  new (
+    type: string,
+    eventInitDict?: ContentVisibilityAutoStateChangeEventInit,
+  ): ContentVisibilityAutoStateChangeEvent;
+};
+
 /**
  * An AudioNode that performs a Linear Convolution on a given AudioBuffer, often used to achieve a reverb effect. A ConvolverNode always has exactly one input and one output.
  *
@@ -6479,6 +6547,19 @@ declare var CustomEvent: {
   new <T>(type: string, eventInitDict?: CustomEventInit<T>): CustomEvent<T>;
 };
 
+/** [MDN Reference](https://developer.mozilla.org/docs/Web/API/CustomStateSet) */
+interface CustomStateSet {
+  forEach(
+    callbackfn: (value: string, key: string, parent: CustomStateSet) => void,
+    thisArg?: any,
+  ): void;
+}
+
+declare var CustomStateSet: {
+  prototype: CustomStateSet;
+  new (): CustomStateSet;
+};
+
 /**
  * An abnormal event (called an exception) which occurs as a result of calling a method or accessing a property of a web API.
  *
@@ -7770,6 +7851,9 @@ interface Document
   createEvent(eventInterface: 'ClipboardEvent'): ClipboardEvent;
   createEvent(eventInterface: 'CloseEvent'): CloseEvent;
   createEvent(eventInterface: 'CompositionEvent'): CompositionEvent;
+  createEvent(
+    eventInterface: 'ContentVisibilityAutoStateChangeEvent',
+  ): ContentVisibilityAutoStateChangeEvent;
   createEvent(eventInterface: 'CustomEvent'): CustomEvent;
   createEvent(eventInterface: 'DeviceMotionEvent'): DeviceMotionEvent;
   createEvent(eventInterface: 'DeviceOrientationEvent'): DeviceOrientationEvent;
@@ -8092,6 +8176,7 @@ interface Document
 declare var Document: {
   prototype: Document;
   new (): Document;
+  parseHTMLUnsafe(html: string): Document;
 };
 
 /**
@@ -8582,6 +8667,7 @@ interface Element
   setAttributeNode(attr: Attr): Attr | null;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/Element/setAttributeNodeNS) */
   setAttributeNodeNS(attr: Attr): Attr | null;
+  setHTMLUnsafe(html: string): void;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/Element/setPointerCapture) */
   setPointerCapture(pointerId: number): void;
   /**
@@ -8662,6 +8748,8 @@ interface ElementInternals extends ARIAMixin {
    * [MDN Reference](https://developer.mozilla.org/docs/Web/API/ElementInternals/shadowRoot)
    */
   readonly shadowRoot: ShadowRoot | null;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/ElementInternals/states) */
+  readonly states: CustomStateSet;
   /**
    * Returns the error message that would be shown to the user if internals's target element was to be checked for validity.
    *
@@ -9560,7 +9648,6 @@ declare var GainNode: {
 
 /**
  * This Gamepad API interface defines an individual gamepad or other controller, allowing access to information such as button presses, axis positions, and id.
- * Available only in secure contexts.
  *
  * [MDN Reference](https://developer.mozilla.org/docs/Web/API/Gamepad)
  */
@@ -9579,7 +9666,7 @@ interface Gamepad {
   readonly mapping: GamepadMappingType;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/Gamepad/timestamp) */
   readonly timestamp: DOMHighResTimeStamp;
-  readonly vibrationActuator: GamepadHapticActuator | null;
+  readonly vibrationActuator: GamepadHapticActuator;
 }
 
 declare var Gamepad: {
@@ -9589,7 +9676,6 @@ declare var Gamepad: {
 
 /**
  * An individual button of a gamepad or other controller, allowing access to the current state of different types of buttons available on the control device.
- * Available only in secure contexts.
  *
  * [MDN Reference](https://developer.mozilla.org/docs/Web/API/GamepadButton)
  */
@@ -9609,7 +9695,6 @@ declare var GamepadButton: {
 
 /**
  * This Gamepad API interface contains references to gamepads connected to the system, which is what the gamepad events Window.gamepadconnected and Window.gamepaddisconnected are fired in response to.
- * Available only in secure contexts.
  *
  * [MDN Reference](https://developer.mozilla.org/docs/Web/API/GamepadEvent)
  */
@@ -9629,8 +9714,6 @@ declare var GamepadEvent: {
  * [MDN Reference](https://developer.mozilla.org/docs/Web/API/GamepadHapticActuator)
  */
 interface GamepadHapticActuator {
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/GamepadHapticActuator/type) */
-  readonly type: GamepadHapticActuatorType;
   playEffect(
     type: GamepadHapticEffectType,
     params?: GamepadEffectParameters,
@@ -10414,6 +10497,7 @@ interface HTMLAnchorElement extends HTMLElement, HTMLHyperlinkElementUtils {
    * [MDN Reference](https://developer.mozilla.org/docs/Web/API/HTMLAnchorElement/name)
    */
   name: string;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/HTMLAnchorElement/ping) */
   ping: string;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/HTMLAnchorElement/referrerPolicy) */
   referrerPolicy: string;
@@ -10507,6 +10591,7 @@ interface HTMLAreaElement extends HTMLElement, HTMLHyperlinkElementUtils {
    * [MDN Reference](https://developer.mozilla.org/docs/Web/API/HTMLAreaElement/noHref)
    */
   noHref: boolean;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/HTMLAreaElement/ping) */
   ping: string;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/HTMLAreaElement/referrerPolicy) */
   referrerPolicy: string;
@@ -11376,7 +11461,11 @@ declare var HTMLElement: {
 interface HTMLEmbedElement extends HTMLElement {
   /** @deprecated */
   align: string;
-  /** Sets or retrieves the height of the object. */
+  /**
+   * Sets or retrieves the height of the object.
+   *
+   * [MDN Reference](https://developer.mozilla.org/docs/Web/API/HTMLEmbedElement/height)
+   */
   height: string;
   /**
    * Sets or retrieves the name of the object.
@@ -11386,7 +11475,11 @@ interface HTMLEmbedElement extends HTMLElement {
   /** Sets or retrieves a URL to be loaded by the object. */
   src: string;
   type: string;
-  /** Sets or retrieves the width of the object. */
+  /**
+   * Sets or retrieves the width of the object.
+   *
+   * [MDN Reference](https://developer.mozilla.org/docs/Web/API/HTMLEmbedElement/width)
+   */
   width: string;
   getSVGDocument(): Document | null;
   addEventListener<K extends keyof HTMLElementEventMap>(
@@ -12125,6 +12218,7 @@ interface HTMLIFrameElement extends HTMLElement {
    */
   align: string;
   allow: string;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/HTMLIFrameElement/allowFullscreen) */
   allowFullscreen: boolean;
   /**
    * Retrieves the document object of the page or frame.
@@ -12151,6 +12245,7 @@ interface HTMLIFrameElement extends HTMLElement {
    * [MDN Reference](https://developer.mozilla.org/docs/Web/API/HTMLIFrameElement/height)
    */
   height: string;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/HTMLIFrameElement/loading) */
   loading: string;
   /**
    * Sets or retrieves a URI to a long description of the object.
@@ -12422,7 +12517,11 @@ interface HTMLInputElement extends HTMLElement, PopoverInvokerElement {
   checked: boolean;
   /** Sets or retrieves the state of the check box or radio button. */
   defaultChecked: boolean;
-  /** Sets or retrieves the initial contents of the object. */
+  /**
+   * Sets or retrieves the initial contents of the object.
+   *
+   * [MDN Reference](https://developer.mozilla.org/docs/Web/API/HTMLInputElement/defaultValue)
+   */
   defaultValue: string;
   dirName: string;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/HTMLInputElement/disabled) */
@@ -12481,12 +12580,25 @@ interface HTMLInputElement extends HTMLElement, PopoverInvokerElement {
    * [MDN Reference](https://developer.mozilla.org/docs/Web/API/HTMLInputElement/list)
    */
   readonly list: HTMLDataListElement | null;
-  /** Defines the maximum acceptable value for an input element with type="number".When used with the min and step attributes, lets you control the range and increment (such as only even numbers) that the user can enter into an input field. */
+  /**
+   * Defines the maximum acceptable value for an input element with type="number".When used with the min and step attributes, lets you control the range and increment (such as only even numbers) that the user can enter into an input field.
+   *
+   * [MDN Reference](https://developer.mozilla.org/docs/Web/API/HTMLInputElement/max)
+   */
   max: string;
-  /** Sets or retrieves the maximum number of characters that the user can enter in a text control. */
+  /**
+   * Sets or retrieves the maximum number of characters that the user can enter in a text control.
+   *
+   * [MDN Reference](https://developer.mozilla.org/docs/Web/API/HTMLInputElement/maxLength)
+   */
   maxLength: number;
-  /** Defines the minimum acceptable value for an input element with type="number". When used with the max and step attributes, lets you control the range and increment (such as even numbers only) that the user can enter into an input field. */
+  /**
+   * Defines the minimum acceptable value for an input element with type="number". When used with the max and step attributes, lets you control the range and increment (such as even numbers only) that the user can enter into an input field.
+   *
+   * [MDN Reference](https://developer.mozilla.org/docs/Web/API/HTMLInputElement/min)
+   */
   min: string;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/HTMLInputElement/minLength) */
   minLength: number;
   /**
    * Sets or retrieves the Boolean value indicating whether multiple items can be selected from a list.
@@ -12517,16 +12629,28 @@ interface HTMLInputElement extends HTMLElement, PopoverInvokerElement {
   required: boolean;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/HTMLInputElement/selectionDirection) */
   selectionDirection: 'forward' | 'backward' | 'none' | null;
-  /** Gets or sets the end position or offset of a text selection. */
+  /**
+   * Gets or sets the end position or offset of a text selection.
+   *
+   * [MDN Reference](https://developer.mozilla.org/docs/Web/API/HTMLInputElement/selectionEnd)
+   */
   selectionEnd: number | null;
-  /** Gets or sets the starting position or offset of a text selection. */
+  /**
+   * Gets or sets the starting position or offset of a text selection.
+   *
+   * [MDN Reference](https://developer.mozilla.org/docs/Web/API/HTMLInputElement/selectionStart)
+   */
   selectionStart: number | null;
   size: number;
   /** The address or URL of the a media resource that is to be considered. */
   src: string;
   /** Defines an increment or jump between values that you want to allow the user to enter. When used with the max and min attributes, lets you control the range and increment (for example, allow only even numbers) that the user can enter into an input field. */
   step: string;
-  /** Returns the content type of the object. */
+  /**
+   * Returns the content type of the object.
+   *
+   * [MDN Reference](https://developer.mozilla.org/docs/Web/API/HTMLInputElement/type)
+   */
   type: string;
   /**
    * Sets or retrieves the URL, often with a bookmark extension (#name), to use as a client-side image map.
@@ -12545,7 +12669,11 @@ interface HTMLInputElement extends HTMLElement, PopoverInvokerElement {
    * [MDN Reference](https://developer.mozilla.org/docs/Web/API/HTMLInputElement/validity)
    */
   readonly validity: ValidityState;
-  /** Returns the value of the data at the cursor's current position. */
+  /**
+   * Returns the value of the data at the cursor's current position.
+   *
+   * [MDN Reference](https://developer.mozilla.org/docs/Web/API/HTMLInputElement/value)
+   */
   value: string;
   /** Returns a Date object representing the form control's value, if applicable; otherwise, returns null. Can be set, to change the value. Throws an "InvalidStateError" DOMException if the control isn't date- or time-based. */
   valueAsDate: Date | null;
@@ -13264,16 +13392,31 @@ declare var HTMLMenuElement: {
  * [MDN Reference](https://developer.mozilla.org/docs/Web/API/HTMLMetaElement)
  */
 interface HTMLMetaElement extends HTMLElement {
-  /** Gets or sets meta-information to associate with httpEquiv or name. */
+  /**
+   * Gets or sets meta-information to associate with httpEquiv or name.
+   *
+   * [MDN Reference](https://developer.mozilla.org/docs/Web/API/HTMLMetaElement/content)
+   */
   content: string;
-  /** Gets or sets information used to bind the value of a content attribute of a meta element to an HTTP response header. */
+  /**
+   * Gets or sets information used to bind the value of a content attribute of a meta element to an HTTP response header.
+   *
+   * [MDN Reference](https://developer.mozilla.org/docs/Web/API/HTMLMetaElement/httpEquiv)
+   */
   httpEquiv: string;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/HTMLMetaElement/media) */
   media: string;
-  /** Sets or retrieves the value specified in the content attribute of the meta object. */
+  /**
+   * Sets or retrieves the value specified in the content attribute of the meta object.
+   *
+   * [MDN Reference](https://developer.mozilla.org/docs/Web/API/HTMLMetaElement/name)
+   */
   name: string;
   /**
    * Sets or retrieves a scheme to be used in interpreting the value of a property specified for the object.
    * @deprecated
+   *
+   * [MDN Reference](https://developer.mozilla.org/docs/Web/API/HTMLMetaElement/scheme)
    */
   scheme: string;
   addEventListener<K extends keyof HTMLElementEventMap>(
@@ -13549,6 +13692,7 @@ interface HTMLObjectElement extends HTMLElement {
   type: string;
   /**
    * Sets or retrieves the URL, often with a bookmark extension (#name), to use as a client-side image map.
+   * @deprecated
    *
    * [MDN Reference](https://developer.mozilla.org/docs/Web/API/HTMLObjectElement/useMap)
    */
@@ -14282,7 +14426,7 @@ interface HTMLSelectElement extends HTMLElement {
    *
    * [MDN Reference](https://developer.mozilla.org/docs/Web/API/HTMLSelectElement/type)
    */
-  readonly type: string;
+  readonly type: 'select-one' | 'select-multiple';
   /**
    * Returns the error message that would be displayed if the user submits the form, or an empty string if no error message. It also triggers the standard error message, such as "this is a required field". The result is that the user sees validation messages without actually submitting.
    *
@@ -14429,6 +14573,7 @@ declare var HTMLSlotElement: {
  * [MDN Reference](https://developer.mozilla.org/docs/Web/API/HTMLSourceElement)
  */
 interface HTMLSourceElement extends HTMLElement {
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/HTMLSourceElement/height) */
   height: number;
   /**
    * Gets or sets the intended media type of the media source.
@@ -14452,6 +14597,7 @@ interface HTMLSourceElement extends HTMLElement {
    * [MDN Reference](https://developer.mozilla.org/docs/Web/API/HTMLSourceElement/type)
    */
   type: string;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/HTMLSourceElement/width) */
   width: number;
   addEventListener<K extends keyof HTMLElementEventMap>(
     type: K,
@@ -15309,7 +15455,11 @@ interface HTMLTextAreaElement extends HTMLElement {
   selectionStart: number;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/HTMLTextAreaElement/textLength) */
   readonly textLength: number;
-  /** Retrieves the type of control. */
+  /**
+   * Retrieves the type of control.
+   *
+   * [MDN Reference](https://developer.mozilla.org/docs/Web/API/HTMLTextAreaElement/type)
+   */
   readonly type: string;
   /** Returns the error message that would be displayed if the user submits the form, or an empty string if no error message. It also triggers the standard error message, such as "this is a required field". The result is that the user sees validation messages without actually submitting. */
   readonly validationMessage: string;
@@ -15637,11 +15787,13 @@ interface HTMLVideoElement extends HTMLMediaElement {
    * [MDN Reference](https://developer.mozilla.org/docs/Web/API/HTMLVideoElement/width)
    */
   width: number;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/HTMLVideoElement/cancelVideoFrameCallback) */
   cancelVideoFrameCallback(handle: number): void;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/HTMLVideoElement/getVideoPlaybackQuality) */
   getVideoPlaybackQuality(): VideoPlaybackQuality;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/HTMLVideoElement/requestPictureInPicture) */
   requestPictureInPicture(): Promise<PictureInPictureWindow>;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/HTMLVideoElement/requestVideoFrameCallback) */
   requestVideoFrameCallback(callback: VideoFrameRequestCallback): number;
   addEventListener<K extends keyof HTMLVideoElementEventMap>(
     type: K,
@@ -16898,6 +17050,29 @@ declare var KeyframeEffect: {
   new (source: KeyframeEffect): KeyframeEffect;
 };
 
+/** [MDN Reference](https://developer.mozilla.org/docs/Web/API/LargestContentfulPaint) */
+interface LargestContentfulPaint extends PerformanceEntry {
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/LargestContentfulPaint/element) */
+  readonly element: Element | null;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/LargestContentfulPaint/id) */
+  readonly id: string;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/LargestContentfulPaint/loadTime) */
+  readonly loadTime: DOMHighResTimeStamp;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/LargestContentfulPaint/renderTime) */
+  readonly renderTime: DOMHighResTimeStamp;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/LargestContentfulPaint/size) */
+  readonly size: number;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/LargestContentfulPaint/url) */
+  readonly url: string;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/LargestContentfulPaint/toJSON) */
+  toJSON(): any;
+}
+
+declare var LargestContentfulPaint: {
+  prototype: LargestContentfulPaint;
+  new (): LargestContentfulPaint;
+};
+
 interface LinkStyle {
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/HTMLLinkElement/sheet) */
   readonly sheet: CSSStyleSheet | null;
@@ -25882,6 +26057,8 @@ interface ShadowRootEventMap {
 
 /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/ShadowRoot) */
 interface ShadowRoot extends DocumentFragment, DocumentOrShadowRoot, InnerHTML {
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/ShadowRoot/clonable) */
+  readonly clonable: boolean;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/ShadowRoot/delegatesFocus) */
   readonly delegatesFocus: boolean;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/ShadowRoot/host) */
@@ -25891,6 +26068,7 @@ interface ShadowRoot extends DocumentFragment, DocumentOrShadowRoot, InnerHTML {
   onslotchange: ((this: ShadowRoot, ev: Event) => any) | null;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/ShadowRoot/slotAssignment) */
   readonly slotAssignment: SlotAssignmentMode;
+  setHTMLUnsafe(html: string): void;
   /** Throws a "NotSupportedError" DOMException if context object is a shadow root. */
   addEventListener<K extends keyof ShadowRootEventMap>(
     type: K,
@@ -31735,7 +31913,11 @@ interface Window
   readonly window: Window & typeof globalThis;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/Window/alert) */
   alert(message?: any): void;
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/Window/blur) */
+  /**
+   * @deprecated
+   *
+   * [MDN Reference](https://developer.mozilla.org/docs/Web/API/Window/blur)
+   */
   blur(): void;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/Window/cancelIdleCallback) */
   cancelIdleCallback(handle: number): void;
@@ -33600,7 +33782,11 @@ declare var visualViewport: VisualViewport | null;
 declare var window: Window & typeof globalThis;
 /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/Window/alert) */
 declare function alert(message?: any): void;
-/** [MDN Reference](https://developer.mozilla.org/docs/Web/API/Window/blur) */
+/**
+ * @deprecated
+ *
+ * [MDN Reference](https://developer.mozilla.org/docs/Web/API/Window/blur)
+ */
 declare function blur(): void;
 /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/Window/cancelIdleCallback) */
 declare function cancelIdleCallback(handle: number): void;
@@ -34598,8 +34784,7 @@ type FontDisplay = 'auto' | 'block' | 'fallback' | 'optional' | 'swap';
 type FontFaceLoadStatus = 'error' | 'loaded' | 'loading' | 'unloaded';
 type FontFaceSetLoadStatus = 'loaded' | 'loading';
 type FullscreenNavigationUI = 'auto' | 'hide' | 'show';
-type GamepadHapticActuatorType = 'vibration';
-type GamepadHapticEffectType = 'dual-rumble';
+type GamepadHapticEffectType = 'dual-rumble' | 'trigger-rumble';
 type GamepadHapticsResult = 'complete' | 'preempted';
 type GamepadMappingType = '' | 'standard' | 'xr-standard';
 type GlobalCompositeOperation =
