diff --git a/packages/v5.4/temp/copied/lib.es2023.array.d.ts b/packages/v5.5/temp/copied/lib.es2023.array.d.ts
index 9fc0b83..d19505e 100644
--- a/packages/v5.4/temp/copied/lib.es2023.array.d.ts
+++ b/packages/v5.5/temp/copied/lib.es2023.array.d.ts
@@ -207,7 +207,7 @@ interface Int8Array {
   /**
    * Copies the array and returns the copy with the elements in reverse order.
    */
-  toReversed(): Uint8Array;
+  toReversed(): Int8Array;
 
   /**
    * Copies and sorts the array.
@@ -215,11 +215,11 @@ interface Int8Array {
    * a negative value if the first argument is less than the second argument, zero if they're equal, and a positive
    * value otherwise. If omitted, the elements are sorted in ascending order.
    * ```ts
-   * const myNums = Uint8Array.from([11, 2, 22, 1]);
-   * myNums.toSorted((a, b) => a - b) // Uint8Array(4) [1, 2, 11, 22]
+   * const myNums = Int8Array.from([11, 2, 22, 1]);
+   * myNums.toSorted((a, b) => a - b) // Int8Array(4) [1, 2, 11, 22]
    * ```
    */
-  toSorted(compareFn?: (a: number, b: number) => number): Uint8Array;
+  toSorted(compareFn?: (a: number, b: number) => number): Int8Array;
 
   /**
    * Copies the array and inserts the given number at the provided index.
@@ -228,7 +228,7 @@ interface Int8Array {
    * @param value The value to insert into the copied array.
    * @returns A copy of the original array with the inserted value.
    */
-  with(index: number, value: number): Uint8Array;
+  with(index: number, value: number): Int8Array;
 }
 
 interface Uint8Array {
