diff --git a/packages/v5.4/temp/copied/lib.dom.iterable.d.ts b/packages/v5.5/temp/copied/lib.dom.iterable.d.ts
index 0fd8dbd..24d1936 100644
--- a/packages/v5.4/temp/copied/lib.dom.iterable.d.ts
+++ b/packages/v5.5/temp/copied/lib.dom.iterable.d.ts
@@ -19,6 +19,11 @@ and limitations under the License.
 /// Window Iterable APIs
 /////////////////////////////
 
+interface AbortSignal {
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/AbortSignal/any_static) */
+  any(signals: Iterable<AbortSignal>): AbortSignal;
+}
+
 interface AudioParam {
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/AudioParam/setValueCurveAtTime) */
   setValueCurveAtTime(
@@ -98,6 +103,8 @@ interface CanvasPathDrawingStyles {
   setLineDash(segments: Iterable<number>): void;
 }
 
+interface CustomStateSet extends Set<string> {}
+
 interface DOMRectList {
   [Symbol.iterator](): IterableIterator<DOMRect>;
 }
