diff --git a/packages/v5.0/temp/copied/lib.webworker.iterable.d.ts b/packages/v5.1/temp/copied/lib.webworker.iterable.d.ts
index a99d466..c839a53 100644
--- a/packages/v5.0/temp/copied/lib.webworker.iterable.d.ts
+++ b/packages/v5.1/temp/copied/lib.webworker.iterable.d.ts
@@ -19,11 +19,34 @@ and limitations under the License.
 /// Worker Iterable APIs
 /////////////////////////////
 
+interface CSSNumericArray {
+  [Symbol.iterator](): IterableIterator<CSSNumericValue>;
+  entries(): IterableIterator<[number, CSSNumericValue]>;
+  keys(): IterableIterator<number>;
+  values(): IterableIterator<CSSNumericValue>;
+}
+
+interface CSSTransformValue {
+  [Symbol.iterator](): IterableIterator<CSSTransformComponent>;
+  entries(): IterableIterator<[number, CSSTransformComponent]>;
+  keys(): IterableIterator<number>;
+  values(): IterableIterator<CSSTransformComponent>;
+}
+
+interface CSSUnparsedValue {
+  [Symbol.iterator](): IterableIterator<CSSUnparsedSegment>;
+  entries(): IterableIterator<[number, CSSUnparsedSegment]>;
+  keys(): IterableIterator<number>;
+  values(): IterableIterator<CSSUnparsedSegment>;
+}
+
 interface Cache {
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/Cache/addAll) */
   addAll(requests: Iterable<RequestInfo>): Promise<void>;
 }
 
 interface CanvasPath {
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/CanvasRenderingContext2D/roundRect) */
   roundRect(
     x: number,
     y: number,
@@ -34,6 +57,7 @@ interface CanvasPath {
 }
 
 interface CanvasPathDrawingStyles {
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/CanvasRenderingContext2D/setLineDash) */
   setLineDash(segments: Iterable<number>): void;
 }
 
@@ -68,7 +92,11 @@ interface Headers {
 }
 
 interface IDBDatabase {
-  /** Returns a new transaction with the given mode ("readonly" or "readwrite") and scope which can be a single object store name or an array of names. */
+  /**
+   * Returns a new transaction with the given mode ("readonly" or "readwrite") and scope which can be a single object store name or an array of names.
+   *
+   * [MDN Reference](https://developer.mozilla.org/docs/Web/API/IDBDatabase/transaction)
+   */
   transaction(
     storeNames: string | Iterable<string>,
     mode?: IDBTransactionMode,
@@ -81,6 +109,8 @@ interface IDBObjectStore {
    * Creates a new index in store with the given name, keyPath and options and returns a new IDBIndex. If the keyPath and options define constraints that cannot be satisfied with the data already in store the upgrade transaction will abort with a "ConstraintError" DOMException.
    *
    * Throws an "InvalidStateError" DOMException if not called within an upgrade transaction.
+   *
+   * [MDN Reference](https://developer.mozilla.org/docs/Web/API/IDBObjectStore/createIndex)
    */
   createIndex(
     name: string,
@@ -90,7 +120,11 @@ interface IDBObjectStore {
 }
 
 interface MessageEvent<T = any> {
-  /** @deprecated */
+  /**
+   * @deprecated
+   *
+   * [MDN Reference](https://developer.mozilla.org/docs/Web/API/MessageEvent/initMessageEvent)
+   */
   initMessageEvent(
     type: string,
     bubbles?: boolean,
@@ -103,7 +137,15 @@ interface MessageEvent<T = any> {
   ): void;
 }
 
+interface StylePropertyMapReadOnly {
+  [Symbol.iterator](): IterableIterator<[string, Iterable<CSSStyleValue>]>;
+  entries(): IterableIterator<[string, Iterable<CSSStyleValue>]>;
+  keys(): IterableIterator<string>;
+  values(): IterableIterator<Iterable<CSSStyleValue>>;
+}
+
 interface SubtleCrypto {
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/SubtleCrypto/deriveKey) */
   deriveKey(
     algorithm:
       | AlgorithmIdentifier
@@ -120,6 +162,7 @@ interface SubtleCrypto {
     extractable: boolean,
     keyUsages: Iterable<KeyUsage>,
   ): Promise<CryptoKey>;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/SubtleCrypto/generateKey) */
   generateKey(
     algorithm: RsaHashedKeyGenParams | EcKeyGenParams,
     extractable: boolean,
@@ -135,6 +178,7 @@ interface SubtleCrypto {
     extractable: boolean,
     keyUsages: Iterable<KeyUsage>,
   ): Promise<CryptoKeyPair | CryptoKey>;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/SubtleCrypto/importKey) */
   importKey(
     format: 'jwk',
     keyData: JsonWebKey,
@@ -159,6 +203,7 @@ interface SubtleCrypto {
     extractable: boolean,
     keyUsages: Iterable<KeyUsage>,
   ): Promise<CryptoKey>;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/SubtleCrypto/unwrapKey) */
   unwrapKey(
     format: KeyFormat,
     wrappedKey: BufferSource,
@@ -191,10 +236,12 @@ interface URLSearchParams {
 }
 
 interface WEBGL_draw_buffers {
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WEBGL_draw_buffers/drawBuffersWEBGL) */
   drawBuffersWEBGL(buffers: Iterable<GLenum>): void;
 }
 
 interface WEBGL_multi_draw {
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WEBGL_multi_draw/multiDrawArraysInstancedWEBGL) */
   multiDrawArraysInstancedWEBGL(
     mode: GLenum,
     firstsList: Int32Array | Iterable<GLint>,
@@ -205,6 +252,7 @@ interface WEBGL_multi_draw {
     instanceCountsOffset: GLuint,
     drawcount: GLsizei,
   ): void;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WEBGL_multi_draw/multiDrawArraysWEBGL) */
   multiDrawArraysWEBGL(
     mode: GLenum,
     firstsList: Int32Array | Iterable<GLint>,
@@ -213,6 +261,7 @@ interface WEBGL_multi_draw {
     countsOffset: GLuint,
     drawcount: GLsizei,
   ): void;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WEBGL_multi_draw/multiDrawElementsInstancedWEBGL) */
   multiDrawElementsInstancedWEBGL(
     mode: GLenum,
     countsList: Int32Array | Iterable<GLsizei>,
@@ -224,6 +273,7 @@ interface WEBGL_multi_draw {
     instanceCountsOffset: GLuint,
     drawcount: GLsizei,
   ): void;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WEBGL_multi_draw/multiDrawElementsWEBGL) */
   multiDrawElementsWEBGL(
     mode: GLenum,
     countsList: Int32Array | Iterable<GLsizei>,
@@ -236,35 +286,43 @@ interface WEBGL_multi_draw {
 }
 
 interface WebGL2RenderingContextBase {
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGL2RenderingContext/clearBuffer) */
   clearBufferfv(
     buffer: GLenum,
     drawbuffer: GLint,
     values: Iterable<GLfloat>,
     srcOffset?: GLuint,
   ): void;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGL2RenderingContext/clearBuffer) */
   clearBufferiv(
     buffer: GLenum,
     drawbuffer: GLint,
     values: Iterable<GLint>,
     srcOffset?: GLuint,
   ): void;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGL2RenderingContext/clearBuffer) */
   clearBufferuiv(
     buffer: GLenum,
     drawbuffer: GLint,
     values: Iterable<GLuint>,
     srcOffset?: GLuint,
   ): void;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGL2RenderingContext/drawBuffers) */
   drawBuffers(buffers: Iterable<GLenum>): void;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGL2RenderingContext/getActiveUniforms) */
   getActiveUniforms(
     program: WebGLProgram,
     uniformIndices: Iterable<GLuint>,
     pname: GLenum,
   ): any;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGL2RenderingContext/getUniformIndices) */
   getUniformIndices(
     program: WebGLProgram,
     uniformNames: Iterable<string>,
   ): Iterable<GLuint> | null;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGL2RenderingContext/invalidateFramebuffer) */
   invalidateFramebuffer(target: GLenum, attachments: Iterable<GLenum>): void;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGL2RenderingContext/invalidateSubFramebuffer) */
   invalidateSubFramebuffer(
     target: GLenum,
     attachments: Iterable<GLenum>,
@@ -273,35 +331,41 @@ interface WebGL2RenderingContextBase {
     width: GLsizei,
     height: GLsizei,
   ): void;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGL2RenderingContext/transformFeedbackVaryings) */
   transformFeedbackVaryings(
     program: WebGLProgram,
     varyings: Iterable<string>,
     bufferMode: GLenum,
   ): void;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGL2RenderingContext/uniform) */
   uniform1uiv(
     location: WebGLUniformLocation | null,
     data: Iterable<GLuint>,
     srcOffset?: GLuint,
     srcLength?: GLuint,
   ): void;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGL2RenderingContext/uniform) */
   uniform2uiv(
     location: WebGLUniformLocation | null,
     data: Iterable<GLuint>,
     srcOffset?: GLuint,
     srcLength?: GLuint,
   ): void;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGL2RenderingContext/uniform) */
   uniform3uiv(
     location: WebGLUniformLocation | null,
     data: Iterable<GLuint>,
     srcOffset?: GLuint,
     srcLength?: GLuint,
   ): void;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGL2RenderingContext/uniform) */
   uniform4uiv(
     location: WebGLUniformLocation | null,
     data: Iterable<GLuint>,
     srcOffset?: GLuint,
     srcLength?: GLuint,
   ): void;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGL2RenderingContext/uniformMatrix) */
   uniformMatrix2x3fv(
     location: WebGLUniformLocation | null,
     transpose: GLboolean,
@@ -309,6 +373,7 @@ interface WebGL2RenderingContextBase {
     srcOffset?: GLuint,
     srcLength?: GLuint,
   ): void;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGL2RenderingContext/uniformMatrix) */
   uniformMatrix2x4fv(
     location: WebGLUniformLocation | null,
     transpose: GLboolean,
@@ -316,6 +381,7 @@ interface WebGL2RenderingContextBase {
     srcOffset?: GLuint,
     srcLength?: GLuint,
   ): void;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGL2RenderingContext/uniformMatrix) */
   uniformMatrix3x2fv(
     location: WebGLUniformLocation | null,
     transpose: GLboolean,
@@ -323,6 +389,7 @@ interface WebGL2RenderingContextBase {
     srcOffset?: GLuint,
     srcLength?: GLuint,
   ): void;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGL2RenderingContext/uniformMatrix) */
   uniformMatrix3x4fv(
     location: WebGLUniformLocation | null,
     transpose: GLboolean,
@@ -330,6 +397,7 @@ interface WebGL2RenderingContextBase {
     srcOffset?: GLuint,
     srcLength?: GLuint,
   ): void;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGL2RenderingContext/uniformMatrix) */
   uniformMatrix4x2fv(
     location: WebGLUniformLocation | null,
     transpose: GLboolean,
@@ -337,6 +405,7 @@ interface WebGL2RenderingContextBase {
     srcOffset?: GLuint,
     srcLength?: GLuint,
   ): void;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGL2RenderingContext/uniformMatrix) */
   uniformMatrix4x3fv(
     location: WebGLUniformLocation | null,
     transpose: GLboolean,
@@ -344,59 +413,70 @@ interface WebGL2RenderingContextBase {
     srcOffset?: GLuint,
     srcLength?: GLuint,
   ): void;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGL2RenderingContext/vertexAttribI) */
   vertexAttribI4iv(index: GLuint, values: Iterable<GLint>): void;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGL2RenderingContext/vertexAttribI) */
   vertexAttribI4uiv(index: GLuint, values: Iterable<GLuint>): void;
 }
 
 interface WebGL2RenderingContextOverloads {
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGLRenderingContext/uniform) */
   uniform1fv(
     location: WebGLUniformLocation | null,
     data: Iterable<GLfloat>,
     srcOffset?: GLuint,
     srcLength?: GLuint,
   ): void;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGLRenderingContext/uniform) */
   uniform1iv(
     location: WebGLUniformLocation | null,
     data: Iterable<GLint>,
     srcOffset?: GLuint,
     srcLength?: GLuint,
   ): void;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGLRenderingContext/uniform) */
   uniform2fv(
     location: WebGLUniformLocation | null,
     data: Iterable<GLfloat>,
     srcOffset?: GLuint,
     srcLength?: GLuint,
   ): void;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGLRenderingContext/uniform) */
   uniform2iv(
     location: WebGLUniformLocation | null,
     data: Iterable<GLint>,
     srcOffset?: GLuint,
     srcLength?: GLuint,
   ): void;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGLRenderingContext/uniform) */
   uniform3fv(
     location: WebGLUniformLocation | null,
     data: Iterable<GLfloat>,
     srcOffset?: GLuint,
     srcLength?: GLuint,
   ): void;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGLRenderingContext/uniform) */
   uniform3iv(
     location: WebGLUniformLocation | null,
     data: Iterable<GLint>,
     srcOffset?: GLuint,
     srcLength?: GLuint,
   ): void;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGLRenderingContext/uniform) */
   uniform4fv(
     location: WebGLUniformLocation | null,
     data: Iterable<GLfloat>,
     srcOffset?: GLuint,
     srcLength?: GLuint,
   ): void;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGLRenderingContext/uniform) */
   uniform4iv(
     location: WebGLUniformLocation | null,
     data: Iterable<GLint>,
     srcOffset?: GLuint,
     srcLength?: GLuint,
   ): void;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGLRenderingContext/uniformMatrix) */
   uniformMatrix2fv(
     location: WebGLUniformLocation | null,
     transpose: GLboolean,
@@ -404,6 +484,7 @@ interface WebGL2RenderingContextOverloads {
     srcOffset?: GLuint,
     srcLength?: GLuint,
   ): void;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGLRenderingContext/uniformMatrix) */
   uniformMatrix3fv(
     location: WebGLUniformLocation | null,
     transpose: GLboolean,
@@ -411,6 +492,7 @@ interface WebGL2RenderingContextOverloads {
     srcOffset?: GLuint,
     srcLength?: GLuint,
   ): void;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGLRenderingContext/uniformMatrix) */
   uniformMatrix4fv(
     location: WebGLUniformLocation | null,
     transpose: GLboolean,
@@ -421,31 +503,46 @@ interface WebGL2RenderingContextOverloads {
 }
 
 interface WebGLRenderingContextBase {
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGLRenderingContext/vertexAttrib) */
   vertexAttrib1fv(index: GLuint, values: Iterable<GLfloat>): void;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGLRenderingContext/vertexAttrib) */
   vertexAttrib2fv(index: GLuint, values: Iterable<GLfloat>): void;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGLRenderingContext/vertexAttrib) */
   vertexAttrib3fv(index: GLuint, values: Iterable<GLfloat>): void;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGLRenderingContext/vertexAttrib) */
   vertexAttrib4fv(index: GLuint, values: Iterable<GLfloat>): void;
 }
 
 interface WebGLRenderingContextOverloads {
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGLRenderingContext/uniform) */
   uniform1fv(location: WebGLUniformLocation | null, v: Iterable<GLfloat>): void;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGLRenderingContext/uniform) */
   uniform1iv(location: WebGLUniformLocation | null, v: Iterable<GLint>): void;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGLRenderingContext/uniform) */
   uniform2fv(location: WebGLUniformLocation | null, v: Iterable<GLfloat>): void;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGLRenderingContext/uniform) */
   uniform2iv(location: WebGLUniformLocation | null, v: Iterable<GLint>): void;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGLRenderingContext/uniform) */
   uniform3fv(location: WebGLUniformLocation | null, v: Iterable<GLfloat>): void;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGLRenderingContext/uniform) */
   uniform3iv(location: WebGLUniformLocation | null, v: Iterable<GLint>): void;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGLRenderingContext/uniform) */
   uniform4fv(location: WebGLUniformLocation | null, v: Iterable<GLfloat>): void;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGLRenderingContext/uniform) */
   uniform4iv(location: WebGLUniformLocation | null, v: Iterable<GLint>): void;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGLRenderingContext/uniformMatrix) */
   uniformMatrix2fv(
     location: WebGLUniformLocation | null,
     transpose: GLboolean,
     value: Iterable<GLfloat>,
   ): void;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGLRenderingContext/uniformMatrix) */
   uniformMatrix3fv(
     location: WebGLUniformLocation | null,
     transpose: GLboolean,
     value: Iterable<GLfloat>,
   ): void;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGLRenderingContext/uniformMatrix) */
   uniformMatrix4fv(
     location: WebGLUniformLocation | null,
     transpose: GLboolean,
