diff --git a/packages/v5.0/temp/copied/lib.es5.d.ts b/packages/v5.1/temp/copied/lib.es5.d.ts
index f104f63..4ff861f 100644
--- a/packages/v5.0/temp/copied/lib.es5.d.ts
+++ b/packages/v5.1/temp/copied/lib.es5.d.ts
@@ -358,9 +358,14 @@ interface CallableFunction extends Function {
   /**
    * Calls the function with the specified object as the this value and the elements of specified array as the arguments.
    * @param thisArg The object to be used as the this object.
-   * @param args An array of argument values to be passed to the function.
    */
   apply<T, R>(this: (this: T) => R, thisArg: T): R;
+
+  /**
+   * Calls the function with the specified object as the this value and the elements of specified array as the arguments.
+   * @param thisArg The object to be used as the this object.
+   * @param args An array of argument values to be passed to the function.
+   */
   apply<T, A extends any[], R>(
     this: (this: T, ...args: A) => R,
     thisArg: T,
@@ -382,49 +387,33 @@ interface CallableFunction extends Function {
    * For a given function, creates a bound function that has the same body as the original function.
    * The this object of the bound function is associated with the specified object, and has the specified initial parameters.
    * @param thisArg The object to be used as the this object.
-   * @param args Arguments to bind to the parameters of the function.
    */
   bind<T>(this: T, thisArg: ThisParameterType<T>): OmitThisParameter<T>;
-  bind<T, A0, A extends any[], R>(
-    this: (this: T, arg0: A0, ...args: A) => R,
-    thisArg: T,
-    arg0: A0,
-  ): (...args: A) => R;
-  bind<T, A0, A1, A extends any[], R>(
-    this: (this: T, arg0: A0, arg1: A1, ...args: A) => R,
-    thisArg: T,
-    arg0: A0,
-    arg1: A1,
-  ): (...args: A) => R;
-  bind<T, A0, A1, A2, A extends any[], R>(
-    this: (this: T, arg0: A0, arg1: A1, arg2: A2, ...args: A) => R,
-    thisArg: T,
-    arg0: A0,
-    arg1: A1,
-    arg2: A2,
-  ): (...args: A) => R;
-  bind<T, A0, A1, A2, A3, A extends any[], R>(
-    this: (this: T, arg0: A0, arg1: A1, arg2: A2, arg3: A3, ...args: A) => R,
-    thisArg: T,
-    arg0: A0,
-    arg1: A1,
-    arg2: A2,
-    arg3: A3,
-  ): (...args: A) => R;
-  bind<T, AX, R>(
-    this: (this: T, ...args: AX[]) => R,
+
+  /**
+   * For a given function, creates a bound function that has the same body as the original function.
+   * The this object of the bound function is associated with the specified object, and has the specified initial parameters.
+   * @param thisArg The object to be used as the this object.
+   * @param args Arguments to bind to the parameters of the function.
+   */
+  bind<T, A extends any[], B extends any[], R>(
+    this: (this: T, ...args: [...A, ...B]) => R,
     thisArg: T,
-    ...args: AX[]
-  ): (...args: AX[]) => R;
+    ...args: A
+  ): (...args: B) => R;
 }
 
 interface NewableFunction extends Function {
   /**
    * Calls the function with the specified object as the this value and the elements of specified array as the arguments.
    * @param thisArg The object to be used as the this object.
-   * @param args An array of argument values to be passed to the function.
    */
   apply<T>(this: new () => T, thisArg: T): void;
+  /**
+   * Calls the function with the specified object as the this value and the elements of specified array as the arguments.
+   * @param thisArg The object to be used as the this object.
+   * @param args An array of argument values to be passed to the function.
+   */
   apply<T, A extends any[]>(
     this: new (...args: A) => T,
     thisArg: T,
@@ -446,40 +435,20 @@ interface NewableFunction extends Function {
    * For a given function, creates a bound function that has the same body as the original function.
    * The this object of the bound function is associated with the specified object, and has the specified initial parameters.
    * @param thisArg The object to be used as the this object.
-   * @param args Arguments to bind to the parameters of the function.
    */
   bind<T>(this: T, thisArg: any): T;
-  bind<A0, A extends any[], R>(
-    this: new (arg0: A0, ...args: A) => R,
-    thisArg: any,
-    arg0: A0,
-  ): new (...args: A) => R;
-  bind<A0, A1, A extends any[], R>(
-    this: new (arg0: A0, arg1: A1, ...args: A) => R,
-    thisArg: any,
-    arg0: A0,
-    arg1: A1,
-  ): new (...args: A) => R;
-  bind<A0, A1, A2, A extends any[], R>(
-    this: new (arg0: A0, arg1: A1, arg2: A2, ...args: A) => R,
-    thisArg: any,
-    arg0: A0,
-    arg1: A1,
-    arg2: A2,
-  ): new (...args: A) => R;
-  bind<A0, A1, A2, A3, A extends any[], R>(
-    this: new (arg0: A0, arg1: A1, arg2: A2, arg3: A3, ...args: A) => R,
-    thisArg: any,
-    arg0: A0,
-    arg1: A1,
-    arg2: A2,
-    arg3: A3,
-  ): new (...args: A) => R;
-  bind<AX, R>(
-    this: new (...args: AX[]) => R,
+
+  /**
+   * For a given function, creates a bound function that has the same body as the original function.
+   * The this object of the bound function is associated with the specified object, and has the specified initial parameters.
+   * @param thisArg The object to be used as the this object.
+   * @param args Arguments to bind to the parameters of the function.
+   */
+  bind<A extends any[], B extends any[], R>(
+    this: new (...args: [...A, ...B]) => R,
     thisArg: any,
-    ...args: AX[]
-  ): new (...args: AX[]) => R;
+    ...args: A
+  ): new (...args: B) => R;
 }
 
 interface IArguments {
@@ -2121,7 +2090,7 @@ interface DataView {
 interface DataViewConstructor {
   readonly prototype: DataView;
   new (
-    buffer: ArrayBufferLike,
+    buffer: ArrayBufferLike & { BYTES_PER_ELEMENT?: never },
     byteOffset?: number,
     byteLength?: number,
   ): DataView;
@@ -2159,10 +2128,10 @@ interface Int8Array {
    * @param target If target is negative, it is treated as length+target where length is the
    * length of the array.
    * @param start If start is negative, it is treated as length+start. If end is negative, it
-   * is treated as length+end.
+   * is treated as length+end. If start is omitted, `0` is used.
    * @param end If not specified, length of the this object is used as its default value.
    */
-  copyWithin(target: number, start: number, end?: number): this;
+  copyWithin(target: number, start?: number, end?: number): this;
 
   /**
    * Determines whether all the members of an array satisfy the specified test.
@@ -2514,10 +2483,10 @@ interface Uint8Array {
    * @param target If target is negative, it is treated as length+target where length is the
    * length of the array.
    * @param start If start is negative, it is treated as length+start. If end is negative, it
-   * is treated as length+end.
+   * is treated as length+end. If start is omitted, `0` is used.
    * @param end If not specified, length of the this object is used as its default value.
    */
-  copyWithin(target: number, start: number, end?: number): this;
+  copyWithin(target: number, start?: number, end?: number): this;
 
   /**
    * Determines whether all the members of an array satisfy the specified test.
@@ -2870,10 +2839,10 @@ interface Uint8ClampedArray {
    * @param target If target is negative, it is treated as length+target where length is the
    * length of the array.
    * @param start If start is negative, it is treated as length+start. If end is negative, it
-   * is treated as length+end.
+   * is treated as length+end. If start is omitted, `0` is used.
    * @param end If not specified, length of the this object is used as its default value.
    */
-  copyWithin(target: number, start: number, end?: number): this;
+  copyWithin(target: number, start?: number, end?: number): this;
 
   /**
    * Determines whether all the members of an array satisfy the specified test.
@@ -3250,10 +3219,10 @@ interface Int16Array {
    * @param target If target is negative, it is treated as length+target where length is the
    * length of the array.
    * @param start If start is negative, it is treated as length+start. If end is negative, it
-   * is treated as length+end.
+   * is treated as length+end. If start is omitted, `0` is used.
    * @param end If not specified, length of the this object is used as its default value.
    */
-  copyWithin(target: number, start: number, end?: number): this;
+  copyWithin(target: number, start?: number, end?: number): this;
 
   /**
    * Determines whether all the members of an array satisfy the specified test.
@@ -3605,10 +3574,10 @@ interface Uint16Array {
    * @param target If target is negative, it is treated as length+target where length is the
    * length of the array.
    * @param start If start is negative, it is treated as length+start. If end is negative, it
-   * is treated as length+end.
+   * is treated as length+end. If start is omitted, `0` is used.
    * @param end If not specified, length of the this object is used as its default value.
    */
-  copyWithin(target: number, start: number, end?: number): this;
+  copyWithin(target: number, start?: number, end?: number): this;
 
   /**
    * Determines whether all the members of an array satisfy the specified test.
@@ -3960,10 +3929,10 @@ interface Int32Array {
    * @param target If target is negative, it is treated as length+target where length is the
    * length of the array.
    * @param start If start is negative, it is treated as length+start. If end is negative, it
-   * is treated as length+end.
+   * is treated as length+end. If start is omitted, `0` is used.
    * @param end If not specified, length of the this object is used as its default value.
    */
-  copyWithin(target: number, start: number, end?: number): this;
+  copyWithin(target: number, start?: number, end?: number): this;
 
   /**
    * Determines whether all the members of an array satisfy the specified test.
@@ -4316,10 +4285,10 @@ interface Uint32Array {
    * @param target If target is negative, it is treated as length+target where length is the
    * length of the array.
    * @param start If start is negative, it is treated as length+start. If end is negative, it
-   * is treated as length+end.
+   * is treated as length+end. If start is omitted, `0` is used.
    * @param end If not specified, length of the this object is used as its default value.
    */
-  copyWithin(target: number, start: number, end?: number): this;
+  copyWithin(target: number, start?: number, end?: number): this;
 
   /**
    * Determines whether all the members of an array satisfy the specified test.
@@ -4671,10 +4640,10 @@ interface Float32Array {
    * @param target If target is negative, it is treated as length+target where length is the
    * length of the array.
    * @param start If start is negative, it is treated as length+start. If end is negative, it
-   * is treated as length+end.
+   * is treated as length+end. If start is omitted, `0` is used.
    * @param end If not specified, length of the this object is used as its default value.
    */
-  copyWithin(target: number, start: number, end?: number): this;
+  copyWithin(target: number, start?: number, end?: number): this;
 
   /**
    * Determines whether all the members of an array satisfy the specified test.
@@ -5027,10 +4996,10 @@ interface Float64Array {
    * @param target If target is negative, it is treated as length+target where length is the
    * length of the array.
    * @param start If start is negative, it is treated as length+start. If end is negative, it
-   * is treated as length+end.
+   * is treated as length+end. If start is omitted, `0` is used.
    * @param end If not specified, length of the this object is used as its default value.
    */
-  copyWithin(target: number, start: number, end?: number): this;
+  copyWithin(target: number, start?: number, end?: number): this;
 
   /**
    * Determines whether all the members of an array satisfy the specified test.
@@ -5288,12 +5257,21 @@ interface Float64Array {
   sort(compareFn?: (a: number, b: number) => number): this;
 
   /**
+   * Gets a new Float64Array view of the ArrayBuffer store for this array, referencing the elements
    * at begin, inclusive, up to end, exclusive.
    * @param begin The index of the beginning of the array.
    * @param end The index of the end of the array.
    */
   subarray(begin?: number, end?: number): Float64Array;
 
+  /**
+   * Converts a number to a string by using the current locale.
+   */
+  toLocaleString(): string;
+
+  /**
+   * Returns a string representation of an array.
+   */
   toString(): string;
 
   /** Returns the primitive value of the specified object. */
@@ -5349,11 +5327,30 @@ declare var Float64Array: Float64ArrayConstructor;
 
 declare namespace Intl {
   interface CollatorOptions {
-    usage?: string | undefined;
-    localeMatcher?: string | undefined;
+    usage?: 'sort' | 'search' | undefined;
+    localeMatcher?: 'lookup' | 'best fit' | undefined;
     numeric?: boolean | undefined;
-    caseFirst?: string | undefined;
-    sensitivity?: string | undefined;
+    caseFirst?: 'upper' | 'lower' | 'false' | undefined;
+    sensitivity?: 'base' | 'accent' | 'case' | 'variant' | undefined;
+    collation?:
+      | 'big5han'
+      | 'compat'
+      | 'dict'
+      | 'direct'
+      | 'ducet'
+      | 'emoji'
+      | 'eor'
+      | 'gb2312'
+      | 'phonebk'
+      | 'phonetic'
+      | 'pinyin'
+      | 'reformed'
+      | 'searchjl'
+      | 'stroke'
+      | 'trad'
+      | 'unihan'
+      | 'zhuyin'
+      | undefined;
     ignorePunctuation?: boolean | undefined;
   }
 
