diff --git experimental/packages/apps/event-schedule-app/src/functions/map-key/index.ts apps/event-schedule-app/src/functions/map-key/index.mts
index 5012b9b3a..a4bcbb635 100644
--- experimental/packages/apps/event-schedule-app/src/functions/map-key/index.ts
+++ apps/event-schedule-app/src/functions/map-key/index.mts
@@ -1,4 +1,4 @@
-export * from './answer-selection-map-key';
-export * from './datetime-range-map-key';
-export * from './hours-minutes-map-key';
-export * from './ymd-to-map-key';
+export * from './answer-selection-map-key.mjs';
+export * from './datetime-range-map-key.mjs';
+export * from './hours-minutes-map-key.mjs';
+export * from './ymd-to-map-key.mjs';
