diff --git a/packages/v5.9/output-branded/lib-files/lib.es2020.intl.d.ts b/packages/v6.0/output-branded/lib-files/lib.es2020.intl.d.ts
index 047ed02a..7df92126 100644
--- a/packages/v5.9/output-branded/lib-files/lib.es2020.intl.d.ts
+++ b/packages/v6.0/output-branded/lib-files/lib.es2020.intl.d.ts
@@ -1,5 +1,3 @@
-/// <reference no-default-lib="true"/>
-
 /// <reference path="./lib.es2018.intl.d.ts" />
 declare namespace Intl {
   /**
