diff --git a/packages/v5.9/output-branded/lib-files/lib.es2015.d.ts b/packages/v6.0/output-branded/lib-files/lib.es2015.d.ts
index 5d3de1c9..d03c57f1 100644
--- a/packages/v5.9/output-branded/lib-files/lib.es2015.d.ts
+++ b/packages/v6.0/output-branded/lib-files/lib.es2015.d.ts
@@ -1,5 +1,3 @@
-/// <reference no-default-lib="true"/>
-
 /// <reference path="./lib.es5.d.ts" />
 /// <reference path="./lib.es2015.core.d.ts" />
 /// <reference path="./lib.es2015.collection.d.ts" />
