diff --git a/packages/v5.9/output-branded/lib-files/lib.es2024.sharedmemory.d.ts b/packages/v6.0/output-branded/lib-files/lib.es2024.sharedmemory.d.ts
index d8c940fb..2b64da96 100644
--- a/packages/v5.9/output-branded/lib-files/lib.es2024.sharedmemory.d.ts
+++ b/packages/v6.0/output-branded/lib-files/lib.es2024.sharedmemory.d.ts
@@ -1,5 +1,3 @@
-/// <reference no-default-lib="true"/>
-
 /// <reference path="./lib.es2020.bigint.d.ts" />
 
 interface Atomics {
