diff --git a/packages/v5.9/output-branded/lib-files/lib.es2024.object.d.ts b/packages/v6.0/output-branded/lib-files/lib.es2024.object.d.ts
index 14e43f29..2b9daa6c 100644
--- a/packages/v5.9/output-branded/lib-files/lib.es2024.object.d.ts
+++ b/packages/v6.0/output-branded/lib-files/lib.es2024.object.d.ts
@@ -1,5 +1,3 @@
-/// <reference no-default-lib="true"/>
-
 interface ObjectConstructor {
   /**
    * Groups members of an iterable according to the return value of the passed callback.
