diff --git a/packages/v5.9/output-branded/lib-files/lib.es2015.iterable.d.ts b/packages/v6.0/output-branded/lib-files/lib.es2015.iterable.d.ts
index 799c95f0..9a5c6a96 100644
--- a/packages/v5.9/output-branded/lib-files/lib.es2015.iterable.d.ts
+++ b/packages/v6.0/output-branded/lib-files/lib.es2015.iterable.d.ts
@@ -1,5 +1,3 @@
-/// <reference no-default-lib="true"/>
-
 /// <reference path="./lib.es2015.symbol.d.ts" />
 
 interface SymbolConstructor {
@@ -192,8 +190,8 @@ interface MapConstructor {
 interface WeakMap<K extends WeakKey, V> {}
 
 interface WeakMapConstructor {
-  new <K extends WeakKey, V>(
-    iterable: Iterable<readonly [K, V]>,
+  new <K extends WeakKey = WeakKey, V = unknown>(
+    iterable?: Iterable<readonly [K, V]> | null,
   ): WeakMap<K, V>;
 }
 
