diff --git a/packages/v5.9/output-branded/lib-files/lib.es2017.date.d.ts b/packages/v6.0/output-branded/lib-files/lib.es2017.date.d.ts
index 9af06173..96d43507 100644
--- a/packages/v5.9/output-branded/lib-files/lib.es2017.date.d.ts
+++ b/packages/v6.0/output-branded/lib-files/lib.es2017.date.d.ts
@@ -1,5 +1,3 @@
-/// <reference no-default-lib="true"/>
-
 interface DateConstructor {
   /**
    * Returns the number of milliseconds between midnight, January 1, 1970 Universal Coordinated Time (UTC) (or GMT) and the specified date.
