diff --git a/packages/v5.9/output-branded/lib-files/lib.es2018.asynciterable.d.ts b/packages/v6.0/output-branded/lib-files/lib.es2018.asynciterable.d.ts
index bc643700..19ae8e2e 100644
--- a/packages/v5.9/output-branded/lib-files/lib.es2018.asynciterable.d.ts
+++ b/packages/v6.0/output-branded/lib-files/lib.es2018.asynciterable.d.ts
@@ -1,5 +1,3 @@
-/// <reference no-default-lib="true"/>
-
 /// <reference path="./lib.es2015.symbol.d.ts" />
 /// <reference path="./lib.es2015.iterable.d.ts" />
 
