diff --git a/packages/v5.9/output-branded/lib-files/lib.es2018.promise.d.ts b/packages/v6.0/output-branded/lib-files/lib.es2018.promise.d.ts
index 386b2958..b3e8cc28 100644
--- a/packages/v5.9/output-branded/lib-files/lib.es2018.promise.d.ts
+++ b/packages/v6.0/output-branded/lib-files/lib.es2018.promise.d.ts
@@ -1,5 +1,3 @@
-/// <reference no-default-lib="true"/>
-
 /**
  * Represents the completion of an asynchronous operation
  */
