diff --git a/packages/v5.9/output-branded/lib-files/lib.es2015.generator.d.ts b/packages/v6.0/output-branded/lib-files/lib.es2015.generator.d.ts
index 29e2b648..46be4f1c 100644
--- a/packages/v5.9/output-branded/lib-files/lib.es2015.generator.d.ts
+++ b/packages/v6.0/output-branded/lib-files/lib.es2015.generator.d.ts
@@ -1,5 +1,3 @@
-/// <reference no-default-lib="true"/>
-
 /// <reference path="./lib.es2015.iterable.d.ts" />
 
 interface Generator<
