diff --git a/packages/v5.9/output-branded/lib-files/lib.es2015.collection.d.ts b/packages/v6.0/output-branded/lib-files/lib.es2015.collection.d.ts
index 108bb988..84b8b8ff 100644
--- a/packages/v5.9/output-branded/lib-files/lib.es2015.collection.d.ts
+++ b/packages/v6.0/output-branded/lib-files/lib.es2015.collection.d.ts
@@ -1,6 +1,7 @@
-/// <reference no-default-lib="true"/>
-
 interface Map<K, V> {
+  /**
+   * Removes all elements from the Map.
+   */
   clear(): void;
   /**
    * @returns true if an element in the Map existed and has been removed, or false if the element does not exist.
@@ -83,7 +84,9 @@ interface Set<T> {
    * Appends a new element with a specified value to the end of the Set.
    */
   add(value: T): this;
-
+  /**
+   * Removes all elements from the Set.
+   */
   clear(): void;
   /**
    * Removes a specified value from the Set.
