diff --git a/packages/v5.9/output-branded/lib-files/lib.es2024.promise.d.ts b/packages/v6.0/output-branded/lib-files/lib.es2024.promise.d.ts
index 34245af3..2ab40c04 100644
--- a/packages/v5.9/output-branded/lib-files/lib.es2024.promise.d.ts
+++ b/packages/v6.0/output-branded/lib-files/lib.es2024.promise.d.ts
@@ -1,5 +1,3 @@
-/// <reference no-default-lib="true"/>
-
 interface PromiseWithResolvers<T> {
   readonly promise: Promise<T>;
   readonly resolve: (value: T | PromiseLike<T>) => void;
