diff --git a/packages/v5.9/output-branded/lib-files/lib.es2017.string.d.ts b/packages/v6.0/output-branded/lib-files/lib.es2017.string.d.ts
index ce499b0f..fd102d3f 100644
--- a/packages/v5.9/output-branded/lib-files/lib.es2017.string.d.ts
+++ b/packages/v6.0/output-branded/lib-files/lib.es2017.string.d.ts
@@ -1,5 +1,3 @@
-/// <reference no-default-lib="true"/>
-
 interface String {
   /**
    * Pads the current string with a given string (possibly repeated) so that the resulting string reaches a given length.
