diff --git a/packages/v5.9/output-branded/lib-files/lib.es2023.intl.d.ts b/packages/v6.0/output-branded/lib-files/lib.es2023.intl.d.ts
index 38aa32cf..8198408e 100644
--- a/packages/v5.9/output-branded/lib-files/lib.es2023.intl.d.ts
+++ b/packages/v6.0/output-branded/lib-files/lib.es2023.intl.d.ts
@@ -1,5 +1,3 @@
-/// <reference no-default-lib="true"/>
-
 declare namespace Intl {
   interface NumberFormatOptionsUseGroupingRegistry {
     readonly min2: never;
@@ -11,6 +9,12 @@ declare namespace Intl {
     readonly negative: never;
   }
 
+  interface NumberFormatRangePartTypeRegistry extends NumberFormatPartTypeRegistry {
+    readonly approximatelySign: never;
+  }
+
+  type NumberFormatRangePartTypes = keyof NumberFormatRangePartTypeRegistry;
+
   interface NumberFormatOptions {
     readonly roundingPriority?:
       | 'auto'
@@ -79,7 +83,9 @@ declare namespace Intl {
     readonly trailingZeroDisplay: 'auto' | 'stripIfInteger';
   }
 
-  interface NumberRangeFormatPart extends NumberFormatPart {
+  interface NumberRangeFormatPart {
+    readonly type: NumberFormatRangePartTypes;
+    readonly value: string;
     readonly source: 'startRange' | 'endRange' | 'shared';
   }
 
