diff --git experimental/packages/utils/react-blueprintjs-utils/src/components/view/input-view.tsx apps/react-blueprintjs-utils/src/components/view/input-view.tsx
index 95050bf4f..8f3fc743e 100644
--- experimental/packages/utils/react-blueprintjs-utils/src/components/view/input-view.tsx
+++ apps/react-blueprintjs-utils/src/components/view/input-view.tsx
@@ -1,11 +1,12 @@
 import { css } from '@emotion/react';
 import styled from '@emotion/styled';
-import { memoNamed } from '@noshiro/react-utils';
-import { useCallback, useRef } from 'react';
+import * as React from 'react';
+import { memoNamed } from 'react-utils';
+import { type StrictOmit } from 'ts-type-forge';
 
 type Props = Partial<
   Readonly<{
-    inputProps: Omit<
+    inputProps: StrictOmit<
       React.DetailedHTMLProps<
         React.InputHTMLAttributes<HTMLInputElement>,
         HTMLInputElement
@@ -48,20 +49,22 @@ export const InputView = memoNamed<Props>(
     selectOnFocus,
   }) => {
     const inputValueChangeHandler: React.ChangeEventHandler<HTMLInputElement> =
-      useCallback(
+      React.useCallback(
         (ev) => {
           onInputStringChange?.(ev.currentTarget.value);
+
           onInputChange?.(ev);
         },
         [onInputStringChange, onInputChange],
       );
 
-    const inputRef = useRef<HTMLInputElement>(null);
+    const inputRef = React.useRef<HTMLInputElement>(null);
 
     const inputFocusHandler: React.FocusEventHandler<HTMLInputElement> =
-      useCallback(
+      React.useCallback(
         (ev) => {
           onInputFocus?.(ev);
+
           if (selectOnFocus === true) {
             inputRef.current?.select();
           }
