diff --git experimental/packages/apps/event-schedule-app/src/components/atoms/icons/calendar.tsx apps/event-schedule-app/src/components/atoms/icons/calendar.tsx
index 6a08c290c..130e96137 100644
--- experimental/packages/apps/event-schedule-app/src/components/atoms/icons/calendar.tsx
+++ apps/event-schedule-app/src/components/atoms/icons/calendar.tsx
@@ -1,3 +1,5 @@
+import { memoNamed } from 'react-utils';
+
 type Props = Readonly<{
   size?: number;
   color?: string;
