diff --git libs/better-preact-use-state/src/use-state.mts libs/better-preact-use-state/src/use-state.mts
new file mode 100644
index 000000000..bf069abd8
--- /dev/null
+++ libs/better-preact-use-state/src/use-state.mts
@@ -0,0 +1,24 @@
+import * as Preact from 'preact/hooks';
+
+export const useState = <T,>(
+  initialState: T,
+): readonly [
+  state: T,
+  setState: (next: T) => void,
+  setters: Readonly<{
+    updateState: (updateFn: (v: T) => T) => void;
+    resetState: () => void;
+  }>,
+] => {
+  const [state, setState] = Preact.useState<T>(initialState);
+
+  const resetState = Preact.useCallback(() => {
+    setState(() => initialState);
+  }, [initialState]);
+
+  const updateState = Preact.useCallback((updateFn: (v: T) => T) => {
+    setState(updateFn);
+  }, []);
+
+  return [state, setState, { updateState, resetState }];
+};
