diff --git experimental/packages/apps/event-schedule-app/src/store/utils/create-form-error-state.ts apps/event-schedule-app/src/store/utils/create-form-error-state.mts
index 2340bb08b..c2698889b 100644
--- experimental/packages/apps/event-schedule-app/src/store/utils/create-form-error-state.ts
+++ apps/event-schedule-app/src/store/utils/create-form-error-state.mts
@@ -1,13 +1,13 @@
-export const createFormError = <T>(
+import { type InitializedObservable } from 'synstate';
+import { createBooleanState } from 'synstate-react-hooks';
+
+export const createFormError = <T,>(
   value: T,
   valueIsInvalid: (v: T) => boolean,
   onValueChange: (v: T) => void,
-): [InitializedObservable<boolean>, (v: T) => void, () => void] => {
-  const {
-    state: errorShown$,
-    setTrue: show,
-    setFalse: hide,
-  } = createBooleanState(false);
+): readonly [InitializedObservable<boolean>, (v: T) => void, () => void] => {
+  const [, { state: errorShown$, setTrue: show, setFalse: hide }] =
+    createBooleanState(false);
 
   const onBlur = (): void => {
     if (valueIsInvalid(value)) {
@@ -16,14 +16,13 @@ export const createFormError = <T>(
   };
 
   // TODO
-  // useEffect(() => {
-  //   if (!valueIsInvalid(value)) {
-  //     hide();
-  //   }
+  // useEffect(() => { //   if (!valueIsInvalid(value)) { //     hide();
+  // }
   // }, [value, valueIsInvalid, hide]);
 
   const onChangeLocal = (v: T): void => {
     onValueChange(v);
+
     if (valueIsInvalid(v)) {
       show();
     } else {
