diff --git experimental/packages/apps/event-schedule-app/src/constants/dictionary/event-settings-page.ts apps/event-schedule-app/src/constants/dictionary/event-settings-page.mts
index 642069be7..4e760d0ee 100644
--- experimental/packages/apps/event-schedule-app/src/constants/dictionary/event-settings-page.ts
+++ apps/event-schedule-app/src/constants/dictionary/event-settings-page.mts
@@ -1,5 +1,5 @@
-import { daysOfWeek } from '@noshiro/ts-utils-additional';
-import { defaultIconPoint } from '../default-icon-point';
+import { daysOfWeek } from '../../utils-ported/index.mjs';
+import { defaultIconPoint } from '../default-icon-point.mjs';
 
 export const eventSettingsPageDictionary = {
   title: '日程調整',
@@ -111,39 +111,19 @@ export const eventSettingsPageDictionary = {
   },
   diff: {
     title: '変更差分',
-    values: {
-      none: 'なし',
-    },
+    values: { none: 'なし' },
     items: {
       title: 'イベント名：',
       notes: 'ノート：',
       datetimeSpecification: '時刻指定方法：',
-      datetimeRangeList: {
-        title: '候補日程',
-        added: '追加',
-        deleted: '削除',
-      },
+      datetimeRangeList: { title: '候補日程', added: '追加', deleted: '削除' },
       answerDeadline: '回答締切日：',
-      notificationSettings: {
-        title: '通知設定',
-        on: 'オン',
-        off: 'オフ',
-      },
+      notificationSettings: { title: '通知設定', on: 'オン', off: 'オフ' },
       answerIcons: {
         title: '記号',
-        good: {
-          title: '〇',
-          description: '説明：',
-        },
-        fair: {
-          title: '△',
-          description: '説明：',
-          point: '点：',
-        },
-        poor: {
-          title: '✕',
-          description: '説明：',
-        },
+        good: { title: '〇', description: '説明：' },
+        fair: { title: '△', description: '説明：', point: '点：' },
+        poor: { title: '✕', description: '説明：' },
       },
       author: '作成者：',
       timezoneOffsetMinutes: 'タイムゾーン：',
