diff --git experimental/packages/apps/event-schedule-app/src/constants/dictionary/header.ts apps/event-schedule-app/src/constants/dictionary/header.mts
index 12cc85c72..c3157a760 100644
--- experimental/packages/apps/event-schedule-app/src/constants/dictionary/header.ts
+++ apps/event-schedule-app/src/constants/dictionary/header.mts
@@ -2,10 +2,7 @@ export const headerDictionary = {
   createNew: '日程調整を作成',
   list: '日程調整一覧',
   auth: {
-    userName: {
-      prefix: '',
-      suffix: 'さん',
-    },
+    userName: { prefix: '', suffix: 'さん' },
     signIn: 'ログイン',
     register: '新規登録',
     menu: {
