diff --git experimental/packages/apps/event-schedule-app-shared/src/v7/types/record/base/hours-minutes.mts apps/event-schedule-app-shared/src/v7/types/record/base/hours-minutes.mts
index 1a00477ae..c81da7124 100644
--- experimental/packages/apps/event-schedule-app-shared/src/v7/types/record/base/hours-minutes.mts
+++ apps/event-schedule-app-shared/src/v7/types/record/base/hours-minutes.mts
@@ -1,5 +1,5 @@
-import * as t from '@noshiro/io-ts';
-import { DateUtils, Num, type DateType } from '@noshiro/ts-utils';
+import * as t from 'ts-fortress';
+import { DateUtils } from 'ts-fortress-types';
 import { hoursTypeDef, minutesTypeDef } from '../../enum/index.mjs';
 
 export const hoursMinutesTypeDef = t.record({
@@ -15,15 +15,16 @@ export const isHoursMinutes = hoursMinutesTypeDef.is;
 
 export const fillHoursMinutes = hoursMinutesTypeDef.fill;
 
-export const hmFromDate = (date: DateType): HoursMinutes => ({
-  hours: DateUtils.getLocaleHours(date),
-  minutes: DateUtils.getLocaleMinutes(date),
-});
+export const hmFromDate = (date: Date): HoursMinutes =>
+  ({
+    hours: DateUtils.getLocaleHours(date),
+    minutes: DateUtils.getLocaleMinutes(date),
+  }) as const;
 
 export const compareHm = (a: HoursMinutes, b: HoursMinutes): -1 | 0 | 1 => {
-  if (a.hours !== b.hours)
-    return Num.mapNaN2Undefined(Math.sign(a.hours - b.hours)) ?? 0;
-  if (a.minutes !== b.minutes)
-    return Num.mapNaN2Undefined(Math.sign(a.minutes - b.minutes)) ?? 0;
+  if (a.hours !== b.hours) return a.hours < b.hours ? -1 : 1;
+
+  if (a.minutes !== b.minutes) return a.minutes < b.minutes ? -1 : 1;
+
   return 0;
 };
