diff --git experimental/packages/apps/event-schedule-app-shared/src/v7/types/record/base/user.mts apps/event-schedule-app-shared/src/v7/types/record/base/user.mts
index aa60428c4..ad5ed40d1 100644
--- experimental/packages/apps/event-schedule-app-shared/src/v7/types/record/base/user.mts
+++ apps/event-schedule-app-shared/src/v7/types/record/base/user.mts
@@ -1,4 +1,4 @@
-import * as t from '@noshiro/io-ts';
+import * as t from 'ts-fortress';
 import {
   userIdTypeDef,
   userNameTypeDef,
