diff --git a/packages/v5.9/temp/copied/lib.dom.asynciterable.d.ts b/packages/v6.0/temp/copied/lib.dom.asynciterable.d.ts
index 3273a508..a6694dc9 100644
--- a/packages/v5.9/temp/copied/lib.dom.asynciterable.d.ts
+++ b/packages/v6.0/temp/copied/lib.dom.asynciterable.d.ts
@@ -7,48 +7,11 @@ License at http://www.apache.org/licenses/LICENSE-2.0
 THIS CODE IS PROVIDED ON AN *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION ANY IMPLIED
 WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR PURPOSE,
-MERCHANTABLITY OR NON-INFRINGEMENT.
+MERCHANTABILITY OR NON-INFRINGEMENT.
 
 See the Apache Version 2.0 License for specific language governing permissions
 and limitations under the License.
 ***************************************************************************** */
 
-/// <reference no-default-lib="true"/>
-
-/////////////////////////////
-/// Window Async Iterable APIs
-/////////////////////////////
-
-interface FileSystemDirectoryHandleAsyncIterator<T> extends AsyncIteratorObject<
-  T,
-  BuiltinIteratorReturn,
-  unknown
-> {
-  [Symbol.asyncIterator](): FileSystemDirectoryHandleAsyncIterator<T>;
-}
-
-interface FileSystemDirectoryHandle {
-  [Symbol.asyncIterator](): FileSystemDirectoryHandleAsyncIterator<
-    [string, FileSystemHandle]
-  >;
-  entries(): FileSystemDirectoryHandleAsyncIterator<[string, FileSystemHandle]>;
-  keys(): FileSystemDirectoryHandleAsyncIterator<string>;
-  values(): FileSystemDirectoryHandleAsyncIterator<FileSystemHandle>;
-}
-
-interface ReadableStreamAsyncIterator<T> extends AsyncIteratorObject<
-  T,
-  BuiltinIteratorReturn,
-  unknown
-> {
-  [Symbol.asyncIterator](): ReadableStreamAsyncIterator<T>;
-}
-
-interface ReadableStream<R = any> {
-  [Symbol.asyncIterator](
-    options?: ReadableStreamIteratorOptions,
-  ): ReadableStreamAsyncIterator<R>;
-  values(
-    options?: ReadableStreamIteratorOptions,
-  ): ReadableStreamAsyncIterator<R>;
-}
+// This file's contents are now included in the main types file.
+// The file has been left for backward compatibility.
