diff --git experimental/packages/apps/event-schedule-app-shared/src/v7/types/enum/answer-icon-point.mts apps/event-schedule-app-shared/src/v7/types/enum/answer-icon-point.mts
index ca03ef96e..629080b13 100644
--- experimental/packages/apps/event-schedule-app-shared/src/v7/types/enum/answer-icon-point.mts
+++ apps/event-schedule-app-shared/src/v7/types/enum/answer-icon-point.mts
@@ -1,5 +1,6 @@
-import * as t from '@noshiro/io-ts';
-import { isNumber } from '@noshiro/ts-utils';
+/* transformer-ignore */
+import { isNumber } from 'ts-data-forge';
+import * as t from 'ts-fortress';
 
 export const answerIconPointTypeDef = (
   defaultValue: AnswerIconPoint,
