diff --git a/packages/v5.8/output-branded/lib-files/lib.es5.d.ts b/packages/v5.9/output-branded/lib-files/lib.es5.d.ts
index 3d2cc07..5815d08 100644
--- a/packages/v5.8/output-branded/lib-files/lib.es5.d.ts
+++ b/packages/v5.9/output-branded/lib-files/lib.es5.d.ts
@@ -1264,6 +1264,7 @@ interface JSON {
    * @param text A valid JSON string.
    * @param reviver A function that transforms the results. This function is called for each member of the object.
    * If a member contains nested objects, the nested objects are transformed before the parent object is.
+   * @throws {SyntaxError} If `text` is not valid JSON.
    */
   parse(
     text: string,
@@ -1274,6 +1275,7 @@ interface JSON {
    * @param value A JavaScript value, usually an object or array, to be converted.
    * @param replacer A function that transforms the results.
    * @param space Adds indentation, white space, and line break characters to the return-value JSON text to make it easier to read.
+   * @throws {TypeError} If a circular reference or a BigInt value is found.
    */
   stringify(
     value: unknown,
@@ -1285,6 +1287,7 @@ interface JSON {
    * @param value A JavaScript value, usually an object or array, to be converted.
    * @param replacer An array of strings and numbers that acts as an approved list for selecting the object properties that will be stringified.
    * @param space Adds indentation, white space, and line break characters to the return-value JSON text to make it easier to read.
+   * @throws {TypeError} If a circular reference or a BigInt value is found.
    */
   stringify(
     value: unknown,
@@ -1620,7 +1623,9 @@ interface Array<T> {
   /**
    * Removes elements from an array and, if necessary, inserts new elements in their place, returning the deleted elements.
    * @param start The zero-based location in the array from which to start removing elements.
-   * @param deleteCount The number of elements to remove.
+   * @param deleteCount The number of elements to remove. Omitting this argument will remove all elements from the start
+   * paramater location to end of the array. If value of this argument is either a negative number, zero, undefined, or a type
+   * that cannot be converted to an integer, the function will evaluate the argument as zero and not remove any elements.
    * @returns An array containing the elements that were deleted.
    */
   splice(
@@ -1630,7 +1635,9 @@ interface Array<T> {
   /**
    * Removes elements from an array and, if necessary, inserts new elements in their place, returning the deleted elements.
    * @param start The zero-based location in the array from which to start removing elements.
-   * @param deleteCount The number of elements to remove.
+   * @param deleteCount The number of elements to remove. If value of this argument is either a negative number, zero,
+   * undefined, or a type that cannot be converted to an integer, the function will evaluate the argument as zero and
+   * not remove any elements.
    * @param items Elements to insert into the array in place of the deleted elements.
    * @returns An array containing the elements that were deleted.
    */
