diff --git a/packages/v5.8/output-branded/lib-files/lib.webworker.iterable.d.ts b/packages/v5.9/output-branded/lib-files/lib.webworker.iterable.d.ts
index 02b82c4..fe4c358 100644
--- a/packages/v5.8/output-branded/lib-files/lib.webworker.iterable.d.ts
+++ b/packages/v5.9/output-branded/lib-files/lib.webworker.iterable.d.ts
@@ -27,7 +27,11 @@ interface CSSUnparsedValue {
 }
 
 interface Cache {
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/Cache/addAll) */
+  /**
+   * The **`addAll()`** method of the Cache interface takes an array of URLs, retrieves them, and adds the resulting response objects to the given cache.
+   *
+   * [MDN Reference](https://developer.mozilla.org/docs/Web/API/Cache/addAll)
+   */
   addAll(requests: Iterable<RequestInfo>): Promise<void>;
 }
 
@@ -47,6 +51,21 @@ interface CanvasPathDrawingStyles {
   setLineDash(segments: Iterable<number>): void;
 }
 
+interface CookieStoreManager {
+  /**
+   * The **`subscribe()`** method of the CookieStoreManager interface subscribes a ServiceWorkerRegistration to cookie change events.
+   *
+   * [MDN Reference](https://developer.mozilla.org/docs/Web/API/CookieStoreManager/subscribe)
+   */
+  subscribe(subscriptions: Iterable<CookieStoreGetOptions>): Promise<void>;
+  /**
+   * The **`unsubscribe()`** method of the CookieStoreManager interface stops the ServiceWorkerRegistration from receiving previously subscribed events.
+   *
+   * [MDN Reference](https://developer.mozilla.org/docs/Web/API/CookieStoreManager/unsubscribe)
+   */
+  unsubscribe(subscriptions: Iterable<CookieStoreGetOptions>): Promise<void>;
+}
+
 interface DOMStringList {
   [Symbol.iterator](): ArrayIterator<string>;
 }
@@ -95,7 +114,7 @@ interface Headers {
 
 interface IDBDatabase {
   /**
-   * Returns a new transaction with the given mode ("readonly" or "readwrite") and scope which can be a single object store name or an array of names.
+   * The **`transaction`** method of the IDBDatabase interface immediately returns a transaction object (IDBTransaction) containing the IDBTransaction.objectStore method, which you can use to access your object store.
    *
    * [MDN Reference](https://developer.mozilla.org/docs/Web/API/IDBDatabase/transaction)
    */
@@ -108,9 +127,7 @@ interface IDBDatabase {
 
 interface IDBObjectStore {
   /**
-   * Creates a new index in store with the given name, keyPath and options and returns a new IDBIndex. If the keyPath and options define constraints that cannot be satisfied with the data already in store the upgrade transaction will abort with a "ConstraintError" DOMException.
-   *
-   * Throws an "InvalidStateError" DOMException if not called within an upgrade transaction.
+   * The **`createIndex()`** method of the field/column defining a new data point for each database record to contain.
    *
    * [MDN Reference](https://developer.mozilla.org/docs/Web/API/IDBObjectStore/createIndex)
    */
@@ -159,7 +176,11 @@ interface StylePropertyMapReadOnly {
 }
 
 interface SubtleCrypto {
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/SubtleCrypto/deriveKey) */
+  /**
+   * The **`deriveKey()`** method of the SubtleCrypto interface can be used to derive a secret key from a master key.
+   *
+   * [MDN Reference](https://developer.mozilla.org/docs/Web/API/SubtleCrypto/deriveKey)
+   */
   deriveKey(
     algorithm:
       | AlgorithmIdentifier
@@ -176,7 +197,11 @@ interface SubtleCrypto {
     extractable: boolean,
     keyUsages: Iterable<KeyUsage>,
   ): Promise<CryptoKey>;
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/SubtleCrypto/generateKey) */
+  /**
+   * The **`generateKey()`** method of the SubtleCrypto interface is used to generate a new key (for symmetric algorithms) or key pair (for public-key algorithms).
+   *
+   * [MDN Reference](https://developer.mozilla.org/docs/Web/API/SubtleCrypto/generateKey)
+   */
   generateKey(
     algorithm: 'Ed25519' | { readonly name: 'Ed25519' },
     extractable: boolean,
@@ -197,7 +222,11 @@ interface SubtleCrypto {
     extractable: boolean,
     keyUsages: Iterable<KeyUsage>,
   ): Promise<CryptoKeyPair | CryptoKey>;
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/SubtleCrypto/importKey) */
+  /**
+   * The **`importKey()`** method of the SubtleCrypto interface imports a key: that is, it takes as input a key in an external, portable format and gives you a CryptoKey object that you can use in the Web Crypto API.
+   *
+   * [MDN Reference](https://developer.mozilla.org/docs/Web/API/SubtleCrypto/importKey)
+   */
   importKey(
     format: 'jwk',
     keyData: JsonWebKey,
@@ -222,7 +251,11 @@ interface SubtleCrypto {
     extractable: boolean,
     keyUsages: Iterable<KeyUsage>,
   ): Promise<CryptoKey>;
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/SubtleCrypto/unwrapKey) */
+  /**
+   * The **`unwrapKey()`** method of the SubtleCrypto interface 'unwraps' a key.
+   *
+   * [MDN Reference](https://developer.mozilla.org/docs/Web/API/SubtleCrypto/unwrapKey)
+   */
   unwrapKey(
     format: KeyFormat,
     wrappedKey: BufferSource,
@@ -263,50 +296,70 @@ interface URLSearchParams {
 }
 
 interface WEBGL_draw_buffers {
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WEBGL_draw_buffers/drawBuffersWEBGL) */
+  /**
+   * The **`WEBGL_draw_buffers.drawBuffersWEBGL()`** method is part of the WebGL API and allows you to define the draw buffers to which all fragment colors are written.
+   *
+   * [MDN Reference](https://developer.mozilla.org/docs/Web/API/WEBGL_draw_buffers/drawBuffersWEBGL)
+   */
   drawBuffersWEBGL(buffers: Iterable<GLenum>): void;
 }
 
 interface WEBGL_multi_draw {
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WEBGL_multi_draw/multiDrawArraysInstancedWEBGL) */
+  /**
+   * The **`WEBGL_multi_draw.multiDrawArraysInstancedWEBGL()`** method of the WebGL API renders multiple primitives from array data.
+   *
+   * [MDN Reference](https://developer.mozilla.org/docs/Web/API/WEBGL_multi_draw/multiDrawArraysInstancedWEBGL)
+   */
   multiDrawArraysInstancedWEBGL(
     mode: GLenum,
-    firstsList: Int32Array | Iterable<GLint>,
+    firstsList: Int32Array<ArrayBufferLike> | Iterable<GLint>,
     firstsOffset: number,
-    countsList: Int32Array | Iterable<GLsizei>,
+    countsList: Int32Array<ArrayBufferLike> | Iterable<GLsizei>,
     countsOffset: number,
-    instanceCountsList: Int32Array | Iterable<GLsizei>,
+    instanceCountsList: Int32Array<ArrayBufferLike> | Iterable<GLsizei>,
     instanceCountsOffset: number,
     drawcount: GLsizei,
   ): void;
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WEBGL_multi_draw/multiDrawArraysWEBGL) */
+  /**
+   * The **`WEBGL_multi_draw.multiDrawArraysWEBGL()`** method of the WebGL API renders multiple primitives from array data.
+   *
+   * [MDN Reference](https://developer.mozilla.org/docs/Web/API/WEBGL_multi_draw/multiDrawArraysWEBGL)
+   */
   multiDrawArraysWEBGL(
     mode: GLenum,
-    firstsList: Int32Array | Iterable<GLint>,
+    firstsList: Int32Array<ArrayBufferLike> | Iterable<GLint>,
     firstsOffset: number,
-    countsList: Int32Array | Iterable<GLsizei>,
+    countsList: Int32Array<ArrayBufferLike> | Iterable<GLsizei>,
     countsOffset: number,
     drawcount: GLsizei,
   ): void;
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WEBGL_multi_draw/multiDrawElementsInstancedWEBGL) */
+  /**
+   * The **`WEBGL_multi_draw.multiDrawElementsInstancedWEBGL()`** method of the WebGL API renders multiple primitives from array data.
+   *
+   * [MDN Reference](https://developer.mozilla.org/docs/Web/API/WEBGL_multi_draw/multiDrawElementsInstancedWEBGL)
+   */
   multiDrawElementsInstancedWEBGL(
     mode: GLenum,
-    countsList: Int32Array | Iterable<GLsizei>,
+    countsList: Int32Array<ArrayBufferLike> | Iterable<GLsizei>,
     countsOffset: number,
     type: GLenum,
-    offsetsList: Int32Array | Iterable<GLsizei>,
+    offsetsList: Int32Array<ArrayBufferLike> | Iterable<GLsizei>,
     offsetsOffset: number,
-    instanceCountsList: Int32Array | Iterable<GLsizei>,
+    instanceCountsList: Int32Array<ArrayBufferLike> | Iterable<GLsizei>,
     instanceCountsOffset: number,
     drawcount: GLsizei,
   ): void;
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WEBGL_multi_draw/multiDrawElementsWEBGL) */
+  /**
+   * The **`WEBGL_multi_draw.multiDrawElementsWEBGL()`** method of the WebGL API renders multiple primitives from array data.
+   *
+   * [MDN Reference](https://developer.mozilla.org/docs/Web/API/WEBGL_multi_draw/multiDrawElementsWEBGL)
+   */
   multiDrawElementsWEBGL(
     mode: GLenum,
-    countsList: Int32Array | Iterable<GLsizei>,
+    countsList: Int32Array<ArrayBufferLike> | Iterable<GLsizei>,
     countsOffset: number,
     type: GLenum,
-    offsetsList: Int32Array | Iterable<GLsizei>,
+    offsetsList: Int32Array<ArrayBufferLike> | Iterable<GLsizei>,
     offsetsOffset: number,
     drawcount: GLsizei,
   ): void;
@@ -346,7 +399,7 @@ interface WebGL2RenderingContextBase {
   getUniformIndices(
     program: WebGLProgram,
     uniformNames: Iterable<string>,
-  ): Iterable<GLuint> | null;
+  ): readonly GLuint[] | null;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGL2RenderingContext/invalidateFramebuffer) */
   invalidateFramebuffer(target: GLenum, attachments: Iterable<GLenum>): void;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGL2RenderingContext/invalidateSubFramebuffer) */
