diff --git experimental/packages/utils/react-blueprintjs-utils/src/constants/common/mixins.mts apps/react-blueprintjs-utils/src/constants/common/mixins.mts
index 75c0c265a..ddaad34d6 100644
--- experimental/packages/utils/react-blueprintjs-utils/src/constants/common/mixins.mts
+++ apps/react-blueprintjs-utils/src/constants/common/mixins.mts
@@ -44,7 +44,7 @@ export const ptDarkIntentTextColors = {
 } as const;
 
 export const intentColor = (intentName: IntentKey): string =>
-  `color: ${ptIntentColors[intentName]});`;
+  `color: ${ptIntentColors[intentName]});` as const;
 
 export const positionAll = (position: string, value: string) =>
   ({
@@ -57,10 +57,10 @@ export const positionAll = (position: string, value: string) =>
 
 export const baseTypography = [
   `font-size: ${ptFontSizePx}px;`,
-  `font-weight: 400;`,
-  `letter-spacing: 0;`,
+  'font-weight: 400;',
+  'letter-spacing: 0;',
   `line-height: ${ptLineHeightPx}px;`,
-  `text-transform: none;`,
+  'text-transform: none;',
 ].join('\n');
 
 export const runningTypography = [
@@ -96,7 +96,7 @@ export const borderShadow = (
   alpha: number,
   colorHex: string = black,
   sizePx = 1,
-): string => `0 0 0 ${sizePx}px ${hexToRgba(colorHex, alpha)}`;
+): string => `0 0 0 ${sizePx}px ${hexToRgba(colorHex, alpha)}` as const;
 
 // returns the padding necessary to center text in a container of the given height.
 // default line-height is that of base typography, 18px assuming 14px font-size.
