diff --git experimental/packages/apps/lambda-calculus-interpreter-core/src/types/lambda-term.test.mts apps/lambda-calculus-interpreter-core/src/types/lambda-term.test.mts
index 4fbafd9a0..ca7c4da4a 100644
--- experimental/packages/apps/lambda-calculus-interpreter-core/src/types/lambda-term.test.mts
+++ apps/lambda-calculus-interpreter-core/src/types/lambda-term.test.mts
@@ -1,4 +1,9 @@
-import { expectType } from '@noshiro/ts-utils';
+import { expectType } from 'ts-data-forge';
+import {
+  type DeepReadonly,
+  type FixedLengthTuple,
+  type MutableFixedLengthTuple,
+} from 'ts-type-forge';
 import {
   type LambdaAbstraction,
   type LambdaApplication,
@@ -12,33 +17,56 @@ describe('test types', () => {
   });
 
   type l1 = 'x';
-  type l2 = readonly ['x', 'x'];
-  type l3 = readonly ['x', ['y', 'y']];
-  type l4 = DeepReadonly<[['x', ['y', 'y']], ['x', ['y', 'y']]]>;
-  type l5 = DeepReadonly<['lambda', 'x', ['x', ['y', 'y']]]>;
+
+  type l2 = FixedLengthTuple<2, 'x'>;
+
+  type l3 = readonly ['x', MutableFixedLengthTuple<2, 'y'>];
+
+  type l4 = DeepReadonly<
+    MutableFixedLengthTuple<2, ['x', MutableFixedLengthTuple<2, 'y'>]>
+  >;
+
+  type l5 = DeepReadonly<
+    ['lambda', 'x', ['x', MutableFixedLengthTuple<2, 'y'>]]
+  >;
 
   expectType<l1, LambdaTerm>('<=');
+
   expectType<l1, Variable>('<=');
+
   expectType<l1, LambdaApplication>('!<=');
+
   expectType<l1, LambdaAbstraction>('!<=');
 
   expectType<l2, LambdaTerm>('<=');
+
   expectType<l2, Variable>('!<=');
+
   expectType<l2, LambdaApplication>('<=');
+
   expectType<l2, LambdaAbstraction>('!<=');
 
   expectType<l3, LambdaTerm>('<=');
+
   expectType<l3, Variable>('!<=');
+
   expectType<l3, LambdaApplication>('<=');
+
   expectType<l3, LambdaAbstraction>('!<=');
 
   expectType<l4, LambdaTerm>('<=');
+
   expectType<l4, Variable>('!<=');
+
   expectType<l4, LambdaApplication>('<=');
+
   expectType<l4, LambdaAbstraction>('!<=');
 
   expectType<l5, LambdaTerm>('<=');
+
   expectType<l5, Variable>('!<=');
+
   expectType<l5, LambdaApplication>('!<=');
+
   expectType<l5, LambdaAbstraction>('<=');
 });
