diff --git a/packages/v5.6/output-branded/lib-files/lib.esnext.string.d.ts b/packages/v5.6/output-branded/lib-files/lib.esnext.string.d.ts
deleted file mode 100644
index db807cd..0000000
--- a/packages/v5.6/output-branded/lib-files/lib.esnext.string.d.ts
+++ /dev/null
@@ -1,14 +0,0 @@
-/// <reference no-default-lib="true"/>
-/// <reference types="ts-type-forge/global" />
-
-interface String {
-  /**
-   * Returns true if all leading surrogates and trailing surrogates appear paired and in order.
-   */
-  isWellFormed(): boolean;
-
-  /**
-   * Returns a string where all lone or out-of-order surrogates have been replaced by the Unicode replacement character (U+FFFD).
-   */
-  toWellFormed(): string;
-}
