diff --git a/packages/v5.6/output-branded/lib-files/lib.es2016.array.include.d.ts b/packages/v5.7/output-branded/lib-files/lib.es2016.array.include.d.ts
index 390d6c8..0c23e50 100644
--- a/packages/v5.6/output-branded/lib-files/lib.es2016.array.include.d.ts
+++ b/packages/v5.7/output-branded/lib-files/lib.es2016.array.include.d.ts
@@ -25,7 +25,7 @@ interface ReadonlyArray<T> {
   ): searchElement is T;
 }
 
-interface Int8Array {
+interface Int8Array<TArrayBuffer extends ArrayBufferLike> {
   /**
    * Determines whether an array includes a certain element, returning true or false as appropriate.
    * @param searchElement The element to search for.
@@ -37,7 +37,7 @@ interface Int8Array {
   ): boolean;
 }
 
-interface Uint8Array {
+interface Uint8Array<TArrayBuffer extends ArrayBufferLike> {
   /**
    * Determines whether an array includes a certain element, returning true or false as appropriate.
    * @param searchElement The element to search for.
@@ -49,7 +49,7 @@ interface Uint8Array {
   ): boolean;
 }
 
-interface Uint8ClampedArray {
+interface Uint8ClampedArray<TArrayBuffer extends ArrayBufferLike> {
   /**
    * Determines whether an array includes a certain element, returning true or false as appropriate.
    * @param searchElement The element to search for.
@@ -61,7 +61,7 @@ interface Uint8ClampedArray {
   ): boolean;
 }
 
-interface Int16Array {
+interface Int16Array<TArrayBuffer extends ArrayBufferLike> {
   /**
    * Determines whether an array includes a certain element, returning true or false as appropriate.
    * @param searchElement The element to search for.
@@ -73,7 +73,7 @@ interface Int16Array {
   ): boolean;
 }
 
-interface Uint16Array {
+interface Uint16Array<TArrayBuffer extends ArrayBufferLike> {
   /**
    * Determines whether an array includes a certain element, returning true or false as appropriate.
    * @param searchElement The element to search for.
@@ -85,7 +85,7 @@ interface Uint16Array {
   ): boolean;
 }
 
-interface Int32Array {
+interface Int32Array<TArrayBuffer extends ArrayBufferLike> {
   /**
    * Determines whether an array includes a certain element, returning true or false as appropriate.
    * @param searchElement The element to search for.
@@ -97,7 +97,7 @@ interface Int32Array {
   ): boolean;
 }
 
-interface Uint32Array {
+interface Uint32Array<TArrayBuffer extends ArrayBufferLike> {
   /**
    * Determines whether an array includes a certain element, returning true or false as appropriate.
    * @param searchElement The element to search for.
@@ -109,7 +109,7 @@ interface Uint32Array {
   ): boolean;
 }
 
-interface Float32Array {
+interface Float32Array<TArrayBuffer extends ArrayBufferLike> {
   /**
    * Determines whether an array includes a certain element, returning true or false as appropriate.
    * @param searchElement The element to search for.
@@ -121,7 +121,7 @@ interface Float32Array {
   ): boolean;
 }
 
-interface Float64Array {
+interface Float64Array<TArrayBuffer extends ArrayBufferLike> {
   /**
    * Determines whether an array includes a certain element, returning true or false as appropriate.
    * @param searchElement The element to search for.
