diff --git a/packages/v5.6/output-branded/lib-files/lib.es2021.weakref.d.ts b/packages/v5.7/output-branded/lib-files/lib.es2021.weakref.d.ts
index 4aa7dfb..78bee5e 100644
--- a/packages/v5.6/output-branded/lib-files/lib.es2021.weakref.d.ts
+++ b/packages/v5.7/output-branded/lib-files/lib.es2021.weakref.d.ts
@@ -1,6 +1,8 @@
 /// <reference no-default-lib="true"/>
 /// <reference types="ts-type-forge/global" />
 
+/// <reference path="./lib.es2015.symbol.wellknown.d.ts" />
+
 interface WeakRef<T extends WeakKey> {
   readonly [Symbol.toStringTag]: 'WeakRef';
 
