diff --git a/packages/v5.6/output-branded/lib-files/lib.es2015.iterable.d.ts b/packages/v5.7/output-branded/lib-files/lib.es2015.iterable.d.ts
index 7596242..70f0a98 100644
--- a/packages/v5.6/output-branded/lib-files/lib.es2015.iterable.d.ts
+++ b/packages/v5.7/output-branded/lib-files/lib.es2015.iterable.d.ts
@@ -288,7 +288,7 @@ interface String {
   [Symbol.iterator](): StringIterator<string>;
 }
 
-interface Int8Array {
+interface Int8Array<TArrayBuffer extends ArrayBufferLike> {
   [Symbol.iterator](): ArrayIterator<Int8>;
   /**
    * Returns an array of key, value pairs for every entry in the array
@@ -305,7 +305,7 @@ interface Int8Array {
 }
 
 interface Int8ArrayConstructor {
-  new (elements: Iterable<number>): Int8Array;
+  new (elements: Iterable<number>): Int8Array<ArrayBuffer>;
 
   /**
    * Creates an array from an array-like or iterable object.
@@ -313,14 +313,21 @@ interface Int8ArrayConstructor {
    * @param mapfn A mapping function to call on every element of the array.
    * @param thisArg Value of 'this' used to invoke the mapfn.
    */
-  from<T extends number>(
+  from<T extends number>(arrayLike: Iterable<T>): Int8Array<ArrayBuffer>;
+  /**
+   * Creates an array from an array-like or iterable object.
+   * @param arrayLike An array-like or iterable object to convert to an array.
+   * @param mapfn A mapping function to call on every element of the array.
+   * @param thisArg Value of 'this' used to invoke the mapfn.
+   */
+  from<T>(
     arrayLike: Iterable<T>,
     mapfn?: (v: T, k: NumberType.TypedArraySize) => Int8,
     thisArg?: unknown,
-  ): Int8Array;
+  ): Int8Array<ArrayBuffer>;
 }
 
-interface Uint8Array {
+interface Uint8Array<TArrayBuffer extends ArrayBufferLike> {
   [Symbol.iterator](): ArrayIterator<Uint8>;
   /**
    * Returns an array of key, value pairs for every entry in the array
@@ -337,7 +344,7 @@ interface Uint8Array {
 }
 
 interface Uint8ArrayConstructor {
-  new (elements: Iterable<number>): Uint8Array;
+  new (elements: Iterable<number>): Uint8Array<ArrayBuffer>;
 
   /**
    * Creates an array from an array-like or iterable object.
@@ -345,14 +352,21 @@ interface Uint8ArrayConstructor {
    * @param mapfn A mapping function to call on every element of the array.
    * @param thisArg Value of 'this' used to invoke the mapfn.
    */
-  from<T extends number>(
+  from<T extends number>(arrayLike: Iterable<T>): Uint8Array<ArrayBuffer>;
+  /**
+   * Creates an array from an array-like or iterable object.
+   * @param arrayLike An array-like or iterable object to convert to an array.
+   * @param mapfn A mapping function to call on every element of the array.
+   * @param thisArg Value of 'this' used to invoke the mapfn.
+   */
+  from<T>(
     arrayLike: Iterable<T>,
     mapfn?: (v: T, k: NumberType.TypedArraySize) => Uint8,
     thisArg?: unknown,
-  ): Uint8Array;
+  ): Uint8Array<ArrayBuffer>;
 }
 
-interface Uint8ClampedArray {
+interface Uint8ClampedArray<TArrayBuffer extends ArrayBufferLike> {
   [Symbol.iterator](): ArrayIterator<Uint8>;
   /**
    * Returns an array of key, value pairs for every entry in the array
@@ -371,7 +385,7 @@ interface Uint8ClampedArray {
 }
 
 interface Uint8ClampedArrayConstructor {
-  new (elements: Iterable<number>): Uint8ClampedArray;
+  new (elements: Iterable<number>): Uint8ClampedArray<ArrayBuffer>;
 
   /**
    * Creates an array from an array-like or iterable object.
@@ -381,12 +395,21 @@ interface Uint8ClampedArrayConstructor {
    */
   from<T extends number>(
     arrayLike: Iterable<T>,
+  ): Uint8ClampedArray<ArrayBuffer>;
+  /**
+   * Creates an array from an array-like or iterable object.
+   * @param arrayLike An array-like or iterable object to convert to an array.
+   * @param mapfn A mapping function to call on every element of the array.
+   * @param thisArg Value of 'this' used to invoke the mapfn.
+   */
+  from<T>(
+    arrayLike: Iterable<T>,
     mapfn?: (v: T, k: NumberType.TypedArraySize) => Uint8,
     thisArg?: unknown,
-  ): Uint8ClampedArray;
+  ): Uint8ClampedArray<ArrayBuffer>;
 }
 
-interface Int16Array {
+interface Int16Array<TArrayBuffer extends ArrayBufferLike> {
   [Symbol.iterator](): ArrayIterator<Int16>;
   /**
    * Returns an array of key, value pairs for every entry in the array
@@ -405,7 +428,7 @@ interface Int16Array {
 }
 
 interface Int16ArrayConstructor {
-  new (elements: Iterable<number>): Int16Array;
+  new (elements: Iterable<number>): Int16Array<ArrayBuffer>;
 
   /**
    * Creates an array from an array-like or iterable object.
@@ -413,14 +436,21 @@ interface Int16ArrayConstructor {
    * @param mapfn A mapping function to call on every element of the array.
    * @param thisArg Value of 'this' used to invoke the mapfn.
    */
-  from<T extends number>(
+  from<T extends number>(arrayLike: Iterable<T>): Int16Array<ArrayBuffer>;
+  /**
+   * Creates an array from an array-like or iterable object.
+   * @param arrayLike An array-like or iterable object to convert to an array.
+   * @param mapfn A mapping function to call on every element of the array.
+   * @param thisArg Value of 'this' used to invoke the mapfn.
+   */
+  from<T>(
     arrayLike: Iterable<T>,
     mapfn?: (v: T, k: NumberType.TypedArraySize) => Int16,
     thisArg?: unknown,
-  ): Int16Array;
+  ): Int16Array<ArrayBuffer>;
 }
 
-interface Uint16Array {
+interface Uint16Array<TArrayBuffer extends ArrayBufferLike> {
   [Symbol.iterator](): ArrayIterator<Uint16>;
   /**
    * Returns an array of key, value pairs for every entry in the array
@@ -437,7 +467,7 @@ interface Uint16Array {
 }
 
 interface Uint16ArrayConstructor {
-  new (elements: Iterable<number>): Uint16Array;
+  new (elements: Iterable<number>): Uint16Array<ArrayBuffer>;
 
   /**
    * Creates an array from an array-like or iterable object.
@@ -445,14 +475,21 @@ interface Uint16ArrayConstructor {
    * @param mapfn A mapping function to call on every element of the array.
    * @param thisArg Value of 'this' used to invoke the mapfn.
    */
-  from<T extends number>(
+  from<T extends number>(arrayLike: Iterable<T>): Uint16Array<ArrayBuffer>;
+  /**
+   * Creates an array from an array-like or iterable object.
+   * @param arrayLike An array-like or iterable object to convert to an array.
+   * @param mapfn A mapping function to call on every element of the array.
+   * @param thisArg Value of 'this' used to invoke the mapfn.
+   */
+  from<T>(
     arrayLike: Iterable<T>,
     mapfn?: (v: T, k: NumberType.TypedArraySize) => Uint16,
     thisArg?: unknown,
-  ): Uint16Array;
+  ): Uint16Array<ArrayBuffer>;
 }
 
-interface Int32Array {
+interface Int32Array<TArrayBuffer extends ArrayBufferLike> {
   [Symbol.iterator](): ArrayIterator<Int32>;
   /**
    * Returns an array of key, value pairs for every entry in the array
@@ -469,7 +506,7 @@ interface Int32Array {
 }
 
 interface Int32ArrayConstructor {
-  new (elements: Iterable<number>): Int32Array;
+  new (elements: Iterable<number>): Int32Array<ArrayBuffer>;
 
   /**
    * Creates an array from an array-like or iterable object.
@@ -477,14 +514,21 @@ interface Int32ArrayConstructor {
    * @param mapfn A mapping function to call on every element of the array.
    * @param thisArg Value of 'this' used to invoke the mapfn.
    */
-  from<T extends number>(
+  from<T extends number>(arrayLike: Iterable<T>): Int32Array<ArrayBuffer>;
+  /**
+   * Creates an array from an array-like or iterable object.
+   * @param arrayLike An array-like or iterable object to convert to an array.
+   * @param mapfn A mapping function to call on every element of the array.
+   * @param thisArg Value of 'this' used to invoke the mapfn.
+   */
+  from<T>(
     arrayLike: Iterable<T>,
     mapfn?: (v: T, k: NumberType.TypedArraySize) => Int32,
     thisArg?: unknown,
-  ): Int32Array;
+  ): Int32Array<ArrayBuffer>;
 }
 
-interface Uint32Array {
+interface Uint32Array<TArrayBuffer extends ArrayBufferLike> {
   [Symbol.iterator](): ArrayIterator<Uint32>;
   /**
    * Returns an array of key, value pairs for every entry in the array
@@ -501,7 +545,7 @@ interface Uint32Array {
 }
 
 interface Uint32ArrayConstructor {
-  new (elements: Iterable<number>): Uint32Array;
+  new (elements: Iterable<number>): Uint32Array<ArrayBuffer>;
 
   /**
    * Creates an array from an array-like or iterable object.
@@ -509,14 +553,21 @@ interface Uint32ArrayConstructor {
    * @param mapfn A mapping function to call on every element of the array.
    * @param thisArg Value of 'this' used to invoke the mapfn.
    */
-  from<T extends number>(
+  from<T extends number>(arrayLike: Iterable<T>): Uint32Array<ArrayBuffer>;
+  /**
+   * Creates an array from an array-like or iterable object.
+   * @param arrayLike An array-like or iterable object to convert to an array.
+   * @param mapfn A mapping function to call on every element of the array.
+   * @param thisArg Value of 'this' used to invoke the mapfn.
+   */
+  from<T>(
     arrayLike: Iterable<T>,
     mapfn?: (v: T, k: NumberType.TypedArraySize) => Uint32,
     thisArg?: unknown,
-  ): Uint32Array;
+  ): Uint32Array<ArrayBuffer>;
 }
 
-interface Float32Array {
+interface Float32Array<TArrayBuffer extends ArrayBufferLike> {
   [Symbol.iterator](): ArrayIterator<Float32>;
   /**
    * Returns an array of key, value pairs for every entry in the array
@@ -533,7 +584,7 @@ interface Float32Array {
 }
 
 interface Float32ArrayConstructor {
-  new (elements: Iterable<number>): Float32Array;
+  new (elements: Iterable<number>): Float32Array<ArrayBuffer>;
 
   /**
    * Creates an array from an array-like or iterable object.
@@ -541,14 +592,21 @@ interface Float32ArrayConstructor {
    * @param mapfn A mapping function to call on every element of the array.
    * @param thisArg Value of 'this' used to invoke the mapfn.
    */
-  from<T extends number>(
+  from<T extends number>(arrayLike: Iterable<T>): Float32Array<ArrayBuffer>;
+  /**
+   * Creates an array from an array-like or iterable object.
+   * @param arrayLike An array-like or iterable object to convert to an array.
+   * @param mapfn A mapping function to call on every element of the array.
+   * @param thisArg Value of 'this' used to invoke the mapfn.
+   */
+  from<T>(
     arrayLike: Iterable<T>,
     mapfn?: (v: T, k: NumberType.TypedArraySize) => Float32,
     thisArg?: unknown,
-  ): Float32Array;
+  ): Float32Array<ArrayBuffer>;
 }
 
-interface Float64Array {
+interface Float64Array<TArrayBuffer extends ArrayBufferLike> {
   [Symbol.iterator](): ArrayIterator<Float64>;
   /**
    * Returns an array of key, value pairs for every entry in the array
@@ -565,7 +623,7 @@ interface Float64Array {
 }
 
 interface Float64ArrayConstructor {
-  new (elements: Iterable<number>): Float64Array;
+  new (elements: Iterable<number>): Float64Array<ArrayBuffer>;
 
   /**
    * Creates an array from an array-like or iterable object.
@@ -573,9 +631,16 @@ interface Float64ArrayConstructor {
    * @param mapfn A mapping function to call on every element of the array.
    * @param thisArg Value of 'this' used to invoke the mapfn.
    */
-  from<T extends number>(
+  from<T extends number>(arrayLike: Iterable<T>): Float64Array<ArrayBuffer>;
+  /**
+   * Creates an array from an array-like or iterable object.
+   * @param arrayLike An array-like or iterable object to convert to an array.
+   * @param mapfn A mapping function to call on every element of the array.
+   * @param thisArg Value of 'this' used to invoke the mapfn.
+   */
+  from<T>(
     arrayLike: Iterable<T>,
     mapfn?: (v: T, k: NumberType.TypedArraySize) => Float64,
     thisArg?: unknown,
-  ): Float64Array;
+  ): Float64Array<ArrayBuffer>;
 }
