diff --git a/packages/v5.6/output-branded/lib-files/lib.es2022.error.d.ts b/packages/v5.7/output-branded/lib-files/lib.es2022.error.d.ts
index df5275b..00f2edc 100644
--- a/packages/v5.6/output-branded/lib-files/lib.es2022.error.d.ts
+++ b/packages/v5.7/output-branded/lib-files/lib.es2022.error.d.ts
@@ -1,6 +1,8 @@
 /// <reference no-default-lib="true"/>
 /// <reference types="ts-type-forge/global" />
 
+/// <reference path="./lib.es2021.promise.d.ts" />
+
 interface ErrorOptions {
   readonly cause?: unknown;
 }
