diff --git a/packages/v5.7/output-branded/lib-files/lib.es2024.arraybuffer.d.ts b/packages/v5.7/output-branded/lib-files/lib.es2024.arraybuffer.d.ts
new file mode 100644
index 0000000..e2aaa92
--- /dev/null
+++ b/packages/v5.7/output-branded/lib-files/lib.es2024.arraybuffer.d.ts
@@ -0,0 +1,59 @@
+/// <reference no-default-lib="true"/>
+/// <reference types="ts-type-forge/global" />
+
+interface ArrayBuffer {
+  /**
+   * If this ArrayBuffer is resizable, returns the maximum byte length given during construction; returns the byte length if not.
+   *
+   * [MDN](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/ArrayBuffer/maxByteLength)
+   */
+  get maxByteLength(): NumberType.TypedArraySize;
+
+  /**
+   * Returns true if this ArrayBuffer can be resized.
+   *
+   * [MDN](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/ArrayBuffer/resizable)
+   */
+  get resizable(): boolean;
+
+  /**
+   * Resizes the ArrayBuffer to the specified size (in bytes).
+   *
+   * [MDN](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/ArrayBuffer/resize)
+   */
+  resize(newByteLength?: NumberType.TypedArraySizeArgNonNegative): void;
+
+  /**
+   * Returns a boolean indicating whether or not this buffer has been detached (transferred).
+   *
+   * [MDN](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/ArrayBuffer/detached)
+   */
+  get detached(): boolean;
+
+  /**
+   * Creates a new ArrayBuffer with the same byte content as this buffer, then detaches this buffer.
+   *
+   * [MDN](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/ArrayBuffer/transfer)
+   */
+  transfer(
+    newByteLength?: NumberType.TypedArraySizeArgNonNegative,
+  ): ArrayBuffer;
+
+  /**
+   * Creates a new non-resizable ArrayBuffer with the same byte content as this buffer, then detaches this buffer.
+   *
+   * [MDN](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/ArrayBuffer/transferToFixedLength)
+   */
+  transferToFixedLength(
+    newByteLength?: NumberType.TypedArraySizeArgNonNegative,
+  ): ArrayBuffer;
+}
+
+interface ArrayBufferConstructor {
+  new (
+    byteLength: NumberType.TypedArraySizeArgNonNegative,
+    options?: {
+      readonly maxByteLength?: NumberType.TypedArraySizeArgNonNegative;
+    },
+  ): ArrayBuffer;
+}
