diff --git experimental/packages/apps/housing-loan-calculator-app/src/types/calculated-values.ts apps/housing-loan-calculator-app/src/types/calculated-values.mts
index a875a146c..40c2b7b3b 100644
--- experimental/packages/apps/housing-loan-calculator-app/src/types/calculated-values.ts
+++ apps/housing-loan-calculator-app/src/types/calculated-values.mts
@@ -1,4 +1,5 @@
-import { type Yen } from './yen';
+import { type DeepReadonly } from 'ts-type-forge';
+import { type Yen } from './yen.mjs';
 
 export type CalculatedValues = DeepReadonly<{
   /** 借入残高（円） */
