diff --git a/packages/v5.2/temp/copied/lib.dom.d.ts b/packages/v5.3/temp/copied/lib.dom.d.ts
index db208a0..5d1e3a1 100644
--- a/packages/v5.2/temp/copied/lib.dom.d.ts
+++ b/packages/v5.3/temp/copied/lib.dom.d.ts
@@ -1438,6 +1438,7 @@ interface RTCInboundRtpStreamStats extends RTCReceivedRtpStreamStats {
   keyFramesDecoded?: number;
   kind: string;
   lastPacketReceivedTimestamp?: DOMHighResTimeStamp;
+  mid?: string;
   nackCount?: number;
   packetsDiscarded?: number;
   pliCount?: number;
@@ -1452,6 +1453,7 @@ interface RTCInboundRtpStreamStats extends RTCReceivedRtpStreamStats {
   totalSamplesDuration?: number;
   totalSamplesReceived?: number;
   totalSquaredInterFrameDelay?: number;
+  trackIdentifier: string;
 }
 
 interface RTCLocalSessionDescriptionInit {
@@ -26503,7 +26505,16 @@ interface SubtleCrypto {
     format: Exclude<KeyFormat, 'jwk'>,
     key: CryptoKey,
   ): Promise<ArrayBuffer>;
+  exportKey(
+    format: KeyFormat,
+    key: CryptoKey,
+  ): Promise<ArrayBuffer | JsonWebKey>;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/SubtleCrypto/generateKey) */
+  generateKey(
+    algorithm: 'Ed25519',
+    extractable: boolean,
+    keyUsages: ReadonlyArray<'sign' | 'verify'>,
+  ): Promise<CryptoKeyPair>;
   generateKey(
     algorithm: RsaHashedKeyGenParams | EcKeyGenParams,
     extractable: boolean,
