diff --git experimental/packages/apps/lambda-calculus-interpreter-core/src/functions/parser/macro/number.mts apps/lambda-calculus-interpreter-core/src/functions/parser/macro/number.mts
index 0bae6c545..f57e870ef 100644
--- experimental/packages/apps/lambda-calculus-interpreter-core/src/functions/parser/macro/number.mts
+++ apps/lambda-calculus-interpreter-core/src/functions/parser/macro/number.mts
@@ -1,7 +1,8 @@
+import { type SafeUint } from 'ts-type-forge';
 import { splitToTokens } from '../split-to-tokens.mjs';
 
 const numberStr = (n: SafeUint): string =>
-  `(lambda s.(lambda z. ${'(s'.repeat(n)} z${')'.repeat(n)}))`;
+  `(lambda s.(lambda z. ${'(s'.repeat(n)} z${')'.repeat(n)}))` as const;
 
 export const numberLambdaTerm = (n: SafeUint): readonly string[] =>
   splitToTokens(numberStr(n));
