diff --git a/packages/v5.6/temp/copied/lib.es2023.array.d.ts b/packages/v5.7/temp/copied/lib.es2023.array.d.ts
index d19505e..8082514 100644
--- a/packages/v5.6/temp/copied/lib.es2023.array.d.ts
+++ b/packages/v5.7/temp/copied/lib.es2023.array.d.ts
@@ -171,7 +171,7 @@ interface ReadonlyArray<T> {
   with(index: number, value: T): T[];
 }
 
-interface Int8Array {
+interface Int8Array<TArrayBuffer extends ArrayBufferLike> {
   /**
    * Returns the value of the last element in the array where predicate is true, and undefined
    * otherwise.
@@ -182,11 +182,11 @@ interface Int8Array {
    * predicate. If it is not provided, undefined is used instead.
    */
   findLast<S extends number>(
-    predicate: (value: number, index: number, array: Int8Array) => value is S,
+    predicate: (value: number, index: number, array: this) => value is S,
     thisArg?: any,
   ): S | undefined;
   findLast(
-    predicate: (value: number, index: number, array: Int8Array) => unknown,
+    predicate: (value: number, index: number, array: this) => unknown,
     thisArg?: any,
   ): number | undefined;
 
@@ -200,14 +200,14 @@ interface Int8Array {
    * predicate. If it is not provided, undefined is used instead.
    */
   findLastIndex(
-    predicate: (value: number, index: number, array: Int8Array) => unknown,
+    predicate: (value: number, index: number, array: this) => unknown,
     thisArg?: any,
   ): number;
 
   /**
    * Copies the array and returns the copy with the elements in reverse order.
    */
-  toReversed(): Int8Array;
+  toReversed(): Int8Array<ArrayBuffer>;
 
   /**
    * Copies and sorts the array.
@@ -215,11 +215,13 @@ interface Int8Array {
    * a negative value if the first argument is less than the second argument, zero if they're equal, and a positive
    * value otherwise. If omitted, the elements are sorted in ascending order.
    * ```ts
-   * const myNums = Int8Array.from([11, 2, 22, 1]);
-   * myNums.toSorted((a, b) => a - b) // Int8Array(4) [1, 2, 11, 22]
+   * const myNums = Int8Array<Buffer>.from([11, 2, 22, 1]);
+   * myNums.toSorted((a, b) => a - b) // Int8Array<Buffer>(4) [1, 2, 11, 22]
    * ```
    */
-  toSorted(compareFn?: (a: number, b: number) => number): Int8Array;
+  toSorted(
+    compareFn?: (a: number, b: number) => number,
+  ): Int8Array<ArrayBuffer>;
 
   /**
    * Copies the array and inserts the given number at the provided index.
@@ -228,10 +230,10 @@ interface Int8Array {
    * @param value The value to insert into the copied array.
    * @returns A copy of the original array with the inserted value.
    */
-  with(index: number, value: number): Int8Array;
+  with(index: number, value: number): Int8Array<ArrayBuffer>;
 }
 
-interface Uint8Array {
+interface Uint8Array<TArrayBuffer extends ArrayBufferLike> {
   /**
    * Returns the value of the last element in the array where predicate is true, and undefined
    * otherwise.
@@ -242,11 +244,11 @@ interface Uint8Array {
    * predicate. If it is not provided, undefined is used instead.
    */
   findLast<S extends number>(
-    predicate: (value: number, index: number, array: Uint8Array) => value is S,
+    predicate: (value: number, index: number, array: this) => value is S,
     thisArg?: any,
   ): S | undefined;
   findLast(
-    predicate: (value: number, index: number, array: Uint8Array) => unknown,
+    predicate: (value: number, index: number, array: this) => unknown,
     thisArg?: any,
   ): number | undefined;
 
@@ -260,14 +262,14 @@ interface Uint8Array {
    * predicate. If it is not provided, undefined is used instead.
    */
   findLastIndex(
-    predicate: (value: number, index: number, array: Uint8Array) => unknown,
+    predicate: (value: number, index: number, array: this) => unknown,
     thisArg?: any,
   ): number;
 
   /**
    * Copies the array and returns the copy with the elements in reverse order.
    */
-  toReversed(): Uint8Array;
+  toReversed(): Uint8Array<ArrayBuffer>;
 
   /**
    * Copies and sorts the array.
@@ -275,11 +277,13 @@ interface Uint8Array {
    * a negative value if the first argument is less than the second argument, zero if they're equal, and a positive
    * value otherwise. If omitted, the elements are sorted in ascending order.
    * ```ts
-   * const myNums = Uint8Array.from([11, 2, 22, 1]);
-   * myNums.toSorted((a, b) => a - b) // Uint8Array(4) [1, 2, 11, 22]
+   * const myNums = Uint8Array<Buffer>.from([11, 2, 22, 1]);
+   * myNums.toSorted((a, b) => a - b) // Uint8Array<Buffer>(4) [1, 2, 11, 22]
    * ```
    */
-  toSorted(compareFn?: (a: number, b: number) => number): Uint8Array;
+  toSorted(
+    compareFn?: (a: number, b: number) => number,
+  ): Uint8Array<ArrayBuffer>;
 
   /**
    * Copies the array and inserts the given number at the provided index.
@@ -288,10 +292,10 @@ interface Uint8Array {
    * @param value The value to insert into the copied array.
    * @returns A copy of the original array with the inserted value.
    */
-  with(index: number, value: number): Uint8Array;
+  with(index: number, value: number): Uint8Array<ArrayBuffer>;
 }
 
-interface Uint8ClampedArray {
+interface Uint8ClampedArray<TArrayBuffer extends ArrayBufferLike> {
   /**
    * Returns the value of the last element in the array where predicate is true, and undefined
    * otherwise.
@@ -302,19 +306,11 @@ interface Uint8ClampedArray {
    * predicate. If it is not provided, undefined is used instead.
    */
   findLast<S extends number>(
-    predicate: (
-      value: number,
-      index: number,
-      array: Uint8ClampedArray,
-    ) => value is S,
+    predicate: (value: number, index: number, array: this) => value is S,
     thisArg?: any,
   ): S | undefined;
   findLast(
-    predicate: (
-      value: number,
-      index: number,
-      array: Uint8ClampedArray,
-    ) => unknown,
+    predicate: (value: number, index: number, array: this) => unknown,
     thisArg?: any,
   ): number | undefined;
 
@@ -328,18 +324,14 @@ interface Uint8ClampedArray {
    * predicate. If it is not provided, undefined is used instead.
    */
   findLastIndex(
-    predicate: (
-      value: number,
-      index: number,
-      array: Uint8ClampedArray,
-    ) => unknown,
+    predicate: (value: number, index: number, array: this) => unknown,
     thisArg?: any,
   ): number;
 
   /**
    * Copies the array and returns the copy with the elements in reverse order.
    */
-  toReversed(): Uint8ClampedArray;
+  toReversed(): Uint8ClampedArray<ArrayBuffer>;
 
   /**
    * Copies and sorts the array.
@@ -347,11 +339,13 @@ interface Uint8ClampedArray {
    * a negative value if the first argument is less than the second argument, zero if they're equal, and a positive
    * value otherwise. If omitted, the elements are sorted in ascending order.
    * ```ts
-   * const myNums = Uint8ClampedArray.from([11, 2, 22, 1]);
-   * myNums.toSorted((a, b) => a - b) // Uint8ClampedArray(4) [1, 2, 11, 22]
+   * const myNums = Uint8ClampedArray<Buffer>.from([11, 2, 22, 1]);
+   * myNums.toSorted((a, b) => a - b) // Uint8ClampedArray<Buffer>(4) [1, 2, 11, 22]
    * ```
    */
-  toSorted(compareFn?: (a: number, b: number) => number): Uint8ClampedArray;
+  toSorted(
+    compareFn?: (a: number, b: number) => number,
+  ): Uint8ClampedArray<ArrayBuffer>;
 
   /**
    * Copies the array and inserts the given number at the provided index.
@@ -360,10 +354,10 @@ interface Uint8ClampedArray {
    * @param value The value to insert into the copied array.
    * @returns A copy of the original array with the inserted value.
    */
-  with(index: number, value: number): Uint8ClampedArray;
+  with(index: number, value: number): Uint8ClampedArray<ArrayBuffer>;
 }
 
-interface Int16Array {
+interface Int16Array<TArrayBuffer extends ArrayBufferLike> {
   /**
    * Returns the value of the last element in the array where predicate is true, and undefined
    * otherwise.
@@ -374,11 +368,11 @@ interface Int16Array {
    * predicate. If it is not provided, undefined is used instead.
    */
   findLast<S extends number>(
-    predicate: (value: number, index: number, array: Int16Array) => value is S,
+    predicate: (value: number, index: number, array: this) => value is S,
     thisArg?: any,
   ): S | undefined;
   findLast(
-    predicate: (value: number, index: number, array: Int16Array) => unknown,
+    predicate: (value: number, index: number, array: this) => unknown,
     thisArg?: any,
   ): number | undefined;
 
@@ -392,14 +386,14 @@ interface Int16Array {
    * predicate. If it is not provided, undefined is used instead.
    */
   findLastIndex(
-    predicate: (value: number, index: number, array: Int16Array) => unknown,
+    predicate: (value: number, index: number, array: this) => unknown,
     thisArg?: any,
   ): number;
 
   /**
    * Copies the array and returns the copy with the elements in reverse order.
    */
-  toReversed(): Int16Array;
+  toReversed(): Int16Array<ArrayBuffer>;
 
   /**
    * Copies and sorts the array.
@@ -407,11 +401,13 @@ interface Int16Array {
    * a negative value if the first argument is less than the second argument, zero if they're equal, and a positive
    * value otherwise. If omitted, the elements are sorted in ascending order.
    * ```ts
-   * const myNums = Int16Array.from([11, 2, -22, 1]);
-   * myNums.toSorted((a, b) => a - b) // Int16Array(4) [-22, 1, 2, 11]
+   * const myNums = Int16Array<Buffer>.from([11, 2, -22, 1]);
+   * myNums.toSorted((a, b) => a - b) // Int16Array<Buffer>(4) [-22, 1, 2, 11]
    * ```
    */
-  toSorted(compareFn?: (a: number, b: number) => number): Int16Array;
+  toSorted(
+    compareFn?: (a: number, b: number) => number,
+  ): Int16Array<ArrayBuffer>;
 
   /**
    * Copies the array and inserts the given number at the provided index.
@@ -420,10 +416,10 @@ interface Int16Array {
    * @param value The value to insert into the copied array.
    * @returns A copy of the original array with the inserted value.
    */
-  with(index: number, value: number): Int16Array;
+  with(index: number, value: number): Int16Array<ArrayBuffer>;
 }
 
-interface Uint16Array {
+interface Uint16Array<TArrayBuffer extends ArrayBufferLike> {
   /**
    * Returns the value of the last element in the array where predicate is true, and undefined
    * otherwise.
@@ -434,11 +430,11 @@ interface Uint16Array {
    * predicate. If it is not provided, undefined is used instead.
    */
   findLast<S extends number>(
-    predicate: (value: number, index: number, array: Uint16Array) => value is S,
+    predicate: (value: number, index: number, array: this) => value is S,
     thisArg?: any,
   ): S | undefined;
   findLast(
-    predicate: (value: number, index: number, array: Uint16Array) => unknown,
+    predicate: (value: number, index: number, array: this) => unknown,
     thisArg?: any,
   ): number | undefined;
 
@@ -452,14 +448,14 @@ interface Uint16Array {
    * predicate. If it is not provided, undefined is used instead.
    */
   findLastIndex(
-    predicate: (value: number, index: number, array: Uint16Array) => unknown,
+    predicate: (value: number, index: number, array: this) => unknown,
     thisArg?: any,
   ): number;
 
   /**
    * Copies the array and returns the copy with the elements in reverse order.
    */
-  toReversed(): Uint16Array;
+  toReversed(): Uint16Array<ArrayBuffer>;
 
   /**
    * Copies and sorts the array.
@@ -467,11 +463,13 @@ interface Uint16Array {
    * a negative value if the first argument is less than the second argument, zero if they're equal, and a positive
    * value otherwise. If omitted, the elements are sorted in ascending order.
    * ```ts
-   * const myNums = Uint16Array.from([11, 2, 22, 1]);
-   * myNums.toSorted((a, b) => a - b) // Uint16Array(4) [1, 2, 11, 22]
+   * const myNums = Uint16Array<Buffer>.from([11, 2, 22, 1]);
+   * myNums.toSorted((a, b) => a - b) // Uint16Array<Buffer>(4) [1, 2, 11, 22]
    * ```
    */
-  toSorted(compareFn?: (a: number, b: number) => number): Uint16Array;
+  toSorted(
+    compareFn?: (a: number, b: number) => number,
+  ): Uint16Array<ArrayBuffer>;
 
   /**
    * Copies the array and inserts the given number at the provided index.
@@ -480,10 +478,10 @@ interface Uint16Array {
    * @param value The value to insert into the copied array.
    * @returns A copy of the original array with the inserted value.
    */
-  with(index: number, value: number): Uint16Array;
+  with(index: number, value: number): Uint16Array<ArrayBuffer>;
 }
 
-interface Int32Array {
+interface Int32Array<TArrayBuffer extends ArrayBufferLike> {
   /**
    * Returns the value of the last element in the array where predicate is true, and undefined
    * otherwise.
@@ -494,11 +492,11 @@ interface Int32Array {
    * predicate. If it is not provided, undefined is used instead.
    */
   findLast<S extends number>(
-    predicate: (value: number, index: number, array: Int32Array) => value is S,
+    predicate: (value: number, index: number, array: this) => value is S,
     thisArg?: any,
   ): S | undefined;
   findLast(
-    predicate: (value: number, index: number, array: Int32Array) => unknown,
+    predicate: (value: number, index: number, array: this) => unknown,
     thisArg?: any,
   ): number | undefined;
 
@@ -512,14 +510,14 @@ interface Int32Array {
    * predicate. If it is not provided, undefined is used instead.
    */
   findLastIndex(
-    predicate: (value: number, index: number, array: Int32Array) => unknown,
+    predicate: (value: number, index: number, array: this) => unknown,
     thisArg?: any,
   ): number;
 
   /**
    * Copies the array and returns the copy with the elements in reverse order.
    */
-  toReversed(): Int32Array;
+  toReversed(): Int32Array<ArrayBuffer>;
 
   /**
    * Copies and sorts the array.
@@ -527,11 +525,13 @@ interface Int32Array {
    * a negative value if the first argument is less than the second argument, zero if they're equal, and a positive
    * value otherwise. If omitted, the elements are sorted in ascending order.
    * ```ts
-   * const myNums = Int32Array.from([11, 2, -22, 1]);
-   * myNums.toSorted((a, b) => a - b) // Int32Array(4) [-22, 1, 2, 11]
+   * const myNums = Int32Array<Buffer>.from([11, 2, -22, 1]);
+   * myNums.toSorted((a, b) => a - b) // Int32Array<Buffer>(4) [-22, 1, 2, 11]
    * ```
    */
-  toSorted(compareFn?: (a: number, b: number) => number): Int32Array;
+  toSorted(
+    compareFn?: (a: number, b: number) => number,
+  ): Int32Array<ArrayBuffer>;
 
   /**
    * Copies the array and inserts the given number at the provided index.
@@ -540,10 +540,10 @@ interface Int32Array {
    * @param value The value to insert into the copied array.
    * @returns A copy of the original array with the inserted value.
    */
-  with(index: number, value: number): Int32Array;
+  with(index: number, value: number): Int32Array<ArrayBuffer>;
 }
 
-interface Uint32Array {
+interface Uint32Array<TArrayBuffer extends ArrayBufferLike> {
   /**
    * Returns the value of the last element in the array where predicate is true, and undefined
    * otherwise.
@@ -554,11 +554,11 @@ interface Uint32Array {
    * predicate. If it is not provided, undefined is used instead.
    */
   findLast<S extends number>(
-    predicate: (value: number, index: number, array: Uint32Array) => value is S,
+    predicate: (value: number, index: number, array: this) => value is S,
     thisArg?: any,
   ): S | undefined;
   findLast(
-    predicate: (value: number, index: number, array: Uint32Array) => unknown,
+    predicate: (value: number, index: number, array: this) => unknown,
     thisArg?: any,
   ): number | undefined;
 
@@ -572,14 +572,14 @@ interface Uint32Array {
    * predicate. If it is not provided, undefined is used instead.
    */
   findLastIndex(
-    predicate: (value: number, index: number, array: Uint32Array) => unknown,
+    predicate: (value: number, index: number, array: this) => unknown,
     thisArg?: any,
   ): number;
 
   /**
    * Copies the array and returns the copy with the elements in reverse order.
    */
-  toReversed(): Uint32Array;
+  toReversed(): Uint32Array<ArrayBuffer>;
 
   /**
    * Copies and sorts the array.
@@ -587,11 +587,13 @@ interface Uint32Array {
    * a negative value if the first argument is less than the second argument, zero if they're equal, and a positive
    * value otherwise. If omitted, the elements are sorted in ascending order.
    * ```ts
-   * const myNums = Uint32Array.from([11, 2, 22, 1]);
-   * myNums.toSorted((a, b) => a - b) // Uint32Array(4) [1, 2, 11, 22]
+   * const myNums = Uint32Array<Buffer>.from([11, 2, 22, 1]);
+   * myNums.toSorted((a, b) => a - b) // Uint32Array<Buffer>(4) [1, 2, 11, 22]
    * ```
    */
-  toSorted(compareFn?: (a: number, b: number) => number): Uint32Array;
+  toSorted(
+    compareFn?: (a: number, b: number) => number,
+  ): Uint32Array<ArrayBuffer>;
 
   /**
    * Copies the array and inserts the given number at the provided index.
@@ -600,10 +602,10 @@ interface Uint32Array {
    * @param value The value to insert into the copied array.
    * @returns A copy of the original array with the inserted value.
    */
-  with(index: number, value: number): Uint32Array;
+  with(index: number, value: number): Uint32Array<ArrayBuffer>;
 }
 
-interface Float32Array {
+interface Float32Array<TArrayBuffer extends ArrayBufferLike> {
   /**
    * Returns the value of the last element in the array where predicate is true, and undefined
    * otherwise.
@@ -614,15 +616,11 @@ interface Float32Array {
    * predicate. If it is not provided, undefined is used instead.
    */
   findLast<S extends number>(
-    predicate: (
-      value: number,
-      index: number,
-      array: Float32Array,
-    ) => value is S,
+    predicate: (value: number, index: number, array: this) => value is S,
     thisArg?: any,
   ): S | undefined;
   findLast(
-    predicate: (value: number, index: number, array: Float32Array) => unknown,
+    predicate: (value: number, index: number, array: this) => unknown,
     thisArg?: any,
   ): number | undefined;
 
@@ -636,14 +634,14 @@ interface Float32Array {
    * predicate. If it is not provided, undefined is used instead.
    */
   findLastIndex(
-    predicate: (value: number, index: number, array: Float32Array) => unknown,
+    predicate: (value: number, index: number, array: this) => unknown,
     thisArg?: any,
   ): number;
 
   /**
    * Copies the array and returns the copy with the elements in reverse order.
    */
-  toReversed(): Float32Array;
+  toReversed(): Float32Array<ArrayBuffer>;
 
   /**
    * Copies and sorts the array.
@@ -651,11 +649,13 @@ interface Float32Array {
    * a negative value if the first argument is less than the second argument, zero if they're equal, and a positive
    * value otherwise. If omitted, the elements are sorted in ascending order.
    * ```ts
-   * const myNums = Float32Array.from([11.25, 2, -22.5, 1]);
-   * myNums.toSorted((a, b) => a - b) // Float32Array(4) [-22.5, 1, 2, 11.5]
+   * const myNums = Float32Array<Buffer>.from([11.25, 2, -22.5, 1]);
+   * myNums.toSorted((a, b) => a - b) // Float32Array<Buffer>(4) [-22.5, 1, 2, 11.5]
    * ```
    */
-  toSorted(compareFn?: (a: number, b: number) => number): Float32Array;
+  toSorted(
+    compareFn?: (a: number, b: number) => number,
+  ): Float32Array<ArrayBuffer>;
 
   /**
    * Copies the array and inserts the given number at the provided index.
@@ -664,10 +664,10 @@ interface Float32Array {
    * @param value The value to insert into the copied array.
    * @returns A copy of the original array with the inserted value.
    */
-  with(index: number, value: number): Float32Array;
+  with(index: number, value: number): Float32Array<ArrayBuffer>;
 }
 
-interface Float64Array {
+interface Float64Array<TArrayBuffer extends ArrayBufferLike> {
   /**
    * Returns the value of the last element in the array where predicate is true, and undefined
    * otherwise.
@@ -678,15 +678,11 @@ interface Float64Array {
    * predicate. If it is not provided, undefined is used instead.
    */
   findLast<S extends number>(
-    predicate: (
-      value: number,
-      index: number,
-      array: Float64Array,
-    ) => value is S,
+    predicate: (value: number, index: number, array: this) => value is S,
     thisArg?: any,
   ): S | undefined;
   findLast(
-    predicate: (value: number, index: number, array: Float64Array) => unknown,
+    predicate: (value: number, index: number, array: this) => unknown,
     thisArg?: any,
   ): number | undefined;
 
@@ -700,14 +696,14 @@ interface Float64Array {
    * predicate. If it is not provided, undefined is used instead.
    */
   findLastIndex(
-    predicate: (value: number, index: number, array: Float64Array) => unknown,
+    predicate: (value: number, index: number, array: this) => unknown,
     thisArg?: any,
   ): number;
 
   /**
    * Copies the array and returns the copy with the elements in reverse order.
    */
-  toReversed(): Float64Array;
+  toReversed(): Float64Array<ArrayBuffer>;
 
   /**
    * Copies and sorts the array.
@@ -715,11 +711,13 @@ interface Float64Array {
    * a negative value if the first argument is less than the second argument, zero if they're equal, and a positive
    * value otherwise. If omitted, the elements are sorted in ascending order.
    * ```ts
-   * const myNums = Float64Array.from([11.25, 2, -22.5, 1]);
-   * myNums.toSorted((a, b) => a - b) // Float64Array(4) [-22.5, 1, 2, 11.5]
+   * const myNums = Float64Array<Buffer>.from([11.25, 2, -22.5, 1]);
+   * myNums.toSorted((a, b) => a - b) // Float64Array<Buffer>(4) [-22.5, 1, 2, 11.5]
    * ```
    */
-  toSorted(compareFn?: (a: number, b: number) => number): Float64Array;
+  toSorted(
+    compareFn?: (a: number, b: number) => number,
+  ): Float64Array<ArrayBuffer>;
 
   /**
    * Copies the array and inserts the given number at the provided index.
@@ -728,10 +726,10 @@ interface Float64Array {
    * @param value The value to insert into the copied array.
    * @returns A copy of the original array with the inserted value.
    */
-  with(index: number, value: number): Float64Array;
+  with(index: number, value: number): Float64Array<ArrayBuffer>;
 }
 
-interface BigInt64Array {
+interface BigInt64Array<TArrayBuffer extends ArrayBufferLike> {
   /**
    * Returns the value of the last element in the array where predicate is true, and undefined
    * otherwise.
@@ -742,15 +740,11 @@ interface BigInt64Array {
    * predicate. If it is not provided, undefined is used instead.
    */
   findLast<S extends bigint>(
-    predicate: (
-      value: bigint,
-      index: number,
-      array: BigInt64Array,
-    ) => value is S,
+    predicate: (value: bigint, index: number, array: this) => value is S,
     thisArg?: any,
   ): S | undefined;
   findLast(
-    predicate: (value: bigint, index: number, array: BigInt64Array) => unknown,
+    predicate: (value: bigint, index: number, array: this) => unknown,
     thisArg?: any,
   ): bigint | undefined;
 
@@ -764,14 +758,14 @@ interface BigInt64Array {
    * predicate. If it is not provided, undefined is used instead.
    */
   findLastIndex(
-    predicate: (value: bigint, index: number, array: BigInt64Array) => unknown,
+    predicate: (value: bigint, index: number, array: this) => unknown,
     thisArg?: any,
   ): number;
 
   /**
    * Copies the array and returns the copy with the elements in reverse order.
    */
-  toReversed(): BigInt64Array;
+  toReversed(): BigInt64Array<ArrayBuffer>;
 
   /**
    * Copies and sorts the array.
@@ -779,11 +773,13 @@ interface BigInt64Array {
    * a negative value if the first argument is less than the second argument, zero if they're equal, and a positive
    * value otherwise. If omitted, the elements are sorted in ascending order.
    * ```ts
-   * const myNums = BigInt64Array.from([11n, 2n, -22n, 1n]);
-   * myNums.toSorted((a, b) => Number(a - b)) // BigInt64Array(4) [-22n, 1n, 2n, 11n]
+   * const myNums = BigInt64Array<Buffer>.from([11n, 2n, -22n, 1n]);
+   * myNums.toSorted((a, b) => Number(a - b)) // BigInt64Array<Buffer>(4) [-22n, 1n, 2n, 11n]
    * ```
    */
-  toSorted(compareFn?: (a: bigint, b: bigint) => number): BigInt64Array;
+  toSorted(
+    compareFn?: (a: bigint, b: bigint) => number,
+  ): BigInt64Array<ArrayBuffer>;
 
   /**
    * Copies the array and inserts the given bigint at the provided index.
@@ -792,10 +788,10 @@ interface BigInt64Array {
    * @param value The value to insert into the copied array.
    * @returns A copy of the original array with the inserted value.
    */
-  with(index: number, value: bigint): BigInt64Array;
+  with(index: number, value: bigint): BigInt64Array<ArrayBuffer>;
 }
 
-interface BigUint64Array {
+interface BigUint64Array<TArrayBuffer extends ArrayBufferLike> {
   /**
    * Returns the value of the last element in the array where predicate is true, and undefined
    * otherwise.
@@ -806,15 +802,11 @@ interface BigUint64Array {
    * predicate. If it is not provided, undefined is used instead.
    */
   findLast<S extends bigint>(
-    predicate: (
-      value: bigint,
-      index: number,
-      array: BigUint64Array,
-    ) => value is S,
+    predicate: (value: bigint, index: number, array: this) => value is S,
     thisArg?: any,
   ): S | undefined;
   findLast(
-    predicate: (value: bigint, index: number, array: BigUint64Array) => unknown,
+    predicate: (value: bigint, index: number, array: this) => unknown,
     thisArg?: any,
   ): bigint | undefined;
 
@@ -828,14 +820,14 @@ interface BigUint64Array {
    * predicate. If it is not provided, undefined is used instead.
    */
   findLastIndex(
-    predicate: (value: bigint, index: number, array: BigUint64Array) => unknown,
+    predicate: (value: bigint, index: number, array: this) => unknown,
     thisArg?: any,
   ): number;
 
   /**
    * Copies the array and returns the copy with the elements in reverse order.
    */
-  toReversed(): BigUint64Array;
+  toReversed(): BigUint64Array<ArrayBuffer>;
 
   /**
    * Copies and sorts the array.
@@ -843,11 +835,13 @@ interface BigUint64Array {
    * a negative value if the first argument is less than the second argument, zero if they're equal, and a positive
    * value otherwise. If omitted, the elements are sorted in ascending order.
    * ```ts
-   * const myNums = BigUint64Array.from([11n, 2n, 22n, 1n]);
-   * myNums.toSorted((a, b) => Number(a - b)) // BigUint64Array(4) [1n, 2n, 11n, 22n]
+   * const myNums = BigUint64Array<Buffer>.from([11n, 2n, 22n, 1n]);
+   * myNums.toSorted((a, b) => Number(a - b)) // BigUint64Array<Buffer>(4) [1n, 2n, 11n, 22n]
    * ```
    */
-  toSorted(compareFn?: (a: bigint, b: bigint) => number): BigUint64Array;
+  toSorted(
+    compareFn?: (a: bigint, b: bigint) => number,
+  ): BigUint64Array<ArrayBuffer>;
 
   /**
    * Copies the array and inserts the given bigint at the provided index.
@@ -856,5 +850,5 @@ interface BigUint64Array {
    * @param value The value to insert into the copied array.
    * @returns A copy of the original array with the inserted value.
    */
-  with(index: number, value: bigint): BigUint64Array;
+  with(index: number, value: bigint): BigUint64Array<ArrayBuffer>;
 }
