diff --git a/packages/v5.6/temp/copied/lib.webworker.d.ts b/packages/v5.7/temp/copied/lib.webworker.d.ts
index 5d33ae5..676d1ca 100644
--- a/packages/v5.6/temp/copied/lib.webworker.d.ts
+++ b/packages/v5.7/temp/copied/lib.webworker.d.ts
@@ -64,6 +64,59 @@ interface AudioConfiguration {
   spatialRendering?: boolean;
 }
 
+interface AudioDataCopyToOptions {
+  format?: AudioSampleFormat;
+  frameCount?: number;
+  frameOffset?: number;
+  planeIndex: number;
+}
+
+interface AudioDataInit {
+  data: BufferSource;
+  format: AudioSampleFormat;
+  numberOfChannels: number;
+  numberOfFrames: number;
+  sampleRate: number;
+  timestamp: number;
+  transfer?: ArrayBuffer[];
+}
+
+interface AudioDecoderConfig {
+  codec: string;
+  description?: BufferSource;
+  numberOfChannels: number;
+  sampleRate: number;
+}
+
+interface AudioDecoderInit {
+  error: WebCodecsErrorCallback;
+  output: AudioDataOutputCallback;
+}
+
+interface AudioDecoderSupport {
+  config?: AudioDecoderConfig;
+  supported?: boolean;
+}
+
+interface AudioEncoderConfig {
+  bitrate?: number;
+  bitrateMode?: BitrateMode;
+  codec: string;
+  numberOfChannels: number;
+  opus?: OpusEncoderConfig;
+  sampleRate: number;
+}
+
+interface AudioEncoderInit {
+  error: WebCodecsErrorCallback;
+  output: EncodedAudioChunkOutputCallback;
+}
+
+interface AudioEncoderSupport {
+  config?: AudioEncoderConfig;
+  supported?: boolean;
+}
+
 interface AvcEncoderConfig {
   format?: AvcBitstreamFormat;
 }
@@ -180,6 +233,18 @@ interface EcdsaParams extends Algorithm {
   hash: HashAlgorithmIdentifier;
 }
 
+interface EncodedAudioChunkInit {
+  data: AllowSharedBufferSource;
+  duration?: number;
+  timestamp: number;
+  transfer?: ArrayBuffer[];
+  type: EncodedAudioChunkType;
+}
+
+interface EncodedAudioChunkMetadata {
+  decoderConfig?: AudioDecoderConfig;
+}
+
 interface EncodedVideoChunkInit {
   data: AllowSharedBufferSource;
   duration?: number;
@@ -446,6 +511,15 @@ interface NotificationOptions {
   tag?: string;
 }
 
+interface OpusEncoderConfig {
+  complexity?: number;
+  format?: OpusBitstreamFormat;
+  frameDuration?: number;
+  packetlossperc?: number;
+  usedtx?: boolean;
+  useinbandfec?: boolean;
+}
+
 interface Pbkdf2Params extends Algorithm {
   hash: HashAlgorithmIdentifier;
   iterations: number;
@@ -768,6 +842,7 @@ interface VideoConfiguration {
   colorGamut?: ColorGamut;
   contentType: string;
   framerate: number;
+  hasAlphaChannel?: boolean;
   hdrMetadataType?: HdrMetadataType;
   height: number;
   scalabilityMode?: string;
@@ -803,6 +878,7 @@ interface VideoEncoderConfig {
   bitrate?: number;
   bitrateMode?: VideoEncoderBitrateMode;
   codec: string;
+  contentHint?: string;
   displayHeight?: number;
   displayWidth?: number;
   framerate?: number;
@@ -814,9 +890,14 @@ interface VideoEncoderConfig {
 }
 
 interface VideoEncoderEncodeOptions {
+  avc?: VideoEncoderEncodeOptionsForAvc;
   keyFrame?: boolean;
 }
 
+interface VideoEncoderEncodeOptionsForAvc {
+  quantizer?: number | null;
+}
+
 interface VideoEncoderInit {
   error: WebCodecsErrorCallback;
   output: EncodedVideoChunkOutputCallback;
@@ -841,6 +922,8 @@ interface VideoFrameBufferInit {
 }
 
 interface VideoFrameCopyToOptions {
+  colorSpace?: PredefinedColorSpace;
+  format?: VideoPixelFormat;
   layout?: PlaneLayout[];
   rect?: DOMRectInit;
 }
@@ -1051,6 +1134,148 @@ interface AnimationFrameProvider {
   requestAnimationFrame(callback: FrameRequestCallback): number;
 }
 
+/** [MDN Reference](https://developer.mozilla.org/docs/Web/API/AudioData) */
+interface AudioData {
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/AudioData/duration) */
+  readonly duration: number;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/AudioData/format) */
+  readonly format: AudioSampleFormat | null;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/AudioData/numberOfChannels) */
+  readonly numberOfChannels: number;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/AudioData/numberOfFrames) */
+  readonly numberOfFrames: number;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/AudioData/sampleRate) */
+  readonly sampleRate: number;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/AudioData/timestamp) */
+  readonly timestamp: number;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/AudioData/allocationSize) */
+  allocationSize(options: AudioDataCopyToOptions): number;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/AudioData/clone) */
+  clone(): AudioData;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/AudioData/close) */
+  close(): void;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/AudioData/copyTo) */
+  copyTo(
+    destination: AllowSharedBufferSource,
+    options: AudioDataCopyToOptions,
+  ): void;
+}
+
+declare var AudioData: {
+  prototype: AudioData;
+  new (init: AudioDataInit): AudioData;
+};
+
+interface AudioDecoderEventMap {
+  dequeue: Event;
+}
+
+/**
+ * Available only in secure contexts.
+ *
+ * [MDN Reference](https://developer.mozilla.org/docs/Web/API/AudioDecoder)
+ */
+interface AudioDecoder extends EventTarget {
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/AudioDecoder/decodeQueueSize) */
+  readonly decodeQueueSize: number;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/AudioDecoder/dequeue_event) */
+  ondequeue: ((this: AudioDecoder, ev: Event) => any) | null;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/AudioDecoder/state) */
+  readonly state: CodecState;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/AudioDecoder/close) */
+  close(): void;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/AudioDecoder/configure) */
+  configure(config: AudioDecoderConfig): void;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/AudioDecoder/decode) */
+  decode(chunk: EncodedAudioChunk): void;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/AudioDecoder/flush) */
+  flush(): Promise<void>;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/AudioDecoder/reset) */
+  reset(): void;
+  addEventListener<K extends keyof AudioDecoderEventMap>(
+    type: K,
+    listener: (this: AudioDecoder, ev: AudioDecoderEventMap[K]) => any,
+    options?: boolean | AddEventListenerOptions,
+  ): void;
+  addEventListener(
+    type: string,
+    listener: EventListenerOrEventListenerObject,
+    options?: boolean | AddEventListenerOptions,
+  ): void;
+  removeEventListener<K extends keyof AudioDecoderEventMap>(
+    type: K,
+    listener: (this: AudioDecoder, ev: AudioDecoderEventMap[K]) => any,
+    options?: boolean | EventListenerOptions,
+  ): void;
+  removeEventListener(
+    type: string,
+    listener: EventListenerOrEventListenerObject,
+    options?: boolean | EventListenerOptions,
+  ): void;
+}
+
+declare var AudioDecoder: {
+  prototype: AudioDecoder;
+  new (init: AudioDecoderInit): AudioDecoder;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/AudioDecoder/isConfigSupported_static) */
+  isConfigSupported(config: AudioDecoderConfig): Promise<AudioDecoderSupport>;
+};
+
+interface AudioEncoderEventMap {
+  dequeue: Event;
+}
+
+/**
+ * Available only in secure contexts.
+ *
+ * [MDN Reference](https://developer.mozilla.org/docs/Web/API/AudioEncoder)
+ */
+interface AudioEncoder extends EventTarget {
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/AudioEncoder/encodeQueueSize) */
+  readonly encodeQueueSize: number;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/AudioEncoder/dequeue_event) */
+  ondequeue: ((this: AudioEncoder, ev: Event) => any) | null;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/AudioEncoder/state) */
+  readonly state: CodecState;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/AudioEncoder/close) */
+  close(): void;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/AudioEncoder/configure) */
+  configure(config: AudioEncoderConfig): void;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/AudioEncoder/encode) */
+  encode(data: AudioData): void;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/AudioEncoder/flush) */
+  flush(): Promise<void>;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/AudioEncoder/reset) */
+  reset(): void;
+  addEventListener<K extends keyof AudioEncoderEventMap>(
+    type: K,
+    listener: (this: AudioEncoder, ev: AudioEncoderEventMap[K]) => any,
+    options?: boolean | AddEventListenerOptions,
+  ): void;
+  addEventListener(
+    type: string,
+    listener: EventListenerOrEventListenerObject,
+    options?: boolean | AddEventListenerOptions,
+  ): void;
+  removeEventListener<K extends keyof AudioEncoderEventMap>(
+    type: K,
+    listener: (this: AudioEncoder, ev: AudioEncoderEventMap[K]) => any,
+    options?: boolean | EventListenerOptions,
+  ): void;
+  removeEventListener(
+    type: string,
+    listener: EventListenerOrEventListenerObject,
+    options?: boolean | EventListenerOptions,
+  ): void;
+}
+
+declare var AudioEncoder: {
+  prototype: AudioEncoder;
+  new (init: AudioEncoderInit): AudioEncoder;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/AudioEncoder/isConfigSupported_static) */
+  isConfigSupported(config: AudioEncoderConfig): Promise<AudioEncoderSupport>;
+};
+
 /**
  * A file-like object of immutable, raw data. Blobs represent data that isn't necessarily in a JavaScript-native format. The File interface is based on Blob, inheriting blob functionality and expanding it to support files on the user's system.
  *
@@ -1063,6 +1288,8 @@ interface Blob {
   readonly type: string;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/Blob/arrayBuffer) */
   arrayBuffer(): Promise<ArrayBuffer>;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/Blob/bytes) */
+  bytes(): Promise<Uint8Array>;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/Blob/slice) */
   slice(start?: number, end?: number, contentType?: string): Blob;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/Blob/stream) */
@@ -1085,6 +1312,8 @@ interface Body {
   arrayBuffer(): Promise<ArrayBuffer>;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/Request/blob) */
   blob(): Promise<Blob>;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/Request/bytes) */
+  bytes(): Promise<Uint8Array>;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/Request/formData) */
   formData(): Promise<FormData>;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/Request/json) */
@@ -2016,7 +2245,10 @@ declare var CloseEvent: {
 };
 
 /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/CompressionStream) */
-interface CompressionStream extends GenericTransformStream {}
+interface CompressionStream extends GenericTransformStream {
+  readonly readable: ReadableStream<Uint8Array>;
+  readonly writable: WritableStream<BufferSource>;
+}
 
 declare var CompressionStream: {
   prototype: CompressionStream;
@@ -2190,27 +2422,49 @@ declare var DOMException: {
 
 /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMMatrix) */
 interface DOMMatrix extends DOMMatrixReadOnly {
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMMatrix#instance_properties) */
   a: number;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMMatrix#instance_properties) */
   b: number;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMMatrix#instance_properties) */
   c: number;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMMatrix#instance_properties) */
   d: number;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMMatrix#instance_properties) */
   e: number;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMMatrix#instance_properties) */
   f: number;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMMatrix#instance_properties) */
   m11: number;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMMatrix#instance_properties) */
   m12: number;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMMatrix#instance_properties) */
   m13: number;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMMatrix#instance_properties) */
   m14: number;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMMatrix#instance_properties) */
   m21: number;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMMatrix#instance_properties) */
   m22: number;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMMatrix#instance_properties) */
   m23: number;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMMatrix#instance_properties) */
   m24: number;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMMatrix#instance_properties) */
   m31: number;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMMatrix#instance_properties) */
   m32: number;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMMatrix#instance_properties) */
   m33: number;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMMatrix#instance_properties) */
   m34: number;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMMatrix#instance_properties) */
   m41: number;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMMatrix#instance_properties) */
   m42: number;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMMatrix#instance_properties) */
   m43: number;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMMatrix#instance_properties) */
   m44: number;
   invertSelf(): DOMMatrix;
   multiplySelf(other?: DOMMatrixInit): DOMMatrix;
@@ -2252,29 +2506,51 @@ declare var DOMMatrix: {
 
 /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMMatrixReadOnly) */
 interface DOMMatrixReadOnly {
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMMatrixReadOnly#instance_properties) */
   readonly a: number;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMMatrixReadOnly#instance_properties) */
   readonly b: number;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMMatrixReadOnly#instance_properties) */
   readonly c: number;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMMatrixReadOnly#instance_properties) */
   readonly d: number;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMMatrixReadOnly#instance_properties) */
   readonly e: number;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMMatrixReadOnly#instance_properties) */
   readonly f: number;
   readonly is2D: boolean;
   readonly isIdentity: boolean;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMMatrixReadOnly#instance_properties) */
   readonly m11: number;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMMatrixReadOnly#instance_properties) */
   readonly m12: number;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMMatrixReadOnly#instance_properties) */
   readonly m13: number;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMMatrixReadOnly#instance_properties) */
   readonly m14: number;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMMatrixReadOnly#instance_properties) */
   readonly m21: number;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMMatrixReadOnly#instance_properties) */
   readonly m22: number;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMMatrixReadOnly#instance_properties) */
   readonly m23: number;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMMatrixReadOnly#instance_properties) */
   readonly m24: number;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMMatrixReadOnly#instance_properties) */
   readonly m31: number;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMMatrixReadOnly#instance_properties) */
   readonly m32: number;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMMatrixReadOnly#instance_properties) */
   readonly m33: number;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMMatrixReadOnly#instance_properties) */
   readonly m34: number;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMMatrixReadOnly#instance_properties) */
   readonly m41: number;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMMatrixReadOnly#instance_properties) */
   readonly m42: number;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMMatrixReadOnly#instance_properties) */
   readonly m43: number;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMMatrixReadOnly#instance_properties) */
   readonly m44: number;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMMatrixReadOnly/flipX) */
   flipX(): DOMMatrix;
@@ -2468,7 +2744,10 @@ declare var DOMStringList: {
 };
 
 /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DecompressionStream) */
-interface DecompressionStream extends GenericTransformStream {}
+interface DecompressionStream extends GenericTransformStream {
+  readonly readable: ReadableStream<Uint8Array>;
+  readonly writable: WritableStream<BufferSource>;
+}
 
 declare var DecompressionStream: {
   prototype: DecompressionStream;
@@ -2478,7 +2757,7 @@ declare var DecompressionStream: {
 interface DedicatedWorkerGlobalScopeEventMap extends WorkerGlobalScopeEventMap {
   message: MessageEvent;
   messageerror: MessageEvent;
-  rtctransform: Event;
+  rtctransform: RTCTransformEvent;
 }
 
 /**
@@ -2503,7 +2782,9 @@ interface DedicatedWorkerGlobalScope
     | ((this: DedicatedWorkerGlobalScope, ev: MessageEvent) => any)
     | null;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DedicatedWorkerGlobalScope/rtctransform_event) */
-  onrtctransform: ((this: DedicatedWorkerGlobalScope, ev: Event) => any) | null;
+  onrtctransform:
+    | ((this: DedicatedWorkerGlobalScope, ev: RTCTransformEvent) => any)
+    | null;
   /**
    * Aborts dedicatedWorkerGlobal.
    *
@@ -2626,6 +2907,25 @@ interface EXT_texture_norm16 {
   readonly RGBA16_SNORM_EXT: 0x8f9b;
 }
 
+/** [MDN Reference](https://developer.mozilla.org/docs/Web/API/EncodedAudioChunk) */
+interface EncodedAudioChunk {
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/EncodedAudioChunk/byteLength) */
+  readonly byteLength: number;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/EncodedAudioChunk/duration) */
+  readonly duration: number | null;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/EncodedAudioChunk/timestamp) */
+  readonly timestamp: number;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/EncodedAudioChunk/type) */
+  readonly type: EncodedAudioChunkType;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/EncodedAudioChunk/copyTo) */
+  copyTo(destination: AllowSharedBufferSource): void;
+}
+
+declare var EncodedAudioChunk: {
+  prototype: EncodedAudioChunk;
+  new (init: EncodedAudioChunkInit): EncodedAudioChunk;
+};
+
 /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/EncodedVideoChunk) */
 interface EncodedVideoChunk {
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/EncodedVideoChunk/byteLength) */
@@ -3298,25 +3598,25 @@ declare var FontFace: {
   prototype: FontFace;
   new (
     family: string,
-    source: string | BinaryData,
+    source: string | BufferSource,
     descriptors?: FontFaceDescriptors,
   ): FontFace;
 };
 
 interface FontFaceSetEventMap {
-  loading: Event;
-  loadingdone: Event;
-  loadingerror: Event;
+  loading: FontFaceSetLoadEvent;
+  loadingdone: FontFaceSetLoadEvent;
+  loadingerror: FontFaceSetLoadEvent;
 }
 
 /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/FontFaceSet) */
 interface FontFaceSet extends EventTarget {
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/FontFaceSet/loading_event) */
-  onloading: ((this: FontFaceSet, ev: Event) => any) | null;
+  onloading: ((this: FontFaceSet, ev: FontFaceSetLoadEvent) => any) | null;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/FontFaceSet/loadingdone_event) */
-  onloadingdone: ((this: FontFaceSet, ev: Event) => any) | null;
+  onloadingdone: ((this: FontFaceSet, ev: FontFaceSetLoadEvent) => any) | null;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/FontFaceSet/loadingerror_event) */
-  onloadingerror: ((this: FontFaceSet, ev: Event) => any) | null;
+  onloadingerror: ((this: FontFaceSet, ev: FontFaceSetLoadEvent) => any) | null;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/FontFaceSet/ready) */
   readonly ready: Promise<FontFaceSet>;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/FontFaceSet/status) */
@@ -3353,7 +3653,7 @@ interface FontFaceSet extends EventTarget {
 
 declare var FontFaceSet: {
   prototype: FontFaceSet;
-  new (initialFaces: FontFace[]): FontFaceSet;
+  new (): FontFaceSet;
 };
 
 /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/FontFaceSetLoadEvent) */
@@ -4171,7 +4471,7 @@ interface IDBTransaction extends EventTarget {
   /**
    * Returns a list of the names of object stores in the transaction's scope. For an upgrade transaction this is all object stores in the database.
    *
-   * [MDN Reference](https://developer.mozilla.org/docs/Web/API/IDBTransaction/ObjectStoreNames)
+   * [MDN Reference](https://developer.mozilla.org/docs/Web/API/IDBTransaction/objectStoreNames)
    */
   readonly objectStoreNames: DOMStringList;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/IDBTransaction/abort_event) */
@@ -4819,7 +5119,7 @@ interface OES_vertex_array_object {
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/OES_vertex_array_object/bindVertexArrayOES) */
   bindVertexArrayOES(arrayObject: WebGLVertexArrayObjectOES | null): void;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/OES_vertex_array_object/createVertexArrayOES) */
-  createVertexArrayOES(): WebGLVertexArrayObjectOES | null;
+  createVertexArrayOES(): WebGLVertexArrayObjectOES;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/OES_vertex_array_object/deleteVertexArrayOES) */
   deleteVertexArrayOES(arrayObject: WebGLVertexArrayObjectOES | null): void;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/OES_vertex_array_object/isVertexArrayOES) */
@@ -4953,6 +5253,7 @@ interface OffscreenCanvasRenderingContext2D
     CanvasText,
     CanvasTextDrawingStyles,
     CanvasTransform {
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/CanvasRenderingContext2D/canvas) */
   readonly canvas: OffscreenCanvas;
 }
 
@@ -5166,6 +5467,8 @@ interface PerformanceResourceTiming extends PerformanceEntry {
   readonly responseEnd: DOMHighResTimeStamp;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/PerformanceResourceTiming/responseStart) */
   readonly responseStart: DOMHighResTimeStamp;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/PerformanceResourceTiming/responseStatus) */
+  readonly responseStatus: number;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/PerformanceResourceTiming/secureConnectionStart) */
   readonly secureConnectionStart: DOMHighResTimeStamp;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/PerformanceResourceTiming/serverTiming) */
@@ -5337,6 +5640,8 @@ interface PushMessageData {
   arrayBuffer(): ArrayBuffer;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/PushMessageData/blob) */
   blob(): Blob;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/PushMessageData/bytes) */
+  bytes(): Uint8Array;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/PushMessageData/json) */
   json(): any;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/PushMessageData/text) */
@@ -5528,7 +5833,7 @@ interface ReadableStreamBYOBReader extends ReadableStreamGenericReader {
 
 declare var ReadableStreamBYOBReader: {
   prototype: ReadableStreamBYOBReader;
-  new (stream: ReadableStream): ReadableStreamBYOBReader;
+  new (stream: ReadableStream<Uint8Array>): ReadableStreamBYOBReader;
 };
 
 /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/ReadableStreamBYOBRequest) */
@@ -6195,7 +6500,7 @@ interface SubtleCrypto {
       | HkdfParams
       | Pbkdf2Params,
     baseKey: CryptoKey,
-    length: number,
+    length?: number | null,
   ): Promise<ArrayBuffer>;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/SubtleCrypto/deriveKey) */
   deriveKey(
@@ -7763,13 +8068,13 @@ interface WebGL2RenderingContextBase {
     height: GLsizei,
   ): void;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGL2RenderingContext/createQuery) */
-  createQuery(): WebGLQuery | null;
+  createQuery(): WebGLQuery;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGL2RenderingContext/createSampler) */
-  createSampler(): WebGLSampler | null;
+  createSampler(): WebGLSampler;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGL2RenderingContext/createTransformFeedback) */
-  createTransformFeedback(): WebGLTransformFeedback | null;
+  createTransformFeedback(): WebGLTransformFeedback;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGL2RenderingContext/createVertexArray) */
-  createVertexArray(): WebGLVertexArrayObject | null;
+  createVertexArray(): WebGLVertexArrayObject;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGL2RenderingContext/deleteQuery) */
   deleteQuery(query: WebGLQuery | null): void;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGL2RenderingContext/deleteSampler) */
@@ -9208,17 +9513,17 @@ interface WebGLRenderingContextBase {
     height: GLsizei,
   ): void;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGLRenderingContext/createBuffer) */
-  createBuffer(): WebGLBuffer | null;
+  createBuffer(): WebGLBuffer;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGLRenderingContext/createFramebuffer) */
-  createFramebuffer(): WebGLFramebuffer | null;
+  createFramebuffer(): WebGLFramebuffer;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGLRenderingContext/createProgram) */
-  createProgram(): WebGLProgram | null;
+  createProgram(): WebGLProgram;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGLRenderingContext/createRenderbuffer) */
-  createRenderbuffer(): WebGLRenderbuffer | null;
+  createRenderbuffer(): WebGLRenderbuffer;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGLRenderingContext/createShader) */
   createShader(type: GLenum): WebGLShader | null;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGLRenderingContext/createTexture) */
-  createTexture(): WebGLTexture | null;
+  createTexture(): WebGLTexture;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGLRenderingContext/cullFace) */
   cullFace(mode: GLenum): void;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGLRenderingContext/deleteBuffer) */
@@ -10341,7 +10646,7 @@ interface WindowOrWorkerGlobalScope {
     sh: number,
     options?: ImageBitmapOptions,
   ): Promise<ImageBitmap>;
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/fetch) */
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/Window/fetch) */
   fetch(input: RequestInfo | URL, init?: RequestInit): Promise<Response>;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/queueMicrotask) */
   queueMicrotask(callback: VoidFunction): void;
@@ -10893,7 +11198,7 @@ interface Console {
   clear(): void;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/console/count_static) */
   count(label?: string): void;
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/console/countreset_static) */
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/console/countReset_static) */
   countReset(label?: string): void;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/console/debug_static) */
   debug(...data: any[]): void;
@@ -10905,9 +11210,9 @@ interface Console {
   error(...data: any[]): void;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/console/group_static) */
   group(...data: any[]): void;
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/console/groupcollapsed_static) */
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/console/groupCollapsed_static) */
   groupCollapsed(...data: any[]): void;
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/console/groupend_static) */
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/console/groupEnd_static) */
   groupEnd(): void;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/console/info_static) */
   info(...data: any[]): void;
@@ -10917,9 +11222,9 @@ interface Console {
   table(tabularData?: any, properties?: string[]): void;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/console/time_static) */
   time(label?: string): void;
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/console/timeend_static) */
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/console/timeEnd_static) */
   timeEnd(label?: string): void;
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/console/timelog_static) */
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/console/timeLog_static) */
   timeLog(label?: string, ...data: any[]): void;
   timeStamp(label?: string): void;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/console/trace_static) */
@@ -11099,6 +11404,14 @@ declare namespace WebAssembly {
   function validate(bytes: BufferSource): boolean;
 }
 
+interface AudioDataOutputCallback {
+  (output: AudioData): void;
+}
+
+interface EncodedAudioChunkOutputCallback {
+  (output: EncodedAudioChunk, metadata?: EncodedAudioChunkMetadata): void;
+}
+
 interface EncodedVideoChunkOutputCallback {
   (chunk: EncodedVideoChunk, metadata?: EncodedVideoChunkMetadata): void;
 }
@@ -11207,7 +11520,7 @@ declare var onmessageerror:
   | null;
 /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DedicatedWorkerGlobalScope/rtctransform_event) */
 declare var onrtctransform:
-  | ((this: DedicatedWorkerGlobalScope, ev: Event) => any)
+  | ((this: DedicatedWorkerGlobalScope, ev: RTCTransformEvent) => any)
   | null;
 /**
  * Aborts dedicatedWorkerGlobal.
@@ -11326,7 +11639,7 @@ declare function createImageBitmap(
   sh: number,
   options?: ImageBitmapOptions,
 ): Promise<ImageBitmap>;
-/** [MDN Reference](https://developer.mozilla.org/docs/Web/API/fetch) */
+/** [MDN Reference](https://developer.mozilla.org/docs/Web/API/Window/fetch) */
 declare function fetch(
   input: RequestInfo | URL,
   init?: RequestInit,
@@ -11389,7 +11702,6 @@ declare function removeEventListener(
 type AlgorithmIdentifier = Algorithm | string;
 type AllowSharedBufferSource = ArrayBuffer | ArrayBufferView;
 type BigInteger = Uint8Array;
-type BinaryData = ArrayBuffer | ArrayBufferView;
 type BlobPart = BufferSource | Blob | string;
 type BodyInit = ReadableStream | XMLHttpRequestBodyInit;
 type BufferSource = ArrayBufferView | ArrayBuffer;
@@ -11452,6 +11764,7 @@ type Transferable =
   | ReadableStream
   | WritableStream
   | TransformStream
+  | AudioData
   | VideoFrame
   | ArrayBuffer;
 type Uint32List = Uint32Array | GLuint[];
@@ -11462,8 +11775,18 @@ type XMLHttpRequestBodyInit =
   | URLSearchParams
   | string;
 type AlphaOption = 'discard' | 'keep';
+type AudioSampleFormat =
+  | 'f32'
+  | 'f32-planar'
+  | 's16'
+  | 's16-planar'
+  | 's32'
+  | 's32-planar'
+  | 'u8'
+  | 'u8-planar';
 type AvcBitstreamFormat = 'annexb' | 'avc';
 type BinaryType = 'arraybuffer' | 'blob';
+type BitrateMode = 'constant' | 'variable';
 type CSSMathOperator =
   | 'clamp'
   | 'invert'
@@ -11522,6 +11845,7 @@ type ColorGamut = 'p3' | 'rec2020' | 'srgb';
 type ColorSpaceConversion = 'default' | 'none';
 type CompressionFormat = 'deflate' | 'deflate-raw' | 'gzip';
 type DocumentVisibilityState = 'hidden' | 'visible';
+type EncodedAudioChunkType = 'delta' | 'key';
 type EncodedVideoChunkType = 'delta' | 'key';
 type EndingType = 'native' | 'transparent';
 type FileSystemHandleKind = 'directory' | 'file';
@@ -11590,13 +11914,15 @@ type OffscreenRenderingContextId =
   | 'webgl'
   | 'webgl2'
   | 'webgpu';
+type OpusBitstreamFormat = 'ogg' | 'opus';
 type PermissionName =
   | 'geolocation'
+  | 'midi'
   | 'notifications'
   | 'persistent-storage'
   | 'push'
   | 'screen-wake-lock'
-  | 'xr-spatial-tracking';
+  | 'storage-access';
 type PermissionState = 'denied' | 'granted' | 'prompt';
 type PredefinedColorSpace = 'display-p3' | 'srgb';
 type PremultiplyAlpha = 'default' | 'none' | 'premultiply';
