diff --git a/packages/v5.6/temp/copied/lib.es2015.core.d.ts b/packages/v5.7/temp/copied/lib.es2015.core.d.ts
index ce8c9de..9d91ddd 100644
--- a/packages/v5.6/temp/copied/lib.es2015.core.d.ts
+++ b/packages/v5.7/temp/copied/lib.es2015.core.d.ts
@@ -595,63 +595,63 @@ interface StringConstructor {
   ): string;
 }
 
-interface Int8Array {
+interface Int8Array<TArrayBuffer extends ArrayBufferLike> {
   toLocaleString(
     locales: string | string[],
     options?: Intl.NumberFormatOptions,
   ): string;
 }
 
-interface Uint8Array {
+interface Uint8Array<TArrayBuffer extends ArrayBufferLike> {
   toLocaleString(
     locales: string | string[],
     options?: Intl.NumberFormatOptions,
   ): string;
 }
 
-interface Uint8ClampedArray {
+interface Uint8ClampedArray<TArrayBuffer extends ArrayBufferLike> {
   toLocaleString(
     locales: string | string[],
     options?: Intl.NumberFormatOptions,
   ): string;
 }
 
-interface Int16Array {
+interface Int16Array<TArrayBuffer extends ArrayBufferLike> {
   toLocaleString(
     locales: string | string[],
     options?: Intl.NumberFormatOptions,
   ): string;
 }
 
-interface Uint16Array {
+interface Uint16Array<TArrayBuffer extends ArrayBufferLike> {
   toLocaleString(
     locales: string | string[],
     options?: Intl.NumberFormatOptions,
   ): string;
 }
 
-interface Int32Array {
+interface Int32Array<TArrayBuffer extends ArrayBufferLike> {
   toLocaleString(
     locales: string | string[],
     options?: Intl.NumberFormatOptions,
   ): string;
 }
 
-interface Uint32Array {
+interface Uint32Array<TArrayBuffer extends ArrayBufferLike> {
   toLocaleString(
     locales: string | string[],
     options?: Intl.NumberFormatOptions,
   ): string;
 }
 
-interface Float32Array {
+interface Float32Array<TArrayBuffer extends ArrayBufferLike> {
   toLocaleString(
     locales: string | string[],
     options?: Intl.NumberFormatOptions,
   ): string;
 }
 
-interface Float64Array {
+interface Float64Array<TArrayBuffer extends ArrayBufferLike> {
   toLocaleString(
     locales: string | string[],
     options?: Intl.NumberFormatOptions,
