diff --git a/packages/v5.6/temp/copied/lib.es2017.typedarrays.d.ts b/packages/v5.7/temp/copied/lib.es2017.typedarrays.d.ts
index 28b7f46..4170584 100644
--- a/packages/v5.6/temp/copied/lib.es2017.typedarrays.d.ts
+++ b/packages/v5.7/temp/copied/lib.es2017.typedarrays.d.ts
@@ -16,37 +16,37 @@ and limitations under the License.
 /// <reference no-default-lib="true"/>
 
 interface Int8ArrayConstructor {
-  new (): Int8Array;
+  new (): Int8Array<ArrayBuffer>;
 }
 
 interface Uint8ArrayConstructor {
-  new (): Uint8Array;
+  new (): Uint8Array<ArrayBuffer>;
 }
 
 interface Uint8ClampedArrayConstructor {
-  new (): Uint8ClampedArray;
+  new (): Uint8ClampedArray<ArrayBuffer>;
 }
 
 interface Int16ArrayConstructor {
-  new (): Int16Array;
+  new (): Int16Array<ArrayBuffer>;
 }
 
 interface Uint16ArrayConstructor {
-  new (): Uint16Array;
+  new (): Uint16Array<ArrayBuffer>;
 }
 
 interface Int32ArrayConstructor {
-  new (): Int32Array;
+  new (): Int32Array<ArrayBuffer>;
 }
 
 interface Uint32ArrayConstructor {
-  new (): Uint32Array;
+  new (): Uint32Array<ArrayBuffer>;
 }
 
 interface Float32ArrayConstructor {
-  new (): Float32Array;
+  new (): Float32Array<ArrayBuffer>;
 }
 
 interface Float64ArrayConstructor {
-  new (): Float64Array;
+  new (): Float64Array<ArrayBuffer>;
 }
