diff --git a/packages/v5.6/temp/copied/lib.es5.d.ts b/packages/v5.7/temp/copied/lib.es5.d.ts
index dda327f..666b067 100644
--- a/packages/v5.6/temp/copied/lib.es5.d.ts
+++ b/packages/v5.7/temp/copied/lib.es5.d.ts
@@ -1947,7 +1947,7 @@ interface ArrayBuffer {
   /**
    * Returns a section of an ArrayBuffer.
    */
-  slice(begin: number, end?: number): ArrayBuffer;
+  slice(begin?: number, end?: number): ArrayBuffer;
 }
 
 /**
@@ -1965,25 +1965,27 @@ interface ArrayBufferConstructor {
 }
 declare var ArrayBuffer: ArrayBufferConstructor;
 
-interface ArrayBufferView {
+interface ArrayBufferView<
+  TArrayBuffer extends ArrayBufferLike = ArrayBufferLike,
+> {
   /**
    * The ArrayBuffer instance referenced by the array.
    */
-  buffer: ArrayBufferLike;
+  readonly buffer: TArrayBuffer;
 
   /**
    * The length in bytes of the array.
    */
-  byteLength: number;
+  readonly byteLength: number;
 
   /**
    * The offset in bytes of the array.
    */
-  byteOffset: number;
+  readonly byteOffset: number;
 }
 
-interface DataView {
-  readonly buffer: ArrayBuffer;
+interface DataView<TArrayBuffer extends ArrayBufferLike = ArrayBufferLike> {
+  readonly buffer: TArrayBuffer;
   readonly byteLength: number;
   readonly byteOffset: number;
   /**
@@ -2109,14 +2111,13 @@ interface DataView {
    */
   setUint32(byteOffset: number, value: number, littleEndian?: boolean): void;
 }
-
 interface DataViewConstructor {
-  readonly prototype: DataView;
-  new (
-    buffer: ArrayBufferLike & { BYTES_PER_ELEMENT?: never },
+  readonly prototype: DataView<ArrayBufferLike>;
+  new <TArrayBuffer extends ArrayBufferLike & { BYTES_PER_ELEMENT?: never }>(
+    buffer: TArrayBuffer,
     byteOffset?: number,
     byteLength?: number,
-  ): DataView;
+  ): DataView<TArrayBuffer>;
 }
 declare var DataView: DataViewConstructor;
 
@@ -2124,7 +2125,7 @@ declare var DataView: DataViewConstructor;
  * A typed array of 8-bit integer values. The contents are initialized to 0. If the requested
  * number of bytes could not be allocated an exception is raised.
  */
-interface Int8Array {
+interface Int8Array<TArrayBuffer extends ArrayBufferLike = ArrayBufferLike> {
   /**
    * The size in bytes of each element in the array.
    */
@@ -2133,7 +2134,7 @@ interface Int8Array {
   /**
    * The ArrayBuffer instance referenced by the array.
    */
-  readonly buffer: ArrayBufferLike;
+  readonly buffer: TArrayBuffer;
 
   /**
    * The length in bytes of the array.
@@ -2165,7 +2166,7 @@ interface Int8Array {
    * If thisArg is omitted, undefined is used as the this value.
    */
   every(
-    predicate: (value: number, index: number, array: Int8Array) => unknown,
+    predicate: (value: number, index: number, array: this) => unknown,
     thisArg?: any,
   ): boolean;
 
@@ -2187,9 +2188,9 @@ interface Int8Array {
    * If thisArg is omitted, undefined is used as the this value.
    */
   filter(
-    predicate: (value: number, index: number, array: Int8Array) => any,
+    predicate: (value: number, index: number, array: this) => any,
     thisArg?: any,
-  ): Int8Array;
+  ): Int8Array<ArrayBuffer>;
 
   /**
    * Returns the value of the first element in the array where predicate is true, and undefined
@@ -2201,7 +2202,7 @@ interface Int8Array {
    * predicate. If it is not provided, undefined is used instead.
    */
   find(
-    predicate: (value: number, index: number, obj: Int8Array) => boolean,
+    predicate: (value: number, index: number, obj: this) => boolean,
     thisArg?: any,
   ): number | undefined;
 
@@ -2215,7 +2216,7 @@ interface Int8Array {
    * predicate. If it is not provided, undefined is used instead.
    */
   findIndex(
-    predicate: (value: number, index: number, obj: Int8Array) => boolean,
+    predicate: (value: number, index: number, obj: this) => boolean,
     thisArg?: any,
   ): number;
 
@@ -2227,7 +2228,7 @@ interface Int8Array {
    * If thisArg is omitted, undefined is used as the this value.
    */
   forEach(
-    callbackfn: (value: number, index: number, array: Int8Array) => void,
+    callbackfn: (value: number, index: number, array: this) => void,
     thisArg?: any,
   ): void;
 
@@ -2268,9 +2269,9 @@ interface Int8Array {
    * If thisArg is omitted, undefined is used as the this value.
    */
   map(
-    callbackfn: (value: number, index: number, array: Int8Array) => number,
+    callbackfn: (value: number, index: number, array: this) => number,
     thisArg?: any,
-  ): Int8Array;
+  ): Int8Array<ArrayBuffer>;
 
   /**
    * Calls the specified callback function for all the elements in an array. The return value of
@@ -2287,7 +2288,7 @@ interface Int8Array {
       previousValue: number,
       currentValue: number,
       currentIndex: number,
-      array: Int8Array,
+      array: this,
     ) => number,
   ): number;
   reduce(
@@ -2295,7 +2296,7 @@ interface Int8Array {
       previousValue: number,
       currentValue: number,
       currentIndex: number,
-      array: Int8Array,
+      array: this,
     ) => number,
     initialValue: number,
   ): number;
@@ -2315,7 +2316,7 @@ interface Int8Array {
       previousValue: U,
       currentValue: number,
       currentIndex: number,
-      array: Int8Array,
+      array: this,
     ) => U,
     initialValue: U,
   ): U;
@@ -2335,7 +2336,7 @@ interface Int8Array {
       previousValue: number,
       currentValue: number,
       currentIndex: number,
-      array: Int8Array,
+      array: this,
     ) => number,
   ): number;
   reduceRight(
@@ -2343,7 +2344,7 @@ interface Int8Array {
       previousValue: number,
       currentValue: number,
       currentIndex: number,
-      array: Int8Array,
+      array: this,
     ) => number,
     initialValue: number,
   ): number;
@@ -2363,7 +2364,7 @@ interface Int8Array {
       previousValue: U,
       currentValue: number,
       currentIndex: number,
-      array: Int8Array,
+      array: this,
     ) => U,
     initialValue: U,
   ): U;
@@ -2371,7 +2372,7 @@ interface Int8Array {
   /**
    * Reverses the elements in an Array.
    */
-  reverse(): Int8Array;
+  reverse(): this;
 
   /**
    * Sets a value or an array of values.
@@ -2385,7 +2386,7 @@ interface Int8Array {
    * @param start The beginning of the specified portion of the array.
    * @param end The end of the specified portion of the array. This is exclusive of the element at the index 'end'.
    */
-  slice(start?: number, end?: number): Int8Array;
+  slice(start?: number, end?: number): Int8Array<ArrayBuffer>;
 
   /**
    * Determines whether the specified callback function returns true for any element of an array.
@@ -2396,7 +2397,7 @@ interface Int8Array {
    * If thisArg is omitted, undefined is used as the this value.
    */
   some(
-    predicate: (value: number, index: number, array: Int8Array) => unknown,
+    predicate: (value: number, index: number, array: this) => unknown,
     thisArg?: any,
   ): boolean;
 
@@ -2417,7 +2418,7 @@ interface Int8Array {
    * @param begin The index of the beginning of the array.
    * @param end The index of the end of the array.
    */
-  subarray(begin?: number, end?: number): Int8Array;
+  subarray(begin?: number, end?: number): Int8Array<TArrayBuffer>;
 
   /**
    * Converts a number to a string by using the current locale.
@@ -2430,19 +2431,20 @@ interface Int8Array {
   toString(): string;
 
   /** Returns the primitive value of the specified object. */
-  valueOf(): Int8Array;
+  valueOf(): this;
 
   [index: number]: number;
 }
 interface Int8ArrayConstructor {
-  readonly prototype: Int8Array;
-  new (length: number): Int8Array;
-  new (array: ArrayLike<number> | ArrayBufferLike): Int8Array;
-  new (
-    buffer: ArrayBufferLike,
+  readonly prototype: Int8Array<ArrayBufferLike>;
+  new (length: number): Int8Array<ArrayBuffer>;
+  new (array: ArrayLike<number>): Int8Array<ArrayBuffer>;
+  new <TArrayBuffer extends ArrayBufferLike = ArrayBuffer>(
+    buffer: TArrayBuffer,
     byteOffset?: number,
     length?: number,
-  ): Int8Array;
+  ): Int8Array<TArrayBuffer>;
+  new (array: ArrayLike<number> | ArrayBuffer): Int8Array<ArrayBuffer>;
 
   /**
    * The size in bytes of each element in the array.
@@ -2453,13 +2455,13 @@ interface Int8ArrayConstructor {
    * Returns a new array from a set of elements.
    * @param items A set of elements to include in the new array object.
    */
-  of(...items: number[]): Int8Array;
+  of(...items: number[]): Int8Array<ArrayBuffer>;
 
   /**
    * Creates an array from an array-like or iterable object.
    * @param arrayLike An array-like or iterable object to convert to an array.
    */
-  from(arrayLike: ArrayLike<number>): Int8Array;
+  from(arrayLike: ArrayLike<number>): Int8Array<ArrayBuffer>;
 
   /**
    * Creates an array from an array-like or iterable object.
@@ -2471,7 +2473,7 @@ interface Int8ArrayConstructor {
     arrayLike: ArrayLike<T>,
     mapfn: (v: T, k: number) => number,
     thisArg?: any,
-  ): Int8Array;
+  ): Int8Array<ArrayBuffer>;
 }
 declare var Int8Array: Int8ArrayConstructor;
 
@@ -2479,7 +2481,7 @@ declare var Int8Array: Int8ArrayConstructor;
  * A typed array of 8-bit unsigned integer values. The contents are initialized to 0. If the
  * requested number of bytes could not be allocated an exception is raised.
  */
-interface Uint8Array {
+interface Uint8Array<TArrayBuffer extends ArrayBufferLike = ArrayBufferLike> {
   /**
    * The size in bytes of each element in the array.
    */
@@ -2488,7 +2490,7 @@ interface Uint8Array {
   /**
    * The ArrayBuffer instance referenced by the array.
    */
-  readonly buffer: ArrayBufferLike;
+  readonly buffer: TArrayBuffer;
 
   /**
    * The length in bytes of the array.
@@ -2520,7 +2522,7 @@ interface Uint8Array {
    * If thisArg is omitted, undefined is used as the this value.
    */
   every(
-    predicate: (value: number, index: number, array: Uint8Array) => unknown,
+    predicate: (value: number, index: number, array: this) => unknown,
     thisArg?: any,
   ): boolean;
 
@@ -2542,9 +2544,9 @@ interface Uint8Array {
    * If thisArg is omitted, undefined is used as the this value.
    */
   filter(
-    predicate: (value: number, index: number, array: Uint8Array) => any,
+    predicate: (value: number, index: number, array: this) => any,
     thisArg?: any,
-  ): Uint8Array;
+  ): Uint8Array<ArrayBuffer>;
 
   /**
    * Returns the value of the first element in the array where predicate is true, and undefined
@@ -2556,7 +2558,7 @@ interface Uint8Array {
    * predicate. If it is not provided, undefined is used instead.
    */
   find(
-    predicate: (value: number, index: number, obj: Uint8Array) => boolean,
+    predicate: (value: number, index: number, obj: this) => boolean,
     thisArg?: any,
   ): number | undefined;
 
@@ -2570,7 +2572,7 @@ interface Uint8Array {
    * predicate. If it is not provided, undefined is used instead.
    */
   findIndex(
-    predicate: (value: number, index: number, obj: Uint8Array) => boolean,
+    predicate: (value: number, index: number, obj: this) => boolean,
     thisArg?: any,
   ): number;
 
@@ -2582,7 +2584,7 @@ interface Uint8Array {
    * If thisArg is omitted, undefined is used as the this value.
    */
   forEach(
-    callbackfn: (value: number, index: number, array: Uint8Array) => void,
+    callbackfn: (value: number, index: number, array: this) => void,
     thisArg?: any,
   ): void;
 
@@ -2623,9 +2625,9 @@ interface Uint8Array {
    * If thisArg is omitted, undefined is used as the this value.
    */
   map(
-    callbackfn: (value: number, index: number, array: Uint8Array) => number,
+    callbackfn: (value: number, index: number, array: this) => number,
     thisArg?: any,
-  ): Uint8Array;
+  ): Uint8Array<ArrayBuffer>;
 
   /**
    * Calls the specified callback function for all the elements in an array. The return value of
@@ -2642,7 +2644,7 @@ interface Uint8Array {
       previousValue: number,
       currentValue: number,
       currentIndex: number,
-      array: Uint8Array,
+      array: this,
     ) => number,
   ): number;
   reduce(
@@ -2650,7 +2652,7 @@ interface Uint8Array {
       previousValue: number,
       currentValue: number,
       currentIndex: number,
-      array: Uint8Array,
+      array: this,
     ) => number,
     initialValue: number,
   ): number;
@@ -2670,7 +2672,7 @@ interface Uint8Array {
       previousValue: U,
       currentValue: number,
       currentIndex: number,
-      array: Uint8Array,
+      array: this,
     ) => U,
     initialValue: U,
   ): U;
@@ -2690,7 +2692,7 @@ interface Uint8Array {
       previousValue: number,
       currentValue: number,
       currentIndex: number,
-      array: Uint8Array,
+      array: this,
     ) => number,
   ): number;
   reduceRight(
@@ -2698,7 +2700,7 @@ interface Uint8Array {
       previousValue: number,
       currentValue: number,
       currentIndex: number,
-      array: Uint8Array,
+      array: this,
     ) => number,
     initialValue: number,
   ): number;
@@ -2718,7 +2720,7 @@ interface Uint8Array {
       previousValue: U,
       currentValue: number,
       currentIndex: number,
-      array: Uint8Array,
+      array: this,
     ) => U,
     initialValue: U,
   ): U;
@@ -2726,7 +2728,7 @@ interface Uint8Array {
   /**
    * Reverses the elements in an Array.
    */
-  reverse(): Uint8Array;
+  reverse(): this;
 
   /**
    * Sets a value or an array of values.
@@ -2740,7 +2742,7 @@ interface Uint8Array {
    * @param start The beginning of the specified portion of the array.
    * @param end The end of the specified portion of the array. This is exclusive of the element at the index 'end'.
    */
-  slice(start?: number, end?: number): Uint8Array;
+  slice(start?: number, end?: number): Uint8Array<ArrayBuffer>;
 
   /**
    * Determines whether the specified callback function returns true for any element of an array.
@@ -2751,7 +2753,7 @@ interface Uint8Array {
    * If thisArg is omitted, undefined is used as the this value.
    */
   some(
-    predicate: (value: number, index: number, array: Uint8Array) => unknown,
+    predicate: (value: number, index: number, array: this) => unknown,
     thisArg?: any,
   ): boolean;
 
@@ -2772,7 +2774,7 @@ interface Uint8Array {
    * @param begin The index of the beginning of the array.
    * @param end The index of the end of the array.
    */
-  subarray(begin?: number, end?: number): Uint8Array;
+  subarray(begin?: number, end?: number): Uint8Array<TArrayBuffer>;
 
   /**
    * Converts a number to a string by using the current locale.
@@ -2785,20 +2787,20 @@ interface Uint8Array {
   toString(): string;
 
   /** Returns the primitive value of the specified object. */
-  valueOf(): Uint8Array;
+  valueOf(): this;
 
   [index: number]: number;
 }
-
 interface Uint8ArrayConstructor {
-  readonly prototype: Uint8Array;
-  new (length: number): Uint8Array;
-  new (array: ArrayLike<number> | ArrayBufferLike): Uint8Array;
-  new (
-    buffer: ArrayBufferLike,
+  readonly prototype: Uint8Array<ArrayBufferLike>;
+  new (length: number): Uint8Array<ArrayBuffer>;
+  new (array: ArrayLike<number>): Uint8Array<ArrayBuffer>;
+  new <TArrayBuffer extends ArrayBufferLike = ArrayBuffer>(
+    buffer: TArrayBuffer,
     byteOffset?: number,
     length?: number,
-  ): Uint8Array;
+  ): Uint8Array<TArrayBuffer>;
+  new (array: ArrayLike<number> | ArrayBuffer): Uint8Array<ArrayBuffer>;
 
   /**
    * The size in bytes of each element in the array.
@@ -2809,13 +2811,13 @@ interface Uint8ArrayConstructor {
    * Returns a new array from a set of elements.
    * @param items A set of elements to include in the new array object.
    */
-  of(...items: number[]): Uint8Array;
+  of(...items: number[]): Uint8Array<ArrayBuffer>;
 
   /**
    * Creates an array from an array-like or iterable object.
    * @param arrayLike An array-like or iterable object to convert to an array.
    */
-  from(arrayLike: ArrayLike<number>): Uint8Array;
+  from(arrayLike: ArrayLike<number>): Uint8Array<ArrayBuffer>;
 
   /**
    * Creates an array from an array-like or iterable object.
@@ -2827,7 +2829,7 @@ interface Uint8ArrayConstructor {
     arrayLike: ArrayLike<T>,
     mapfn: (v: T, k: number) => number,
     thisArg?: any,
-  ): Uint8Array;
+  ): Uint8Array<ArrayBuffer>;
 }
 declare var Uint8Array: Uint8ArrayConstructor;
 
@@ -2835,7 +2837,9 @@ declare var Uint8Array: Uint8ArrayConstructor;
  * A typed array of 8-bit unsigned integer (clamped) values. The contents are initialized to 0.
  * If the requested number of bytes could not be allocated an exception is raised.
  */
-interface Uint8ClampedArray {
+interface Uint8ClampedArray<
+  TArrayBuffer extends ArrayBufferLike = ArrayBufferLike,
+> {
   /**
    * The size in bytes of each element in the array.
    */
@@ -2844,7 +2848,7 @@ interface Uint8ClampedArray {
   /**
    * The ArrayBuffer instance referenced by the array.
    */
-  readonly buffer: ArrayBufferLike;
+  readonly buffer: TArrayBuffer;
 
   /**
    * The length in bytes of the array.
@@ -2876,11 +2880,7 @@ interface Uint8ClampedArray {
    * If thisArg is omitted, undefined is used as the this value.
    */
   every(
-    predicate: (
-      value: number,
-      index: number,
-      array: Uint8ClampedArray,
-    ) => unknown,
+    predicate: (value: number, index: number, array: this) => unknown,
     thisArg?: any,
   ): boolean;
 
@@ -2902,9 +2902,9 @@ interface Uint8ClampedArray {
    * If thisArg is omitted, undefined is used as the this value.
    */
   filter(
-    predicate: (value: number, index: number, array: Uint8ClampedArray) => any,
+    predicate: (value: number, index: number, array: this) => any,
     thisArg?: any,
-  ): Uint8ClampedArray;
+  ): Uint8ClampedArray<ArrayBuffer>;
 
   /**
    * Returns the value of the first element in the array where predicate is true, and undefined
@@ -2916,11 +2916,7 @@ interface Uint8ClampedArray {
    * predicate. If it is not provided, undefined is used instead.
    */
   find(
-    predicate: (
-      value: number,
-      index: number,
-      obj: Uint8ClampedArray,
-    ) => boolean,
+    predicate: (value: number, index: number, obj: this) => boolean,
     thisArg?: any,
   ): number | undefined;
 
@@ -2934,11 +2930,7 @@ interface Uint8ClampedArray {
    * predicate. If it is not provided, undefined is used instead.
    */
   findIndex(
-    predicate: (
-      value: number,
-      index: number,
-      obj: Uint8ClampedArray,
-    ) => boolean,
+    predicate: (value: number, index: number, obj: this) => boolean,
     thisArg?: any,
   ): number;
 
@@ -2950,11 +2942,7 @@ interface Uint8ClampedArray {
    * If thisArg is omitted, undefined is used as the this value.
    */
   forEach(
-    callbackfn: (
-      value: number,
-      index: number,
-      array: Uint8ClampedArray,
-    ) => void,
+    callbackfn: (value: number, index: number, array: this) => void,
     thisArg?: any,
   ): void;
 
@@ -2995,13 +2983,9 @@ interface Uint8ClampedArray {
    * If thisArg is omitted, undefined is used as the this value.
    */
   map(
-    callbackfn: (
-      value: number,
-      index: number,
-      array: Uint8ClampedArray,
-    ) => number,
+    callbackfn: (value: number, index: number, array: this) => number,
     thisArg?: any,
-  ): Uint8ClampedArray;
+  ): Uint8ClampedArray<ArrayBuffer>;
 
   /**
    * Calls the specified callback function for all the elements in an array. The return value of
@@ -3018,7 +3002,7 @@ interface Uint8ClampedArray {
       previousValue: number,
       currentValue: number,
       currentIndex: number,
-      array: Uint8ClampedArray,
+      array: this,
     ) => number,
   ): number;
   reduce(
@@ -3026,7 +3010,7 @@ interface Uint8ClampedArray {
       previousValue: number,
       currentValue: number,
       currentIndex: number,
-      array: Uint8ClampedArray,
+      array: this,
     ) => number,
     initialValue: number,
   ): number;
@@ -3046,7 +3030,7 @@ interface Uint8ClampedArray {
       previousValue: U,
       currentValue: number,
       currentIndex: number,
-      array: Uint8ClampedArray,
+      array: this,
     ) => U,
     initialValue: U,
   ): U;
@@ -3066,7 +3050,7 @@ interface Uint8ClampedArray {
       previousValue: number,
       currentValue: number,
       currentIndex: number,
-      array: Uint8ClampedArray,
+      array: this,
     ) => number,
   ): number;
   reduceRight(
@@ -3074,7 +3058,7 @@ interface Uint8ClampedArray {
       previousValue: number,
       currentValue: number,
       currentIndex: number,
-      array: Uint8ClampedArray,
+      array: this,
     ) => number,
     initialValue: number,
   ): number;
@@ -3094,7 +3078,7 @@ interface Uint8ClampedArray {
       previousValue: U,
       currentValue: number,
       currentIndex: number,
-      array: Uint8ClampedArray,
+      array: this,
     ) => U,
     initialValue: U,
   ): U;
@@ -3102,7 +3086,7 @@ interface Uint8ClampedArray {
   /**
    * Reverses the elements in an Array.
    */
-  reverse(): Uint8ClampedArray;
+  reverse(): this;
 
   /**
    * Sets a value or an array of values.
@@ -3116,7 +3100,7 @@ interface Uint8ClampedArray {
    * @param start The beginning of the specified portion of the array.
    * @param end The end of the specified portion of the array. This is exclusive of the element at the index 'end'.
    */
-  slice(start?: number, end?: number): Uint8ClampedArray;
+  slice(start?: number, end?: number): Uint8ClampedArray<ArrayBuffer>;
 
   /**
    * Determines whether the specified callback function returns true for any element of an array.
@@ -3127,11 +3111,7 @@ interface Uint8ClampedArray {
    * If thisArg is omitted, undefined is used as the this value.
    */
   some(
-    predicate: (
-      value: number,
-      index: number,
-      array: Uint8ClampedArray,
-    ) => unknown,
+    predicate: (value: number, index: number, array: this) => unknown,
     thisArg?: any,
   ): boolean;
 
@@ -3152,7 +3132,7 @@ interface Uint8ClampedArray {
    * @param begin The index of the beginning of the array.
    * @param end The index of the end of the array.
    */
-  subarray(begin?: number, end?: number): Uint8ClampedArray;
+  subarray(begin?: number, end?: number): Uint8ClampedArray<TArrayBuffer>;
 
   /**
    * Converts a number to a string by using the current locale.
@@ -3165,20 +3145,20 @@ interface Uint8ClampedArray {
   toString(): string;
 
   /** Returns the primitive value of the specified object. */
-  valueOf(): Uint8ClampedArray;
+  valueOf(): this;
 
   [index: number]: number;
 }
-
 interface Uint8ClampedArrayConstructor {
-  readonly prototype: Uint8ClampedArray;
-  new (length: number): Uint8ClampedArray;
-  new (array: ArrayLike<number> | ArrayBufferLike): Uint8ClampedArray;
-  new (
-    buffer: ArrayBufferLike,
+  readonly prototype: Uint8ClampedArray<ArrayBufferLike>;
+  new (length: number): Uint8ClampedArray<ArrayBuffer>;
+  new (array: ArrayLike<number>): Uint8ClampedArray<ArrayBuffer>;
+  new <TArrayBuffer extends ArrayBufferLike = ArrayBuffer>(
+    buffer: TArrayBuffer,
     byteOffset?: number,
     length?: number,
-  ): Uint8ClampedArray;
+  ): Uint8ClampedArray<TArrayBuffer>;
+  new (array: ArrayLike<number> | ArrayBuffer): Uint8ClampedArray<ArrayBuffer>;
 
   /**
    * The size in bytes of each element in the array.
@@ -3189,13 +3169,13 @@ interface Uint8ClampedArrayConstructor {
    * Returns a new array from a set of elements.
    * @param items A set of elements to include in the new array object.
    */
-  of(...items: number[]): Uint8ClampedArray;
+  of(...items: number[]): Uint8ClampedArray<ArrayBuffer>;
 
   /**
    * Creates an array from an array-like or iterable object.
    * @param arrayLike An array-like or iterable object to convert to an array.
    */
-  from(arrayLike: ArrayLike<number>): Uint8ClampedArray;
+  from(arrayLike: ArrayLike<number>): Uint8ClampedArray<ArrayBuffer>;
 
   /**
    * Creates an array from an array-like or iterable object.
@@ -3207,7 +3187,7 @@ interface Uint8ClampedArrayConstructor {
     arrayLike: ArrayLike<T>,
     mapfn: (v: T, k: number) => number,
     thisArg?: any,
-  ): Uint8ClampedArray;
+  ): Uint8ClampedArray<ArrayBuffer>;
 }
 declare var Uint8ClampedArray: Uint8ClampedArrayConstructor;
 
@@ -3215,7 +3195,7 @@ declare var Uint8ClampedArray: Uint8ClampedArrayConstructor;
  * A typed array of 16-bit signed integer values. The contents are initialized to 0. If the
  * requested number of bytes could not be allocated an exception is raised.
  */
-interface Int16Array {
+interface Int16Array<TArrayBuffer extends ArrayBufferLike = ArrayBufferLike> {
   /**
    * The size in bytes of each element in the array.
    */
@@ -3224,7 +3204,7 @@ interface Int16Array {
   /**
    * The ArrayBuffer instance referenced by the array.
    */
-  readonly buffer: ArrayBufferLike;
+  readonly buffer: TArrayBuffer;
 
   /**
    * The length in bytes of the array.
@@ -3256,7 +3236,7 @@ interface Int16Array {
    * If thisArg is omitted, undefined is used as the this value.
    */
   every(
-    predicate: (value: number, index: number, array: Int16Array) => unknown,
+    predicate: (value: number, index: number, array: this) => unknown,
     thisArg?: any,
   ): boolean;
 
@@ -3278,9 +3258,9 @@ interface Int16Array {
    * If thisArg is omitted, undefined is used as the this value.
    */
   filter(
-    predicate: (value: number, index: number, array: Int16Array) => any,
+    predicate: (value: number, index: number, array: this) => any,
     thisArg?: any,
-  ): Int16Array;
+  ): Int16Array<ArrayBuffer>;
 
   /**
    * Returns the value of the first element in the array where predicate is true, and undefined
@@ -3292,7 +3272,7 @@ interface Int16Array {
    * predicate. If it is not provided, undefined is used instead.
    */
   find(
-    predicate: (value: number, index: number, obj: Int16Array) => boolean,
+    predicate: (value: number, index: number, obj: this) => boolean,
     thisArg?: any,
   ): number | undefined;
 
@@ -3306,7 +3286,7 @@ interface Int16Array {
    * predicate. If it is not provided, undefined is used instead.
    */
   findIndex(
-    predicate: (value: number, index: number, obj: Int16Array) => boolean,
+    predicate: (value: number, index: number, obj: this) => boolean,
     thisArg?: any,
   ): number;
 
@@ -3318,7 +3298,7 @@ interface Int16Array {
    * If thisArg is omitted, undefined is used as the this value.
    */
   forEach(
-    callbackfn: (value: number, index: number, array: Int16Array) => void,
+    callbackfn: (value: number, index: number, array: this) => void,
     thisArg?: any,
   ): void;
   /**
@@ -3358,9 +3338,9 @@ interface Int16Array {
    * If thisArg is omitted, undefined is used as the this value.
    */
   map(
-    callbackfn: (value: number, index: number, array: Int16Array) => number,
+    callbackfn: (value: number, index: number, array: this) => number,
     thisArg?: any,
-  ): Int16Array;
+  ): Int16Array<ArrayBuffer>;
 
   /**
    * Calls the specified callback function for all the elements in an array. The return value of
@@ -3377,7 +3357,7 @@ interface Int16Array {
       previousValue: number,
       currentValue: number,
       currentIndex: number,
-      array: Int16Array,
+      array: this,
     ) => number,
   ): number;
   reduce(
@@ -3385,7 +3365,7 @@ interface Int16Array {
       previousValue: number,
       currentValue: number,
       currentIndex: number,
-      array: Int16Array,
+      array: this,
     ) => number,
     initialValue: number,
   ): number;
@@ -3405,7 +3385,7 @@ interface Int16Array {
       previousValue: U,
       currentValue: number,
       currentIndex: number,
-      array: Int16Array,
+      array: this,
     ) => U,
     initialValue: U,
   ): U;
@@ -3425,7 +3405,7 @@ interface Int16Array {
       previousValue: number,
       currentValue: number,
       currentIndex: number,
-      array: Int16Array,
+      array: this,
     ) => number,
   ): number;
   reduceRight(
@@ -3433,7 +3413,7 @@ interface Int16Array {
       previousValue: number,
       currentValue: number,
       currentIndex: number,
-      array: Int16Array,
+      array: this,
     ) => number,
     initialValue: number,
   ): number;
@@ -3453,7 +3433,7 @@ interface Int16Array {
       previousValue: U,
       currentValue: number,
       currentIndex: number,
-      array: Int16Array,
+      array: this,
     ) => U,
     initialValue: U,
   ): U;
@@ -3461,7 +3441,7 @@ interface Int16Array {
   /**
    * Reverses the elements in an Array.
    */
-  reverse(): Int16Array;
+  reverse(): this;
 
   /**
    * Sets a value or an array of values.
@@ -3475,7 +3455,7 @@ interface Int16Array {
    * @param start The beginning of the specified portion of the array.
    * @param end The end of the specified portion of the array. This is exclusive of the element at the index 'end'.
    */
-  slice(start?: number, end?: number): Int16Array;
+  slice(start?: number, end?: number): Int16Array<ArrayBuffer>;
 
   /**
    * Determines whether the specified callback function returns true for any element of an array.
@@ -3486,7 +3466,7 @@ interface Int16Array {
    * If thisArg is omitted, undefined is used as the this value.
    */
   some(
-    predicate: (value: number, index: number, array: Int16Array) => unknown,
+    predicate: (value: number, index: number, array: this) => unknown,
     thisArg?: any,
   ): boolean;
 
@@ -3507,7 +3487,7 @@ interface Int16Array {
    * @param begin The index of the beginning of the array.
    * @param end The index of the end of the array.
    */
-  subarray(begin?: number, end?: number): Int16Array;
+  subarray(begin?: number, end?: number): Int16Array<TArrayBuffer>;
 
   /**
    * Converts a number to a string by using the current locale.
@@ -3520,20 +3500,20 @@ interface Int16Array {
   toString(): string;
 
   /** Returns the primitive value of the specified object. */
-  valueOf(): Int16Array;
+  valueOf(): this;
 
   [index: number]: number;
 }
-
 interface Int16ArrayConstructor {
-  readonly prototype: Int16Array;
-  new (length: number): Int16Array;
-  new (array: ArrayLike<number> | ArrayBufferLike): Int16Array;
-  new (
-    buffer: ArrayBufferLike,
+  readonly prototype: Int16Array<ArrayBufferLike>;
+  new (length: number): Int16Array<ArrayBuffer>;
+  new (array: ArrayLike<number>): Int16Array<ArrayBuffer>;
+  new <TArrayBuffer extends ArrayBufferLike = ArrayBuffer>(
+    buffer: TArrayBuffer,
     byteOffset?: number,
     length?: number,
-  ): Int16Array;
+  ): Int16Array<TArrayBuffer>;
+  new (array: ArrayLike<number> | ArrayBuffer): Int16Array<ArrayBuffer>;
 
   /**
    * The size in bytes of each element in the array.
@@ -3544,13 +3524,13 @@ interface Int16ArrayConstructor {
    * Returns a new array from a set of elements.
    * @param items A set of elements to include in the new array object.
    */
-  of(...items: number[]): Int16Array;
+  of(...items: number[]): Int16Array<ArrayBuffer>;
 
   /**
    * Creates an array from an array-like or iterable object.
    * @param arrayLike An array-like or iterable object to convert to an array.
    */
-  from(arrayLike: ArrayLike<number>): Int16Array;
+  from(arrayLike: ArrayLike<number>): Int16Array<ArrayBuffer>;
 
   /**
    * Creates an array from an array-like or iterable object.
@@ -3562,7 +3542,7 @@ interface Int16ArrayConstructor {
     arrayLike: ArrayLike<T>,
     mapfn: (v: T, k: number) => number,
     thisArg?: any,
-  ): Int16Array;
+  ): Int16Array<ArrayBuffer>;
 }
 declare var Int16Array: Int16ArrayConstructor;
 
@@ -3570,7 +3550,7 @@ declare var Int16Array: Int16ArrayConstructor;
  * A typed array of 16-bit unsigned integer values. The contents are initialized to 0. If the
  * requested number of bytes could not be allocated an exception is raised.
  */
-interface Uint16Array {
+interface Uint16Array<TArrayBuffer extends ArrayBufferLike = ArrayBufferLike> {
   /**
    * The size in bytes of each element in the array.
    */
@@ -3579,7 +3559,7 @@ interface Uint16Array {
   /**
    * The ArrayBuffer instance referenced by the array.
    */
-  readonly buffer: ArrayBufferLike;
+  readonly buffer: TArrayBuffer;
 
   /**
    * The length in bytes of the array.
@@ -3611,7 +3591,7 @@ interface Uint16Array {
    * If thisArg is omitted, undefined is used as the this value.
    */
   every(
-    predicate: (value: number, index: number, array: Uint16Array) => unknown,
+    predicate: (value: number, index: number, array: this) => unknown,
     thisArg?: any,
   ): boolean;
 
@@ -3633,9 +3613,9 @@ interface Uint16Array {
    * If thisArg is omitted, undefined is used as the this value.
    */
   filter(
-    predicate: (value: number, index: number, array: Uint16Array) => any,
+    predicate: (value: number, index: number, array: this) => any,
     thisArg?: any,
-  ): Uint16Array;
+  ): Uint16Array<ArrayBuffer>;
 
   /**
    * Returns the value of the first element in the array where predicate is true, and undefined
@@ -3647,7 +3627,7 @@ interface Uint16Array {
    * predicate. If it is not provided, undefined is used instead.
    */
   find(
-    predicate: (value: number, index: number, obj: Uint16Array) => boolean,
+    predicate: (value: number, index: number, obj: this) => boolean,
     thisArg?: any,
   ): number | undefined;
 
@@ -3661,7 +3641,7 @@ interface Uint16Array {
    * predicate. If it is not provided, undefined is used instead.
    */
   findIndex(
-    predicate: (value: number, index: number, obj: Uint16Array) => boolean,
+    predicate: (value: number, index: number, obj: this) => boolean,
     thisArg?: any,
   ): number;
 
@@ -3673,7 +3653,7 @@ interface Uint16Array {
    * If thisArg is omitted, undefined is used as the this value.
    */
   forEach(
-    callbackfn: (value: number, index: number, array: Uint16Array) => void,
+    callbackfn: (value: number, index: number, array: this) => void,
     thisArg?: any,
   ): void;
 
@@ -3714,9 +3694,9 @@ interface Uint16Array {
    * If thisArg is omitted, undefined is used as the this value.
    */
   map(
-    callbackfn: (value: number, index: number, array: Uint16Array) => number,
+    callbackfn: (value: number, index: number, array: this) => number,
     thisArg?: any,
-  ): Uint16Array;
+  ): Uint16Array<ArrayBuffer>;
 
   /**
    * Calls the specified callback function for all the elements in an array. The return value of
@@ -3733,7 +3713,7 @@ interface Uint16Array {
       previousValue: number,
       currentValue: number,
       currentIndex: number,
-      array: Uint16Array,
+      array: this,
     ) => number,
   ): number;
   reduce(
@@ -3741,7 +3721,7 @@ interface Uint16Array {
       previousValue: number,
       currentValue: number,
       currentIndex: number,
-      array: Uint16Array,
+      array: this,
     ) => number,
     initialValue: number,
   ): number;
@@ -3761,7 +3741,7 @@ interface Uint16Array {
       previousValue: U,
       currentValue: number,
       currentIndex: number,
-      array: Uint16Array,
+      array: this,
     ) => U,
     initialValue: U,
   ): U;
@@ -3781,7 +3761,7 @@ interface Uint16Array {
       previousValue: number,
       currentValue: number,
       currentIndex: number,
-      array: Uint16Array,
+      array: this,
     ) => number,
   ): number;
   reduceRight(
@@ -3789,7 +3769,7 @@ interface Uint16Array {
       previousValue: number,
       currentValue: number,
       currentIndex: number,
-      array: Uint16Array,
+      array: this,
     ) => number,
     initialValue: number,
   ): number;
@@ -3809,7 +3789,7 @@ interface Uint16Array {
       previousValue: U,
       currentValue: number,
       currentIndex: number,
-      array: Uint16Array,
+      array: this,
     ) => U,
     initialValue: U,
   ): U;
@@ -3817,7 +3797,7 @@ interface Uint16Array {
   /**
    * Reverses the elements in an Array.
    */
-  reverse(): Uint16Array;
+  reverse(): this;
 
   /**
    * Sets a value or an array of values.
@@ -3831,7 +3811,7 @@ interface Uint16Array {
    * @param start The beginning of the specified portion of the array.
    * @param end The end of the specified portion of the array. This is exclusive of the element at the index 'end'.
    */
-  slice(start?: number, end?: number): Uint16Array;
+  slice(start?: number, end?: number): Uint16Array<ArrayBuffer>;
 
   /**
    * Determines whether the specified callback function returns true for any element of an array.
@@ -3842,7 +3822,7 @@ interface Uint16Array {
    * If thisArg is omitted, undefined is used as the this value.
    */
   some(
-    predicate: (value: number, index: number, array: Uint16Array) => unknown,
+    predicate: (value: number, index: number, array: this) => unknown,
     thisArg?: any,
   ): boolean;
 
@@ -3863,7 +3843,7 @@ interface Uint16Array {
    * @param begin The index of the beginning of the array.
    * @param end The index of the end of the array.
    */
-  subarray(begin?: number, end?: number): Uint16Array;
+  subarray(begin?: number, end?: number): Uint16Array<TArrayBuffer>;
 
   /**
    * Converts a number to a string by using the current locale.
@@ -3876,20 +3856,20 @@ interface Uint16Array {
   toString(): string;
 
   /** Returns the primitive value of the specified object. */
-  valueOf(): Uint16Array;
+  valueOf(): this;
 
   [index: number]: number;
 }
-
 interface Uint16ArrayConstructor {
-  readonly prototype: Uint16Array;
-  new (length: number): Uint16Array;
-  new (array: ArrayLike<number> | ArrayBufferLike): Uint16Array;
-  new (
-    buffer: ArrayBufferLike,
+  readonly prototype: Uint16Array<ArrayBufferLike>;
+  new (length: number): Uint16Array<ArrayBuffer>;
+  new (array: ArrayLike<number>): Uint16Array<ArrayBuffer>;
+  new <TArrayBuffer extends ArrayBufferLike = ArrayBuffer>(
+    buffer: TArrayBuffer,
     byteOffset?: number,
     length?: number,
-  ): Uint16Array;
+  ): Uint16Array<TArrayBuffer>;
+  new (array: ArrayLike<number> | ArrayBuffer): Uint16Array<ArrayBuffer>;
 
   /**
    * The size in bytes of each element in the array.
@@ -3900,13 +3880,13 @@ interface Uint16ArrayConstructor {
    * Returns a new array from a set of elements.
    * @param items A set of elements to include in the new array object.
    */
-  of(...items: number[]): Uint16Array;
+  of(...items: number[]): Uint16Array<ArrayBuffer>;
 
   /**
    * Creates an array from an array-like or iterable object.
    * @param arrayLike An array-like or iterable object to convert to an array.
    */
-  from(arrayLike: ArrayLike<number>): Uint16Array;
+  from(arrayLike: ArrayLike<number>): Uint16Array<ArrayBuffer>;
 
   /**
    * Creates an array from an array-like or iterable object.
@@ -3918,14 +3898,14 @@ interface Uint16ArrayConstructor {
     arrayLike: ArrayLike<T>,
     mapfn: (v: T, k: number) => number,
     thisArg?: any,
-  ): Uint16Array;
+  ): Uint16Array<ArrayBuffer>;
 }
 declare var Uint16Array: Uint16ArrayConstructor;
 /**
  * A typed array of 32-bit signed integer values. The contents are initialized to 0. If the
  * requested number of bytes could not be allocated an exception is raised.
  */
-interface Int32Array {
+interface Int32Array<TArrayBuffer extends ArrayBufferLike = ArrayBufferLike> {
   /**
    * The size in bytes of each element in the array.
    */
@@ -3934,7 +3914,7 @@ interface Int32Array {
   /**
    * The ArrayBuffer instance referenced by the array.
    */
-  readonly buffer: ArrayBufferLike;
+  readonly buffer: TArrayBuffer;
 
   /**
    * The length in bytes of the array.
@@ -3966,7 +3946,7 @@ interface Int32Array {
    * If thisArg is omitted, undefined is used as the this value.
    */
   every(
-    predicate: (value: number, index: number, array: Int32Array) => unknown,
+    predicate: (value: number, index: number, array: this) => unknown,
     thisArg?: any,
   ): boolean;
 
@@ -3988,9 +3968,9 @@ interface Int32Array {
    * If thisArg is omitted, undefined is used as the this value.
    */
   filter(
-    predicate: (value: number, index: number, array: Int32Array) => any,
+    predicate: (value: number, index: number, array: this) => any,
     thisArg?: any,
-  ): Int32Array;
+  ): Int32Array<ArrayBuffer>;
 
   /**
    * Returns the value of the first element in the array where predicate is true, and undefined
@@ -4002,7 +3982,7 @@ interface Int32Array {
    * predicate. If it is not provided, undefined is used instead.
    */
   find(
-    predicate: (value: number, index: number, obj: Int32Array) => boolean,
+    predicate: (value: number, index: number, obj: this) => boolean,
     thisArg?: any,
   ): number | undefined;
 
@@ -4016,7 +3996,7 @@ interface Int32Array {
    * predicate. If it is not provided, undefined is used instead.
    */
   findIndex(
-    predicate: (value: number, index: number, obj: Int32Array) => boolean,
+    predicate: (value: number, index: number, obj: this) => boolean,
     thisArg?: any,
   ): number;
 
@@ -4028,7 +4008,7 @@ interface Int32Array {
    * If thisArg is omitted, undefined is used as the this value.
    */
   forEach(
-    callbackfn: (value: number, index: number, array: Int32Array) => void,
+    callbackfn: (value: number, index: number, array: this) => void,
     thisArg?: any,
   ): void;
 
@@ -4069,9 +4049,9 @@ interface Int32Array {
    * If thisArg is omitted, undefined is used as the this value.
    */
   map(
-    callbackfn: (value: number, index: number, array: Int32Array) => number,
+    callbackfn: (value: number, index: number, array: this) => number,
     thisArg?: any,
-  ): Int32Array;
+  ): Int32Array<ArrayBuffer>;
 
   /**
    * Calls the specified callback function for all the elements in an array. The return value of
@@ -4088,7 +4068,7 @@ interface Int32Array {
       previousValue: number,
       currentValue: number,
       currentIndex: number,
-      array: Int32Array,
+      array: this,
     ) => number,
   ): number;
   reduce(
@@ -4096,7 +4076,7 @@ interface Int32Array {
       previousValue: number,
       currentValue: number,
       currentIndex: number,
-      array: Int32Array,
+      array: this,
     ) => number,
     initialValue: number,
   ): number;
@@ -4116,7 +4096,7 @@ interface Int32Array {
       previousValue: U,
       currentValue: number,
       currentIndex: number,
-      array: Int32Array,
+      array: this,
     ) => U,
     initialValue: U,
   ): U;
@@ -4136,7 +4116,7 @@ interface Int32Array {
       previousValue: number,
       currentValue: number,
       currentIndex: number,
-      array: Int32Array,
+      array: this,
     ) => number,
   ): number;
   reduceRight(
@@ -4144,7 +4124,7 @@ interface Int32Array {
       previousValue: number,
       currentValue: number,
       currentIndex: number,
-      array: Int32Array,
+      array: this,
     ) => number,
     initialValue: number,
   ): number;
@@ -4164,7 +4144,7 @@ interface Int32Array {
       previousValue: U,
       currentValue: number,
       currentIndex: number,
-      array: Int32Array,
+      array: this,
     ) => U,
     initialValue: U,
   ): U;
@@ -4172,7 +4152,7 @@ interface Int32Array {
   /**
    * Reverses the elements in an Array.
    */
-  reverse(): Int32Array;
+  reverse(): this;
 
   /**
    * Sets a value or an array of values.
@@ -4186,7 +4166,7 @@ interface Int32Array {
    * @param start The beginning of the specified portion of the array.
    * @param end The end of the specified portion of the array. This is exclusive of the element at the index 'end'.
    */
-  slice(start?: number, end?: number): Int32Array;
+  slice(start?: number, end?: number): Int32Array<ArrayBuffer>;
 
   /**
    * Determines whether the specified callback function returns true for any element of an array.
@@ -4197,7 +4177,7 @@ interface Int32Array {
    * If thisArg is omitted, undefined is used as the this value.
    */
   some(
-    predicate: (value: number, index: number, array: Int32Array) => unknown,
+    predicate: (value: number, index: number, array: this) => unknown,
     thisArg?: any,
   ): boolean;
 
@@ -4218,7 +4198,7 @@ interface Int32Array {
    * @param begin The index of the beginning of the array.
    * @param end The index of the end of the array.
    */
-  subarray(begin?: number, end?: number): Int32Array;
+  subarray(begin?: number, end?: number): Int32Array<TArrayBuffer>;
 
   /**
    * Converts a number to a string by using the current locale.
@@ -4231,20 +4211,20 @@ interface Int32Array {
   toString(): string;
 
   /** Returns the primitive value of the specified object. */
-  valueOf(): Int32Array;
+  valueOf(): this;
 
   [index: number]: number;
 }
-
 interface Int32ArrayConstructor {
-  readonly prototype: Int32Array;
-  new (length: number): Int32Array;
-  new (array: ArrayLike<number> | ArrayBufferLike): Int32Array;
-  new (
-    buffer: ArrayBufferLike,
+  readonly prototype: Int32Array<ArrayBufferLike>;
+  new (length: number): Int32Array<ArrayBuffer>;
+  new (array: ArrayLike<number>): Int32Array<ArrayBuffer>;
+  new <TArrayBuffer extends ArrayBufferLike = ArrayBuffer>(
+    buffer: TArrayBuffer,
     byteOffset?: number,
     length?: number,
-  ): Int32Array;
+  ): Int32Array<TArrayBuffer>;
+  new (array: ArrayLike<number> | ArrayBuffer): Int32Array<ArrayBuffer>;
 
   /**
    * The size in bytes of each element in the array.
@@ -4255,13 +4235,13 @@ interface Int32ArrayConstructor {
    * Returns a new array from a set of elements.
    * @param items A set of elements to include in the new array object.
    */
-  of(...items: number[]): Int32Array;
+  of(...items: number[]): Int32Array<ArrayBuffer>;
 
   /**
    * Creates an array from an array-like or iterable object.
    * @param arrayLike An array-like or iterable object to convert to an array.
    */
-  from(arrayLike: ArrayLike<number>): Int32Array;
+  from(arrayLike: ArrayLike<number>): Int32Array<ArrayBuffer>;
 
   /**
    * Creates an array from an array-like or iterable object.
@@ -4273,7 +4253,7 @@ interface Int32ArrayConstructor {
     arrayLike: ArrayLike<T>,
     mapfn: (v: T, k: number) => number,
     thisArg?: any,
-  ): Int32Array;
+  ): Int32Array<ArrayBuffer>;
 }
 declare var Int32Array: Int32ArrayConstructor;
 
@@ -4281,7 +4261,7 @@ declare var Int32Array: Int32ArrayConstructor;
  * A typed array of 32-bit unsigned integer values. The contents are initialized to 0. If the
  * requested number of bytes could not be allocated an exception is raised.
  */
-interface Uint32Array {
+interface Uint32Array<TArrayBuffer extends ArrayBufferLike = ArrayBufferLike> {
   /**
    * The size in bytes of each element in the array.
    */
@@ -4290,7 +4270,7 @@ interface Uint32Array {
   /**
    * The ArrayBuffer instance referenced by the array.
    */
-  readonly buffer: ArrayBufferLike;
+  readonly buffer: TArrayBuffer;
 
   /**
    * The length in bytes of the array.
@@ -4322,7 +4302,7 @@ interface Uint32Array {
    * If thisArg is omitted, undefined is used as the this value.
    */
   every(
-    predicate: (value: number, index: number, array: Uint32Array) => unknown,
+    predicate: (value: number, index: number, array: this) => unknown,
     thisArg?: any,
   ): boolean;
 
@@ -4344,9 +4324,9 @@ interface Uint32Array {
    * If thisArg is omitted, undefined is used as the this value.
    */
   filter(
-    predicate: (value: number, index: number, array: Uint32Array) => any,
+    predicate: (value: number, index: number, array: this) => any,
     thisArg?: any,
-  ): Uint32Array;
+  ): Uint32Array<ArrayBuffer>;
 
   /**
    * Returns the value of the first element in the array where predicate is true, and undefined
@@ -4358,7 +4338,7 @@ interface Uint32Array {
    * predicate. If it is not provided, undefined is used instead.
    */
   find(
-    predicate: (value: number, index: number, obj: Uint32Array) => boolean,
+    predicate: (value: number, index: number, obj: this) => boolean,
     thisArg?: any,
   ): number | undefined;
 
@@ -4372,7 +4352,7 @@ interface Uint32Array {
    * predicate. If it is not provided, undefined is used instead.
    */
   findIndex(
-    predicate: (value: number, index: number, obj: Uint32Array) => boolean,
+    predicate: (value: number, index: number, obj: this) => boolean,
     thisArg?: any,
   ): number;
 
@@ -4384,7 +4364,7 @@ interface Uint32Array {
    * If thisArg is omitted, undefined is used as the this value.
    */
   forEach(
-    callbackfn: (value: number, index: number, array: Uint32Array) => void,
+    callbackfn: (value: number, index: number, array: this) => void,
     thisArg?: any,
   ): void;
   /**
@@ -4424,9 +4404,9 @@ interface Uint32Array {
    * If thisArg is omitted, undefined is used as the this value.
    */
   map(
-    callbackfn: (value: number, index: number, array: Uint32Array) => number,
+    callbackfn: (value: number, index: number, array: this) => number,
     thisArg?: any,
-  ): Uint32Array;
+  ): Uint32Array<ArrayBuffer>;
 
   /**
    * Calls the specified callback function for all the elements in an array. The return value of
@@ -4443,7 +4423,7 @@ interface Uint32Array {
       previousValue: number,
       currentValue: number,
       currentIndex: number,
-      array: Uint32Array,
+      array: this,
     ) => number,
   ): number;
   reduce(
@@ -4451,7 +4431,7 @@ interface Uint32Array {
       previousValue: number,
       currentValue: number,
       currentIndex: number,
-      array: Uint32Array,
+      array: this,
     ) => number,
     initialValue: number,
   ): number;
@@ -4471,7 +4451,7 @@ interface Uint32Array {
       previousValue: U,
       currentValue: number,
       currentIndex: number,
-      array: Uint32Array,
+      array: this,
     ) => U,
     initialValue: U,
   ): U;
@@ -4491,7 +4471,7 @@ interface Uint32Array {
       previousValue: number,
       currentValue: number,
       currentIndex: number,
-      array: Uint32Array,
+      array: this,
     ) => number,
   ): number;
   reduceRight(
@@ -4499,7 +4479,7 @@ interface Uint32Array {
       previousValue: number,
       currentValue: number,
       currentIndex: number,
-      array: Uint32Array,
+      array: this,
     ) => number,
     initialValue: number,
   ): number;
@@ -4519,7 +4499,7 @@ interface Uint32Array {
       previousValue: U,
       currentValue: number,
       currentIndex: number,
-      array: Uint32Array,
+      array: this,
     ) => U,
     initialValue: U,
   ): U;
@@ -4527,7 +4507,7 @@ interface Uint32Array {
   /**
    * Reverses the elements in an Array.
    */
-  reverse(): Uint32Array;
+  reverse(): this;
 
   /**
    * Sets a value or an array of values.
@@ -4541,7 +4521,7 @@ interface Uint32Array {
    * @param start The beginning of the specified portion of the array.
    * @param end The end of the specified portion of the array. This is exclusive of the element at the index 'end'.
    */
-  slice(start?: number, end?: number): Uint32Array;
+  slice(start?: number, end?: number): Uint32Array<ArrayBuffer>;
 
   /**
    * Determines whether the specified callback function returns true for any element of an array.
@@ -4552,7 +4532,7 @@ interface Uint32Array {
    * If thisArg is omitted, undefined is used as the this value.
    */
   some(
-    predicate: (value: number, index: number, array: Uint32Array) => unknown,
+    predicate: (value: number, index: number, array: this) => unknown,
     thisArg?: any,
   ): boolean;
 
@@ -4573,7 +4553,7 @@ interface Uint32Array {
    * @param begin The index of the beginning of the array.
    * @param end The index of the end of the array.
    */
-  subarray(begin?: number, end?: number): Uint32Array;
+  subarray(begin?: number, end?: number): Uint32Array<TArrayBuffer>;
 
   /**
    * Converts a number to a string by using the current locale.
@@ -4586,20 +4566,20 @@ interface Uint32Array {
   toString(): string;
 
   /** Returns the primitive value of the specified object. */
-  valueOf(): Uint32Array;
+  valueOf(): this;
 
   [index: number]: number;
 }
-
 interface Uint32ArrayConstructor {
-  readonly prototype: Uint32Array;
-  new (length: number): Uint32Array;
-  new (array: ArrayLike<number> | ArrayBufferLike): Uint32Array;
-  new (
-    buffer: ArrayBufferLike,
+  readonly prototype: Uint32Array<ArrayBufferLike>;
+  new (length: number): Uint32Array<ArrayBuffer>;
+  new (array: ArrayLike<number>): Uint32Array<ArrayBuffer>;
+  new <TArrayBuffer extends ArrayBufferLike = ArrayBuffer>(
+    buffer: TArrayBuffer,
     byteOffset?: number,
     length?: number,
-  ): Uint32Array;
+  ): Uint32Array<TArrayBuffer>;
+  new (array: ArrayLike<number> | ArrayBuffer): Uint32Array<ArrayBuffer>;
 
   /**
    * The size in bytes of each element in the array.
@@ -4610,13 +4590,13 @@ interface Uint32ArrayConstructor {
    * Returns a new array from a set of elements.
    * @param items A set of elements to include in the new array object.
    */
-  of(...items: number[]): Uint32Array;
+  of(...items: number[]): Uint32Array<ArrayBuffer>;
 
   /**
    * Creates an array from an array-like or iterable object.
    * @param arrayLike An array-like or iterable object to convert to an array.
    */
-  from(arrayLike: ArrayLike<number>): Uint32Array;
+  from(arrayLike: ArrayLike<number>): Uint32Array<ArrayBuffer>;
 
   /**
    * Creates an array from an array-like or iterable object.
@@ -4628,7 +4608,7 @@ interface Uint32ArrayConstructor {
     arrayLike: ArrayLike<T>,
     mapfn: (v: T, k: number) => number,
     thisArg?: any,
-  ): Uint32Array;
+  ): Uint32Array<ArrayBuffer>;
 }
 declare var Uint32Array: Uint32ArrayConstructor;
 
@@ -4636,7 +4616,7 @@ declare var Uint32Array: Uint32ArrayConstructor;
  * A typed array of 32-bit float values. The contents are initialized to 0. If the requested number
  * of bytes could not be allocated an exception is raised.
  */
-interface Float32Array {
+interface Float32Array<TArrayBuffer extends ArrayBufferLike = ArrayBufferLike> {
   /**
    * The size in bytes of each element in the array.
    */
@@ -4645,7 +4625,7 @@ interface Float32Array {
   /**
    * The ArrayBuffer instance referenced by the array.
    */
-  readonly buffer: ArrayBufferLike;
+  readonly buffer: TArrayBuffer;
 
   /**
    * The length in bytes of the array.
@@ -4677,7 +4657,7 @@ interface Float32Array {
    * If thisArg is omitted, undefined is used as the this value.
    */
   every(
-    predicate: (value: number, index: number, array: Float32Array) => unknown,
+    predicate: (value: number, index: number, array: this) => unknown,
     thisArg?: any,
   ): boolean;
 
@@ -4699,9 +4679,9 @@ interface Float32Array {
    * If thisArg is omitted, undefined is used as the this value.
    */
   filter(
-    predicate: (value: number, index: number, array: Float32Array) => any,
+    predicate: (value: number, index: number, array: this) => any,
     thisArg?: any,
-  ): Float32Array;
+  ): Float32Array<ArrayBuffer>;
 
   /**
    * Returns the value of the first element in the array where predicate is true, and undefined
@@ -4713,7 +4693,7 @@ interface Float32Array {
    * predicate. If it is not provided, undefined is used instead.
    */
   find(
-    predicate: (value: number, index: number, obj: Float32Array) => boolean,
+    predicate: (value: number, index: number, obj: this) => boolean,
     thisArg?: any,
   ): number | undefined;
 
@@ -4727,7 +4707,7 @@ interface Float32Array {
    * predicate. If it is not provided, undefined is used instead.
    */
   findIndex(
-    predicate: (value: number, index: number, obj: Float32Array) => boolean,
+    predicate: (value: number, index: number, obj: this) => boolean,
     thisArg?: any,
   ): number;
 
@@ -4739,7 +4719,7 @@ interface Float32Array {
    * If thisArg is omitted, undefined is used as the this value.
    */
   forEach(
-    callbackfn: (value: number, index: number, array: Float32Array) => void,
+    callbackfn: (value: number, index: number, array: this) => void,
     thisArg?: any,
   ): void;
 
@@ -4780,9 +4760,9 @@ interface Float32Array {
    * If thisArg is omitted, undefined is used as the this value.
    */
   map(
-    callbackfn: (value: number, index: number, array: Float32Array) => number,
+    callbackfn: (value: number, index: number, array: this) => number,
     thisArg?: any,
-  ): Float32Array;
+  ): Float32Array<ArrayBuffer>;
 
   /**
    * Calls the specified callback function for all the elements in an array. The return value of
@@ -4799,7 +4779,7 @@ interface Float32Array {
       previousValue: number,
       currentValue: number,
       currentIndex: number,
-      array: Float32Array,
+      array: this,
     ) => number,
   ): number;
   reduce(
@@ -4807,7 +4787,7 @@ interface Float32Array {
       previousValue: number,
       currentValue: number,
       currentIndex: number,
-      array: Float32Array,
+      array: this,
     ) => number,
     initialValue: number,
   ): number;
@@ -4827,7 +4807,7 @@ interface Float32Array {
       previousValue: U,
       currentValue: number,
       currentIndex: number,
-      array: Float32Array,
+      array: this,
     ) => U,
     initialValue: U,
   ): U;
@@ -4847,7 +4827,7 @@ interface Float32Array {
       previousValue: number,
       currentValue: number,
       currentIndex: number,
-      array: Float32Array,
+      array: this,
     ) => number,
   ): number;
   reduceRight(
@@ -4855,7 +4835,7 @@ interface Float32Array {
       previousValue: number,
       currentValue: number,
       currentIndex: number,
-      array: Float32Array,
+      array: this,
     ) => number,
     initialValue: number,
   ): number;
@@ -4875,7 +4855,7 @@ interface Float32Array {
       previousValue: U,
       currentValue: number,
       currentIndex: number,
-      array: Float32Array,
+      array: this,
     ) => U,
     initialValue: U,
   ): U;
@@ -4883,7 +4863,7 @@ interface Float32Array {
   /**
    * Reverses the elements in an Array.
    */
-  reverse(): Float32Array;
+  reverse(): this;
 
   /**
    * Sets a value or an array of values.
@@ -4897,7 +4877,7 @@ interface Float32Array {
    * @param start The beginning of the specified portion of the array.
    * @param end The end of the specified portion of the array. This is exclusive of the element at the index 'end'.
    */
-  slice(start?: number, end?: number): Float32Array;
+  slice(start?: number, end?: number): Float32Array<ArrayBuffer>;
 
   /**
    * Determines whether the specified callback function returns true for any element of an array.
@@ -4908,7 +4888,7 @@ interface Float32Array {
    * If thisArg is omitted, undefined is used as the this value.
    */
   some(
-    predicate: (value: number, index: number, array: Float32Array) => unknown,
+    predicate: (value: number, index: number, array: this) => unknown,
     thisArg?: any,
   ): boolean;
 
@@ -4929,7 +4909,7 @@ interface Float32Array {
    * @param begin The index of the beginning of the array.
    * @param end The index of the end of the array.
    */
-  subarray(begin?: number, end?: number): Float32Array;
+  subarray(begin?: number, end?: number): Float32Array<TArrayBuffer>;
 
   /**
    * Converts a number to a string by using the current locale.
@@ -4942,20 +4922,20 @@ interface Float32Array {
   toString(): string;
 
   /** Returns the primitive value of the specified object. */
-  valueOf(): Float32Array;
+  valueOf(): this;
 
   [index: number]: number;
 }
-
 interface Float32ArrayConstructor {
-  readonly prototype: Float32Array;
-  new (length: number): Float32Array;
-  new (array: ArrayLike<number> | ArrayBufferLike): Float32Array;
-  new (
-    buffer: ArrayBufferLike,
+  readonly prototype: Float32Array<ArrayBufferLike>;
+  new (length: number): Float32Array<ArrayBuffer>;
+  new (array: ArrayLike<number>): Float32Array<ArrayBuffer>;
+  new <TArrayBuffer extends ArrayBufferLike = ArrayBuffer>(
+    buffer: TArrayBuffer,
     byteOffset?: number,
     length?: number,
-  ): Float32Array;
+  ): Float32Array<TArrayBuffer>;
+  new (array: ArrayLike<number> | ArrayBuffer): Float32Array<ArrayBuffer>;
 
   /**
    * The size in bytes of each element in the array.
@@ -4966,13 +4946,13 @@ interface Float32ArrayConstructor {
    * Returns a new array from a set of elements.
    * @param items A set of elements to include in the new array object.
    */
-  of(...items: number[]): Float32Array;
+  of(...items: number[]): Float32Array<ArrayBuffer>;
 
   /**
    * Creates an array from an array-like or iterable object.
    * @param arrayLike An array-like or iterable object to convert to an array.
    */
-  from(arrayLike: ArrayLike<number>): Float32Array;
+  from(arrayLike: ArrayLike<number>): Float32Array<ArrayBuffer>;
 
   /**
    * Creates an array from an array-like or iterable object.
@@ -4984,7 +4964,7 @@ interface Float32ArrayConstructor {
     arrayLike: ArrayLike<T>,
     mapfn: (v: T, k: number) => number,
     thisArg?: any,
-  ): Float32Array;
+  ): Float32Array<ArrayBuffer>;
 }
 declare var Float32Array: Float32ArrayConstructor;
 
@@ -4992,7 +4972,7 @@ declare var Float32Array: Float32ArrayConstructor;
  * A typed array of 64-bit float values. The contents are initialized to 0. If the requested
  * number of bytes could not be allocated an exception is raised.
  */
-interface Float64Array {
+interface Float64Array<TArrayBuffer extends ArrayBufferLike = ArrayBufferLike> {
   /**
    * The size in bytes of each element in the array.
    */
@@ -5001,7 +4981,7 @@ interface Float64Array {
   /**
    * The ArrayBuffer instance referenced by the array.
    */
-  readonly buffer: ArrayBufferLike;
+  readonly buffer: TArrayBuffer;
 
   /**
    * The length in bytes of the array.
@@ -5033,7 +5013,7 @@ interface Float64Array {
    * If thisArg is omitted, undefined is used as the this value.
    */
   every(
-    predicate: (value: number, index: number, array: Float64Array) => unknown,
+    predicate: (value: number, index: number, array: this) => unknown,
     thisArg?: any,
   ): boolean;
 
@@ -5055,9 +5035,9 @@ interface Float64Array {
    * If thisArg is omitted, undefined is used as the this value.
    */
   filter(
-    predicate: (value: number, index: number, array: Float64Array) => any,
+    predicate: (value: number, index: number, array: this) => any,
     thisArg?: any,
-  ): Float64Array;
+  ): Float64Array<ArrayBuffer>;
 
   /**
    * Returns the value of the first element in the array where predicate is true, and undefined
@@ -5069,7 +5049,7 @@ interface Float64Array {
    * predicate. If it is not provided, undefined is used instead.
    */
   find(
-    predicate: (value: number, index: number, obj: Float64Array) => boolean,
+    predicate: (value: number, index: number, obj: this) => boolean,
     thisArg?: any,
   ): number | undefined;
 
@@ -5083,7 +5063,7 @@ interface Float64Array {
    * predicate. If it is not provided, undefined is used instead.
    */
   findIndex(
-    predicate: (value: number, index: number, obj: Float64Array) => boolean,
+    predicate: (value: number, index: number, obj: this) => boolean,
     thisArg?: any,
   ): number;
 
@@ -5095,7 +5075,7 @@ interface Float64Array {
    * If thisArg is omitted, undefined is used as the this value.
    */
   forEach(
-    callbackfn: (value: number, index: number, array: Float64Array) => void,
+    callbackfn: (value: number, index: number, array: this) => void,
     thisArg?: any,
   ): void;
 
@@ -5136,9 +5116,9 @@ interface Float64Array {
    * If thisArg is omitted, undefined is used as the this value.
    */
   map(
-    callbackfn: (value: number, index: number, array: Float64Array) => number,
+    callbackfn: (value: number, index: number, array: this) => number,
     thisArg?: any,
-  ): Float64Array;
+  ): Float64Array<ArrayBuffer>;
 
   /**
    * Calls the specified callback function for all the elements in an array. The return value of
@@ -5155,7 +5135,7 @@ interface Float64Array {
       previousValue: number,
       currentValue: number,
       currentIndex: number,
-      array: Float64Array,
+      array: this,
     ) => number,
   ): number;
   reduce(
@@ -5163,7 +5143,7 @@ interface Float64Array {
       previousValue: number,
       currentValue: number,
       currentIndex: number,
-      array: Float64Array,
+      array: this,
     ) => number,
     initialValue: number,
   ): number;
@@ -5183,7 +5163,7 @@ interface Float64Array {
       previousValue: U,
       currentValue: number,
       currentIndex: number,
-      array: Float64Array,
+      array: this,
     ) => U,
     initialValue: U,
   ): U;
@@ -5203,7 +5183,7 @@ interface Float64Array {
       previousValue: number,
       currentValue: number,
       currentIndex: number,
-      array: Float64Array,
+      array: this,
     ) => number,
   ): number;
   reduceRight(
@@ -5211,7 +5191,7 @@ interface Float64Array {
       previousValue: number,
       currentValue: number,
       currentIndex: number,
-      array: Float64Array,
+      array: this,
     ) => number,
     initialValue: number,
   ): number;
@@ -5231,7 +5211,7 @@ interface Float64Array {
       previousValue: U,
       currentValue: number,
       currentIndex: number,
-      array: Float64Array,
+      array: this,
     ) => U,
     initialValue: U,
   ): U;
@@ -5239,7 +5219,7 @@ interface Float64Array {
   /**
    * Reverses the elements in an Array.
    */
-  reverse(): Float64Array;
+  reverse(): this;
 
   /**
    * Sets a value or an array of values.
@@ -5253,7 +5233,7 @@ interface Float64Array {
    * @param start The beginning of the specified portion of the array.
    * @param end The end of the specified portion of the array. This is exclusive of the element at the index 'end'.
    */
-  slice(start?: number, end?: number): Float64Array;
+  slice(start?: number, end?: number): Float64Array<ArrayBuffer>;
 
   /**
    * Determines whether the specified callback function returns true for any element of an array.
@@ -5264,7 +5244,7 @@ interface Float64Array {
    * If thisArg is omitted, undefined is used as the this value.
    */
   some(
-    predicate: (value: number, index: number, array: Float64Array) => unknown,
+    predicate: (value: number, index: number, array: this) => unknown,
     thisArg?: any,
   ): boolean;
 
@@ -5285,7 +5265,7 @@ interface Float64Array {
    * @param begin The index of the beginning of the array.
    * @param end The index of the end of the array.
    */
-  subarray(begin?: number, end?: number): Float64Array;
+  subarray(begin?: number, end?: number): Float64Array<TArrayBuffer>;
 
   /**
    * Converts a number to a string by using the current locale.
@@ -5298,20 +5278,20 @@ interface Float64Array {
   toString(): string;
 
   /** Returns the primitive value of the specified object. */
-  valueOf(): Float64Array;
+  valueOf(): this;
 
   [index: number]: number;
 }
-
 interface Float64ArrayConstructor {
-  readonly prototype: Float64Array;
-  new (length: number): Float64Array;
-  new (array: ArrayLike<number> | ArrayBufferLike): Float64Array;
-  new (
-    buffer: ArrayBufferLike,
+  readonly prototype: Float64Array<ArrayBufferLike>;
+  new (length: number): Float64Array<ArrayBuffer>;
+  new (array: ArrayLike<number>): Float64Array<ArrayBuffer>;
+  new <TArrayBuffer extends ArrayBufferLike = ArrayBuffer>(
+    buffer: TArrayBuffer,
     byteOffset?: number,
     length?: number,
-  ): Float64Array;
+  ): Float64Array<TArrayBuffer>;
+  new (array: ArrayLike<number> | ArrayBuffer): Float64Array<ArrayBuffer>;
 
   /**
    * The size in bytes of each element in the array.
@@ -5322,13 +5302,13 @@ interface Float64ArrayConstructor {
    * Returns a new array from a set of elements.
    * @param items A set of elements to include in the new array object.
    */
-  of(...items: number[]): Float64Array;
+  of(...items: number[]): Float64Array<ArrayBuffer>;
 
   /**
    * Creates an array from an array-like or iterable object.
    * @param arrayLike An array-like or iterable object to convert to an array.
    */
-  from(arrayLike: ArrayLike<number>): Float64Array;
+  from(arrayLike: ArrayLike<number>): Float64Array<ArrayBuffer>;
 
   /**
    * Creates an array from an array-like or iterable object.
@@ -5340,7 +5320,7 @@ interface Float64ArrayConstructor {
     arrayLike: ArrayLike<T>,
     mapfn: (v: T, k: number) => number,
     thisArg?: any,
-  ): Float64Array;
+  ): Float64Array<ArrayBuffer>;
 }
 declare var Float64Array: Float64ArrayConstructor;
 
