diff --git a/packages/v5.6/temp/copied/lib.es2020.string.d.ts b/packages/v5.7/temp/copied/lib.es2020.string.d.ts
index ad45314..7da0ebd 100644
--- a/packages/v5.6/temp/copied/lib.es2020.string.d.ts
+++ b/packages/v5.7/temp/copied/lib.es2020.string.d.ts
@@ -15,6 +15,8 @@ and limitations under the License.
 
 /// <reference no-default-lib="true"/>
 
+/// <reference lib="es2015.iterable" />
+/// <reference lib="es2020.intl" />
 /// <reference lib="es2020.symbol.wellknown" />
 
 interface String {
