diff --git a/packages/v5.6/temp/copied/lib.es2016.array.include.d.ts b/packages/v5.7/temp/copied/lib.es2016.array.include.d.ts
index 137657d..faa8002 100644
--- a/packages/v5.6/temp/copied/lib.es2016.array.include.d.ts
+++ b/packages/v5.7/temp/copied/lib.es2016.array.include.d.ts
@@ -33,7 +33,7 @@ interface ReadonlyArray<T> {
   includes(searchElement: T, fromIndex?: number): boolean;
 }
 
-interface Int8Array {
+interface Int8Array<TArrayBuffer extends ArrayBufferLike> {
   /**
    * Determines whether an array includes a certain element, returning true or false as appropriate.
    * @param searchElement The element to search for.
@@ -42,7 +42,7 @@ interface Int8Array {
   includes(searchElement: number, fromIndex?: number): boolean;
 }
 
-interface Uint8Array {
+interface Uint8Array<TArrayBuffer extends ArrayBufferLike> {
   /**
    * Determines whether an array includes a certain element, returning true or false as appropriate.
    * @param searchElement The element to search for.
@@ -51,7 +51,7 @@ interface Uint8Array {
   includes(searchElement: number, fromIndex?: number): boolean;
 }
 
-interface Uint8ClampedArray {
+interface Uint8ClampedArray<TArrayBuffer extends ArrayBufferLike> {
   /**
    * Determines whether an array includes a certain element, returning true or false as appropriate.
    * @param searchElement The element to search for.
@@ -60,7 +60,7 @@ interface Uint8ClampedArray {
   includes(searchElement: number, fromIndex?: number): boolean;
 }
 
-interface Int16Array {
+interface Int16Array<TArrayBuffer extends ArrayBufferLike> {
   /**
    * Determines whether an array includes a certain element, returning true or false as appropriate.
    * @param searchElement The element to search for.
@@ -69,7 +69,7 @@ interface Int16Array {
   includes(searchElement: number, fromIndex?: number): boolean;
 }
 
-interface Uint16Array {
+interface Uint16Array<TArrayBuffer extends ArrayBufferLike> {
   /**
    * Determines whether an array includes a certain element, returning true or false as appropriate.
    * @param searchElement The element to search for.
@@ -78,7 +78,7 @@ interface Uint16Array {
   includes(searchElement: number, fromIndex?: number): boolean;
 }
 
-interface Int32Array {
+interface Int32Array<TArrayBuffer extends ArrayBufferLike> {
   /**
    * Determines whether an array includes a certain element, returning true or false as appropriate.
    * @param searchElement The element to search for.
@@ -87,7 +87,7 @@ interface Int32Array {
   includes(searchElement: number, fromIndex?: number): boolean;
 }
 
-interface Uint32Array {
+interface Uint32Array<TArrayBuffer extends ArrayBufferLike> {
   /**
    * Determines whether an array includes a certain element, returning true or false as appropriate.
    * @param searchElement The element to search for.
@@ -96,7 +96,7 @@ interface Uint32Array {
   includes(searchElement: number, fromIndex?: number): boolean;
 }
 
-interface Float32Array {
+interface Float32Array<TArrayBuffer extends ArrayBufferLike> {
   /**
    * Determines whether an array includes a certain element, returning true or false as appropriate.
    * @param searchElement The element to search for.
@@ -105,7 +105,7 @@ interface Float32Array {
   includes(searchElement: number, fromIndex?: number): boolean;
 }
 
-interface Float64Array {
+interface Float64Array<TArrayBuffer extends ArrayBufferLike> {
   /**
    * Determines whether an array includes a certain element, returning true or false as appropriate.
    * @param searchElement The element to search for.
