diff --git a/packages/v5.6/temp/copied/lib.es2016.intl.d.ts b/packages/v5.7/temp/copied/lib.es2016.intl.d.ts
index 6b536b3..9a26f46 100644
--- a/packages/v5.6/temp/copied/lib.es2016.intl.d.ts
+++ b/packages/v5.7/temp/copied/lib.es2016.intl.d.ts
@@ -21,7 +21,7 @@ declare namespace Intl {
    * the canonical locale names. Duplicates will be omitted and elements
    * will be validated as structurally valid language tags.
    *
-   * [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/getCanonicalLocales)
+   * [MDN](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/Intl/getCanonicalLocales)
    *
    * @param locale A list of String values for which to get the canonical locale names
    * @returns An array containing the canonical and validated locale names.
