diff --git a/packages/v5.6/temp/copied/lib.esnext.array.d.ts b/packages/v5.7/temp/copied/lib.esnext.array.d.ts
index f844678..ee1d9b3 100644
--- a/packages/v5.6/temp/copied/lib.esnext.array.d.ts
+++ b/packages/v5.7/temp/copied/lib.esnext.array.d.ts
@@ -37,7 +37,7 @@ interface ArrayConstructor {
    */
   fromAsync<T, U>(
     iterableOrArrayLike: AsyncIterable<T> | Iterable<T> | ArrayLike<T>,
-    mapFn: (value: Awaited<T>) => U,
+    mapFn: (value: Awaited<T>, index: number) => U,
     thisArg?: any,
   ): Promise<Awaited<U>[]>;
 }
