diff --git a/packages/v5.0/output-branded/lib-files/lib.es2022.intl.d.ts b/packages/v5.1/output-branded/lib-files/lib.es2022.intl.d.ts
index 5ae1977..b4d3d90 100644
--- a/packages/v5.0/output-branded/lib-files/lib.es2022.intl.d.ts
+++ b/packages/v5.1/output-branded/lib-files/lib.es2022.intl.d.ts
@@ -96,4 +96,21 @@ declare namespace Intl {
       options?: Pick<SegmenterOptions, 'localeMatcher'>,
     ): readonly BCP47LanguageTag[];
   };
+
+  /**
+   * Returns a sorted array of the supported collation, calendar, currency, numbering system, timezones, and units by the implementation.
+   * [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/supportedValuesOf)
+   *
+   * @param key A string indicating the category of values to return.
+   * @returns A sorted array of the supported values.
+   */
+  function supportedValuesOf(
+    key:
+      | 'calendar'
+      | 'collation'
+      | 'currency'
+      | 'numberingSystem'
+      | 'timeZone'
+      | 'unit',
+  ): readonly string[];
 }
