diff --git experimental/packages/apps/event-schedule-app/src/components/atoms/icon.tsx apps/event-schedule-app/src/components/atoms/icon.tsx
index 00c4ac830..fc8d756f1 100644
--- experimental/packages/apps/event-schedule-app/src/components/atoms/icon.tsx
+++ apps/event-schedule-app/src/components/atoms/icon.tsx
@@ -1,5 +1,7 @@
 import { Icon } from '@blueprintjs/core';
-import { CircleIcon, CloseIcon, TriangleIcon } from './icons';
+import * as React from 'react';
+import { memoNamed } from 'react-utils';
+import { CircleIcon, CloseIcon, TriangleIcon } from './icons/index.mjs';
 
 type Props = Readonly<{
   iconName: AnswerIconIdWithNone;
@@ -10,7 +12,7 @@ type Props = Readonly<{
 export const CustomIcon = memoNamed<Props>(
   'MyIcon',
   ({ iconName, color, size }) => {
-    const icon = useMemo<React.JSX.Element>(() => {
+    const icon = React.useMemo<React.JSX.Element>(() => {
       switch (iconName) {
         case 'good':
           return <CircleIcon color={color} size={size} />;
