diff --git experimental/packages/apps/poll-discord-app/src/functions/convert-rp30-args-to-rp-args.mts apps/poll-discord-app/src/functions/convert-rp30-args-to-rp-args.mts
index 7d8bff2c4..316ab747b 100644
--- experimental/packages/apps/poll-discord-app/src/functions/convert-rp30-args-to-rp-args.mts
+++ apps/poll-discord-app/src/functions/convert-rp30-args-to-rp-args.mts
@@ -1,4 +1,6 @@
-import { daysOfWeekList } from '@noshiro/ts-utils-additional';
+import { Arr, Result, pipe } from 'ts-data-forge';
+import { type DeepReadonly } from 'ts-type-forge';
+import { DateUtils, daysOfWeekList } from '../utils/index.mjs';
 import {
   rp3060ParseCommand,
   rp3060dParseCommand,
@@ -15,7 +17,7 @@ export const convertRp60ArgToRpArgsShared = (
 > => {
   const argsConverted: readonly string[] = pipe(
     Arr.range(arg1AsNumber, arg2AsNumber),
-  ).chain((list) => list.map((hour) => `${hour}:00-${hour + 1}:00`)).value;
+  ).map((list) => list.map((hour) => `${hour}:00-${hour + 1}:00`)).value;
 
   return Result.ok({ title, args: argsConverted });
 };
@@ -53,9 +55,10 @@ export const convertRp60dArgToRpArgs = (
   const [arg1AsNumber, arg2AsNumber] = res.value;
 
   const td = DateUtils.today();
+
   const title = `${DateUtils.getLocaleMonth(td)}/${DateUtils.getLocaleDate(
     td,
-  )}（${daysOfWeekList.jp[DateUtils.getLocaleDayOfWeek(td)].abbr}）`;
+  )}（${daysOfWeekList.jp[DateUtils.getLocaleDayOfWeek(td)].abbr}）` as const;
 
   return convertRp60ArgToRpArgsShared(title, arg1AsNumber, arg2AsNumber);
 };
@@ -70,8 +73,8 @@ export const convertRp30ArgToRpArgsShared = (
 > => {
   const argsConverted: readonly string[] = pipe(
     Arr.range(arg1AsNumber, arg2AsNumber),
-  ).chain((list) =>
-    Arr.flatMap(list, (hour) => [
+  ).map(
+    Arr.flatMap((hour) => [
       `${hour}:00-${hour}:30`,
       `${hour}:30-${hour + 1}:00`,
     ]),
@@ -110,9 +113,10 @@ export const convertRp30dArgToRpArgs = (
   const [arg1AsNumber, arg2AsNumber] = res.value;
 
   const td = DateUtils.today();
+
   const title = `${DateUtils.getLocaleMonth(td)}/${DateUtils.getLocaleDate(
     td,
-  )}（${daysOfWeekList.jp[DateUtils.getLocaleDayOfWeek(td)].abbr}）`;
+  )}（${daysOfWeekList.jp[DateUtils.getLocaleDayOfWeek(td)].abbr}）` as const;
 
   return convertRp30ArgToRpArgsShared(title, arg1AsNumber, arg2AsNumber);
 };
