diff --git experimental/packages/apps/cant-stop-probability-app/src/types/dice-value.ts apps/cant-stop-probability-app/src/types/dice-value.mts
index f83996fce..848551354 100644
--- experimental/packages/apps/cant-stop-probability-app/src/types/dice-value.ts
+++ apps/cant-stop-probability-app/src/types/dice-value.mts
@@ -1 +1,3 @@
-export type DiceValue = UintRange<1, 7>;
+import { type UintRangeInclusive } from 'ts-type-forge';
+
+export type DiceValue = UintRangeInclusive<1, 6>;
