diff --git a/packages/v5.7/output-branded/lib-files/lib.es2023.array.d.ts b/packages/v5.8/output-branded/lib-files/lib.es2023.array.d.ts
index 523d7da..70172c8 100644
--- a/packages/v5.7/output-branded/lib-files/lib.es2023.array.d.ts
+++ b/packages/v5.8/output-branded/lib-files/lib.es2023.array.d.ts
@@ -55,7 +55,7 @@ interface Array<T> {
    * Returns a copy of an array with its elements sorted.
    * @param compareFn Function used to determine the order of the elements. It is expected to return
    * a negative value if the first argument is less than the second argument, zero if they're equal, and a positive
-   * value otherwise. If omitted, the elements are sorted in ascending, ASCII character order.
+   * value otherwise. If omitted, the elements are sorted in ascending, UTF-16 code unit order.
    * ```ts
    * [11, 2, 22, 1].toSorted((a, b) => a - b) // [1, 2, 11, 22]
    * ```
@@ -152,7 +152,7 @@ interface ReadonlyArray<T> {
    * Returns a copy of an array with its elements sorted.
    * @param compareFn Function used to determine the order of the elements. It is expected to return
    * a negative value if the first argument is less than the second argument, zero if they're equal, and a positive
-   * value otherwise. If omitted, the elements are sorted in ascending, ASCII character order.
+   * value otherwise. If omitted, the elements are sorted in ascending, UTF-16 code unit order.
    * ```ts
    * [11, 2, 22, 1].toSorted((a, b) => a - b) // [1, 2, 11, 22]
    * ```
@@ -251,8 +251,8 @@ interface Int8Array<TArrayBuffer extends ArrayBufferLike> {
    * a negative value if the first argument is less than the second argument, zero if they're equal, and a positive
    * value otherwise. If omitted, the elements are sorted in ascending order.
    * ```ts
-   * const myNums = Int8Array<Buffer>.from([11, 2, 22, 1]);
-   * myNums.toSorted((a, b) => a - b) // Int8Array<Buffer>(4) [1, 2, 11, 22]
+   * const myNums = Int8Array.from([11, 2, 22, 1]);
+   * myNums.toSorted((a, b) => a - b) // Int8Array(4) [1, 2, 11, 22]
    * ```
    */
   toSorted(compareFn?: (a: Int8, b: Int8) => number): Int8Array<ArrayBuffer>;
@@ -328,8 +328,8 @@ interface Uint8Array<TArrayBuffer extends ArrayBufferLike> {
    * a negative value if the first argument is less than the second argument, zero if they're equal, and a positive
    * value otherwise. If omitted, the elements are sorted in ascending order.
    * ```ts
-   * const myNums = Uint8Array<Buffer>.from([11, 2, 22, 1]);
-   * myNums.toSorted((a, b) => a - b) // Uint8Array<Buffer>(4) [1, 2, 11, 22]
+   * const myNums = Uint8Array.from([11, 2, 22, 1]);
+   * myNums.toSorted((a, b) => a - b) // Uint8Array(4) [1, 2, 11, 22]
    * ```
    */
   toSorted(compareFn?: (a: Uint8, b: Uint8) => number): Uint8Array<ArrayBuffer>;
@@ -405,8 +405,8 @@ interface Uint8ClampedArray<TArrayBuffer extends ArrayBufferLike> {
    * a negative value if the first argument is less than the second argument, zero if they're equal, and a positive
    * value otherwise. If omitted, the elements are sorted in ascending order.
    * ```ts
-   * const myNums = Uint8ClampedArray<Buffer>.from([11, 2, 22, 1]);
-   * myNums.toSorted((a, b) => a - b) // Uint8ClampedArray<Buffer>(4) [1, 2, 11, 22]
+   * const myNums = Uint8ClampedArray.from([11, 2, 22, 1]);
+   * myNums.toSorted((a, b) => a - b) // Uint8ClampedArray(4) [1, 2, 11, 22]
    * ```
    */
   toSorted(
@@ -484,8 +484,8 @@ interface Int16Array<TArrayBuffer extends ArrayBufferLike> {
    * a negative value if the first argument is less than the second argument, zero if they're equal, and a positive
    * value otherwise. If omitted, the elements are sorted in ascending order.
    * ```ts
-   * const myNums = Int16Array<Buffer>.from([11, 2, -22, 1]);
-   * myNums.toSorted((a, b) => a - b) // Int16Array<Buffer>(4) [-22, 1, 2, 11]
+   * const myNums = Int16Array.from([11, 2, -22, 1]);
+   * myNums.toSorted((a, b) => a - b) // Int16Array(4) [-22, 1, 2, 11]
    * ```
    */
   toSorted(compareFn?: (a: Int16, b: Int16) => number): Int16Array<ArrayBuffer>;
@@ -561,8 +561,8 @@ interface Uint16Array<TArrayBuffer extends ArrayBufferLike> {
    * a negative value if the first argument is less than the second argument, zero if they're equal, and a positive
    * value otherwise. If omitted, the elements are sorted in ascending order.
    * ```ts
-   * const myNums = Uint16Array<Buffer>.from([11, 2, 22, 1]);
-   * myNums.toSorted((a, b) => a - b) // Uint16Array<Buffer>(4) [1, 2, 11, 22]
+   * const myNums = Uint16Array.from([11, 2, 22, 1]);
+   * myNums.toSorted((a, b) => a - b) // Uint16Array(4) [1, 2, 11, 22]
    * ```
    */
   toSorted(
@@ -640,8 +640,8 @@ interface Int32Array<TArrayBuffer extends ArrayBufferLike> {
    * a negative value if the first argument is less than the second argument, zero if they're equal, and a positive
    * value otherwise. If omitted, the elements are sorted in ascending order.
    * ```ts
-   * const myNums = Int32Array<Buffer>.from([11, 2, -22, 1]);
-   * myNums.toSorted((a, b) => a - b) // Int32Array<Buffer>(4) [-22, 1, 2, 11]
+   * const myNums = Int32Array.from([11, 2, -22, 1]);
+   * myNums.toSorted((a, b) => a - b) // Int32Array(4) [-22, 1, 2, 11]
    * ```
    */
   toSorted(compareFn?: (a: Int32, b: Int32) => number): Int32Array<ArrayBuffer>;
@@ -717,8 +717,8 @@ interface Uint32Array<TArrayBuffer extends ArrayBufferLike> {
    * a negative value if the first argument is less than the second argument, zero if they're equal, and a positive
    * value otherwise. If omitted, the elements are sorted in ascending order.
    * ```ts
-   * const myNums = Uint32Array<Buffer>.from([11, 2, 22, 1]);
-   * myNums.toSorted((a, b) => a - b) // Uint32Array<Buffer>(4) [1, 2, 11, 22]
+   * const myNums = Uint32Array.from([11, 2, 22, 1]);
+   * myNums.toSorted((a, b) => a - b) // Uint32Array(4) [1, 2, 11, 22]
    * ```
    */
   toSorted(
@@ -796,8 +796,8 @@ interface Float32Array<TArrayBuffer extends ArrayBufferLike> {
    * a negative value if the first argument is less than the second argument, zero if they're equal, and a positive
    * value otherwise. If omitted, the elements are sorted in ascending order.
    * ```ts
-   * const myNums = Float32Array<Buffer>.from([11.25, 2, -22.5, 1]);
-   * myNums.toSorted((a, b) => a - b) // Float32Array<Buffer>(4) [-22.5, 1, 2, 11.5]
+   * const myNums = Float32Array.from([11.25, 2, -22.5, 1]);
+   * myNums.toSorted((a, b) => a - b) // Float32Array(4) [-22.5, 1, 2, 11.5]
    * ```
    */
   toSorted(
@@ -875,8 +875,8 @@ interface Float64Array<TArrayBuffer extends ArrayBufferLike> {
    * a negative value if the first argument is less than the second argument, zero if they're equal, and a positive
    * value otherwise. If omitted, the elements are sorted in ascending order.
    * ```ts
-   * const myNums = Float64Array<Buffer>.from([11.25, 2, -22.5, 1]);
-   * myNums.toSorted((a, b) => a - b) // Float64Array<Buffer>(4) [-22.5, 1, 2, 11.5]
+   * const myNums = Float64Array.from([11.25, 2, -22.5, 1]);
+   * myNums.toSorted((a, b) => a - b) // Float64Array(4) [-22.5, 1, 2, 11.5]
    * ```
    */
   toSorted(
@@ -954,8 +954,8 @@ interface BigInt64Array<TArrayBuffer extends ArrayBufferLike> {
    * a negative value if the first argument is less than the second argument, zero if they're equal, and a positive
    * value otherwise. If omitted, the elements are sorted in ascending order.
    * ```ts
-   * const myNums = BigInt64Array<Buffer>.from([11n, 2n, -22n, 1n]);
-   * myNums.toSorted((a, b) => Number(a - b)) // BigInt64Array<Buffer>(4) [-22n, 1n, 2n, 11n]
+   * const myNums = BigInt64Array.from([11n, 2n, -22n, 1n]);
+   * myNums.toSorted((a, b) => Number(a - b)) // BigInt64Array(4) [-22n, 1n, 2n, 11n]
    * ```
    */
   toSorted(
@@ -1033,8 +1033,8 @@ interface BigUint64Array<TArrayBuffer extends ArrayBufferLike> {
    * a negative value if the first argument is less than the second argument, zero if they're equal, and a positive
    * value otherwise. If omitted, the elements are sorted in ascending order.
    * ```ts
-   * const myNums = BigUint64Array<Buffer>.from([11n, 2n, 22n, 1n]);
-   * myNums.toSorted((a, b) => Number(a - b)) // BigUint64Array<Buffer>(4) [1n, 2n, 11n, 22n]
+   * const myNums = BigUint64Array.from([11n, 2n, 22n, 1n]);
+   * myNums.toSorted((a, b) => Number(a - b)) // BigUint64Array(4) [1n, 2n, 11n, 22n]
    * ```
    */
   toSorted(
