diff --git a/packages/v5.7/output-branded/lib-files/lib.webworker.d.ts b/packages/v5.8/output-branded/lib-files/lib.webworker.d.ts
index bdb4b1a..6456b1f 100644
--- a/packages/v5.7/output-branded/lib-files/lib.webworker.d.ts
+++ b/packages/v5.8/output-branded/lib-files/lib.webworker.d.ts
@@ -276,7 +276,7 @@ interface ExtendableMessageEventInit extends ExtendableEventInit {
 
 interface FetchEventInit extends ExtendableEventInit {
   clientId?: string;
-  handled?: Promise<undefined>;
+  handled?: Promise<void>;
   preloadResponse?: Promise<unknown>;
   replacesClientId?: string;
   request: Request;
@@ -384,6 +384,26 @@ interface ImageDataSettings {
   colorSpace?: PredefinedColorSpace;
 }
 
+interface ImageDecodeOptions {
+  completeFramesOnly?: boolean;
+  frameIndex?: number;
+}
+
+interface ImageDecodeResult {
+  complete: boolean;
+  image: VideoFrame;
+}
+
+interface ImageDecoderInit {
+  colorSpaceConversion?: ColorSpaceConversion;
+  data: ImageBufferSource;
+  desiredHeight?: number;
+  desiredWidth?: number;
+  preferAnimation?: boolean;
+  transfer?: readonly ArrayBuffer[];
+  type: string;
+}
+
 interface ImageEncodeOptions {
   quality?: number;
   type?: string;
@@ -2460,8 +2480,11 @@ interface DOMMatrix extends DOMMatrixReadOnly {
   m43: number;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMMatrix#instance_properties) */
   m44: number;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMMatrix/invertSelf) */
   invertSelf(): DOMMatrix;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMMatrix/multiplySelf) */
   multiplySelf(other?: DOMMatrixInit): DOMMatrix;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMMatrix/preMultiplySelf) */
   preMultiplySelf(other?: DOMMatrixInit): DOMMatrix;
   rotateAxisAngleSelf(
     x?: number,
@@ -2470,6 +2493,7 @@ interface DOMMatrix extends DOMMatrixReadOnly {
     angle?: number,
   ): DOMMatrix;
   rotateFromVectorSelf(x?: number, y?: number): DOMMatrix;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMMatrix/rotateSelf) */
   rotateSelf(rotX?: number, rotY?: number, rotZ?: number): DOMMatrix;
   scale3dSelf(
     scale?: number,
@@ -2485,8 +2509,11 @@ interface DOMMatrix extends DOMMatrixReadOnly {
     originY?: number,
     originZ?: number,
   ): DOMMatrix;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMMatrix/skewXSelf) */
   skewXSelf(sx?: number): DOMMatrix;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMMatrix/skewYSelf) */
   skewYSelf(sy?: number): DOMMatrix;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMMatrix/translateSelf) */
   translateSelf(tx?: number, ty?: number, tz?: number): DOMMatrix;
 }
 
@@ -2512,7 +2539,9 @@ interface DOMMatrixReadOnly {
   readonly e: number;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMMatrixReadOnly#instance_properties) */
   readonly f: number;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMMatrixReadOnly/is2D) */
   readonly is2D: boolean;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMMatrixReadOnly/isIdentity) */
   readonly isIdentity: boolean;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMMatrixReadOnly#instance_properties) */
   readonly m11: number;
@@ -2548,8 +2577,11 @@ interface DOMMatrixReadOnly {
   readonly m44: number;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMMatrixReadOnly/flipX) */
   flipX(): DOMMatrix;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMMatrixReadOnly/flipY) */
   flipY(): DOMMatrix;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMMatrixReadOnly/inverse) */
   inverse(): DOMMatrix;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMMatrixReadOnly/multiply) */
   multiply(other?: DOMMatrixInit): DOMMatrix;
   rotate(rotX?: number, rotY?: number, rotZ?: number): DOMMatrix;
   rotateAxisAngle(
@@ -2578,9 +2610,13 @@ interface DOMMatrixReadOnly {
   scaleNonUniform(scaleX?: number, scaleY?: number): DOMMatrix;
   skewX(sx?: number): DOMMatrix;
   skewY(sy?: number): DOMMatrix;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMMatrixReadOnly/toFloat32Array) */
   toFloat32Array(): Float32Array;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMMatrixReadOnly/toFloat64Array) */
   toFloat64Array(): Float64Array;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMMatrixReadOnly/toJSON) */
   toJSON(): unknown;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMMatrixReadOnly/transformPoint) */
   transformPoint(point?: DOMPointInit): DOMPoint;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMMatrixReadOnly/translate) */
   translate(tx?: number, ty?: number, tz?: number): DOMMatrix;
@@ -2623,6 +2659,7 @@ interface DOMPointReadOnly {
   readonly y: number;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMPointReadOnly/z) */
   readonly z: number;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMPointReadOnly/matrixTransform) */
   matrixTransform(matrix?: DOMMatrixInit): DOMPoint;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMPointReadOnly/toJSON) */
   toJSON(): unknown;
@@ -2637,11 +2674,17 @@ declare const DOMPointReadOnly: {
 
 /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMQuad) */
 interface DOMQuad {
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMQuad/p1) */
   readonly p1: DOMPoint;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMQuad/p2) */
   readonly p2: DOMPoint;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMQuad/p3) */
   readonly p3: DOMPoint;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMQuad/p4) */
   readonly p4: DOMPoint;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMQuad/getBounds) */
   getBounds(): DOMRect;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMQuad/toJSON) */
   toJSON(): unknown;
 }
 
@@ -2659,9 +2702,13 @@ declare const DOMQuad: {
 
 /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMRect) */
 interface DOMRect extends DOMRectReadOnly {
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMRect/height) */
   height: number;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMRect/width) */
   width: number;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMRect/x) */
   x: number;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMRect/y) */
   y: number;
 }
 
@@ -2690,6 +2737,7 @@ interface DOMRectReadOnly {
   readonly x: number;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMRectReadOnly/y) */
   readonly y: number;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DOMRectReadOnly/toJSON) */
   toJSON(): unknown;
 }
 
@@ -2748,7 +2796,8 @@ declare const DecompressionStream: {
   new (format: CompressionFormat): DecompressionStream;
 };
 
-interface DedicatedWorkerGlobalScopeEventMap extends WorkerGlobalScopeEventMap {
+interface DedicatedWorkerGlobalScopeEventMap
+  extends WorkerGlobalScopeEventMap, MessageEventTargetEventMap {
   message: MessageEvent;
   messageerror: MessageEvent;
   rtctransform: RTCTransformEvent;
@@ -2760,21 +2809,16 @@ interface DedicatedWorkerGlobalScopeEventMap extends WorkerGlobalScopeEventMap {
  * [MDN Reference](https://developer.mozilla.org/docs/Web/API/DedicatedWorkerGlobalScope)
  */
 interface DedicatedWorkerGlobalScope
-  extends WorkerGlobalScope, AnimationFrameProvider {
+  extends
+    WorkerGlobalScope,
+    AnimationFrameProvider,
+    MessageEventTarget<DedicatedWorkerGlobalScope> {
   /**
    * Returns dedicatedWorkerGlobal's name, i.e. the value given to the Worker constructor. Primarily useful for debugging.
    *
    * [MDN Reference](https://developer.mozilla.org/docs/Web/API/DedicatedWorkerGlobalScope/name)
    */
   readonly name: string;
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DedicatedWorkerGlobalScope/message_event) */
-  onmessage:
-    | ((this: DedicatedWorkerGlobalScope, ev: MessageEvent) => unknown)
-    | null;
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DedicatedWorkerGlobalScope/messageerror_event) */
-  onmessageerror:
-    | ((this: DedicatedWorkerGlobalScope, ev: MessageEvent) => unknown)
-    | null;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DedicatedWorkerGlobalScope/rtctransform_event) */
   onrtctransform:
     | ((this: DedicatedWorkerGlobalScope, ev: RTCTransformEvent) => unknown)
@@ -2945,10 +2989,15 @@ declare const EncodedVideoChunk: {
  * [MDN Reference](https://developer.mozilla.org/docs/Web/API/ErrorEvent)
  */
 interface ErrorEvent extends Event {
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/ErrorEvent/colno) */
   readonly colno: number;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/ErrorEvent/error) */
   readonly error: unknown;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/ErrorEvent/filename) */
   readonly filename: string;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/ErrorEvent/lineno) */
   readonly lineno: number;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/ErrorEvent/message) */
   readonly message: string;
 }
 
@@ -3276,7 +3325,7 @@ interface FetchEvent extends ExtendableEvent {
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/FetchEvent/clientId) */
   readonly clientId: string;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/FetchEvent/handled) */
-  readonly handled: Promise<undefined>;
+  readonly handled: Promise<void>;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/FetchEvent/preloadResponse) */
   readonly preloadResponse: Promise<unknown>;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/FetchEvent/request) */
@@ -3724,6 +3773,16 @@ declare const FormData: {
   new (): FormData;
 };
 
+/**
+ * Available only in secure contexts.
+ *
+ * [MDN Reference](https://developer.mozilla.org/docs/Web/API/GPUError)
+ */
+interface GPUError {
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/GPUError/message) */
+  readonly message: string;
+}
+
 interface GenericTransformStream {
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/CompressionStream/readable) */
   readonly readable: ReadableStream;
@@ -4640,6 +4699,70 @@ declare const ImageData: {
   ): ImageData;
 };
 
+/**
+ * Available only in secure contexts.
+ *
+ * [MDN Reference](https://developer.mozilla.org/docs/Web/API/ImageDecoder)
+ */
+interface ImageDecoder {
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/ImageDecoder/complete) */
+  readonly complete: boolean;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/ImageDecoder/completed) */
+  readonly completed: Promise<void>;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/ImageDecoder/tracks) */
+  readonly tracks: ImageTrackList;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/ImageDecoder/type) */
+  readonly type: string;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/ImageDecoder/close) */
+  close(): void;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/ImageDecoder/decode) */
+  decode(options?: ImageDecodeOptions): Promise<ImageDecodeResult>;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/ImageDecoder/reset) */
+  reset(): void;
+}
+
+declare const ImageDecoder: {
+  prototype: ImageDecoder;
+  new (init: ImageDecoderInit): ImageDecoder;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/ImageDecoder/isTypeSupported_static) */
+  isTypeSupported(type: string): Promise<boolean>;
+};
+
+/** [MDN Reference](https://developer.mozilla.org/docs/Web/API/ImageTrack) */
+interface ImageTrack {
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/ImageTrack/animated) */
+  readonly animated: boolean;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/ImageTrack/frameCount) */
+  readonly frameCount: number;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/ImageTrack/repetitionCount) */
+  readonly repetitionCount: number;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/ImageTrack/selected) */
+  selected: boolean;
+}
+
+declare const ImageTrack: {
+  prototype: ImageTrack;
+  new (): ImageTrack;
+};
+
+/** [MDN Reference](https://developer.mozilla.org/docs/Web/API/ImageTrackList) */
+interface ImageTrackList {
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/ImageTrackList/length) */
+  readonly length: number;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/ImageTrackList/ready) */
+  readonly ready: Promise<void>;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/ImageTrackList/selectedIndex) */
+  readonly selectedIndex: number;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/ImageTrackList/selectedTrack) */
+  readonly selectedTrack: ImageTrack | null;
+  [index: number]: ImageTrack;
+}
+
+declare const ImageTrackList: {
+  prototype: ImageTrackList;
+  new (): ImageTrackList;
+};
+
 interface ImportMeta {
   url: string;
   resolve(specifier: string): string;
@@ -4804,7 +4927,39 @@ declare const MessageEvent: {
   new <T>(type: string, eventInitDict?: MessageEventInit<T>): MessageEvent<T>;
 };
 
-interface MessagePortEventMap {
+interface MessageEventTargetEventMap {
+  message: MessageEvent;
+  messageerror: MessageEvent;
+}
+
+interface MessageEventTarget<T> {
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DedicatedWorkerGlobalScope/message_event) */
+  onmessage: ((this: T, ev: MessageEvent) => unknown) | null;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DedicatedWorkerGlobalScope/messageerror_event) */
+  onmessageerror: ((this: T, ev: MessageEvent) => unknown) | null;
+  addEventListener<K extends keyof MessageEventTargetEventMap>(
+    type: K,
+    listener: (this: T, ev: MessageEventTargetEventMap[K]) => unknown,
+    options?: boolean | AddEventListenerOptions,
+  ): void;
+  addEventListener(
+    type: string,
+    listener: EventListenerOrEventListenerObject,
+    options?: boolean | AddEventListenerOptions,
+  ): void;
+  removeEventListener<K extends keyof MessageEventTargetEventMap>(
+    type: K,
+    listener: (this: T, ev: MessageEventTargetEventMap[K]) => unknown,
+    options?: boolean | EventListenerOptions,
+  ): void;
+  removeEventListener(
+    type: string,
+    listener: EventListenerOrEventListenerObject,
+    options?: boolean | EventListenerOptions,
+  ): void;
+}
+
+interface MessagePortEventMap extends MessageEventTargetEventMap {
   message: MessageEvent;
   messageerror: MessageEvent;
 }
@@ -4814,11 +4969,7 @@ interface MessagePortEventMap {
  *
  * [MDN Reference](https://developer.mozilla.org/docs/Web/API/MessagePort)
  */
-interface MessagePort extends EventTarget {
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/MessagePort/message_event) */
-  onmessage: ((this: MessagePort, ev: MessageEvent) => unknown) | null;
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/MessagePort/messageerror_event) */
-  onmessageerror: ((this: MessagePort, ev: MessageEvent) => unknown) | null;
+interface MessagePort extends EventTarget, MessageEventTarget<MessagePort> {
   /**
    * Disconnects the port, so that it is no longer active.
    *
@@ -5734,6 +5885,85 @@ declare const PushSubscriptionOptions: {
   new (): PushSubscriptionOptions;
 };
 
+interface RTCDataChannelEventMap {
+  bufferedamountlow: Event;
+  close: Event;
+  closing: Event;
+  error: Event;
+  message: MessageEvent;
+  open: Event;
+}
+
+/** [MDN Reference](https://developer.mozilla.org/docs/Web/API/RTCDataChannel) */
+interface RTCDataChannel extends EventTarget {
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/RTCDataChannel/binaryType) */
+  binaryType: BinaryType;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/RTCDataChannel/bufferedAmount) */
+  readonly bufferedAmount: number;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/RTCDataChannel/bufferedAmountLowThreshold) */
+  bufferedAmountLowThreshold: number;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/RTCDataChannel/id) */
+  readonly id: number | null;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/RTCDataChannel/label) */
+  readonly label: string;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/RTCDataChannel/maxPacketLifeTime) */
+  readonly maxPacketLifeTime: number | null;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/RTCDataChannel/maxRetransmits) */
+  readonly maxRetransmits: number | null;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/RTCDataChannel/negotiated) */
+  readonly negotiated: boolean;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/RTCDataChannel/bufferedamountlow_event) */
+  onbufferedamountlow: ((this: RTCDataChannel, ev: Event) => unknown) | null;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/RTCDataChannel/close_event) */
+  onclose: ((this: RTCDataChannel, ev: Event) => unknown) | null;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/RTCDataChannel/closing_event) */
+  onclosing: ((this: RTCDataChannel, ev: Event) => unknown) | null;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/RTCDataChannel/error_event) */
+  onerror: ((this: RTCDataChannel, ev: Event) => unknown) | null;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/RTCDataChannel/message_event) */
+  onmessage: ((this: RTCDataChannel, ev: MessageEvent) => unknown) | null;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/RTCDataChannel/open_event) */
+  onopen: ((this: RTCDataChannel, ev: Event) => unknown) | null;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/RTCDataChannel/ordered) */
+  readonly ordered: boolean;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/RTCDataChannel/protocol) */
+  readonly protocol: string;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/RTCDataChannel/readyState) */
+  readonly readyState: RTCDataChannelState;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/RTCDataChannel/close) */
+  close(): void;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/RTCDataChannel/send) */
+  send(data: string): void;
+  send(data: Blob): void;
+  send(data: ArrayBuffer): void;
+  send(data: ArrayBufferView): void;
+  addEventListener<K extends keyof RTCDataChannelEventMap>(
+    type: K,
+    listener: (this: RTCDataChannel, ev: RTCDataChannelEventMap[K]) => unknown,
+    options?: boolean | AddEventListenerOptions,
+  ): void;
+  addEventListener(
+    type: string,
+    listener: EventListenerOrEventListenerObject,
+    options?: boolean | AddEventListenerOptions,
+  ): void;
+  removeEventListener<K extends keyof RTCDataChannelEventMap>(
+    type: K,
+    listener: (this: RTCDataChannel, ev: RTCDataChannelEventMap[K]) => unknown,
+    options?: boolean | EventListenerOptions,
+  ): void;
+  removeEventListener(
+    type: string,
+    listener: EventListenerOrEventListenerObject,
+    options?: boolean | EventListenerOptions,
+  ): void;
+}
+
+declare const RTCDataChannel: {
+  prototype: RTCDataChannel;
+  new (): RTCDataChannel;
+};
+
 /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/RTCEncodedAudioFrame) */
 interface RTCEncodedAudioFrame {
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/RTCEncodedAudioFrame/data) */
@@ -5923,7 +6153,7 @@ declare const ReadableStreamDefaultReader: {
 
 interface ReadableStreamGenericReader {
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/ReadableStreamBYOBReader/closed) */
-  readonly closed: Promise<undefined>;
+  readonly closed: Promise<void>;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/ReadableStreamBYOBReader/cancel) */
   cancel(reason?: unknown): Promise<void>;
 }
@@ -6009,7 +6239,11 @@ interface Request extends Body {
    * [MDN Reference](https://developer.mozilla.org/docs/Web/API/Request/integrity)
    */
   readonly integrity: string;
-  /** Returns a boolean indicating whether or not request can outlive the global in which it was created. */
+  /**
+   * Returns a boolean indicating whether or not request can outlive the global in which it was created.
+   *
+   * [MDN Reference](https://developer.mozilla.org/docs/Web/API/Request/keepalive)
+   */
   readonly keepalive: boolean;
   /**
    * Returns request's HTTP method, which is "GET" by default.
@@ -6595,7 +6829,7 @@ interface SubtleCrypto {
   ): Promise<ArrayBuffer | JsonWebKey>;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/SubtleCrypto/generateKey) */
   generateKey(
-    algorithm: 'Ed25519',
+    algorithm: 'Ed25519' | { name: 'Ed25519' },
     extractable: boolean,
     keyUsages: ReadonlyArray<'sign' | 'verify'>,
   ): Promise<CryptoKeyPair>;
@@ -8047,7 +8281,7 @@ interface WebGL2RenderingContextBase {
   ): void;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGL2RenderingContext/clientWaitSync) */
   clientWaitSync(sync: WebGLSync, flags: GLbitfield, timeout: GLuint64): GLenum;
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGLRenderingContext/compressedTexImage2D) */
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGL2RenderingContext/compressedTexImage3D) */
   compressedTexImage3D(
     target: GLenum,
     level: GLint,
@@ -8781,7 +9015,7 @@ interface WebGL2RenderingContextBase {
 }
 
 interface WebGL2RenderingContextOverloads {
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGLRenderingContext/bufferData) */
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGL2RenderingContext/bufferData) */
   bufferData(target: GLenum, size: GLsizeiptr, usage: GLenum): void;
   bufferData(
     target: GLenum,
@@ -8795,7 +9029,7 @@ interface WebGL2RenderingContextOverloads {
     srcOffset: number,
     length?: GLuint,
   ): void;
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGLRenderingContext/bufferSubData) */
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGL2RenderingContext/bufferSubData) */
   bufferSubData(
     target: GLenum,
     dstByteOffset: GLintptr,
@@ -9048,7 +9282,7 @@ interface WebGL2RenderingContextOverloads {
     srcOffset?: number,
     srcLength?: GLuint,
   ): void;
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGLRenderingContext/uniformMatrix) */
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGL2RenderingContext/uniformMatrix) */
   uniformMatrix2fv(
     location: WebGLUniformLocation | null,
     transpose: GLboolean,
@@ -9475,12 +9709,14 @@ declare const WebGLRenderingContext: {
 };
 
 interface WebGLRenderingContextBase {
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGLRenderingContext/drawingBufferColorSpace) */
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGL2RenderingContext/drawingBufferColorSpace) */
   drawingBufferColorSpace: PredefinedColorSpace;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGLRenderingContext/drawingBufferHeight) */
   readonly drawingBufferHeight: GLsizei;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGLRenderingContext/drawingBufferWidth) */
   readonly drawingBufferWidth: GLsizei;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGL2RenderingContext/unpackColorSpace) */
+  unpackColorSpace: PredefinedColorSpace;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGLRenderingContext/activeTexture) */
   activeTexture(texture: GLenum): void;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGLRenderingContext/attachShader) */
@@ -9799,6 +10035,7 @@ interface WebGLRenderingContextBase {
   isShader(shader: WebGLShader | null): GLboolean;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGLRenderingContext/isTexture) */
   isTexture(texture: WebGLTexture | null): GLboolean;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGLRenderingContext/lineWidth) */
   lineWidth(width: GLfloat): void;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGLRenderingContext/linkProgram) */
   linkProgram(program: WebGLProgram): void;
@@ -10558,7 +10795,7 @@ interface WebTransport {
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebTransport/incomingUnidirectionalStreams) */
   readonly incomingUnidirectionalStreams: ReadableStream;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebTransport/ready) */
-  readonly ready: Promise<undefined>;
+  readonly ready: Promise<void>;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebTransport/close) */
   close(closeInfo?: WebTransportCloseInfo): void;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebTransport/createBidirectionalStream) */
@@ -10681,11 +10918,11 @@ interface WindowOrWorkerGlobalScope {
   atob(data: string): string;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/Window/btoa) */
   btoa(data: string): string;
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/clearInterval) */
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/Window/clearInterval) */
   clearInterval(id: number | undefined): void;
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/clearTimeout) */
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/Window/clearTimeout) */
   clearTimeout(id: number | undefined): void;
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/createImageBitmap) */
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/Window/createImageBitmap) */
   createImageBitmap(
     image: ImageBitmapSource,
     options?: ImageBitmapOptions,
@@ -10700,44 +10937,39 @@ interface WindowOrWorkerGlobalScope {
   ): Promise<ImageBitmap>;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/Window/fetch) */
   fetch(input: RequestInfo | URL, init?: RequestInit): Promise<Response>;
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/queueMicrotask) */
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/Window/queueMicrotask) */
   queueMicrotask(callback: VoidFunction): void;
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/reportError) */
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/Window/reportError) */
   reportError(e: unknown): void;
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/setInterval) */
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/Window/setInterval) */
   setInterval(
     handler: TimerHandler,
     timeout?: number,
     ...arguments: readonly unknown[]
   ): number;
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/setTimeout) */
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/Window/setTimeout) */
   setTimeout(
     handler: TimerHandler,
     timeout?: number,
     ...arguments: readonly unknown[]
   ): number;
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/structuredClone) */
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/Window/structuredClone) */
   structuredClone<T = unknown>(
     value: T,
     options?: StructuredSerializeOptions,
   ): T;
 }
 
-interface WorkerEventMap extends AbstractWorkerEventMap {
-  message: MessageEvent;
-  messageerror: MessageEvent;
-}
+interface WorkerEventMap
+  extends AbstractWorkerEventMap, MessageEventTargetEventMap {}
 
 /**
  * This Web Workers API interface represents a background task that can be easily created and can send messages back to its creator. Creating a worker is as simple as calling the Worker() constructor and specifying a script to be run in the worker thread.
  *
  * [MDN Reference](https://developer.mozilla.org/docs/Web/API/Worker)
  */
-interface Worker extends EventTarget, AbstractWorker {
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/Worker/message_event) */
-  onmessage: ((this: Worker, ev: MessageEvent) => unknown) | null;
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/Worker/messageerror_event) */
-  onmessageerror: ((this: Worker, ev: MessageEvent) => unknown) | null;
+interface Worker
+  extends EventTarget, AbstractWorker, MessageEventTarget<Worker> {
   /**
    * Clones message and transmits it to worker's global environment. transfer can be passed as a list of objects that are to be transferred rather than cloned.
    *
@@ -10917,6 +11149,12 @@ interface WorkerNavigator
   readonly mediaCapabilities: MediaCapabilities;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WorkerNavigator/permissions) */
   readonly permissions: Permissions;
+  /**
+   * Available only in secure contexts.
+   *
+   * [MDN Reference](https://developer.mozilla.org/docs/Web/API/WorkerNavigator/serviceWorker)
+   */
+  readonly serviceWorker: ServiceWorkerContainer;
 }
 
 declare const WorkerNavigator: {
@@ -10972,11 +11210,11 @@ declare const WritableStreamDefaultController: {
  */
 interface WritableStreamDefaultWriter<W = unknown> {
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WritableStreamDefaultWriter/closed) */
-  readonly closed: Promise<undefined>;
+  readonly closed: Promise<void>;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WritableStreamDefaultWriter/desiredSize) */
   readonly desiredSize: number | null;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WritableStreamDefaultWriter/ready) */
-  readonly ready: Promise<undefined>;
+  readonly ready: Promise<void>;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WritableStreamDefaultWriter/abort) */
   abort(reason?: unknown): Promise<void>;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WritableStreamDefaultWriter/close) */
@@ -11568,14 +11806,6 @@ interface WebCodecsErrorCallback {
  * [MDN Reference](https://developer.mozilla.org/docs/Web/API/DedicatedWorkerGlobalScope/name)
  */
 declare const name: string;
-/** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DedicatedWorkerGlobalScope/message_event) */
-declare const onmessage:
-  | ((this: DedicatedWorkerGlobalScope, ev: MessageEvent) => unknown)
-  | null;
-/** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DedicatedWorkerGlobalScope/messageerror_event) */
-declare const onmessageerror:
-  | ((this: DedicatedWorkerGlobalScope, ev: MessageEvent) => unknown)
-  | null;
 /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DedicatedWorkerGlobalScope/rtctransform_event) */
 declare const onrtctransform:
   | ((this: DedicatedWorkerGlobalScope, ev: RTCTransformEvent) => unknown)
@@ -11683,11 +11913,11 @@ declare const performance: Performance;
 declare function atob(data: string): string;
 /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/Window/btoa) */
 declare function btoa(data: string): string;
-/** [MDN Reference](https://developer.mozilla.org/docs/Web/API/clearInterval) */
+/** [MDN Reference](https://developer.mozilla.org/docs/Web/API/Window/clearInterval) */
 declare function clearInterval(id: number | undefined): void;
-/** [MDN Reference](https://developer.mozilla.org/docs/Web/API/clearTimeout) */
+/** [MDN Reference](https://developer.mozilla.org/docs/Web/API/Window/clearTimeout) */
 declare function clearTimeout(id: number | undefined): void;
-/** [MDN Reference](https://developer.mozilla.org/docs/Web/API/createImageBitmap) */
+/** [MDN Reference](https://developer.mozilla.org/docs/Web/API/Window/createImageBitmap) */
 declare function createImageBitmap(
   image: ImageBitmapSource,
   options?: ImageBitmapOptions,
@@ -11705,23 +11935,23 @@ declare function fetch(
   input: RequestInfo | URL,
   init?: RequestInit,
 ): Promise<Response>;
-/** [MDN Reference](https://developer.mozilla.org/docs/Web/API/queueMicrotask) */
+/** [MDN Reference](https://developer.mozilla.org/docs/Web/API/Window/queueMicrotask) */
 declare function queueMicrotask(callback: VoidFunction): void;
-/** [MDN Reference](https://developer.mozilla.org/docs/Web/API/reportError) */
+/** [MDN Reference](https://developer.mozilla.org/docs/Web/API/Window/reportError) */
 declare function reportError(e: unknown): void;
-/** [MDN Reference](https://developer.mozilla.org/docs/Web/API/setInterval) */
+/** [MDN Reference](https://developer.mozilla.org/docs/Web/API/Window/setInterval) */
 declare function setInterval(
   handler: TimerHandler,
   timeout?: number,
   ...arguments: readonly unknown[]
 ): number;
-/** [MDN Reference](https://developer.mozilla.org/docs/Web/API/setTimeout) */
+/** [MDN Reference](https://developer.mozilla.org/docs/Web/API/Window/setTimeout) */
 declare function setTimeout(
   handler: TimerHandler,
   timeout?: number,
   ...arguments: readonly unknown[]
 ): number;
-/** [MDN Reference](https://developer.mozilla.org/docs/Web/API/structuredClone) */
+/** [MDN Reference](https://developer.mozilla.org/docs/Web/API/Window/structuredClone) */
 declare function structuredClone<T = unknown>(
   value: T,
   options?: StructuredSerializeOptions,
@@ -11730,6 +11960,14 @@ declare function structuredClone<T = unknown>(
 declare function cancelAnimationFrame(handle: number): void;
 /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DedicatedWorkerGlobalScope/requestAnimationFrame) */
 declare function requestAnimationFrame(callback: FrameRequestCallback): number;
+/** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DedicatedWorkerGlobalScope/message_event) */
+declare const onmessage:
+  | ((this: DedicatedWorkerGlobalScope, ev: MessageEvent) => unknown)
+  | null;
+/** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DedicatedWorkerGlobalScope/messageerror_event) */
+declare const onmessageerror:
+  | ((this: DedicatedWorkerGlobalScope, ev: MessageEvent) => unknown)
+  | null;
 declare function addEventListener<
   K extends keyof DedicatedWorkerGlobalScopeEventMap,
 >(
@@ -11801,6 +12039,7 @@ type IDBValidKey =
   | BufferSource
   | readonly IDBValidKey[];
 type ImageBitmapSource = CanvasImageSource | Blob | ImageData;
+type ImageBufferSource = AllowSharedBufferSource | ReadableStream;
 type Int32List = Int32Array | readonly GLint[];
 type MessageEventSource = MessagePort | ServiceWorker;
 type NamedCurve = string;
@@ -11835,6 +12074,7 @@ type Transferable =
   | TransformStream
   | AudioData
   | VideoFrame
+  | RTCDataChannel
   | ArrayBuffer;
 type Uint32List = Uint32Array | readonly GLuint[];
 type XMLHttpRequestBodyInit =
@@ -11985,7 +12225,9 @@ type OffscreenRenderingContextId =
   | 'webgpu';
 type OpusBitstreamFormat = 'ogg' | 'opus';
 type PermissionName =
+  | 'camera'
   | 'geolocation'
+  | 'microphone'
   | 'midi'
   | 'notifications'
   | 'persistent-storage'
@@ -11996,6 +12238,7 @@ type PermissionState = 'denied' | 'granted' | 'prompt';
 type PredefinedColorSpace = 'display-p3' | 'srgb';
 type PremultiplyAlpha = 'default' | 'none' | 'premultiply';
 type PushEncryptionKeyName = 'auth' | 'p256dh';
+type RTCDataChannelState = 'closed' | 'closing' | 'connecting' | 'open';
 type RTCEncodedVideoFrameType = 'delta' | 'empty' | 'key';
 type ReadableStreamReaderMode = 'byob';
 type ReadableStreamType = 'bytes';
