diff --git experimental/packages/apps/event-schedule-app-shared/src/v6/types/record/datetime-range.mts apps/event-schedule-app-shared/src/v6/types/record/datetime-range.mts
index a8048b63c..1107f4f9e 100644
--- experimental/packages/apps/event-schedule-app-shared/src/v6/types/record/datetime-range.mts
+++ apps/event-schedule-app-shared/src/v6/types/record/datetime-range.mts
@@ -1,4 +1,5 @@
-import { isRecord, Obj } from '@noshiro/ts-utils';
+import { hasKey, isRecord } from 'ts-data-forge';
+import { hasKeyValue } from '../../../utils/index.mjs';
 import {
   compareYmd,
   fillYearMonthDate,
@@ -26,28 +27,32 @@ export const datetimeRangeDefaultValue = {
 
 export const isDatetimeRange = (a: unknown): a is DatetimeRange =>
   isRecord(a) &&
-  Obj.hasKeyValue(a, 'ymd', isYearMonthDate) &&
-  Obj.hasKeyValue(a, 'timeRange', isTimeRange);
+  hasKeyValue(a, 'ymd', isYearMonthDate) &&
+  hasKeyValue(a, 'timeRange', isTimeRange);
 
 const d = datetimeRangeDefaultValue;
 
 export const fillDatetimeRange = (a?: unknown): DatetimeRange =>
   a === undefined || !isRecord(a)
     ? d
-    : {
-        ymd: Object.hasOwn(a, 'ymd') ? fillYearMonthDate(a.ymd) : d.ymd,
-        timeRange: Object.hasOwn(a, 'timeRange')
+    : ({
+        ymd: hasKey(a, 'ymd') ? fillYearMonthDate(a.ymd) : d.ymd,
+        timeRange: hasKey(a, 'timeRange')
           ? fillTimeRange(a.timeRange)
           : d.timeRange,
-      };
+      } as const);
 
 export const compareDatetimeRange = (
   a: DatetimeRange,
   b: DatetimeRange,
 ): number => {
   const compareYmdResult = compareYmd(a.ymd, b.ymd);
+
   if (compareYmdResult !== 0) return compareYmdResult;
+
   const compareTimeRangeResult = compareTimeRange(a.timeRange, b.timeRange);
+
   if (compareTimeRangeResult !== 0) return compareTimeRangeResult;
+
   return 0;
 };
