diff --git a/packages/v5.3/output/lib-files/lib.es5.d.ts b/packages/v5.4/output/lib-files/lib.es5.d.ts
index 4b41f44..111caed 100644
--- a/packages/v5.3/output/lib-files/lib.es5.d.ts
+++ b/packages/v5.4/output/lib-files/lib.es5.d.ts
@@ -714,6 +714,7 @@ interface ImportCallOptions {
 
 /**
  * The type for the `assert` property of the optional second argument to `import()`.
+ * @deprecated
  */
 interface ImportAssertions {
   readonly [key: string]: string;
@@ -1937,6 +1938,11 @@ type Capitalize<S extends string> = intrinsic;
  */
 type Uncapitalize<S extends string> = intrinsic;
 
+/**
+ * Marker for non-inference type position
+ */
+type NoInfer<T> = intrinsic;
+
 /**
  * Marker for contextual 'this' type
  */
@@ -5404,7 +5410,8 @@ declare namespace Intl {
     compare(x: string, y: string): number;
     resolvedOptions(): ResolvedCollatorOptions;
   }
-  const Collator: {
+
+  interface CollatorConstructor {
     new (
       locales?: string | readonly string[],
       options?: CollatorOptions,
@@ -5414,7 +5421,9 @@ declare namespace Intl {
       locales: string | readonly string[],
       options?: CollatorOptions,
     ): string[];
-  };
+  }
+
+  const Collator: CollatorConstructor;
 
   interface NumberFormatOptions {
     readonly localeMatcher?: string | undefined;
@@ -5446,7 +5455,8 @@ declare namespace Intl {
     format(value: number): string;
     resolvedOptions(): ResolvedNumberFormatOptions;
   }
-  const NumberFormat: {
+
+  interface NumberFormatConstructor {
     new (
       locales?: string | readonly string[],
       options?: NumberFormatOptions,
@@ -5460,7 +5470,9 @@ declare namespace Intl {
       options?: NumberFormatOptions,
     ): string[];
     readonly prototype: NumberFormat;
-  };
+  }
+
+  const NumberFormat: NumberFormatConstructor;
 
   interface DateTimeFormatOptions {
     readonly localeMatcher?: 'best fit' | 'lookup' | undefined;
@@ -5512,7 +5524,8 @@ declare namespace Intl {
     format(date?: Date | number): string;
     resolvedOptions(): ResolvedDateTimeFormatOptions;
   }
-  const DateTimeFormat: {
+
+  interface DateTimeFormatConstructor {
     new (
       locales?: string | readonly string[],
       options?: DateTimeFormatOptions,
@@ -5526,7 +5539,9 @@ declare namespace Intl {
       options?: DateTimeFormatOptions,
     ): string[];
     readonly prototype: DateTimeFormat;
-  };
+  }
+
+  const DateTimeFormat: DateTimeFormatConstructor;
 }
 
 interface String {
