diff --git a/packages/v5.3/output/lib-files/lib.webworker.d.ts b/packages/v5.4/output/lib-files/lib.webworker.d.ts
index d9eebdc..17e2395 100644
--- a/packages/v5.3/output/lib-files/lib.webworker.d.ts
+++ b/packages/v5.4/output/lib-files/lib.webworker.d.ts
@@ -167,7 +167,7 @@ interface EcdsaParams extends Algorithm {
 }
 
 interface EncodedVideoChunkInit {
-  data: BufferSource;
+  data: AllowSharedBufferSource;
   duration?: number;
   timestamp: number;
   type: EncodedVideoChunkType;
@@ -251,7 +251,6 @@ interface FontFaceDescriptors {
   stretch?: string;
   style?: string;
   unicodeRange?: string;
-  variant?: string;
   weight?: string;
 }
 
@@ -416,32 +415,21 @@ interface NavigationPreloadState {
   headerValue?: string;
 }
 
-interface NotificationAction {
-  action: string;
-  icon?: string;
-  title: string;
-}
-
 interface NotificationEventInit extends ExtendableEventInit {
   action?: string;
   notification: Notification;
 }
 
 interface NotificationOptions {
-  actions?: readonly NotificationAction[];
   badge?: string;
   body?: string;
   data?: unknown;
   dir?: NotificationDirection;
   icon?: string;
-  image?: string;
   lang?: string;
-  renotify?: boolean;
   requireInteraction?: boolean;
   silent?: boolean | null;
   tag?: string;
-  timestamp?: EpochTimeStamp;
-  vibrate?: VibratePattern;
 }
 
 interface Pbkdf2Params extends Algorithm {
@@ -519,16 +507,21 @@ interface QueuingStrategyInit {
 
 interface RTCEncodedAudioFrameMetadata {
   contributingSources?: readonly number[];
+  payloadType?: number;
+  sequenceNumber?: number;
   synchronizationSource?: number;
 }
 
 interface RTCEncodedVideoFrameMetadata {
+  contributingSources?: readonly number[];
   dependencies?: readonly number[];
   frameId?: number;
   height?: number;
+  payloadType?: number;
   spatialIndex?: number;
   synchronizationSource?: number;
   temporalIndex?: number;
+  timestamp?: number;
   width?: number;
 }
 
@@ -589,6 +582,7 @@ interface RequestInit {
   method?: string;
   /** A string to indicate whether the request will use CORS, or will be restricted to same-origin URLs. Sets request's mode. */
   mode?: RequestMode;
+  priority?: RequestPriority;
   /** A string indicating whether request follows redirects, results in an error upon encountering a redirect, or returns the redirect (in an opaque fashion). Sets request's redirect. */
   redirect?: RequestRedirect;
   /** A string whose value is a same-origin URL, "about:client", or the empty string, to set request's referrer. */
@@ -761,7 +755,7 @@ interface VideoDecoderConfig {
   codedHeight?: number;
   codedWidth?: number;
   colorSpace?: VideoColorSpaceInit;
-  description?: BufferSource;
+  description?: AllowSharedBufferSource;
   displayAspectHeight?: number;
   displayAspectWidth?: number;
   hardwareAcceleration?: HardwareAcceleration;
@@ -874,7 +868,7 @@ interface WebTransportOptions {
 }
 
 interface WebTransportSendStreamOptions {
-  sendOrder?: number | null;
+  sendOrder?: number;
 }
 
 interface WorkerOptions {
@@ -1024,7 +1018,9 @@ interface AbstractWorker {
 }
 
 interface AnimationFrameProvider {
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DedicatedWorkerGlobalScope/cancelAnimationFrame) */
   cancelAnimationFrame(handle: number): void;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DedicatedWorkerGlobalScope/requestAnimationFrame) */
   requestAnimationFrame(callback: FrameRequestCallback): number;
 }
 
@@ -1872,10 +1868,20 @@ interface CanvasTextDrawingStyles {
   font: string;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/CanvasRenderingContext2D/fontKerning) */
   fontKerning: CanvasFontKerning;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/CanvasRenderingContext2D/fontStretch) */
+  fontStretch: CanvasFontStretch;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/CanvasRenderingContext2D/fontVariantCaps) */
+  fontVariantCaps: CanvasFontVariantCaps;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/CanvasRenderingContext2D/letterSpacing) */
+  letterSpacing: string;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/CanvasRenderingContext2D/textAlign) */
   textAlign: CanvasTextAlign;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/CanvasRenderingContext2D/textBaseline) */
   textBaseline: CanvasTextBaseline;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/CanvasRenderingContext2D/textRendering) */
+  textRendering: CanvasTextRendering;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/CanvasRenderingContext2D/wordSpacing) */
+  wordSpacing: string;
 }
 
 interface CanvasTransform {
@@ -2498,6 +2504,7 @@ declare const DecompressionStream: {
 interface DedicatedWorkerGlobalScopeEventMap extends WorkerGlobalScopeEventMap {
   message: MessageEvent;
   messageerror: MessageEvent;
+  rtctransform: Event;
 }
 
 /**
@@ -2521,6 +2528,10 @@ interface DedicatedWorkerGlobalScope
   onmessageerror:
     | ((this: DedicatedWorkerGlobalScope, ev: MessageEvent) => unknown)
     | null;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DedicatedWorkerGlobalScope/rtctransform_event) */
+  onrtctransform:
+    | ((this: DedicatedWorkerGlobalScope, ev: Event) => unknown)
+    | null;
   /**
    * Aborts dedicatedWorkerGlobal.
    *
@@ -2654,7 +2665,7 @@ interface EncodedVideoChunk {
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/EncodedVideoChunk/type) */
   readonly type: EncodedVideoChunkType;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/EncodedVideoChunk/copyTo) */
-  copyTo(destination: BufferSource): void;
+  copyTo(destination: AllowSharedBufferSource): void;
 }
 
 declare const EncodedVideoChunk: {
@@ -3111,7 +3122,11 @@ interface FileReader extends EventTarget {
   abort(): void;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/FileReader/readAsArrayBuffer) */
   readAsArrayBuffer(blob: Blob): void;
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/FileReader/readAsBinaryString) */
+  /**
+   * @deprecated
+   *
+   * [MDN Reference](https://developer.mozilla.org/docs/Web/API/FileReader/readAsBinaryString)
+   */
   readAsBinaryString(blob: Blob): void;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/FileReader/readAsDataURL) */
   readAsDataURL(blob: Blob): void;
@@ -3318,8 +3333,6 @@ interface FontFace {
   style: string;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/FontFace/unicodeRange) */
   unicodeRange: string;
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/FontFace/variant) */
-  variant: string;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/FontFace/weight) */
   weight: string;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/FontFace/load) */
@@ -4673,6 +4686,8 @@ interface NotificationEventMap {
  * [MDN Reference](https://developer.mozilla.org/docs/Web/API/Notification)
  */
 interface Notification extends EventTarget {
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/Notification/badge) */
+  readonly badge: string;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/Notification/body) */
   readonly body: string;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/Notification/data) */
@@ -4691,6 +4706,8 @@ interface Notification extends EventTarget {
   onerror: ((this: Notification, ev: Event) => unknown) | null;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/Notification/show_event) */
   onshow: ((this: Notification, ev: Event) => unknown) | null;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/Notification/requireInteraction) */
+  readonly requireInteraction: boolean;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/Notification/silent) */
   readonly silent: boolean | null;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/Notification/tag) */
@@ -5425,9 +5442,13 @@ declare const PushSubscriptionOptions: {
   new (): PushSubscriptionOptions;
 };
 
+/** [MDN Reference](https://developer.mozilla.org/docs/Web/API/RTCEncodedAudioFrame) */
 interface RTCEncodedAudioFrame {
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/RTCEncodedAudioFrame/data) */
   data: ArrayBuffer;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/RTCEncodedAudioFrame/timestamp) */
   readonly timestamp: number;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/RTCEncodedAudioFrame/getMetadata) */
   getMetadata(): RTCEncodedAudioFrameMetadata;
 }
 
@@ -5436,10 +5457,15 @@ declare const RTCEncodedAudioFrame: {
   new (): RTCEncodedAudioFrame;
 };
 
+/** [MDN Reference](https://developer.mozilla.org/docs/Web/API/RTCEncodedVideoFrame) */
 interface RTCEncodedVideoFrame {
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/RTCEncodedVideoFrame/data) */
   data: ArrayBuffer;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/RTCEncodedVideoFrame/timestamp) */
   readonly timestamp: number;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/RTCEncodedVideoFrame/type) */
   readonly type: RTCEncodedVideoFrameType;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/RTCEncodedVideoFrame/getMetadata) */
   getMetadata(): RTCEncodedVideoFrameMetadata;
 }
 
@@ -5448,6 +5474,36 @@ declare const RTCEncodedVideoFrame: {
   new (): RTCEncodedVideoFrame;
 };
 
+/** [MDN Reference](https://developer.mozilla.org/docs/Web/API/RTCRtpScriptTransformer) */
+interface RTCRtpScriptTransformer extends EventTarget {
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/RTCRtpScriptTransformer/options) */
+  readonly options: unknown;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/RTCRtpScriptTransformer/readable) */
+  readonly readable: ReadableStream;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/RTCRtpScriptTransformer/writable) */
+  readonly writable: WritableStream;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/RTCRtpScriptTransformer/generateKeyFrame) */
+  generateKeyFrame(rid?: string): Promise<number>;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/RTCRtpScriptTransformer/sendKeyFrameRequest) */
+  sendKeyFrameRequest(): Promise<void>;
+}
+
+declare const RTCRtpScriptTransformer: {
+  prototype: RTCRtpScriptTransformer;
+  new (): RTCRtpScriptTransformer;
+};
+
+/** [MDN Reference](https://developer.mozilla.org/docs/Web/API/RTCTransformEvent) */
+interface RTCTransformEvent extends Event {
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/RTCTransformEvent/transformer) */
+  readonly transformer: RTCRtpScriptTransformer;
+}
+
+declare const RTCTransformEvent: {
+  prototype: RTCTransformEvent;
+  new (): RTCTransformEvent;
+};
+
 /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/ReadableByteStreamController) */
 interface ReadableByteStreamController {
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/ReadableByteStreamController/byobRequest) */
@@ -5862,6 +5918,7 @@ interface ServiceWorkerContainer extends EventTarget {
   onmessage:
     | ((this: ServiceWorkerContainer, ev: MessageEvent) => unknown)
     | null;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/ServiceWorkerContainer/messageerror_event) */
   onmessageerror:
     | ((this: ServiceWorkerContainer, ev: MessageEvent) => unknown)
     | null;
@@ -5967,6 +6024,7 @@ interface ServiceWorkerGlobalScope extends WorkerGlobalScope {
     | null;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/ServiceWorkerGlobalScope/registration) */
   readonly registration: ServiceWorkerRegistration;
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/ServiceWorkerGlobalScope/serviceWorker) */
   readonly serviceWorker: ServiceWorker;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/ServiceWorkerGlobalScope/skipWaiting) */
   skipWaiting(): Promise<void>;
@@ -6480,6 +6538,24 @@ interface TextMetrics {
    * [MDN Reference](https://developer.mozilla.org/docs/Web/API/TextMetrics/actualBoundingBoxRight)
    */
   readonly actualBoundingBoxRight: number;
+  /**
+   * Returns the measurement described below.
+   *
+   * [MDN Reference](https://developer.mozilla.org/docs/Web/API/TextMetrics/alphabeticBaseline)
+   */
+  readonly alphabeticBaseline: number;
+  /**
+   * Returns the measurement described below.
+   *
+   * [MDN Reference](https://developer.mozilla.org/docs/Web/API/TextMetrics/emHeightAscent)
+   */
+  readonly emHeightAscent: number;
+  /**
+   * Returns the measurement described below.
+   *
+   * [MDN Reference](https://developer.mozilla.org/docs/Web/API/TextMetrics/emHeightDescent)
+   */
+  readonly emHeightDescent: number;
   /**
    * Returns the measurement described below.
    *
@@ -6492,6 +6568,18 @@ interface TextMetrics {
    * [MDN Reference](https://developer.mozilla.org/docs/Web/API/TextMetrics/fontBoundingBoxDescent)
    */
   readonly fontBoundingBoxDescent: number;
+  /**
+   * Returns the measurement described below.
+   *
+   * [MDN Reference](https://developer.mozilla.org/docs/Web/API/TextMetrics/hangingBaseline)
+   */
+  readonly hangingBaseline: number;
+  /**
+   * Returns the measurement described below.
+   *
+   * [MDN Reference](https://developer.mozilla.org/docs/Web/API/TextMetrics/ideographicBaseline)
+   */
+  readonly ideographicBaseline: number;
   /**
    * Returns the measurement described below.
    *
@@ -6799,7 +6887,7 @@ interface VideoFrame {
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/VideoFrame/close) */
   close(): void;
   copyTo(
-    destination: BufferSource,
+    destination: AllowSharedBufferSource,
     options?: VideoFrameCopyToOptions,
   ): Promise<PlaneLayout[]>;
 }
@@ -6807,7 +6895,7 @@ interface VideoFrame {
 declare const VideoFrame: {
   prototype: VideoFrame;
   new (image: CanvasImageSource, init?: VideoFrameInit): VideoFrame;
-  new (data: BufferSource, init: VideoFrameBufferInit): VideoFrame;
+  new (data: AllowSharedBufferSource, init: VideoFrameBufferInit): VideoFrame;
 };
 
 /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WEBGL_color_buffer_float) */
@@ -6977,42 +7065,42 @@ interface WEBGL_multi_draw {
   multiDrawArraysInstancedWEBGL(
     mode: GLenum,
     firstsList: Int32Array | readonly GLint[],
-    firstsOffset: GLuint,
+    firstsOffset: number,
     countsList: Int32Array | readonly GLsizei[],
-    countsOffset: GLuint,
+    countsOffset: number,
     instanceCountsList: Int32Array | readonly GLsizei[],
-    instanceCountsOffset: GLuint,
+    instanceCountsOffset: number,
     drawcount: GLsizei,
   ): void;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WEBGL_multi_draw/multiDrawArraysWEBGL) */
   multiDrawArraysWEBGL(
     mode: GLenum,
     firstsList: Int32Array | readonly GLint[],
-    firstsOffset: GLuint,
+    firstsOffset: number,
     countsList: Int32Array | readonly GLsizei[],
-    countsOffset: GLuint,
+    countsOffset: number,
     drawcount: GLsizei,
   ): void;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WEBGL_multi_draw/multiDrawElementsInstancedWEBGL) */
   multiDrawElementsInstancedWEBGL(
     mode: GLenum,
     countsList: Int32Array | readonly GLsizei[],
-    countsOffset: GLuint,
+    countsOffset: number,
     type: GLenum,
     offsetsList: Int32Array | readonly GLsizei[],
-    offsetsOffset: GLuint,
+    offsetsOffset: number,
     instanceCountsList: Int32Array | readonly GLsizei[],
-    instanceCountsOffset: GLuint,
+    instanceCountsOffset: number,
     drawcount: GLsizei,
   ): void;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WEBGL_multi_draw/multiDrawElementsWEBGL) */
   multiDrawElementsWEBGL(
     mode: GLenum,
     countsList: Int32Array | readonly GLsizei[],
-    countsOffset: GLuint,
+    countsOffset: number,
     type: GLenum,
     offsetsList: Int32Array | readonly GLsizei[],
-    offsetsOffset: GLuint,
+    offsetsOffset: number,
     drawcount: GLsizei,
   ): void;
 }
@@ -7039,7 +7127,6 @@ declare const WebGL2RenderingContext: {
   readonly STENCIL: 0x1802;
   readonly RED: 0x1903;
   readonly RGB8: 0x8051;
-  readonly RGBA8: 0x8058;
   readonly RGB10_A2: 0x8059;
   readonly TEXTURE_BINDING_3D: 0x806a;
   readonly UNPACK_SKIP_IMAGES: 0x806d;
@@ -7550,6 +7637,7 @@ declare const WebGL2RenderingContext: {
   readonly RENDERBUFFER: 0x8d41;
   readonly RGBA4: 0x8056;
   readonly RGB5_A1: 0x8057;
+  readonly RGBA8: 0x8058;
   readonly RGB565: 0x8d62;
   readonly DEPTH_COMPONENT16: 0x81a5;
   readonly STENCIL_INDEX8: 0x8d48;
@@ -7641,21 +7729,21 @@ interface WebGL2RenderingContextBase {
     buffer: GLenum,
     drawbuffer: GLint,
     values: Float32List,
-    srcOffset?: GLuint,
+    srcOffset?: number,
   ): void;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGL2RenderingContext/clearBuffer) */
   clearBufferiv(
     buffer: GLenum,
     drawbuffer: GLint,
     values: Int32List,
-    srcOffset?: GLuint,
+    srcOffset?: number,
   ): void;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGL2RenderingContext/clearBuffer) */
   clearBufferuiv(
     buffer: GLenum,
     drawbuffer: GLint,
     values: Uint32List,
-    srcOffset?: GLuint,
+    srcOffset?: number,
   ): void;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGL2RenderingContext/clientWaitSync) */
   clientWaitSync(sync: WebGLSync, flags: GLbitfield, timeout: GLuint64): GLenum;
@@ -7680,7 +7768,7 @@ interface WebGL2RenderingContextBase {
     depth: GLsizei,
     border: GLint,
     srcData: ArrayBufferView,
-    srcOffset?: GLuint,
+    srcOffset?: number,
     srcLengthOverride?: GLuint,
   ): void;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGL2RenderingContext/compressedTexSubImage3D) */
@@ -7708,7 +7796,7 @@ interface WebGL2RenderingContextBase {
     depth: GLsizei,
     format: GLenum,
     srcData: ArrayBufferView,
-    srcOffset?: GLuint,
+    srcOffset?: number,
     srcLengthOverride?: GLuint,
   ): void;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGL2RenderingContext/copyBufferSubData) */
@@ -7811,7 +7899,7 @@ interface WebGL2RenderingContextBase {
     target: GLenum,
     srcByteOffset: GLintptr,
     dstBuffer: ArrayBufferView,
-    dstOffset?: GLuint,
+    dstOffset?: number,
     length?: GLuint,
   ): void;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGL2RenderingContext/getFragDataLocation) */
@@ -7931,7 +8019,7 @@ interface WebGL2RenderingContextBase {
     format: GLenum,
     type: GLenum,
     srcData: ArrayBufferView,
-    srcOffset: GLuint,
+    srcOffset: number,
   ): void;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGL2RenderingContext/texStorage2D) */
   texStorage2D(
@@ -7989,7 +8077,7 @@ interface WebGL2RenderingContextBase {
     format: GLenum,
     type: GLenum,
     srcData: ArrayBufferView | null,
-    srcOffset?: GLuint,
+    srcOffset?: number,
   ): void;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGL2RenderingContext/transformFeedbackVaryings) */
   transformFeedbackVaryings(
@@ -8003,7 +8091,7 @@ interface WebGL2RenderingContextBase {
   uniform1uiv(
     location: WebGLUniformLocation | null,
     data: Uint32List,
-    srcOffset?: GLuint,
+    srcOffset?: number,
     srcLength?: GLuint,
   ): void;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGL2RenderingContext/uniform) */
@@ -8016,7 +8104,7 @@ interface WebGL2RenderingContextBase {
   uniform2uiv(
     location: WebGLUniformLocation | null,
     data: Uint32List,
-    srcOffset?: GLuint,
+    srcOffset?: number,
     srcLength?: GLuint,
   ): void;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGL2RenderingContext/uniform) */
@@ -8030,7 +8118,7 @@ interface WebGL2RenderingContextBase {
   uniform3uiv(
     location: WebGLUniformLocation | null,
     data: Uint32List,
-    srcOffset?: GLuint,
+    srcOffset?: number,
     srcLength?: GLuint,
   ): void;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGL2RenderingContext/uniform) */
@@ -8045,7 +8133,7 @@ interface WebGL2RenderingContextBase {
   uniform4uiv(
     location: WebGLUniformLocation | null,
     data: Uint32List,
-    srcOffset?: GLuint,
+    srcOffset?: number,
     srcLength?: GLuint,
   ): void;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGL2RenderingContext/uniformBlockBinding) */
@@ -8059,7 +8147,7 @@ interface WebGL2RenderingContextBase {
     location: WebGLUniformLocation | null,
     transpose: GLboolean,
     data: Float32List,
-    srcOffset?: GLuint,
+    srcOffset?: number,
     srcLength?: GLuint,
   ): void;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGL2RenderingContext/uniformMatrix) */
@@ -8067,7 +8155,7 @@ interface WebGL2RenderingContextBase {
     location: WebGLUniformLocation | null,
     transpose: GLboolean,
     data: Float32List,
-    srcOffset?: GLuint,
+    srcOffset?: number,
     srcLength?: GLuint,
   ): void;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGL2RenderingContext/uniformMatrix) */
@@ -8075,7 +8163,7 @@ interface WebGL2RenderingContextBase {
     location: WebGLUniformLocation | null,
     transpose: GLboolean,
     data: Float32List,
-    srcOffset?: GLuint,
+    srcOffset?: number,
     srcLength?: GLuint,
   ): void;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGL2RenderingContext/uniformMatrix) */
@@ -8083,7 +8171,7 @@ interface WebGL2RenderingContextBase {
     location: WebGLUniformLocation | null,
     transpose: GLboolean,
     data: Float32List,
-    srcOffset?: GLuint,
+    srcOffset?: number,
     srcLength?: GLuint,
   ): void;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGL2RenderingContext/uniformMatrix) */
@@ -8091,7 +8179,7 @@ interface WebGL2RenderingContextBase {
     location: WebGLUniformLocation | null,
     transpose: GLboolean,
     data: Float32List,
-    srcOffset?: GLuint,
+    srcOffset?: number,
     srcLength?: GLuint,
   ): void;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGL2RenderingContext/uniformMatrix) */
@@ -8099,7 +8187,7 @@ interface WebGL2RenderingContextBase {
     location: WebGLUniformLocation | null,
     transpose: GLboolean,
     data: Float32List,
-    srcOffset?: GLuint,
+    srcOffset?: number,
     srcLength?: GLuint,
   ): void;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGL2RenderingContext/vertexAttribDivisor) */
@@ -8140,7 +8228,6 @@ interface WebGL2RenderingContextBase {
   readonly STENCIL: 0x1802;
   readonly RED: 0x1903;
   readonly RGB8: 0x8051;
-  readonly RGBA8: 0x8058;
   readonly RGB10_A2: 0x8059;
   readonly TEXTURE_BINDING_3D: 0x806a;
   readonly UNPACK_SKIP_IMAGES: 0x806d;
@@ -8396,25 +8483,29 @@ interface WebGL2RenderingContextBase {
 interface WebGL2RenderingContextOverloads {
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGLRenderingContext/bufferData) */
   bufferData(target: GLenum, size: GLsizeiptr, usage: GLenum): void;
-  bufferData(target: GLenum, srcData: BufferSource | null, usage: GLenum): void;
+  bufferData(
+    target: GLenum,
+    srcData: AllowSharedBufferSource | null,
+    usage: GLenum,
+  ): void;
   bufferData(
     target: GLenum,
     srcData: ArrayBufferView,
     usage: GLenum,
-    srcOffset: GLuint,
+    srcOffset: number,
     length?: GLuint,
   ): void;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGLRenderingContext/bufferSubData) */
   bufferSubData(
     target: GLenum,
     dstByteOffset: GLintptr,
-    srcData: BufferSource,
+    srcData: AllowSharedBufferSource,
   ): void;
   bufferSubData(
     target: GLenum,
     dstByteOffset: GLintptr,
     srcData: ArrayBufferView,
-    srcOffset: GLuint,
+    srcOffset: number,
     length?: GLuint,
   ): void;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGLRenderingContext/compressedTexImage2D) */
@@ -8436,7 +8527,7 @@ interface WebGL2RenderingContextOverloads {
     height: GLsizei,
     border: GLint,
     srcData: ArrayBufferView,
-    srcOffset?: GLuint,
+    srcOffset?: number,
     srcLengthOverride?: GLuint,
   ): void;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGLRenderingContext/compressedTexSubImage2D) */
@@ -8460,7 +8551,7 @@ interface WebGL2RenderingContextOverloads {
     height: GLsizei,
     format: GLenum,
     srcData: ArrayBufferView,
-    srcOffset?: GLuint,
+    srcOffset?: number,
     srcLengthOverride?: GLuint,
   ): void;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGLRenderingContext/readPixels) */
@@ -8490,7 +8581,7 @@ interface WebGL2RenderingContextOverloads {
     format: GLenum,
     type: GLenum,
     dstData: ArrayBufferView,
-    dstOffset: GLuint,
+    dstOffset: number,
   ): void;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGLRenderingContext/texImage2D) */
   texImage2D(
@@ -8544,7 +8635,7 @@ interface WebGL2RenderingContextOverloads {
     format: GLenum,
     type: GLenum,
     srcData: ArrayBufferView,
-    srcOffset: GLuint,
+    srcOffset: number,
   ): void;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGLRenderingContext/texSubImage2D) */
   texSubImage2D(
@@ -8599,62 +8690,62 @@ interface WebGL2RenderingContextOverloads {
     format: GLenum,
     type: GLenum,
     srcData: ArrayBufferView,
-    srcOffset: GLuint,
+    srcOffset: number,
   ): void;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGLRenderingContext/uniform) */
   uniform1fv(
     location: WebGLUniformLocation | null,
     data: Float32List,
-    srcOffset?: GLuint,
+    srcOffset?: number,
     srcLength?: GLuint,
   ): void;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGLRenderingContext/uniform) */
   uniform1iv(
     location: WebGLUniformLocation | null,
     data: Int32List,
-    srcOffset?: GLuint,
+    srcOffset?: number,
     srcLength?: GLuint,
   ): void;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGLRenderingContext/uniform) */
   uniform2fv(
     location: WebGLUniformLocation | null,
     data: Float32List,
-    srcOffset?: GLuint,
+    srcOffset?: number,
     srcLength?: GLuint,
   ): void;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGLRenderingContext/uniform) */
   uniform2iv(
     location: WebGLUniformLocation | null,
     data: Int32List,
-    srcOffset?: GLuint,
+    srcOffset?: number,
     srcLength?: GLuint,
   ): void;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGLRenderingContext/uniform) */
   uniform3fv(
     location: WebGLUniformLocation | null,
     data: Float32List,
-    srcOffset?: GLuint,
+    srcOffset?: number,
     srcLength?: GLuint,
   ): void;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGLRenderingContext/uniform) */
   uniform3iv(
     location: WebGLUniformLocation | null,
     data: Int32List,
-    srcOffset?: GLuint,
+    srcOffset?: number,
     srcLength?: GLuint,
   ): void;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGLRenderingContext/uniform) */
   uniform4fv(
     location: WebGLUniformLocation | null,
     data: Float32List,
-    srcOffset?: GLuint,
+    srcOffset?: number,
     srcLength?: GLuint,
   ): void;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGLRenderingContext/uniform) */
   uniform4iv(
     location: WebGLUniformLocation | null,
     data: Int32List,
-    srcOffset?: GLuint,
+    srcOffset?: number,
     srcLength?: GLuint,
   ): void;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGLRenderingContext/uniformMatrix) */
@@ -8662,7 +8753,7 @@ interface WebGL2RenderingContextOverloads {
     location: WebGLUniformLocation | null,
     transpose: GLboolean,
     data: Float32List,
-    srcOffset?: GLuint,
+    srcOffset?: number,
     srcLength?: GLuint,
   ): void;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGLRenderingContext/uniformMatrix) */
@@ -8670,7 +8761,7 @@ interface WebGL2RenderingContextOverloads {
     location: WebGLUniformLocation | null,
     transpose: GLboolean,
     data: Float32List,
-    srcOffset?: GLuint,
+    srcOffset?: number,
     srcLength?: GLuint,
   ): void;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGLRenderingContext/uniformMatrix) */
@@ -8678,7 +8769,7 @@ interface WebGL2RenderingContextOverloads {
     location: WebGLUniformLocation | null,
     transpose: GLboolean,
     data: Float32List,
-    srcOffset?: GLuint,
+    srcOffset?: number,
     srcLength?: GLuint,
   ): void;
 }
@@ -9044,6 +9135,7 @@ declare const WebGLRenderingContext: {
   readonly RENDERBUFFER: 0x8d41;
   readonly RGBA4: 0x8056;
   readonly RGB5_A1: 0x8057;
+  readonly RGBA8: 0x8058;
   readonly RGB565: 0x8d62;
   readonly DEPTH_COMPONENT16: 0x81a5;
   readonly STENCIL_INDEX8: 0x8d48;
@@ -9791,6 +9883,7 @@ interface WebGLRenderingContextBase {
   readonly RENDERBUFFER: 0x8d41;
   readonly RGBA4: 0x8056;
   readonly RGB5_A1: 0x8057;
+  readonly RGBA8: 0x8058;
   readonly RGB565: 0x8d62;
   readonly DEPTH_COMPONENT16: 0x81a5;
   readonly STENCIL_INDEX8: 0x8d48;
@@ -9832,9 +9925,17 @@ interface WebGLRenderingContextBase {
 interface WebGLRenderingContextOverloads {
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGLRenderingContext/bufferData) */
   bufferData(target: GLenum, size: GLsizeiptr, usage: GLenum): void;
-  bufferData(target: GLenum, data: BufferSource | null, usage: GLenum): void;
+  bufferData(
+    target: GLenum,
+    data: AllowSharedBufferSource | null,
+    usage: GLenum,
+  ): void;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGLRenderingContext/bufferSubData) */
-  bufferSubData(target: GLenum, offset: GLintptr, data: BufferSource): void;
+  bufferSubData(
+    target: GLenum,
+    offset: GLintptr,
+    data: AllowSharedBufferSource,
+  ): void;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebGLRenderingContext/compressedTexImage2D) */
   compressedTexImage2D(
     target: GLenum,
@@ -10200,13 +10301,13 @@ interface WebTransportDatagramDuplexStream {
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebTransportDatagramDuplexStream/incomingHighWaterMark) */
   incomingHighWaterMark: number;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebTransportDatagramDuplexStream/incomingMaxAge) */
-  incomingMaxAge: number;
+  incomingMaxAge: number | null;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebTransportDatagramDuplexStream/maxDatagramSize) */
   readonly maxDatagramSize: number;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebTransportDatagramDuplexStream/outgoingHighWaterMark) */
   outgoingHighWaterMark: number;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebTransportDatagramDuplexStream/outgoingMaxAge) */
-  outgoingMaxAge: number;
+  outgoingMaxAge: number | null;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebTransportDatagramDuplexStream/readable) */
   readonly readable: ReadableStream;
   /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/WebTransportDatagramDuplexStream/writable) */
@@ -10843,44 +10944,44 @@ declare const XMLHttpRequestUpload: {
 
 /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/console) */
 interface Console {
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/console/assert) */
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/console/assert_static) */
   assert(condition?: boolean, ...data: readonly unknown[]): void;
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/console/clear) */
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/console/clear_static) */
   clear(): void;
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/console/count) */
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/console/count_static) */
   count(label?: string): void;
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/console/countReset) */
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/console/countReset_static) */
   countReset(label?: string): void;
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/console/debug) */
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/console/debug_static) */
   debug(...data: readonly unknown[]): void;
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/console/dir) */
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/console/dir_static) */
   dir(item?: unknown, options?: unknown): void;
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/console/dirxml) */
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/console/dirxml_static) */
   dirxml(...data: readonly unknown[]): void;
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/console/error) */
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/console/error_static) */
   error(...data: readonly unknown[]): void;
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/console/group) */
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/console/group_static) */
   group(...data: readonly unknown[]): void;
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/console/groupCollapsed) */
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/console/groupCollapsed_static) */
   groupCollapsed(...data: readonly unknown[]): void;
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/console/groupEnd) */
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/console/groupEnd_static) */
   groupEnd(): void;
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/console/info) */
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/console/info_static) */
   info(...data: readonly unknown[]): void;
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/console/log) */
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/console/log_static) */
   log(...data: readonly unknown[]): void;
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/console/table) */
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/console/table_static) */
   table(tabularData?: unknown, properties?: readonly string[]): void;
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/console/time) */
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/console/time_static) */
   time(label?: string): void;
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/console/timeEnd) */
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/console/timeEnd_static) */
   timeEnd(label?: string): void;
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/console/timeLog) */
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/console/timeLog_static) */
   timeLog(label?: string, ...data: readonly unknown[]): void;
   timeStamp(label?: string): void;
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/console/trace) */
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/console/trace_static) */
   trace(...data: readonly unknown[]): void;
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/console/warn) */
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/console/warn_static) */
   warn(...data: readonly unknown[]): void;
 }
 
@@ -10895,11 +10996,11 @@ declare namespace WebAssembly {
     (message?: string): CompileError;
   };
 
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/WebAssembly/Global) */
+  /** [MDN Reference](https://developer.mozilla.org/docs/WebAssembly/JavaScript_interface/Global) */
   interface Global<T extends ValueType = ValueType> {
-    /** [MDN Reference](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/WebAssembly/Global/value) */
+    /** [MDN Reference](https://developer.mozilla.org/docs/WebAssembly/JavaScript_interface/Global/value) */
     value: ValueTypeMap[T];
-    /** [MDN Reference](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/WebAssembly/Global/valueOf) */
+    /** [MDN Reference](https://developer.mozilla.org/docs/WebAssembly/JavaScript_interface/Global/valueOf) */
     valueOf(): ValueTypeMap[T];
   }
 
@@ -10911,9 +11012,9 @@ declare namespace WebAssembly {
     ): Global<T>;
   };
 
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/WebAssembly/Instance) */
+  /** [MDN Reference](https://developer.mozilla.org/docs/WebAssembly/JavaScript_interface/Instance) */
   interface Instance {
-    /** [MDN Reference](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/WebAssembly/Instance/exports) */
+    /** [MDN Reference](https://developer.mozilla.org/docs/WebAssembly/JavaScript_interface/Instance/exports) */
     readonly exports: Exports;
   }
 
@@ -10930,11 +11031,11 @@ declare namespace WebAssembly {
     (message?: string): LinkError;
   };
 
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/WebAssembly/Memory) */
+  /** [MDN Reference](https://developer.mozilla.org/docs/WebAssembly/JavaScript_interface/Memory) */
   interface Memory {
-    /** [MDN Reference](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/WebAssembly/Memory/buffer) */
+    /** [MDN Reference](https://developer.mozilla.org/docs/WebAssembly/JavaScript_interface/Memory/buffer) */
     readonly buffer: ArrayBuffer;
-    /** [MDN Reference](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/WebAssembly/Memory/grow) */
+    /** [MDN Reference](https://developer.mozilla.org/docs/WebAssembly/JavaScript_interface/Memory/grow) */
     grow(delta: number): number;
   }
 
@@ -10943,17 +11044,17 @@ declare namespace WebAssembly {
     new (descriptor: MemoryDescriptor): Memory;
   };
 
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/WebAssembly/Module) */
+  /** [MDN Reference](https://developer.mozilla.org/docs/WebAssembly/JavaScript_interface/Module) */
   interface Module {}
 
   const Module: {
     prototype: Module;
     new (bytes: BufferSource): Module;
-    /** [MDN Reference](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/WebAssembly/Module/customSections) */
+    /** [MDN Reference](https://developer.mozilla.org/docs/WebAssembly/JavaScript_interface/Module/customSections_static) */
     customSections(moduleObject: Module, sectionName: string): ArrayBuffer[];
-    /** [MDN Reference](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/WebAssembly/Module/exports) */
+    /** [MDN Reference](https://developer.mozilla.org/docs/WebAssembly/JavaScript_interface/Module/exports_static) */
     exports(moduleObject: Module): ModuleExportDescriptor[];
-    /** [MDN Reference](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/WebAssembly/Module/imports) */
+    /** [MDN Reference](https://developer.mozilla.org/docs/WebAssembly/JavaScript_interface/Module/imports_static) */
     imports(moduleObject: Module): ModuleImportDescriptor[];
   };
 
@@ -10965,15 +11066,15 @@ declare namespace WebAssembly {
     (message?: string): RuntimeError;
   };
 
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/WebAssembly/Table) */
+  /** [MDN Reference](https://developer.mozilla.org/docs/WebAssembly/JavaScript_interface/Table) */
   interface Table {
-    /** [MDN Reference](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/WebAssembly/Table/length) */
+    /** [MDN Reference](https://developer.mozilla.org/docs/WebAssembly/JavaScript_interface/Table/length) */
     readonly length: number;
-    /** [MDN Reference](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/WebAssembly/Table/get) */
+    /** [MDN Reference](https://developer.mozilla.org/docs/WebAssembly/JavaScript_interface/Table/get) */
     get(index: number): unknown;
-    /** [MDN Reference](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/WebAssembly/Table/grow) */
+    /** [MDN Reference](https://developer.mozilla.org/docs/WebAssembly/JavaScript_interface/Table/grow) */
     grow(delta: number, value?: unknown): number;
-    /** [MDN Reference](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/WebAssembly/Table/set) */
+    /** [MDN Reference](https://developer.mozilla.org/docs/WebAssembly/JavaScript_interface/Table/set) */
     set(index: number, value?: unknown): void;
   }
 
@@ -11033,13 +11134,13 @@ declare namespace WebAssembly {
   type Imports = Record<string, ModuleImports>;
   type ModuleImports = Record<string, ImportValue>;
   type ValueType = keyof ValueTypeMap;
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/WebAssembly/compile) */
+  /** [MDN Reference](https://developer.mozilla.org/docs/WebAssembly/JavaScript_interface/compile_static) */
   function compile(bytes: BufferSource): Promise<Module>;
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/WebAssembly/compileStreaming) */
+  /** [MDN Reference](https://developer.mozilla.org/docs/WebAssembly/JavaScript_interface/compileStreaming_static) */
   function compileStreaming(
     source: Response | PromiseLike<Response>,
   ): Promise<Module>;
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/WebAssembly/instantiate) */
+  /** [MDN Reference](https://developer.mozilla.org/docs/WebAssembly/JavaScript_interface/instantiate_static) */
   function instantiate(
     bytes: BufferSource,
     importObject?: Imports,
@@ -11048,12 +11149,12 @@ declare namespace WebAssembly {
     moduleObject: Module,
     importObject?: Imports,
   ): Promise<Instance>;
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/WebAssembly/instantiateStreaming) */
+  /** [MDN Reference](https://developer.mozilla.org/docs/WebAssembly/JavaScript_interface/instantiateStreaming_static) */
   function instantiateStreaming(
     source: Response | PromiseLike<Response>,
     importObject?: Imports,
   ): Promise<WebAssemblyInstantiatedSource>;
-  /** [MDN Reference](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/WebAssembly/validate) */
+  /** [MDN Reference](https://developer.mozilla.org/docs/WebAssembly/JavaScript_interface/validate_static) */
   function validate(bytes: BufferSource): boolean;
 }
 
@@ -11163,6 +11264,10 @@ declare const onmessage:
 declare const onmessageerror:
   | ((this: DedicatedWorkerGlobalScope, ev: MessageEvent) => unknown)
   | null;
+/** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DedicatedWorkerGlobalScope/rtctransform_event) */
+declare const onrtctransform:
+  | ((this: DedicatedWorkerGlobalScope, ev: Event) => unknown)
+  | null;
 /**
  * Aborts dedicatedWorkerGlobal.
  *
@@ -11307,7 +11412,9 @@ declare function structuredClone<T = unknown>(
   value: T,
   options?: StructuredSerializeOptions,
 ): T;
+/** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DedicatedWorkerGlobalScope/cancelAnimationFrame) */
 declare function cancelAnimationFrame(handle: number): void;
+/** [MDN Reference](https://developer.mozilla.org/docs/Web/API/DedicatedWorkerGlobalScope/requestAnimationFrame) */
 declare function requestAnimationFrame(callback: FrameRequestCallback): number;
 declare function addEventListener<
   K extends keyof DedicatedWorkerGlobalScopeEventMap,
@@ -11415,7 +11522,6 @@ type Transferable =
   | VideoFrame
   | ArrayBuffer;
 type Uint32List = Uint32Array | readonly GLuint[];
-type VibratePattern = number | readonly number[];
 type XMLHttpRequestBodyInit =
   | Blob
   | BufferSource
@@ -11605,6 +11711,7 @@ type RequestDestination =
   | 'worker'
   | 'xslt';
 type RequestMode = 'cors' | 'navigate' | 'no-cors' | 'same-origin';
+type RequestPriority = 'auto' | 'high' | 'low';
 type RequestRedirect = 'error' | 'follow' | 'manual';
 type ResizeQuality = 'high' | 'low' | 'medium' | 'pixelated';
 type ResponseType =
