diff --git a/packages/v5.4/output/lib-files/lib.es2016.intl.d.ts b/packages/v5.4/output/lib-files/lib.es2016.intl.d.ts
new file mode 100644
index 0000000..5d6386a
--- /dev/null
+++ b/packages/v5.4/output/lib-files/lib.es2016.intl.d.ts
@@ -0,0 +1,16 @@
+/// <reference no-default-lib="true"/>
+/// <reference types="ts-type-forge/global" />
+
+declare namespace Intl {
+  /**
+   * The `Intl.getCanonicalLocales()` method returns an array containing
+   * the canonical locale names. Duplicates will be omitted and elements
+   * will be validated as structurally valid language tags.
+   *
+   * [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/getCanonicalLocales)
+   *
+   * @param locale A list of String values for which to get the canonical locale names
+   * @returns An array containing the canonical and validated locale names.
+   */
+  function getCanonicalLocales(locale?: string | readonly string[]): string[];
+}
