diff --git a/packages/v5.3/output/lib-files/lib.es2020.intl.d.ts b/packages/v5.4/output/lib-files/lib.es2020.intl.d.ts
index 9c0a1ac..f6f701e 100644
--- a/packages/v5.3/output/lib-files/lib.es2020.intl.d.ts
+++ b/packages/v5.4/output/lib-files/lib.es2020.intl.d.ts
@@ -4,9 +4,11 @@
 /// <reference path="./lib.es2018.intl.d.ts" />
 declare namespace Intl {
   /**
-   * [Unicode BCP 47 Locale Identifiers](https://unicode.org/reports/tr35/#Unicode_Language_and_Locale_Identifiers) definition.
+   * A string that is a valid [Unicode BCP 47 Locale Identifier](https://unicode.org/reports/tr35/#Unicode_locale_identifier).
    *
-   * [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl#locales_argument).
+   * For example: "fa", "es-MX", "zh-Hant-TW".
+   *
+   * See [MDN - Intl - locales argument](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl#locales_argument).
    */
   type UnicodeBCP47LocaleIdentifier = string;
 
@@ -74,16 +76,9 @@ declare namespace Intl {
   type RelativeTimeFormatStyle = 'long' | 'short' | 'narrow';
 
   /**
-   * [BCP 47 language tag](http://tools.ietf.org/html/rfc5646) definition.
+   * The locale or locales to use
    *
-   * [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl#locales_argument).
-   */
-  type BCP47LanguageTag = string;
-
-  /**
-   * The locale(s) to use
-   *
-   * [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl#locales_argument).
+   * See [MDN - Intl - locales argument](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl#locales_argument).
    */
   type LocalesArgument =
     | UnicodeBCP47LocaleIdentifier
@@ -210,9 +205,7 @@ declare namespace Intl {
      * [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/RelativeTimeFormat/RelativeTimeFormat).
      */
     new (
-      locales?:
-        | UnicodeBCP47LocaleIdentifier
-        | readonly UnicodeBCP47LocaleIdentifier[],
+      locales?: LocalesArgument,
       options?: RelativeTimeFormatOptions,
     ): RelativeTimeFormat;
 
@@ -235,9 +228,7 @@ declare namespace Intl {
      * [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/RelativeTimeFormat/supportedLocalesOf).
      */
     supportedLocalesOf(
-      locales?:
-        | UnicodeBCP47LocaleIdentifier
-        | readonly UnicodeBCP47LocaleIdentifier[],
+      locales?: LocalesArgument,
       options?: RelativeTimeFormatOptions,
     ): UnicodeBCP47LocaleIdentifier[];
   };
@@ -318,7 +309,7 @@ declare namespace Intl {
     /** Attempts to remove information about the locale that would be added by calling `Locale.maximize()`. */
     minimize(): Locale;
     /** Returns the locale's full locale identifier string. */
-    toString(): BCP47LanguageTag;
+    toString(): UnicodeBCP47LocaleIdentifier;
   }
 
   /**
@@ -336,7 +327,10 @@ declare namespace Intl {
    * [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/Locale).
    */
   const Locale: {
-    new (tag: BCP47LanguageTag | Locale, options?: LocaleOptions): Locale;
+    new (
+      tag: UnicodeBCP47LocaleIdentifier | Locale,
+      options?: LocaleOptions,
+    ): Locale;
   };
 
   type DisplayNamesFallback = 'code' | 'none';
@@ -429,6 +423,52 @@ declare namespace Intl {
     supportedLocalesOf(
       locales?: LocalesArgument,
       options?: { readonly localeMatcher?: RelativeTimeFormatLocaleMatcher },
-    ): BCP47LanguageTag[];
+    ): UnicodeBCP47LocaleIdentifier[];
   };
+
+  interface CollatorConstructor {
+    new (locales?: LocalesArgument, options?: CollatorOptions): Collator;
+    (locales?: LocalesArgument, options?: CollatorOptions): Collator;
+    supportedLocalesOf(
+      locales: LocalesArgument,
+      options?: CollatorOptions,
+    ): string[];
+  }
+
+  interface DateTimeFormatConstructor {
+    new (
+      locales?: LocalesArgument,
+      options?: DateTimeFormatOptions,
+    ): DateTimeFormat;
+    (
+      locales?: LocalesArgument,
+      options?: DateTimeFormatOptions,
+    ): DateTimeFormat;
+    supportedLocalesOf(
+      locales: LocalesArgument,
+      options?: DateTimeFormatOptions,
+    ): string[];
+  }
+
+  interface NumberFormatConstructor {
+    new (
+      locales?: LocalesArgument,
+      options?: NumberFormatOptions,
+    ): NumberFormat;
+    (locales?: LocalesArgument, options?: NumberFormatOptions): NumberFormat;
+    supportedLocalesOf(
+      locales: LocalesArgument,
+      options?: NumberFormatOptions,
+    ): string[];
+  }
+
+  interface PluralRulesConstructor {
+    new (locales?: LocalesArgument, options?: PluralRulesOptions): PluralRules;
+    (locales?: LocalesArgument, options?: PluralRulesOptions): PluralRules;
+
+    supportedLocalesOf(
+      locales: LocalesArgument,
+      options?: { readonly localeMatcher?: 'lookup' | 'best fit' },
+    ): string[];
+  }
 }
