diff --git experimental/packages/utils/numeric-input-utils/src/utils.mts apps/numeric-input-utils/src/utils.mts
index 63b76cb7d..dcb628ffb 100644
--- experimental/packages/utils/numeric-input-utils/src/utils.mts
+++ apps/numeric-input-utils/src/utils.mts
@@ -1,3 +1,5 @@
+import { Num, Result } from 'ts-data-forge';
+
 export type NumericTypeProperties<N extends number> = Readonly<{
   min: N;
   max: N;
@@ -40,7 +42,8 @@ export const createNumberType = <N extends number>({
 
   const encode = (s: N): string => s.toString();
 
-  const decode = (s: string): N => clampAndRound(Number(s));
+  const decode = (s: string): N =>
+    clampAndRound(Result.unwrapOkOr(Num.safeParseFloat(s), Number.NaN));
 
   return {
     min,
