diff --git experimental/packages/apps/event-schedule-app-shared/src/v2/types/record/event-schedule.mts apps/event-schedule-app-shared/src/v2/types/record/event-schedule.mts
index febd66139..37e5335c5 100644
--- experimental/packages/apps/event-schedule-app-shared/src/v2/types/record/event-schedule.mts
+++ apps/event-schedule-app-shared/src/v2/types/record/event-schedule.mts
@@ -1,4 +1,4 @@
-import { DateUtils } from '@noshiro/ts-utils';
+import { DateUtils } from 'ts-fortress-types';
 import {
   defaultNotificationSettings,
   fillAnswerSymbol,
@@ -64,23 +64,25 @@ export const defaultEventSchedule = {
 } as const satisfies EventSchedule;
 
 const d = defaultEventSchedule;
-export const fillEventSchedule = (p?: PartialEventSchedule): EventSchedule => ({
-  title: p?.title ?? d.title,
-  notes: p?.notes ?? d.notes,
-  datetimeSpecification: p?.datetimeSpecification ?? d.datetimeSpecification,
-  datetimeRangeList: (p?.datetimeRangeList ?? d.datetimeRangeList).map(
-    fillDatetimeRange,
-  ),
-  useAnswerDeadline: p?.useAnswerDeadline ?? d.useAnswerDeadline,
-  answerDeadline: fillYmdhm(p?.answerDeadline ?? d.answerDeadline),
-  customizeSymbolSettings:
-    p?.customizeSymbolSettings ?? d.customizeSymbolSettings,
-  answerSymbolList: (p?.answerSymbolList ?? d.answerSymbolList).map(
-    fillAnswerSymbol,
-  ),
-  useNotification: p?.useNotification ?? d.useNotification,
-  notificationSettings: fillNotificationSettings(
-    p?.notificationSettings ?? d.notificationSettings,
-  ),
-  timezoneOffsetMinutes: p?.timezoneOffsetMinutes ?? d.timezoneOffsetMinutes,
-});
+
+export const fillEventSchedule = (p?: PartialEventSchedule): EventSchedule =>
+  ({
+    title: p?.title ?? d.title,
+    notes: p?.notes ?? d.notes,
+    datetimeSpecification: p?.datetimeSpecification ?? d.datetimeSpecification,
+    datetimeRangeList: (p?.datetimeRangeList ?? d.datetimeRangeList).map(
+      fillDatetimeRange,
+    ),
+    useAnswerDeadline: p?.useAnswerDeadline ?? d.useAnswerDeadline,
+    answerDeadline: fillYmdhm(p?.answerDeadline ?? d.answerDeadline),
+    customizeSymbolSettings:
+      p?.customizeSymbolSettings ?? d.customizeSymbolSettings,
+    answerSymbolList: (p?.answerSymbolList ?? d.answerSymbolList).map(
+      fillAnswerSymbol,
+    ),
+    useNotification: p?.useNotification ?? d.useNotification,
+    notificationSettings: fillNotificationSettings(
+      p?.notificationSettings ?? d.notificationSettings,
+    ),
+    timezoneOffsetMinutes: p?.timezoneOffsetMinutes ?? d.timezoneOffsetMinutes,
+  }) as const;
