diff --git experimental/packages/apps/event-schedule-app-shared/src/v2/types/record/ymdhm.mts apps/event-schedule-app-shared/src/v2/types/record/ymdhm.mts
index e04b60cb5..58fa8a3fb 100644
--- experimental/packages/apps/event-schedule-app-shared/src/v2/types/record/ymdhm.mts
+++ apps/event-schedule-app-shared/src/v2/types/record/ymdhm.mts
@@ -1,11 +1,18 @@
-import { DateUtils, Num, type DateType } from '@noshiro/ts-utils';
+import { DateUtils } from 'ts-fortress-types';
+import {
+  type DateEnum,
+  type HoursEnum,
+  type MinutesEnum,
+  type MonthEnum,
+  type SafeUint,
+} from 'ts-type-forge';
 import {
   defaultHoursMinutes,
   defaultYearMonthDate,
 } from '../../../v1/index.mjs';
 
 export type Ymdhm = Readonly<{
-  year: YearEnum;
+  year: SafeUint;
   month: MonthEnum;
   date: DateEnum;
   hours: HoursEnum;
@@ -20,26 +27,29 @@ export const defaultYmdhm: Ymdhm = {
   date: defaultYearMonthDate.date,
   hours: defaultHoursMinutes.hours,
   minutes: defaultHoursMinutes.minutes,
-};
+} as const;
 
 const d = defaultYmdhm;
-export const fillYmdhm = (p?: PartialYmdhm): Ymdhm => ({
-  year: p?.year ?? d.year,
-  month: p?.month ?? d.month,
-  date: p?.date ?? d.date,
-  hours: p?.hours ?? d.hours,
-  minutes: p?.minutes ?? d.minutes,
-});
-
-export const ymdhmFromDate = (date: RawDateType): Ymdhm => ({
-  year: DateUtils.getLocaleYear(date),
-  month: DateUtils.getLocaleMonth(date),
-  date: DateUtils.getLocaleDate(date),
-  hours: DateUtils.getLocaleHours(date),
-  minutes: DateUtils.getLocaleMinutes(date),
-});
-
-export const ymdhm2Date = (ymdhm: Ymdhm): DateType =>
+
+export const fillYmdhm = (p?: PartialYmdhm): Ymdhm =>
+  ({
+    year: p?.year ?? d.year,
+    month: p?.month ?? d.month,
+    date: p?.date ?? d.date,
+    hours: p?.hours ?? d.hours,
+    minutes: p?.minutes ?? d.minutes,
+  }) as const;
+
+export const ymdhmFromDate = (date: Date): Ymdhm =>
+  ({
+    year: DateUtils.getLocaleYear(date),
+    month: DateUtils.getLocaleMonth(date),
+    date: DateUtils.getLocaleDate(date),
+    hours: DateUtils.getLocaleHours(date),
+    minutes: DateUtils.getLocaleMinutes(date),
+  }) as const;
+
+export const ymdhm2Date = (ymdhm: Ymdhm): Date =>
   DateUtils.create(
     ymdhm.year,
     ymdhm.month,
@@ -49,15 +59,15 @@ export const ymdhm2Date = (ymdhm: Ymdhm): DateType =>
   );
 
 export const compareYmdhm = (a: Ymdhm, b: Ymdhm): -1 | 0 | 1 => {
-  if (a.year !== b.year)
-    return Num.mapNaN2Undefined(Math.sign(a.year - b.year)) ?? 0;
-  if (a.month !== b.month)
-    return Num.mapNaN2Undefined(Math.sign(a.month - b.month)) ?? 0;
-  if (a.date !== b.date)
-    return Num.mapNaN2Undefined(Math.sign(a.date - b.date)) ?? 0;
-  if (a.hours !== b.hours)
-    return Num.mapNaN2Undefined(Math.sign(a.hours - b.hours)) ?? 0;
-  if (a.minutes !== b.minutes)
-    return Num.mapNaN2Undefined(Math.sign(a.minutes - b.minutes)) ?? 0;
+  if (a.year !== b.year) return a.year < b.year ? -1 : 1;
+
+  if (a.month !== b.month) return a.month < b.month ? -1 : 1;
+
+  if (a.date !== b.date) return a.date < b.date ? -1 : 1;
+
+  if (a.hours !== b.hours) return a.hours < b.hours ? -1 : 1;
+
+  if (a.minutes !== b.minutes) return a.minutes < b.minutes ? -1 : 1;
+
   return 0;
 };
