diff --git experimental/packages/utils/react-utils/src/hooks/index.mts apps/react-utils/src/hooks/index.mts
index 356c53b61..40742e95e 100644
--- experimental/packages/utils/react-utils/src/hooks/index.mts
+++ apps/react-utils/src/hooks/index.mts
@@ -5,8 +5,8 @@ export * from './use-alive.mjs';
 export * from './use-async-reducer.mjs';
 export * from './use-debounce-fn.mjs';
 export * from './use-debounce.mjs';
+export * from './use-observable.mjs';
 export * from './use-previous.mjs';
 export * from './use-promise-value.mjs';
 export * from './use-state-with-mapfn.mjs';
-export * from './use-tiny-observable-hooks.mjs';
 export * from './use-value-as-ref.mjs';
